diff -Naur mame-98b0ba18a9109339c2ebf4f5945f6c3575301ba9.orig/src/osd/libretro/libretro-internal/libretro.cpp mame-98b0ba18a9109339c2ebf4f5945f6c3575301ba9/src/osd/libretro/libretro-internal/libretro.cpp
--- mame-98b0ba18a9109339c2ebf4f5945f6c3575301ba9.orig/src/osd/libretro/libretro-internal/libretro.cpp	2021-04-12 16:53:19.000000000 +0100
+++ mame-98b0ba18a9109339c2ebf4f5945f6c3575301ba9/src/osd/libretro/libretro-internal/libretro.cpp	2021-06-22 20:13:34.531629714 +0100
@@ -810,7 +810,7 @@
 	{
 		memory_interface_enumerator iter(mame_machine_manager::instance()->machine()->root_device());
 		for (device_memory_interface &memory : iter)
-			for ( space_index = 0 ; space_index < memory.num_spaces() ; space_index++)
+			for ( space_index = 0 ; space_index < memory.has_space() ; space_index++)
 				if ( memory.has_space(space_index))
 				{
 					auto &space = memory.space(space_index);
@@ -839,7 +839,7 @@
 	{
 		memory_interface_enumerator iter(mame_machine_manager::instance()->machine()->root_device());
 		for (device_memory_interface &memory : iter)
-			for ( space_index = 0 ; space_index < memory.num_spaces() ; space_index++)
+			for ( space_index = 0 ; space_index < memory.has_space() ; space_index++)
 				if ( memory.has_space(space_index))
 				{
 					auto &space = memory.space(space_index);
