diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 70334d9..a0db29d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -60,7 +60,7 @@ include(BuildLibMPEG2)
 
 set (CMAKE_CXX_STANDARD 11)
 
-build_libmpeg2( )
+PKG_SEARCH_MODULE(MPEG2 REQUIRED libmpeg2)
 
 message(STATUS "Target: ${CMAKE_SYSTEM_NAME} ${CMAKE_TARGET_ARCHITECTURES}")
 
@@ -124,7 +124,6 @@ add_subdirectory(timer)
 add_subdirectory(video)
 add_subdirectory(vldp)
 
-add_dependencies( vldp libmpeg2 )
 add_dependencies( ldp-out vldp )
 add_dependencies( game vldp )
 add_dependencies( sound vldp )
diff --git a/src/vldp/vldp.h b/src/vldp/vldp.h
index fe78b32..ed54424 100644
--- a/src/vldp/vldp.h
+++ b/src/vldp/vldp.h
@@ -35,6 +35,8 @@ extern "C" {
 // But CMake does the right thing here 
 #include <SDL3/SDL.h> // only used for threading
 
+#include <mpeg2dec/mpeg2.h>
+
 struct yuv_buf {
     uint8_t *Y;     // Y channel
     uint8_t *U;     // U channel
diff --git a/src/vldp/vldp_internal.h b/src/vldp/vldp_internal.h
index 88450e9..7976e73 100644
--- a/src/vldp/vldp_internal.h
+++ b/src/vldp/vldp_internal.h
@@ -27,7 +27,6 @@
 
 #include "vldp.h" // for the VLDP_BOOL definition and SDL.h
 
-#include <mpeg2.h>
 
 // this is which version of the .dat file format we are using
 #define DAT_VERSION 3
