diff --git a/src/video/video.cpp b/src/video/video.cpp
index 3c974e9..f1dd5a9 100644
--- a/src/video/video.cpp
+++ b/src/video/video.cpp
@@ -2006,12 +2006,12 @@ void vid_blit () {
 
     if (g_bezel_toggle) vid_render_bezels();
 
+    if (g_fRotateDegrees != 0) vid_render_rotate();
+
     if (g_game->get_sinden_border())
         draw_border(g_game->get_sinden_border(),
             g_game->get_sinden_border_color());
 
-    if (g_fRotateDegrees != 0) vid_render_rotate();
-
     SDL_RenderPresent(g_renderer);
 
     if (g_softsboard_needs_update) {
