diff --git a/src/io/input.cpp b/src/io/input.cpp
index 2c44353..0b4c5a7 100644
--- a/src/io/input.cpp
+++ b/src/io/input.cpp
@@ -690,6 +690,9 @@ void SDL_check_input()
         process_event(&event);
     }
 
+    // processes non sdl events
+    process_event(NULL);
+
     // added by JFA for -idleexit
     if (get_idleexit() > 0 && elapsed_ms_time(idle_timer) > get_idleexit())
         set_quitflag();
@@ -727,6 +730,8 @@ void process_event(SDL_Event *event)
 {
     unsigned int i = 0;
 
+    if(event != NULL) {
+
     // by RDG2010
     // make things easier to read...
     SDL_Keycode keyPressed = event->key.keysym.sym;
@@ -931,6 +936,7 @@ void process_event(SDL_Event *event)
     default:
         break;
     }
+    }
 
     if (g_game->get_mouse_enabled())
     {
@@ -940,6 +946,7 @@ void process_event(SDL_Event *event)
 
         } else {
 
+	   if(event != NULL) {
            switch (event->type) {
            case SDL_MOUSEBUTTONDOWN:
                for (i = 0; i < (sizeof(mouse_buttons_map) / sizeof(int)); i++) {
@@ -962,6 +969,7 @@ void process_event(SDL_Event *event)
                        event->motion.xrel, event->motion.yrel, NOMOUSE);
                break;
           }
+	  }
        }
     }
 
