diff --git a/src/joystick/SDL_joystick.c b/src/joystick/SDL_joystick.c
index 69dd289f1..47ad14f52 100644
--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1905,7 +1905,7 @@ void SDL_JoystickUpdate(void)
             if (joystick->delayed_guide_button) {
                 SDL_GameControllerHandleDelayedGuideButton(joystick);
             }
-        }
+
 
         now = SDL_GetTicks();
         if (joystick->rumble_expiration &&
@@ -1927,6 +1927,8 @@ void SDL_JoystickUpdate(void)
             SDL_TICKS_PASSED(now, joystick->trigger_rumble_expiration)) {
             SDL_JoystickRumbleTriggers(joystick, 0, 0, 0);
         }
+
+	}
     }
 
     /* this needs to happen AFTER walking the joystick list above, so that any
