diff --git a/src/joystick/SDL_joystick.c b/src/joystick/SDL_joystick.c
index 26b017bb9..37667ed1c 100644
--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1690,7 +1690,7 @@ SDL_JoystickUpdate(void)
             if (joystick->delayed_guide_button) {
                 SDL_GameControllerHandleDelayedGuideButton(joystick);
             }
-        }
+
 
         if (joystick->rumble_expiration) {
             SDL_LockJoysticks();
@@ -1711,6 +1711,8 @@ SDL_JoystickUpdate(void)
             }
             SDL_UnlockJoysticks();
         }
+
+	}
     }
 
     SDL_LockJoysticks();
