diff --git a/src/joystick/SDL_joystick.c b/src/joystick/SDL_joystick.c
index 6d85571f7..7b2546a53 100644
--- a/src/joystick/SDL_joystick.c
+++ b/src/joystick/SDL_joystick.c
@@ -1494,7 +1494,7 @@ int SDL_JoystickRumble(SDL_Joystick *joystick, Uint16 low_frequency_rumble, Uint
             } else {
                 joystick->rumble_resend = 0;
             }
-        }
+
 
         if (retval == 0) {
             joystick->low_frequency_rumble = low_frequency_rumble;
@@ -1510,6 +1510,8 @@ int SDL_JoystickRumble(SDL_Joystick *joystick, Uint16 low_frequency_rumble, Uint
                 joystick->rumble_resend = 0;
             }
         }
+
+	}
     }
     SDL_UnlockJoysticks();
 
