diff --git a/src/joystick/SDL_gamecontroller.c b/src/joystick/SDL_gamecontroller.c
index c4f8d6542..524034585 100644
--- a/src/joystick/SDL_gamecontroller.c
+++ b/src/joystick/SDL_gamecontroller.c
@@ -1957,13 +1957,13 @@ int SDL_GameControllerInitMappings(void)
 
     SDL_AssertJoysticksLocked();
 
-    pMappingString = s_ControllerMappings[i];
-    while (pMappingString) {
-        SDL_PrivateGameControllerAddMapping(pMappingString, SDL_CONTROLLER_MAPPING_PRIORITY_DEFAULT);
-
-        i++;
-        pMappingString = s_ControllerMappings[i];
-    }
+    //pMappingString = s_ControllerMappings[i];
+    //while (pMappingString) {
+    //    SDL_PrivateGameControllerAddMapping(pMappingString, SDL_CONTROLLER_MAPPING_PRIORITY_DEFAULT);
+    //
+    //    i++;
+    //    pMappingString = s_ControllerMappings[i];
+    //}
 
     if (SDL_GetControllerMappingFilePath(szControllerMapPath, sizeof(szControllerMapPath))) {
         SDL_GameControllerAddMappingsFromFile(szControllerMapPath);
