diff --git a/src/core/linux/SDL_evdev.c b/src/core/linux/SDL_evdev.c
index de1b758db..910b7a063 100644
--- a/src/core/linux/SDL_evdev.c
+++ b/src/core/linux/SDL_evdev.c
@@ -373,6 +373,11 @@ void SDL_EVDEV_Poll(void)
                             SDL_SendKeyboardKey(SDL_PRESSED, scan_code);
                         }
                     }
+
+		    // retry to init in case something failed at init (like tty not yet opened)
+		    if (!_this->kbd) {
+		      _this->kbd = SDL_EVDEV_kbd_init();
+		    }
                     SDL_EVDEV_kbd_keycode(_this->kbd, events[i].code, events[i].value);
                     break;
                 case EV_ABS:
