diff -ruN batocera-emulationstation-f333689ef0383b917dd7d3806940ce14f92e0828.orig/es-core/src/platform.cpp batocera-emulationstation-f333689ef0383b917dd7d3806940ce14f92e0828/es-core/src/platform.cpp
--- batocera-emulationstation-f333689ef0383b917dd7d3806940ce14f92e0828.orig/es-core/src/platform.cpp	2021-05-11 20:16:34.000000000 +0100
+++ batocera-emulationstation-f333689ef0383b917dd7d3806940ce14f92e0828/es-core/src/platform.cpp	2021-05-15 14:36:06.223926542 +0100
@@ -333,10 +333,10 @@
 	{
 		std::string batteryRootPath;
 
-		auto files = Utils::FileSystem::getDirContent("/sys/class/power_supply");
+		auto files = Utils::FileSystem::getDirContent("/sys/kernel");
 		for (auto file : files)
 		{
-			if (Utils::String::toLower(file).find("/bat") != std::string::npos)
+			if (Utils::String::toLower(file).find("/xpi_gamecon") != std::string::npos)
 			{
 				batteryRootPath = file;
 				break;
@@ -348,7 +348,7 @@
 		else
 		{
 			batteryStatusPath = batteryRootPath + "/status";
-			batteryCapacityPath = batteryRootPath + "/capacity";
+			batteryCapacityPath = batteryRootPath + "/percent";
 		}
 	}
 
@@ -361,7 +361,7 @@
 	else
 	{
 		ret.hasBattery = true;
-		ret.isCharging = (Utils::String::replace(Utils::FileSystem::readAllText(batteryStatusPath), "\n", "") != "Discharging");
+		ret.isCharging = (Utils::String::replace(Utils::FileSystem::readAllText(batteryStatusPath), "\n", "") == "198");
 		ret.level = atoi(Utils::FileSystem::readAllText(batteryCapacityPath).c_str());
 	}
 
