diff -ruN batocera-emulationstation-d19e2cfc9f40743bc134a5a6a7786fd7c41c6f85.orig/es-app/src/guis/GuiMenu.cpp batocera-emulationstation-d19e2cfc9f40743bc134a5a6a7786fd7c41c6f85/es-app/src/guis/GuiMenu.cpp
--- batocera-emulationstation-d19e2cfc9f40743bc134a5a6a7786fd7c41c6f85.orig/es-app/src/guis/GuiMenu.cpp	2021-05-17 18:13:47.000000000 +0100
+++ batocera-emulationstation-d19e2cfc9f40743bc134a5a6a7786fd7c41c6f85/es-app/src/guis/GuiMenu.cpp	2021-05-21 22:48:00.791572427 +0100
@@ -152,7 +152,7 @@
 #endif
 
 		addEntry(_("SCRAPE").c_str(), true, [this] { openScraperSettings(); }, "iconScraper");		
-		addEntry(_("UPDATES & DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
+		addEntry(_("DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
 		addEntry(_("SYSTEM SETTINGS").c_str(), true, [this] { openSystemSettings_batocera(); }, "iconSystem");
 	}
 	else
@@ -1055,7 +1055,7 @@
 
 void GuiMenu::openUpdatesSettings()
 {
-	GuiSettings *updateGui = new GuiSettings(mWindow, _("UPDATES & DOWNLOADS").c_str());
+	GuiSettings *updateGui = new GuiSettings(mWindow, _("DOWNLOADS").c_str());
 
 	updateGui->addGroup(_("DOWNLOADS"));
 
@@ -1095,7 +1095,7 @@
 		});
 	}
 
-	updateGui->addGroup(_("SOFTWARE UPDATES"));
+	/*updateGui->addGroup(_("SOFTWARE UPDATES"));
 
 	// Enable updates
 	auto updates_enabled = std::make_shared<SwitchComponent>(mWindow);
@@ -1143,7 +1143,7 @@
 
 			mWindow->pushGui(new GuiUpdate(mWindow));
 		}
-	});
+	});*/
 
 	mWindow->pushGui(updateGui);
 }
