diff -Naur batocera-emulationstation-92a9cc028441722b7d00acc6ad5e72372216a672.orig/es-app/src/guis/GuiMenu.cpp batocera-emulationstation-92a9cc028441722b7d00acc6ad5e72372216a672/es-app/src/guis/GuiMenu.cpp
--- batocera-emulationstation-92a9cc028441722b7d00acc6ad5e72372216a672.orig/es-app/src/guis/GuiMenu.cpp	2021-06-16 06:29:29.000000000 +0100
+++ batocera-emulationstation-92a9cc028441722b7d00acc6ad5e72372216a672/es-app/src/guis/GuiMenu.cpp	2021-06-20 18:49:21.491857874 +0100
@@ -64,6 +64,7 @@
 #define fake_gettext_architecture _("Architecture")
 #define fake_gettext_diskformat   _("Disk format")
 #define fake_gettext_temperature  _("Temperature")
+#define fake_gettext_fan          _("Fan speed")
 #define fake_gettext_avail_memory _("Available memory")
 #define fake_gettext_battery      _("Battery")
 #define fake_gettext_cpu_model    _("Cpu model")
