diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8c494bfdb..643ffd9bbc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -163,7 +163,7 @@ endif()
 include(CCache)
 
 # for revision info
-find_package(Git)
+#find_package(Git)
 if(GIT_FOUND)
   # make sure version information gets re-run when the current Git HEAD changes
   execute_process(WORKING_DIRECTORY ${PROJECT_SOURCE_DIR} COMMAND ${GIT_EXECUTABLE} rev-parse --git-path HEAD
