diff --git a/Source/Core/InputCommon/ControllerInterface/ControllerInterface.h b/Source/Core/InputCommon/ControllerInterface/ControllerInterface.h
index 05a3a99a05..9390d02d73 100644
--- a/Source/Core/InputCommon/ControllerInterface/ControllerInterface.h
+++ b/Source/Core/InputCommon/ControllerInterface/ControllerInterface.h
@@ -35,9 +35,9 @@
 #if defined(HAVE_SDL3)
 #define CIFACE_USE_SDL
 #endif
-#if defined(HAVE_HIDAPI)
-#define CIFACE_USE_STEAMDECK
-#endif
+//#if defined(HAVE_HIDAPI)
+//#define CIFACE_USE_STEAMDECK
+//#endif
 
 namespace ciface
 {
