diff --git a/Source/Core/DolphinQt/CMakeLists.txt b/Source/Core/DolphinQt/CMakeLists.txt
index 9603f2ffd9..d3e93db314 100644
--- a/Source/Core/DolphinQt/CMakeLists.txt
+++ b/Source/Core/DolphinQt/CMakeLists.txt
@@ -15,6 +15,11 @@ set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 
 find_package(Qt6 REQUIRED COMPONENTS Core Gui Widgets Svg)
+
+# GuiPrivate is needed to #include qplatformnativeinterface.h in MainWindow.cpp with Qt 6.10+.
+set(NEED_QT_GUI_PRIVATE_COMPONENT ON)
+find_package(Qt6 REQUIRED COMPONENTS GuiPrivate)
+
 message(STATUS "Found Qt version ${Qt6_VERSION}")
 
 set_property(TARGET Qt6::Core PROPERTY INTERFACE_COMPILE_FEATURES "")
@@ -423,6 +428,13 @@ PRIVATE
   implot
 )
 
+if (NEED_QT_GUI_PRIVATE_COMPONENT)
+  target_link_libraries(dolphin-emu
+  PRIVATE
+    Qt6::GuiPrivate
+  )
+endif()
+
 if (WIN32)
   target_link_libraries(dolphin-emu
   PRIVATE
