diff --git a/libshaderc/CMakeLists.txt b/libshaderc/CMakeLists.txt
index aa86fd5..c331a06 100644
--- a/libshaderc/CMakeLists.txt
+++ b/libshaderc/CMakeLists.txt
@@ -37,6 +37,12 @@ target_compile_definitions(shaderc_shared
     PUBLIC SHADERC_SHAREDLIB
 )
 
+# Use a different shared library name on Linux, to avoid conflicts with
+# the shaderc distributed with the Vulkan SDK.
+if(NOT WIN32)
+  set_target_properties(shaderc_shared PROPERTIES OUTPUT_NAME "shaderc_ds")
+endif()
+
 if(SHADERC_ENABLE_INSTALL)
   install(
     FILES
