diff --git a/xbmc/messaging/ApplicationMessenger.cpp b/xbmc/messaging/ApplicationMessenger.cpp
index 562edde372..34082a62e2 100644
--- a/xbmc/messaging/ApplicationMessenger.cpp
+++ b/xbmc/messaging/ApplicationMessenger.cpp
@@ -224,8 +224,25 @@ void CApplicationMessenger::ProcessMessages()
   }
 }
 
+void fileMsgEvent(unsigned int dwMessage) {
+    switch(dwMessage) {
+      case TMSG_QUIT:
+        system("echo EXIT > /var/run/kodi.msg");
+        break;
+      case TMSG_RESTART:
+        system("echo RESTART > /var/run/kodi.msg");
+        break;
+      case TMSG_POWERDOWN:
+        system("echo SHUTDOWN > /var/run/kodi.msg");
+        break;
+    }
+}
+
 void CApplicationMessenger::ProcessMessage(ThreadMessage *pMsg)
 {
+  // Export event to file
+  fileMsgEvent(pMsg->dwMessage);
+
   //special case for this that we handle ourselves
   if (pMsg->dwMessage == TMSG_CALLBACK)
   {
