diff --git a/cmake/modules/FindFFMPEG.cmake b/cmake/modules/FindFFMPEG.cmake
index 37cf1451a3..6d07bf9bb1 100644
--- a/cmake/modules/FindFFMPEG.cmake
+++ b/cmake/modules/FindFFMPEG.cmake
@@ -305,40 +275,20 @@ else()
 
   # macro for find_library usage
   # arg1: lowercase libname (eg libavcodec, libpostproc, etc)
-  # arg2: version search specifier
-  macro(ffmpeg_find_lib libname libversion)
+  macro(ffmpeg_find_lib libname)
     string(TOUPPER ${libname} libname_UPPER)
     string(REPLACE "lib" "" name ${libname})
 
-    if(WIN32 OR WINDOWS_STORE)
-      find_library(FFMPEG_${libname_UPPER}
-                   NAMES ${name} ${libname}
-                   PATH_SUFFIXES ffmpeg/${libname}
-                   HINTS ${DEPENDS_PATH}/lib ${MINGW_LIBS_DIR}/lib
-                   ${${CORE_SYSTEM_NAME}_SEARCH_CONFIG})
-    else()
-      find_package(PkgConfig REQUIRED)
-
-      # ToDo: We cant use IMPORTED_TARGET yet.
-      # Linux CI fails with gmp related link issues when using the imported target
-      pkg_check_modules(FFMPEG_${libname_UPPER} ${libname}${libversion} ${SEARCH_QUIET})
-
-      # As windows does a simple find_library call, the output is a filename to the library
-      # for the pkgconfig search, we can get the full file path from _LINK_LIBRARIES first element
-      if(FFMPEG_${libname_UPPER}_FOUND)
-        # Retrieve full path name of lib (first element)
-        list(POP_FRONT FFMPEG_${libname_UPPER}_LINK_LIBRARIES FFMPEG_${libname_UPPER})
-      endif()
-    endif()
+    find_library(FFMPEG_${libname_UPPER}
+                 NAMES ${name} ${libname}
+                 PATH_SUFFIXES ffmpeg/${libname}
+                 HINTS ${DEPENDS_PATH}/lib ${MINGW_LIBS_DIR}/lib
+                 ${${CORE_SYSTEM_NAME}_SEARCH_CONFIG})
   endmacro()
 
   foreach(_ffmpeg_pkg IN ITEMS ${FFMPEG_PKGS})
-    string(REGEX REPLACE "[>]?=.*" "" _libname ${_ffmpeg_pkg})
-
-    string(REGEX MATCH "[>]?=" _search_spec ${_ffmpeg_pkg})
-    string(REGEX REPLACE ".*[>]?=" "${_search_spec}" _version ${_ffmpeg_pkg})
-
-    ffmpeg_find_lib(${_libname} ${_version})
+    string(REGEX REPLACE ">=.*" "" _libname ${_ffmpeg_pkg})
+    ffmpeg_find_lib(${_libname})
   endforeach()
 
   # Check all libs are found, and set found.
