diff --git a/xbmc/application/Application.cpp b/xbmc/application/Application.cpp
index fe3967edcf..3100c4a1c6 100644
--- a/xbmc/application/Application.cpp
+++ b/xbmc/application/Application.cpp
@@ -2036,7 +2036,13 @@ bool CApplication::Stop(int exitCode)
     m_frameMoveGuard.unlock();
 
     CVariant vExitCode(CVariant::VariantTypeObject);
-    vExitCode["exitcode"] = exitCode;
+ 
+    // Passing a CEC exit code usually powers off display - avoid that by pretending it's a reboot unless it's a power-off
+    if (exitCode == EXITCODE_POWERDOWN)
+      vExitCode["exitcode"] = exitCode;
+    else
+      vExitCode["exitcode"] = EXITCODE_REBOOT;
+ 
     CServiceBroker::GetAnnouncementManager()->Announce(ANNOUNCEMENT::System, "OnQuit", vExitCode);
 
     // Abort any active screensaver
