diff --git a/src/common/fs/path_util.cpp b/src/common/fs/path_util.cpp
index 4f69db6f5..0ccded20c 100644
--- a/src/common/fs/path_util.cpp
+++ b/src/common/fs/path_util.cpp
@@ -115,21 +115,21 @@ public:
         }
 #endif
 
-        GenerateSuyuPath(SuyuPath::SuyuDir, suyu_path);
-        GenerateSuyuPath(SuyuPath::AmiiboDir, suyu_path / AMIIBO_DIR);
-        GenerateSuyuPath(SuyuPath::CacheDir, suyu_path_cache);
-        GenerateSuyuPath(SuyuPath::ConfigDir, suyu_path_config);
+        GenerateSuyuPath(SuyuPath::SuyuDir, "/userdata/saves/switch/yuzu");
+        GenerateSuyuPath(SuyuPath::AmiiboDir, "/userdata/bios/switch/amiibo");
+        GenerateSuyuPath(SuyuPath::CacheDir, "/userdata/system/cache/yuzu");
+        GenerateSuyuPath(SuyuPath::ConfigDir, "/userdata/system/configs/yuzu");
         GenerateSuyuPath(SuyuPath::CrashDumpsDir, suyu_path / CRASH_DUMPS_DIR);
-        GenerateSuyuPath(SuyuPath::DumpDir, suyu_path / DUMP_DIR);
-        GenerateSuyuPath(SuyuPath::KeysDir, suyu_path / KEYS_DIR);
-        GenerateSuyuPath(SuyuPath::LoadDir, suyu_path / LOAD_DIR);
-        GenerateSuyuPath(SuyuPath::LogDir, suyu_path / LOG_DIR);
-        GenerateSuyuPath(SuyuPath::NANDDir, suyu_path / NAND_DIR);
+        GenerateSuyuPath(SuyuPath::DumpDir, "/userdata/saves/switch/yuzu/dump");
+        GenerateSuyuPath(SuyuPath::KeysDir, "/userdata/bios/switch/keys");
+        GenerateSuyuPath(SuyuPath::LoadDir, "/userdata/saves/switch/yuzu/load");
+        GenerateSuyuPath(SuyuPath::LogDir, "/userdata/system/logs");
+        GenerateSuyuPath(SuyuPath::NANDDir, "/userdata/bios/switch/nand");
         GenerateSuyuPath(SuyuPath::PlayTimeDir, suyu_path / PLAY_TIME_DIR);
-        GenerateSuyuPath(SuyuPath::ScreenshotsDir, suyu_path / SCREENSHOTS_DIR);
-        GenerateSuyuPath(SuyuPath::SDMCDir, suyu_path / SDMC_DIR);
-        GenerateSuyuPath(SuyuPath::ShaderDir, suyu_path / SHADER_DIR);
-        GenerateSuyuPath(SuyuPath::TASDir, suyu_path / TAS_DIR);
+        GenerateSuyuPath(SuyuPath::ScreenshotsDir, "/userdata/screenshots/switch");
+        GenerateSuyuPath(SuyuPath::SDMCDir, "/userdata/bios/switch/sdmc");
+        GenerateSuyuPath(SuyuPath::ShaderDir, "/userdata/saves/switch/yuzu/shaders");
+        GenerateSuyuPath(SuyuPath::TASDir, "/userdata/saves/switch/yuzu/tas");
         GenerateSuyuPath(SuyuPath::IconsDir, suyu_path / ICONS_DIR);
     }
 
