diff --git a/sound/pci/hda/patch_hdmi.c b/sound/pci/hda/patch_hdmi.c
index 9a7793eb16e9..9f08bea98c2a 100644
--- a/sound/pci/hda/patch_hdmi.c
+++ b/sound/pci/hda/patch_hdmi.c
@@ -1981,6 +1981,14 @@ static int hdmi_add_cvt(struct hda_codec *codec, hda_nid_t cvt_nid)
 	if (err < 0)
 		return err;
 
+	/*
+	 * Some HDMI codecs (at least NVIDIA 0x10de000b:0x10de0101:0x100100)
+	 * start transmitting an empty audio stream as soon as PIN_OUT and
+	 * AC_DIG1_ENABLE are enabled, which happens at open() time.
+	 * To avoid that, set format to 0, which is not valid for HDMI.
+	*/
+	snd_hda_codec_write(codec, cvt_nid, 0, AC_VERB_SET_STREAM_FORMAT, 0);
+
 	if (spec->num_cvts < ARRAY_SIZE(spec->cvt_nids))
 		spec->cvt_nids[spec->num_cvts] = cvt_nid;
 	spec->num_cvts++;
@@ -2181,6 +2189,12 @@ static int hdmi_pcm_close(struct hda_pcm_stream *hinfo,
 
 	mutex_lock(&spec->pcm_lock);
 	if (hinfo->nid) {
+		/*
+		 * Make sure no empty audio is output after this point by
+		 * setting stream format to 0, which is not valid for HDMI.
+		*/
+		__snd_hda_codec_cleanup_stream(codec, hinfo->nid, 1);
+
 		pcm_idx = hinfo_to_pcm_index(codec, hinfo);
 		if (snd_BUG_ON(pcm_idx < 0)) {
 			err = -EINVAL;
