From a8fc0f0af944ac24655be34d7e926ade1cbe6946 Mon Sep 17 00:00:00 2001
From: Jason Millard <jsm174@gmail.com>
Date: Wed, 3 Jul 2024 13:18:15 -0400
Subject: [PATCH] misc: add serum.h to installation files

---
 CMakeLists.txt | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index be0d82f..45f9130 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,7 +98,10 @@ if(BUILD_SHARED)
    install(TARGETS serum_shared
       LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
    )
-   install(FILES src/serum-decode.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+   install(FILES
+      src/serum.h
+      src/serum-decode.h
+      DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 
    if(PLATFORM STREQUAL "win" OR PLATFORM STREQUAL "macos" OR PLATFORM STREQUAL "linux")
       add_executable(serum_test
@@ -127,7 +130,10 @@ if(BUILD_STATIC)
    install(TARGETS serum_static
       LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib
    )
-   install(FILES src/serum-decode.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
+   install(FILES
+      src/serum.h
+      src/serum-decode.h
+      DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 
    if(PLATFORM STREQUAL "win" OR PLATFORM STREQUAL "macos" OR PLATFORM STREQUAL "linux")
       add_executable(serum_test_s
