diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index a3e8082c..b9916cab 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -437,7 +437,7 @@ target_compile_definitions(tracy PUBLIC $<$<CONFIG:Debug,RelWithDebInfo>:TRACY_E
 # target_compile_definitions(tracy PUBLIC TRACY_ENABLE)
 
 # Use XDG desktop portals on Linux
-set(NFD_PORTAL ON)
+set(NFD_PORTAL OFF)
 
 # nativefiledialog-extended
 add_subdirectory(nativefiledialog-extended)
