diff --git a/vita3k/gui/src/archive_install_dialog.cpp b/vita3k/gui/src/archive_install_dialog.cpp
index 1e9382d8..3628e926 100644
--- a/vita3k/gui/src/archive_install_dialog.cpp
+++ b/vita3k/gui/src/archive_install_dialog.cpp
@@ -112,10 +112,6 @@ void draw_archive_install_dialog(GuiState &gui, EmuEnvState &emuenv) {
                 result = NFD_PickFolder(nullptr, &archive_path);
             if (result == NFD_OKAY) {
                 state = "install";
-            } else {
-                if (result == host::dialog::filesystem::Result::ERROR)
-                    LOG_CRITICAL("Error initializing file dialog: {}", host::dialog::filesystem::get_error());
-                type.clear();
             }
         } else if (state == "install") {
             std::thread installation([&emuenv, &gui]() {
diff --git a/vita3k/gui/src/license_install_dialog.cpp b/vita3k/gui/src/license_install_dialog.cpp
index e4064df4..4e69fee6 100644
--- a/vita3k/gui/src/license_install_dialog.cpp
+++ b/vita3k/gui/src/license_install_dialog.cpp
@@ -74,11 +74,6 @@ void draw_license_install_dialog(GuiState &gui, EmuEnvState &emuenv) {
                 state = "success";
             else
                 state = "fail";
-        } else {
-            if (result == host::dialog::filesystem::Result::ERROR)
-                LOG_CRITICAL("Error initializing file dialog: {}", host::dialog::filesystem::get_error());
-
-            state.clear();
         }
     } else if (state == "zrif") {
         title = license["enter_zrif_key"];
diff --git a/vita3k/gui/src/pkg_install_dialog.cpp b/vita3k/gui/src/pkg_install_dialog.cpp
index b32122e8..f2456efa 100644
--- a/vita3k/gui/src/pkg_install_dialog.cpp
+++ b/vita3k/gui/src/pkg_install_dialog.cpp
@@ -101,10 +101,6 @@ void draw_pkg_install_dialog(GuiState &gui, EmuEnvState &emuenv) {
                 fs::ifstream binfile(string_utils::utf_to_wide(std::string(license_path)), std::ios::in | std::ios::binary | std::ios::ate);
                 zRIF = rif2zrif(binfile);
                 state = "install";
-            } else {
-                if (result == host::dialog::filesystem::Result::ERROR)
-                    LOG_CRITICAL("Error initializing file dialog: {}", host::dialog::filesystem::get_error());
-                state.clear();
             }
         } else if (state == "zrif") {
             title = lang["enter_zrif_key"];
diff --git a/vita3k/gui/src/settings.cpp b/vita3k/gui/src/settings.cpp
index cd9bbfa9..7a7f23ee 100644
--- a/vita3k/gui/src/settings.cpp
+++ b/vita3k/gui/src/settings.cpp
@@ -608,9 +608,6 @@ void draw_settings(GuiState &gui, EmuEnvState &emuenv) {
                             gui.users[emuenv.io.user_id].start_type = "image";
                             save_user(gui, emuenv, emuenv.io.user_id);
                         }
-                        if (result == host::dialog::filesystem::Result::ERROR) {
-                            LOG_CRITICAL("Error initializing file dialog: {}", host::dialog::filesystem::get_error());
-                        }
                         sub_menu.clear();
                     } else if (sub_menu == "default") {
                         title = theme_background.main["default"];
@@ -674,9 +671,6 @@ void draw_settings(GuiState &gui, EmuEnvState &emuenv) {
                             save_user(gui, emuenv, emuenv.io.user_id);
                         }
                     }
-                    if (result == host::dialog::filesystem::Result::ERROR) {
-                        LOG_CRITICAL("Error initializing file dialog: {}", host::dialog::filesystem::get_error());
-                    }
                 }
                 ImGui::PopStyleVar();
                 ImGui::Columns(1);
