diff --git a/src/backends/unix/system.c b/src/backends/unix/system.c
index 47786f90..0a45b430 100644
--- a/src/backends/unix/system.c
+++ b/src/backends/unix/system.c
@@ -485,7 +485,7 @@ Sys_GetHomeDir(void)
 	static char gdir[MAX_OSPATH];
 	char *home;
 
-	home = getenv("HOME");
+	home = "/userdata/system/configs/";
 
 	if (!home)
 	{
diff --git a/src/common/header/common.h b/src/common/header/common.h
index fc65a104..23d5bc35 100644
--- a/src/common/header/common.h
+++ b/src/common/header/common.h
@@ -51,9 +51,9 @@
  #define CFGDIR "YamagiQ2"
 #else
  #ifndef __HAIKU__
-   #define CFGDIR ".yq2"
+   #define CFGDIR "quake2"
  #else
-   #define CFGDIR "yq2"
+   #define CFGDIR "quake2"
  #endif
 #endif
 
