diff --git a/src/common/header/shared.h b/src/common/header/shared.h
index 9fb97019..6c103b0f 100644
--- a/src/common/header/shared.h
+++ b/src/common/header/shared.h
@@ -369,8 +369,8 @@ qboolean Sys_IsDir(const char *path);
 qboolean Sys_IsFile(const char *path);
 
 /* large block stack allocation routines */
-YQ2_ATTR_MALLOC void *Hunk_Begin(int maxsize);
-YQ2_ATTR_MALLOC void *Hunk_Alloc(int size);
+void *Hunk_Begin(int maxsize);
+void *Hunk_Alloc(int size);
 void Hunk_Free(void *buf);
 int Hunk_End(void);
 
