diff --git a/CMakeLists.txt b/CMakeLists.txt
index a9df2b26..5b9fc4e0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -235,9 +235,6 @@ add_dependencies(openomf copy_shaders)
 
 # Build tools if requested
 set(TOOL_TARGET_NAMES)
-# always build languagetool, for BuildLanguages.cmake
-add_executable(languagetool tools/languagetool/main.c)
-list(APPEND TOOL_TARGET_NAMES languagetool)
 if (USE_TOOLS)
     add_executable(bktool tools/bktool/main.c
         tools/shared/animation_misc.c
@@ -246,6 +243,7 @@ if (USE_TOOLS)
         tools/shared/animation_misc.c
         tools/shared/conversions.c)
     add_executable(soundtool tools/soundtool/main.c)
+    add_executable(languagetool tools/languagetool/main.c)
     add_executable(afdiff tools/afdiff/main.c)
     add_executable(rectool tools/rectool/main.c tools/shared/pilot.c)
     add_executable(pcxtool tools/pcxtool/main.c)
@@ -262,6 +260,7 @@ if (USE_TOOLS)
         bktool
         aftool
         soundtool
+        languagetool
         afdiff
         rectool
         pcxtool
@@ -383,7 +382,12 @@ else()
     message(STATUS "Development: Unit-tests are disabled")
 endif()
 
-include("cmake-scripts/BuildLanguages.cmake")
+if(USE_TOOLS)
+    include("cmake-scripts/BuildLanguages.cmake")
+    message(STATUS "Since tools are enabled, languages will be built.")
+else()
+    message(STATUS "Since tools are disabled, languages will NOT be built!")
+endif()
 
 # Copy some resources to destination resources directory to ease development setup.
 file(COPY resources/openomf.bk resources/gamecontrollerdb.txt DESTINATION resources)
diff --git a/cmake-scripts/BuildLanguages.cmake b/cmake-scripts/BuildLanguages.cmake
index ee5c4f7c..41f058cd 100644
--- a/cmake-scripts/BuildLanguages.cmake
+++ b/cmake-scripts/BuildLanguages.cmake
@@ -15,11 +15,9 @@ endif()
 
 # generate custom target info
 set(BUILD_LANG_COMMANDS)
-set(BUILD_LANG_SOURCES)
 foreach(LANG ${OMF_LANGS})
     set(TXT2 "${PROJECT_SOURCE_DIR}/resources/${LANG}2.TXT")
     set(DAT2 "${CMAKE_CURRENT_BINARY_DIR}/resources/${LANG}.DAT2")
-    list(APPEND BUILD_LANG_SORUCES "${TXT2}")
     list(APPEND BUILD_LANG_COMMANDS
         DEPENDS "${TXT2}"
         BYPRODUCTS "${DAT2}"
@@ -33,7 +31,6 @@ foreach(LANG ${OPENOMF_LANGS})
     set(TXT2 "${PROJECT_SOURCE_DIR}/resources/${LANG}2.TXT")
     set(LNG "${CMAKE_CURRENT_BINARY_DIR}/resources/${LANG}.LNG")
     set(LNG2 "${CMAKE_CURRENT_BINARY_DIR}/resources/${LANG}.LNG2")
-    list(APPEND BUILD_LANG_SORUCES "${TXT}" "${TXT2}")
     list(APPEND BUILD_LANG_COMMANDS
         DEPENDS "${TXT}" "${TXT2}"
         BYPRODUCTS "${LNG}" "{LNG2}"
@@ -51,6 +48,5 @@ add_custom_target(build_languages
     ${BUILD_LANG_COMMANDS}
     COMMAND ${CMAKE_COMMAND} -E echo_append "done"
 )
-target_sources(build_languages PRIVATE ${BUILD_LANG_SORUCES})
 add_dependencies(openomf build_languages)
-add_dependencies(build_languages languagetool)
+add_dependencies(build_languages languagetool)
\ No newline at end of file
