diff --git a/cmake-scripts/BuildLanguages.cmake b/cmake-scripts/BuildLanguages.cmake
index 7cde27d4..47674ac5 100644
--- a/cmake-scripts/BuildLanguages.cmake
+++ b/cmake-scripts/BuildLanguages.cmake
@@ -2,6 +2,8 @@ if(NOT BUILD_LANGUAGES)
     return()
 endif()
 
+set(OMF_COMMAND_WRAPPER "$<TARGET_FILE:languagetool>" CACHE STRING "Optional languagetool tools (cross compile)")
+
 # OMF 2097 Epic Challenge Arena
 set(LANG_STRCOUNT 1013)
 set(OMF_LANGS ENGLISH GERMAN)
@@ -26,7 +28,7 @@ foreach(LANG ${OMF_LANGS})
         DEPENDS "${TXT2}"
         BYPRODUCTS "${DAT2}"
         COMMAND ${CMAKE_COMMAND} -E echo_append "${LANG}, "
-        COMMAND "$<TARGET_FILE:languagetool>" -i "${TXT2}" -o "${DAT2}" --check-count ${LANG2_STRCOUNT}
+        COMMAND ${OMF_COMMAND_WRAPPER} -i "${TXT2}" -o "${DAT2}" --check-count ${LANG2_STRCOUNT}
     )
     install(FILES "${DAT2}" DESTINATION "${LANGUAGE_INSTALL_PATH}")
 endforeach()
@@ -40,8 +42,8 @@ foreach(LANG ${OPENOMF_LANGS})
         DEPENDS "${TXT}" "${TXT2}"
         BYPRODUCTS "${LNG}" "{LNG2}"
         COMMAND ${CMAKE_COMMAND} -E echo_append "${LANG}, "
-        COMMAND "$<TARGET_FILE:languagetool>" -i "${TXT}" -o "${LNG}" --check-count ${LANG_STRCOUNT}
-        COMMAND "$<TARGET_FILE:languagetool>" -i "${TXT2}" -o "${LNG2}" --check-count ${LANG2_STRCOUNT}
+        COMMAND ${OMF_COMMAND_WRAPPER} -i "${TXT}" -o "${LNG}" --check-count ${LANG_STRCOUNT}
+        COMMAND ${OMF_COMMAND_WRAPPER} -i "${TXT2}" -o "${LNG2}" --check-count ${LANG2_STRCOUNT}
     )
     install(FILES "${LNG}" "${LNG2}" DESTINATION "${LANGUAGE_INSTALL_PATH}")
 endforeach()
