diff --git a/cmake-scripts/BuildLanguages.cmake b/cmake-scripts/BuildLanguages.cmake
index ee5c4f7c..7733f5ec 100644
--- a/cmake-scripts/BuildLanguages.cmake
+++ b/cmake-scripts/BuildLanguages.cmake
@@ -5,7 +5,7 @@ set(OMF_LANGS ENGLISH GERMAN)
 set(LANG2_STRCOUNT 1)
 set(OPENOMF_LANGS DANISH)
 
-set(OMF_COMMAND_WRAPPER "" CACHE STRING "Optional wrapper to run languagetool with")
+set(OMF_COMMAND_WRAPPER "$<TARGET_FILE:languagetool>" CACHE STRING "Optional wrapper to run languagetool with")
 
 if(WIN32)
     set(LANGUAGE_INSTALL_PATH "openomf/resources/")
@@ -24,7 +24,7 @@ foreach(LANG ${OMF_LANGS})
         DEPENDS "${TXT2}"
         BYPRODUCTS "${DAT2}"
         COMMAND ${CMAKE_COMMAND} -E echo_append "${LANG}, "
-        COMMAND ${OMF_COMMAND_WRAPPER} "$<TARGET_FILE:languagetool>" -i "${TXT2}" -o "${DAT2}" --check-count ${LANG2_STRCOUNT}
+        COMMAND ${OMF_COMMAND_WRAPPER}  -i "${TXT2}" -o "${DAT2}" --check-count ${LANG2_STRCOUNT}
     )
     install(FILES "${DAT2}" DESTINATION "${LANGUAGE_INSTALL_PATH}")
 endforeach()
@@ -38,8 +38,8 @@ foreach(LANG ${OPENOMF_LANGS})
         DEPENDS "${TXT}" "${TXT2}"
         BYPRODUCTS "${LNG}" "{LNG2}"
         COMMAND ${CMAKE_COMMAND} -E echo_append "${LANG}, "
-        COMMAND ${OMF_COMMAND_WRAPPER} "$<TARGET_FILE:languagetool>" -i "${TXT}" -o "${LNG}" --check-count ${LANG_STRCOUNT}
-        COMMAND ${OMF_COMMAND_WRAPPER} "$<TARGET_FILE:languagetool>" -i "${TXT2}" -o "${LNG2}" --check-count ${LANG2_STRCOUNT}
+        COMMAND ${OMF_COMMAND_WRAPPER}  -i "${TXT}" -o "${LNG}" --check-count ${LANG_STRCOUNT}
+        COMMAND ${OMF_COMMAND_WRAPPER}  -i "${TXT2}" -o "${LNG2}" --check-count ${LANG2_STRCOUNT}
     )
     install(FILES "${LNG}" "${LNG2}" DESTINATION "${LANGUAGE_INSTALL_PATH}")
 endforeach()
