diff --git a/src/game/utils/settings.c b/src/game/utils/settings.c
index f38d8cae..8ebd273f 100644
--- a/src/game/utils/settings.c
+++ b/src/game/utils/settings.c
@@ -73,7 +73,7 @@ const field f_video[] = {
     F_INT(settings_video, framerate_limit, 0),
     F_BOOL(settings_video, vsync, 0),
     F_INT(settings_video, aspect, 0),
-    F_BOOL(settings_video, fullscreen, 0),
+    F_BOOL(settings_video, fullscreen, 1),
     F_INT(settings_video, scaling, 0),
     F_BOOL(settings_video, instant_console, 0),
     F_BOOL(settings_video, crossfade_on, 1),
