--- a/vice/src/arch/sdl/menu_screenshot.c	2025-02-07 22:54:47.367531542 +0100
+++ b/vice/src/arch/sdl/menu_screenshot.c	2024-12-24 20:12:44.000000000 +0100
@@ -695,6 +695,8 @@
 static void uiscreenshot_menu_free(ui_menu_entry_t *menu)
 {
     ui_menu_entry_t *entry = menu;
+    if (entry == NULL)
+        return;
     while (entry->string != NULL) {
         lib_free(entry->string);
         entry++;
