diff --git a/src/menu/menu.cpp b/src/menu/menu.cpp
index ec9f382096..800dcfa678 100644
--- a/src/menu/menu.cpp
+++ b/src/menu/menu.cpp
@@ -666,14 +666,14 @@ bool M_Responder (event_t *ev)
 			ch = ev->data1;
 			switch (ch)
 			{
-			case KEY_JOY1:
+			case KEY_JOY2:
 			case KEY_JOY3:
 			case KEY_JOY15:
 			case KEY_PAD_A:
 				mkey = MKEY_Enter;
 				break;
 
-			case KEY_JOY2:
+			case KEY_JOY1:
 			case KEY_JOY14:
 			case KEY_PAD_B:
 				mkey = MKEY_Back;
