diff --git a/src/common/menu/menu.cpp b/src/common/menu/menu.cpp
index db89dfedd4..1505a232a3 100644
--- a/src/common/menu/menu.cpp
+++ b/src/common/menu/menu.cpp
@@ -702,14 +702,14 @@ bool M_Responder (event_t *ev)
 			ch = ev->data1;
 			switch (ch)
 			{
-			case KEY_JOY1:
+			case KEY_JOY2:
 			case KEY_JOY3:
 			case KEY_JOY15:
 			case KEY_PAD_A:
 				mkey = MKEY_Enter;
 				break;
 
-			case KEY_JOY2:
+			case KEY_JOY1:
 			case KEY_JOY14:
 			case KEY_PAD_B:
 				mkey = MKEY_Back;
@@ -732,14 +732,14 @@ bool M_Responder (event_t *ev)
 
 			case KEY_PAD_DPAD_UP:
 			case KEY_PAD_LTHUMB_UP:
-			case KEY_JOYAXIS2MINUS:
+			//case KEY_JOYAXIS2MINUS:
 			case KEY_JOYPOV1_UP:
 				mkey = MKEY_Up;
 				break;
 
 			case KEY_PAD_DPAD_DOWN:
 			case KEY_PAD_LTHUMB_DOWN:
-			case KEY_JOYAXIS2PLUS:
+			//case KEY_JOYAXIS2PLUS:
 			case KEY_JOYPOV1_DOWN:
 				mkey = MKEY_Down;
 				break;
