diff --git a/CMakeLists.txt b/CMakeLists.txt
index e34eb281e..b87d505de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,9 +12,9 @@ SET(GAMES_SHAREDIR "${APPDIR}" CACHE PATH "Game data root dir")
 SET(DOCDIR ${DATADIR} CACHE PATH "Docs destination")
 ELSE(WIN32)
 SET(SHAREDIR "/usr/share" CACHE PATH "System share dir location")
-SET(GAMES_SHAREDIR "${SHAREDIR}/games/" CACHE PATH "Game data root dir")
+SET(GAMES_SHAREDIR "${SHAREDIR}" CACHE PATH "Game data root dir")
 SET(DATADIR "${GAMES_SHAREDIR}/commandergenius")
-SET(APPDIR games CACHE PATH "Binary destination")
+SET(APPDIR bin CACHE PATH "Binary destination")
 SET(DOCDIR ${DATADIR} CACHE PATH "Docs destination")
 SET(ICONDIR "${SHAREDIR}/icons/hicolor" CACHE PATH "Standard icon installation dir")
 SET(DESKTOPFILESDIR "${SHAREDIR}/applications" CACHE PATH "Application installation dir")
