diff --git a/GsKit/base/interface/FindFile.cpp b/GsKit/base/interface/FindFile.cpp
index 64fb916a5..e3cb62d5b 100644
--- a/GsKit/base/interface/FindFile.cpp
+++ b/GsKit/base/interface/FindFile.cpp
@@ -509,7 +509,7 @@ void InitBaseSearchPaths()
     #ifdef ALTERNATE_HOME
       AddToFileList(&basesearchpaths, ALTERNATE_HOME"/CommanderGenius");
     #else
-      AddToFileList(&basesearchpaths, "${HOME}/.CommanderGenius");
+      AddToFileList(&basesearchpaths, "/userdata/roms/ports/cgenius");
     #endif
 #endif
     AddToFileList(&basesearchpaths, ".");
@@ -805,6 +805,7 @@ bool FileListIncludesExact(const searchpathlist* l, const std::string& f) {
 
 std::string GetHomeDir()
 {
+return "/userdata/system/config";
 #ifndef WIN32
 #if defined(CAANOO) || defined(WIZ) || defined(GP2X) || defined(DINGOO) || defined(PANDORA)
     char* home = getenv("PWD");
