diff --git a/GsKit/base/video/CVidConfig.cpp b/GsKit/base/video/CVidConfig.cpp
index 0afc17ae7..6d5defc8d 100644
--- a/GsKit/base/video/CVidConfig.cpp
+++ b/GsKit/base/video/CVidConfig.cpp
@@ -45,7 +45,7 @@ void CVidConfig::reset()
 #else
     mDisplayRect.dim.x = 1920;
     mDisplayRect.dim.y = 1080;
-    mFullscreen = false;
+    mFullscreen = true;
 #endif
 
     m_ScaleXFilter = VidFilter::NONE;
diff --git a/GsKit/base/video/CVidConfig.h b/GsKit/base/video/CVidConfig.h
index 57ab72ff6..a91f8e256 100644
--- a/GsKit/base/video/CVidConfig.h
+++ b/GsKit/base/video/CVidConfig.h
@@ -60,13 +60,13 @@ public:
 
     int mHorizBorders = 0; // Horizontal border bars for a more CRT feeling
 
-#ifdef __SWITCH__
+//#ifdef __SWITCH__
     bool mFullscreen = true;
     bool mIntegerScaling = false;
-#else
+/*#else
     bool mFullscreen = false;
     bool mIntegerScaling = true;
-#endif
+#endif*/
     VidFilter m_ScaleXFilter = VidFilter::NONE;
     bool m_normal_scale;
 
diff --git a/src/engine/core/CSettings.cpp b/src/engine/core/CSettings.cpp
index 7a4766002..5daa86249 100644
--- a/src/engine/core/CSettings.cpp
+++ b/src/engine/core/CSettings.cpp
@@ -89,8 +89,8 @@ bool CSettings::saveDrvCfg()
             config.WriteString("FileHandling", "SearchPath" + itoa(i), *p);
 
         CVidConfig &VidConf = gVideoDriver.getVidConfig();
-        config.SetKeyword("Video", "fullscreen", VidConf.mFullscreen);
-        config.SetKeyword("Video", "integerScaling", VidConf.mIntegerScaling);
+        config.SetKeyword("Video", "fullscreen", true);
+        config.SetKeyword("Video", "integerScaling", false);
         config.SetKeyword("Video", "OpenGL", VidConf.mOpengl);
 #ifdef USE_VIRTUALPAD
         config.SetKeyword("Video", "VirtPad", VidConf.mVPad);
@@ -235,13 +235,13 @@ bool CSettings::loadDrvCfg()
 
     int value = 0;
 
-#ifdef __SWITCH__
+//#ifdef __SWITCH__
     config.ReadKeyword("Video", "fullscreen", &vidConf.mFullscreen, true);
     config.ReadKeyword("Video", "integerScaling", &vidConf.mIntegerScaling, false);
-#else
-    config.ReadKeyword("Video", "fullscreen", &vidConf.mFullscreen, false);
-    config.ReadKeyword("Video", "integerScaling", &vidConf.mIntegerScaling, true);
-#endif
+/*#else
+//    config.ReadKeyword("Video", "fullscreen", &vidConf.mFullscreen, false);
+//    config.ReadKeyword("Video", "integerScaling", &vidConf.mIntegerScaling, true);
+#endif*/
 
     config.ReadInteger("Video", "scale", &value, 1);
     vidConf.Zoom = static_cast<unsigned short>(value);
