diff --git a/os_dep/linux/ioctl_cfg80211.c b/os_dep/linux/ioctl_cfg80211.c
index 9ec345a..1ef892b 100644
--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -1473,7 +1473,7 @@ exit:
 	return ret;
 }
 #endif /* CONFIG_AP_MODE */
-
+__attribute__((no_sanitize("bounds")))
 static int rtw_cfg80211_set_encryption(struct net_device *dev, struct ieee_param *param)
 {
 	int ret = 0;
@@ -5912,7 +5912,10 @@ static int	cfg80211_rtw_set_channel(struct wiphy *wiphy
 
 static int cfg80211_rtw_set_monitor_channel(struct wiphy *wiphy
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0))
-	, struct cfg80211_chan_def *chandef
+  #if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0))
+        , struct net_device *netdev
+  #endif
+        , struct cfg80211_chan_def *chandef
 #else
 	, struct ieee80211_channel *chan
 	, enum nl80211_channel_type channel_type
