diff --git a/Common.mak b/Common.mak
index 78466b9a7..84e931aae 100644
--- a/Common.mak
+++ b/Common.mak
@@ -574,8 +574,8 @@ ifndef OPTOPT
         else
             ifeq ($(PLATFORM),DARWIN)
                 OPTOPT := -march=core2 -mmmx -msse -msse2 -msse3 -mssse3
-            else
-                OPTOPT := -march=nocona
+#            else
+#                OPTOPT := -march=nocona
             endif
         endif
     endif
@@ -602,6 +602,9 @@ ifndef OPTOPT
     ifeq ($(PLATFORM),WII)
         OPTOPT := -mtune=750
     endif
+    ifneq (0,$(RPI4))
+        OPTOPT := -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math
+    endif
 endif
 
 ifeq ($(PACKAGE_REPOSITORY),0)
@@ -893,7 +896,7 @@ endif
 
 ifeq ($(RENDERTYPE),SDL)
     ifeq ($(SDL_TARGET),2)
-        SDLCONFIG := sdl2-config
+        SDLCONFIG := sdl2
         SDLNAME := SDL2
     else ifeq ($(SDL_TARGET),1)
         SDLCONFIG := sdl-config
@@ -941,8 +944,8 @@ ifeq ($(RENDERTYPE),SDL)
             SDLCONFIG_CFLAGS := $(strip $(subst -Dmain=SDL_main,,$(shell $(SDLCONFIG) --cflags)))
             SDLCONFIG_LIBS := $(strip $(subst -mwindows,,$(shell $(SDLCONFIG) --libs)))
 
-            COMPILERFLAGS += $(SDLCONFIG_CFLAGS)
-            LIBS += $(SDLCONFIG_LIBS)
+            COMPILERFLAGS += `pkg-config sdl2 --cflags`
+            LIBS += `pkg-config sdl2 --libs`
         else
             ifeq ($(SDL_TARGET),1)
                 COMPILERFLAGS += -D_GNU_SOURCE=1
diff --git a/GNUmakefile b/GNUmakefile
index 105f402dd..3d6213f08 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -514,7 +514,7 @@ kenbuild_obj := $(obj)/$(kenbuild)
 kenbuild_cflags := -I$(kenbuild_src)
 
 kenbuild_game := ekenbuild
-kenbuild_editor := ekenbuild-editor
+kenbuild_editor :=
 
 kenbuild_game_deps := audiolib
 
@@ -570,7 +570,7 @@ tekwar_obj := $(obj)/$(tekwar)
 tekwar_cflags := -I$(tekwar_src)
 
 tekwar_game := etekwar
-tekwar_editor := etekwar-editor
+tekwar_editor :=
 
 tekwar_game_proper := ETekWar
 tekwar_editor_proper := ETekWar Editor
@@ -653,7 +653,7 @@ duke3d_game_deps := audiolib mact
 duke3d_editor_deps := audiolib
 
 duke3d_game := eduke32
-duke3d_editor := mapster32
+duke3d_editor :=
 
 ifneq (,$(APPBASENAME))
     duke3d_game := $(APPBASENAME)
@@ -921,7 +921,7 @@ ifneq (0,$(NETCODE))
 endif
 
 rr_game := rednukem
-rr_editor := rrmapster32
+rr_editor :=
 
 ifneq (,$(APPBASENAME))
     rr_game := $(APPBASENAME)
@@ -1048,7 +1048,7 @@ sw_game_deps := audiolib mact
 sw_editor_deps := audiolib
 
 sw_game := voidsw
-sw_editor := wangulator
+sw_editor :=
 
 sw_game_proper := VoidSW
 sw_editor_proper := Wangulator
@@ -1177,7 +1177,7 @@ exhumed_game_deps := duke3d_common_midi audiolib mact
 exhumed_editor_deps := audiolib
 
 exhumed_game := pcexhumed
-exhumed_editor := pcexhumed_editor
+exhumed_editor :=
 
 exhumed_game_proper := PCExhumed
 exhumed_editor_proper := PCExhumed Editor
@@ -1286,7 +1286,7 @@ witchaven_game_deps := duke3d_common_midi audiolib mact hmpplay
 witchaven_editor_deps := audiolib
 
 witchaven_game := ewitchaven
-witchaven_editor := ewitchaven_editor
+witchaven_editor :=
 
 witchaven_game_proper := EWitchaven
 witchaven_editor_proper := EWitchaven Editor
@@ -1415,7 +1415,8 @@ endif
 all: \
     blood \
     rr \
-    exhumed \
+    exhumed
+    
 
 start:
 	$(BUILD_STARTED)
