diff --git a/source/blood/src/config.cpp b/source/blood/src/config.cpp
index 4d5379640..1397b02e1 100644
--- a/source/blood/src/config.cpp
+++ b/source/blood/src/config.cpp
@@ -300,11 +300,11 @@ void CONFIG_SetDefaults(void)
     NumVoices = 64;
 #endif
 
-#ifdef GEKKO
+//#ifdef GEKKO
     gSetup.usejoystick = 1;
-#else
+/*#else
     gSetup.usejoystick = 0;
-#endif
+#endif*/
 
     gSetup.forcesetup       = 1;
     gSetup.noautoload       = 1;
