diff --git a/source/blood/src/config.cpp b/source/blood/src/config.cpp
index 196642ac9..1bd72c25b 100644
--- a/source/blood/src/config.cpp
+++ b/source/blood/src/config.cpp
@@ -306,9 +306,9 @@ void CONFIG_SetDefaults(void)
 #ifdef GEKKO
     gSetup.usejoystick = 1;
 #else
-    gSetup.usejoystick = 0;
+    gSetup.usejoystick = 1;
 #endif
-    gSetup.joystickrumble = 0;
+    gSetup.joystickrumble = 1;
 
     gSetup.forcesetup       = 1;
     gSetup.noautoload       = 1;
