diff --git a/Makefiles/Makefile.UNIX b/Makefiles/Makefile.UNIX
index 62a9d2d..6b90c00 100644
--- a/Makefiles/Makefile.UNIX
+++ b/Makefiles/Makefile.UNIX
@@ -70,7 +70,7 @@ INPUT_CFLAGS = -DSUPERMODEL_MANYMOUSE
 #
 
 PLATFORM_CXXFLAGS = $(SDL2_CFLAGS) $(INPUT_CFLAGS) -O3
-PLATFORM_LDFLAGS = $(SDL2_LIBS) -lGL -lGLU -lz -lm -lstdc++ -lpthread -ldl
+PLATFORM_LDFLAGS = $(SDL2_LIBS) -lGL -lGLU -lz -lm -lstdc++ -lpthread -ldl -ludev
 
 
 ###############################################################################
diff --git a/Makefiles/Rules.inc b/Makefiles/Rules.inc
index c6dd31f..15bda9d 100644
--- a/Makefiles/Rules.inc
+++ b/Makefiles/Rules.inc
@@ -164,6 +164,7 @@ SRC_FILES = \
 	Src/Inputs/Manymouse.c \
 	Src/Inputs/MultiInputSource.cpp \
 	Src/OSD/SDL/SDLInputSystem.cpp \
+	Src/OSD/evdev/EvdevInputSystem.cpp \
 	Src/OSD/SDL/Crosshair.cpp \
 	Src/OSD/Outputs.cpp \
 	Src/Sound/MPEG/MpegAudio.cpp \
diff --git a/Src/OSD/SDL/Main.cpp b/Src/OSD/SDL/Main.cpp
index 3062af4..0b541dd 100644
--- a/Src/OSD/SDL/Main.cpp
+++ b/Src/OSD/SDL/Main.cpp
@@ -2455,6 +2455,8 @@ int main(int argc, char **argv)
     InputSystem = std::shared_ptr<CInputSystem>(new CSDLInputSystem(s_runtime_config, false));
   else if (selectedInputSystem == "sdlgamepad")
     InputSystem = std::shared_ptr<CInputSystem>(new CSDLInputSystem(s_runtime_config, true));
+  else if (selectedInputSystem == "evdev")
+    InputSystem = std::shared_ptr<CInputSystem>(new CEvdevInputSystem(s_runtime_config));
 #ifdef SUPERMODEL_WIN32
   else if (selectedInputSystem == "dinput")
     InputSystem = std::shared_ptr<CInputSystem>(new CDirectInputSystem(s_runtime_config, s_window, false, false));
