diff --git a/src/winetricks b/src/winetricks
index 0a5884a..0357c9d 100755
--- a/src/winetricks
+++ b/src/winetricks
@@ -19916,6 +19916,25 @@ load_winxp()
     w_set_winver winxp
 }
 
+#----------------------------------------------------------------
+
+w_metadata xeniadisclaimer settings \
+    title="Xenia accept disclaimer"
+
+load_xeniadisclaimer()
+{
+    echo "Disabling Xenia disclaimer dialog"
+    cat > "${W_TMP}"/xeniadisclaimer.reg <<_EOF_
+REGEDIT4
+
+[HKEY_CURRENT_USER\\Software\\Xenia]
+"XEFLAGS"=dword:00000001
+
+_EOF_
+    w_try_cd "${W_TMP}"
+    w_try_regedit xeniadisclaimer.reg
+}
+
 #---- Main Program ----
 
 # In GUI mode, allow a user to select an arbitrary executable and start it
