diff --git a/Source/Core/InputCommon/ControllerInterface/ControllerInterface.h b/Source/Core/InputCommon/ControllerInterface/ControllerInterface.h
index e5712f0a4f..aebd072a6a 100644
--- a/Source/Core/InputCommon/ControllerInterface/ControllerInterface.h
+++ b/Source/Core/InputCommon/ControllerInterface/ControllerInterface.h
@@ -34,9 +34,9 @@
 #if defined(HAVE_SDL3)
 #define CIFACE_USE_SDL
 #endif
-#if defined(HAVE_HIDAPI)
-#define CIFACE_USE_STEAMDECK
-#endif
+//#if defined(HAVE_HIDAPI)
+//#define CIFACE_USE_STEAMDECK
+//#endif
 
 namespace ciface
 {
