diff --git a/Source/Core/DolphinQt/MainWindow.cpp b/Source/Core/DolphinQt/MainWindow.cpp
index 42a6a04603..3dcd4a7c7c 100644
--- a/Source/Core/DolphinQt/MainWindow.cpp
+++ b/Source/Core/DolphinQt/MainWindow.cpp
@@ -1966,7 +1966,12 @@ void MainWindow::Show()
   // If the booting of a game was requested on start up, do that now
   if (m_pending_boot != nullptr)
   {
+    m_menu_bar->hide();
+    m_tool_bar->hide();
+    m_game_list->hide();
     StartGame(std::move(m_pending_boot));
     m_pending_boot.reset();
+  } else {
+    m_tool_bar->show();
   }
 }
