diff --git a/Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp b/Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp
index 0b041d8fba..1223a58693 100644
--- a/Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp
+++ b/Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp
@@ -261,6 +261,7 @@ std::shared_ptr<evdevDevice>
 InputBackend::FindDeviceWithUniqueIDAndPhysicalLocation(const char* unique_id,
                                                         const char* physical_location)
 {
+  return nullptr;
   if (!unique_id || !physical_location)
     return nullptr;
 
