diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6ecfdb9bf4..77cb2f2f35 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -662,7 +662,7 @@ add_subdirectory(Externals/implot)
 # A few distributions (mainly Debian-based ones) distribute glslang without linking it against SPIRV-Tools
 find_package(SPIRV-Tools)
 if(SPIRV-Tools_FOUND)
-  dolphin_find_optional_system_library(glslang Externals/glslang 15.0)
+  dolphin_find_optional_system_library(glslang Externals/glslang 16.0)
   target_link_libraries(glslang::glslang INTERFACE SPIRV-Tools)
 else()
   dolphin_add_bundled_library(glslang OFF Externals/glslang)
