diff --git a/Source/Core/Core/Config/MainSettings.cpp b/Source/Core/Core/Config/MainSettings.cpp
index d09955e603..dbee82ca4c 100644
--- a/Source/Core/Core/Config/MainSettings.cpp
+++ b/Source/Core/Core/Config/MainSettings.cpp
@@ -348,6 +348,8 @@ const Info<std::string> MAIN_SKYLANDERS_PATH{{System::Main, "General", "Skylande
                                              ""};
 const Info<bool> MAIN_TIME_TRACKING{{System::Main, "General", "EnablePlayTimeTracking"}, true};
 
+const Info<bool> MAIN_AUTO_INCREMENT_SLOT_COUNT{{System::Main, "General", "AutoIncrementSlot"}, false};
+
 static Info<std::string> MakeISOPathConfigInfo(size_t idx)
 {
   return Config::Info<std::string>{{Config::System::Main, "General", fmt::format("ISOPath{}", idx)},
diff --git a/Source/Core/Core/Config/MainSettings.h b/Source/Core/Core/Config/MainSettings.h
index 5ae26e66a4..c87001c7eb 100644
--- a/Source/Core/Core/Config/MainSettings.h
+++ b/Source/Core/Core/Config/MainSettings.h
@@ -218,6 +218,8 @@ extern const Info<std::string> MAIN_GDB_SOCKET;
 extern const Info<int> MAIN_GDB_PORT;
 extern const Info<int> MAIN_ISO_PATH_COUNT;
 extern const Info<std::string> MAIN_SKYLANDERS_PATH;
+extern const Info<bool> MAIN_AUTO_INCREMENT_SLOT_COUNT;
+
 std::vector<std::string> GetIsoPaths();
 void SetIsoPaths(std::span<const std::string> paths);
 
diff --git a/Source/Core/DolphinQt/MainWindow.cpp b/Source/Core/DolphinQt/MainWindow.cpp
index f89a0684bc..159f6912cf 100644
--- a/Source/Core/DolphinQt/MainWindow.cpp
+++ b/Source/Core/DolphinQt/MainWindow.cpp
@@ -1549,6 +1549,10 @@ void MainWindow::StateLoadSlot()
 
 void MainWindow::StateSaveSlot()
 {
+  if (Config::Get(Config::MAIN_AUTO_INCREMENT_SLOT_COUNT))
+  {
+    m_state_slot++;
+  }
   State::Save(m_system, m_state_slot);
 }
 
