diff --git a/CMake/ScmRevGen.cmake b/CMake/ScmRevGen.cmake
index e1b5bb2d62..7746a7a453 100644
--- a/CMake/ScmRevGen.cmake
+++ b/CMake/ScmRevGen.cmake
@@ -32,7 +32,7 @@ endif()
 
 # version number
 set(DOLPHIN_VERSION_MAJOR "2407")
-set(DOLPHIN_VERSION_MINOR "0")
+set(DOLPHIN_VERSION_MINOR "76")
 set(DOLPHIN_VERSION_PATCH ${DOLPHIN_WC_REVISION})
 
 # If Dolphin is not built from a Git repository, default the version info to
