diff --git a/Makefile b/Makefile
index 661d6ca..3113cd4 100644
--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,8 @@ USE_ASM ?= 1
 RELEASE ?= 1
 
 # Path where game data is installed
-DATADIR ?= /usr/local/share/games/jftekwar
+DATADIR ?= /userdata/roms/ports/etekwar/
+HAVE_GTK ?= 0
 
 ##
 ##
@@ -126,8 +127,8 @@ ifeq ($(PLATFORM),DARWIN)
 endif
 
 ifeq ($(RENDERTYPE),SDL)
-	OURCFLAGS+= $(SDLCONFIG_CFLAGS)
-	OURLDFLAGS+= $(SDLCONFIG_LIBS)
+	OURCFLAGS+= `pkg-config sdl2 --cflags`
+	LIBS+= `pkg-config sdl2 --libs`
 
 	ifeq (1,$(HAVE_GTK))
 		OURCFLAGS+= $(GTKCONFIG_CFLAGS)
@@ -153,7 +154,7 @@ endif
 
 # TARGETS
 .PHONY: clean veryclean all
-all: tekwar$(EXESUFFIX) build$(EXESUFFIX)
+all: tekwar$(EXESUFFIX)
 
 include Makefile.deps
 
diff --git a/jfaudiolib/Makefile b/jfaudiolib/Makefile
index 49c8673..38433ff 100644
--- a/jfaudiolib/Makefile
+++ b/jfaudiolib/Makefile
@@ -39,7 +39,7 @@ else
   LDFLAGS+= -framework Foundation
  endif
  ifneq (0,$(JFAUDIOLIB_HAVE_SDL))
-  JFAUDIOLIB_CPPFLAGS+= -DHAVE_SDL=2 $(shell $(SDL2CONFIG) --cflags)
+  JFAUDIOLIB_CPPFLAGS+= -DHAVE_SDL=2 `pkg-config sdl2 --cflags`
   ifeq (1,$(JFAUDIOLIB_USE_SDLMIXER))
    JFAUDIOLIB_CPPFLAGS+= -DUSE_SDLMIXER
    SOURCES+= src/driver_sdlmixer.c
diff --git a/jfaudiolib/Makefile.shared b/jfaudiolib/Makefile.shared
index aea6279..241491e 100644
--- a/jfaudiolib/Makefile.shared
+++ b/jfaudiolib/Makefile.shared
@@ -22,7 +22,7 @@ else
  endif
  ifneq (,$(shell $(SDL2CONFIG) --version))
   JFAUDIOLIB_HAVE_SDL=2
-  JFAUDIOLIB_LDFLAGS+= $(shell $(SDL2CONFIG) --libs)
+  JFAUDIOLIB_LDFLAGS+= `pkg-config sdl2 --libs`
  endif
  ifeq (yes,$(shell $(PKGCONFIG) --exists alsa && echo yes))
   JFAUDIOLIB_HAVE_ALSA=1
Submodule jfbuild contains modified content
diff --git a/jfbuild/Makefile.shared b/jfbuild/Makefile.shared
index 537dcd4..dedac39 100644
--- a/jfbuild/Makefile.shared
+++ b/jfbuild/Makefile.shared
@@ -74,8 +74,8 @@ endif
 
 ifeq ($(RENDERTYPE),SDL)
 	ifneq ($(SDL2CONFIG),)
-		SDLCONFIG_CFLAGS=$(shell $(SDL2CONFIG) --cflags)
-		SDLCONFIG_LIBS=$(shell $(SDL2CONFIG) --libs)
+		SDLCONFIG_CFLAGS=`pkg-config sdl2 --cflags`
+		SDLCONFIG_LIBS=`pkg-config sdl2 --libs`
 	else
 		$(error No $(SDL2CONFIG) could be found)
 	endif
