diff --git a/src/glbapi.cpp b/src/glbapi.cpp
index cc3500d..476269f 100644
--- a/src/glbapi.cpp
+++ b/src/glbapi.cpp
@@ -9,6 +9,7 @@
 #include "glbapi.h"
 #include "vmemapi.h"
 #include "entypes.h"
+#include "rap.h"
 
 #ifdef _WIN32
 #include <io.h>
@@ -34,7 +35,7 @@ char* strupr(char* s)
 #define _SCOTTGAME
 
 static const char* serial = "32768GLB";
-static char exePath[PATH_MAX];
+static char exePath[PATH_MAX] = "/userdata/roms/ports/raptor/";
 static int num_glbs;
 static KEYFILE g_key;
 static char prefix[5] = "FILE";
@@ -918,4 +919,4 @@ GLB_SaveFile(
 	}
 	
 	fclose(handle);
-}
\ No newline at end of file
+}
diff --git a/src/loadsave.cpp b/src/loadsave.cpp
index 78b2edc..73de059 100644
--- a/src/loadsave.cpp
+++ b/src/loadsave.cpp
@@ -47,8 +47,8 @@ int map_item = -1;
 int curplr_diff = 2;
 int srwpos = 0;
 
-static const char *fmt = "CHAR%04u.FIL";
-static const char *cdfmt = "%sCHAR%04u.FIL";
+static const char *fmt = "/userdata/roms/ports/raptor/CHAR%04u.FIL";
+static const char *cdfmt = "/userdata/roms/ports/raptor/CHAR%04u.FIL";
 
 MAZELEVEL *mapmem;
 CSPRITE *csprite;
@@ -374,7 +374,7 @@ RAP_AreSavedFiles(
         else
             sprintf(temp, fmt, loop);
         
-        if (!access(temp, 0))
+        if (!checkfile(temp))
             return 1;
     }
     
@@ -434,7 +434,7 @@ RAP_FFSaveFile(
         else
             sprintf(temp, fmt, loop);
         
-        if (access(temp, 0) != 0)
+        if (checkfile(temp))
         {
             RAP_ClearPlayer();
             filepos = loop;
@@ -766,7 +766,7 @@ RAP_LoadWin(
         else
             sprintf(temp, fmt, loop);
         
-        if (!access(temp, 0))
+        if (!checkfile(temp))
         {
             if (pos == -1)
                 pos = loop;
@@ -1014,9 +1014,9 @@ RAP_InitLoadSave(
     cdflag = 0;
 
     if(!access("setup.ini", 0))
-        strcpy(g_setup_ini, "setup.ini");
+        strcpy(g_setup_ini, "/usr/share/raptor/SETUP.INI");
     else
-        strcpy(g_setup_ini, "SETUP.INI");
+        strcpy(g_setup_ini, "/usr/share/raptor/SETUP.INI");
 
     return cdpath;
 #endif // _WIN32 || __linux__ || __APPLE__
@@ -1133,4 +1133,4 @@ RAP_CheckFileInPath(
         return 1;
     else
         return 0;
-}
\ No newline at end of file
+}
diff --git a/src/prefapi.cpp b/src/prefapi.cpp
index faab246..df2c2e6 100644
--- a/src/prefapi.cpp
+++ b/src/prefapi.cpp
@@ -2,6 +2,7 @@
 #include <string.h>
 #include "common.h"
 #include "prefapi.h"
+#include "rap.h"
 
 #ifdef _WIN32
 #include <io.h>
diff --git a/src/rap.cpp b/src/rap.cpp
index 1e64e10..0e220a2 100644
--- a/src/rap.cpp
+++ b/src/rap.cpp
@@ -33,6 +33,7 @@
 #include "i_lastscr.h"
 #include "fileids.h"
 #include "entypes.h"
+#include "rap.h"
 
 #ifdef _WIN32
 #include <io.h>
@@ -211,8 +212,8 @@ ShutDown(
 {
     char* mem;
 
-    if (!errcode && !godmode)
-        WIN_Order();
+    //if (!errcode && !godmode)
+    //    WIN_Order();
 
     //IPT_DeInit();
     //DMX_DeInit();
@@ -1256,6 +1257,18 @@ RAP_InitMem(
     GLB_UseVM();
 }
 
+bool checkfile(const char* path)
+{
+	FILE* f = fopen(path, "r");
+	if (f)
+	{
+        fclose(f);
+		return true;
+	} else {
+        return false;
+    }
+}
+
 /***************************************************************************
 main() -
  ***************************************************************************/
@@ -1321,26 +1334,13 @@ main(
     
     cur_diff = 0;
 
-    if (!access("FILE0001.GLB", 0) ||
-        !access("file0001.glb", 0) ||
-        RAP_CheckFileInPath("FILE0001.GLB") ||
-        RAP_CheckFileInPath("file0001.glb"))
+    if (checkfile("/userdata/roms/ports/raptor/FILE0001.GLB"))
         gameflag[0] = 1;
 
-    if (!access("FILE0002.GLB", 0) ||
-        !access("file0002.glb", 0) ||
-        RAP_CheckFileInPath("FILE0002.GLB") ||
-        RAP_CheckFileInPath("file0002.glb"))
+    if (checkfile("/userdata/roms/ports/raptor/FILE0002.GLB"))
         gameflag[1] = 1;
 
-    if ((!access("FILE0003.GLB", 0) && !access("FILE0004.GLB", 0)) || 
-        (!access("file0003.glb", 0) && !access("file0004.glb", 0)) ||
-        (!access("FILE0003.GLB", 0) && !access("file0004.glb", 0)) ||
-        (!access("file0003.glb", 0) && !access("FILE0004.GLB", 0)) ||
-        (RAP_CheckFileInPath("FILE0003.GLB") && RAP_CheckFileInPath("FILE0004.GLB")) ||
-        (RAP_CheckFileInPath("file0003.glb") && RAP_CheckFileInPath("file0004.glb")) ||
-        (RAP_CheckFileInPath("FILE0003.GLB") && RAP_CheckFileInPath("file0004.glb")) ||
-        (RAP_CheckFileInPath("file0003.glb") && RAP_CheckFileInPath("FILE0004.GLB")))
+    if (checkfile("/userdata/roms/ports/raptor/FILE0003.GLB") && checkfile("/userdata/roms/ports/raptor/FILE0004.GLB"))
     {
         gameflag[2] = 1;
         gameflag[3] = 1;
@@ -1357,17 +1357,15 @@ main(
             numfiles++;
     }
 
-    if ((access("FILE0000.GLB", 0) && !RAP_CheckFileInPath("FILE0000.GLB")) &&
-        (access("file0000.glb", 0) && !RAP_CheckFileInPath("file0000.glb")) ||
-        !numfiles)
+    if (!checkfile("/userdata/roms/ports/raptor/FILE0000.GLB") || !numfiles)
     {
-        printf("All game data files NOT FOUND cannot proceed !!\n");
-        SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_ERROR,
-            "Raptor", "All game data files NOT FOUND cannot proceed !!", NULL);
-        exit(0);
+        //printf("All game data files NOT FOUND cannot proceed !!\n");
+        //SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_ERROR,
+        //    "Raptor", "All game data files NOT FOUND cannot proceed !!", NULL);
+        //exit(0);
     }
     
-    printf("Init -\n");
+    //printf("Init -\n");
     EXIT_Install(ShutDown);
     
     // ================================================
@@ -1414,10 +1412,10 @@ main(
     // ================================================
 
     if (access(RAP_SetupFilename(), 0))
-        EXIT_Error("You Must run SETUP.EXE First !!");
+        printf("You Must run SETUP.EXE First !!");
 
     if (!INI_InitPreference(RAP_SetupFilename()))
-        EXIT_Error("SETUP Error");
+        printf("SETUP Error");
 
     fflush(stdout);
     KBD_Install();
diff --git a/src/rap.h b/src/rap.h
index 33b1b8b..6db7c67 100644
--- a/src/rap.h
+++ b/src/rap.h
@@ -173,3 +173,4 @@ void RAP_LoadMap(void);
 int Do_Game(void);
 int RAP_LoadWin(void);
 void ShutDown(int errcode);
+bool checkfile(const char* path);
