diff --git a/src/loadsave.cpp b/src/loadsave.cpp
index 83941e0..6ca5206 100644
--- a/src/loadsave.cpp
+++ b/src/loadsave.cpp
@@ -33,8 +33,8 @@
 #define PATH_MAX MAX_PATH
 #endif
 
-char g_data_path[PATH_MAX];
-char g_setup_path[PATH_MAX];
+static char g_data_path[256] = "/userdata/roms/ports/raptor/";
+static char g_setup_path[256] = "/userdata/system/configs/raptor/";
 
 int hasdatapath = 0;
 
@@ -494,7 +494,7 @@ const char *RAP_DataPath(void)
 {
     memset(g_data_path, 0, sizeof(g_data_path));
     hasdatapath = 0;
-    strcpy(g_setup_path, "SETUP.INI");
+    strcat(g_setup_path, "SETUP.INI");  
     return g_data_path;
 }
 
