diff --git a/cube/CMakeLists.txt b/cube/CMakeLists.txt
index 4087351d..45145612 100644
--- a/cube/CMakeLists.txt
+++ b/cube/CMakeLists.txt
@@ -228,7 +228,7 @@ include_directories(${CUBE_INCLUDE_DIRS})
 if(APPLE)
     include(macOS/cube/cube.cmake)
 elseif(NOT WIN32)
-    if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
+    #if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
         add_executable(vkcube
                        cube.c
                        ${PROJECT_SOURCE_DIR}/cube/cube.vert
@@ -243,7 +243,7 @@ elseif(NOT WIN32)
         if (NEED_RT)
             target_link_libraries(vkcube rt)
         endif()
-    endif()
+    #endif()
 else()
     if(CMAKE_CL_64)
         set(LIB_DIR "Win64")
@@ -280,7 +280,7 @@ endif()
 if(APPLE)
     include(macOS/cubepp/cubepp.cmake)
 elseif(NOT WIN32)
-    if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
+    #if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL ${CMAKE_HOST_SYSTEM_PROCESSOR})
         add_executable(vkcubepp
                        cube.cpp
                        ${PROJECT_SOURCE_DIR}/cube/cube.vert
@@ -290,7 +290,7 @@ elseif(NOT WIN32)
                        ${OPTIONAL_WAYLAND_DATA_FILES})
         target_link_libraries(vkcubepp Vulkan::Vulkan)
         target_compile_definitions(vkcubepp PUBLIC ${CUBE_PLATFORM})
-    endif()
+    #endif()
 else()
     if(CMAKE_CL_64)
         set(LIB_DIR "Win64")
