diff --git a/common/Pcsx2Defs.h b/common/Pcsx2Defs.h
index d11e2e001..31469e7e5 100644
--- a/common/Pcsx2Defs.h
+++ b/common/Pcsx2Defs.h
@@ -121,7 +121,7 @@ static constexpr unsigned int __pagemask = __pagesize - 1;
 
 	#define _inline __inline__ __attribute__((unused))
 	#ifdef NDEBUG
-		#define __forceinline __attribute__((always_inline, unused))
+		#define __forceinline __attribute__((unused))
 	#else
 		#define __forceinline __attribute__((unused))
 	#endif
