diff --git a/pcsx2/Pcsx2Config.cpp b/pcsx2/Pcsx2Config.cpp
index 51ec1f788..ba8a09011 100644
--- a/pcsx2/Pcsx2Config.cpp
+++ b/pcsx2/Pcsx2Config.cpp
@@ -2253,18 +2253,18 @@ bool EmuFolders::SetDataDirectory(Error* error)
 
 void EmuFolders::SetDefaults(SettingsInterface& si)
 {
-	si.SetStringValue("Folders", "Bios", "bios");
-	si.SetStringValue("Folders", "Snapshots", "snaps");
-	si.SetStringValue("Folders", "Savestates", "sstates");
-	si.SetStringValue("Folders", "MemoryCards", "memcards");
-	si.SetStringValue("Folders", "Logs", "logs");
-	si.SetStringValue("Folders", "Cheats", "cheats");
-	si.SetStringValue("Folders", "Patches", "patches");
+	si.SetStringValue("Folders", "Bios", "/userdata/bios/ps2");
+	si.SetStringValue("Folders", "Snapshots", "/userdata/screenshots/ps2");
+	si.SetStringValue("Folders", "Savestates", "/userdata/saves/ps2");
+	si.SetStringValue("Folders", "MemoryCards", "/userdata/saves/ps2");
+	si.SetStringValue("Folders", "Logs", "/userdata/system/logs");
+	si.SetStringValue("Folders", "Cheats", "/userdata/cheats/ps2/cheats");
+	si.SetStringValue("Folders", "Patches", "/usr/bin/pcsx2/resources");
 	si.SetStringValue("Folders", "UserResources", "resources");
-	si.SetStringValue("Folders", "Cache", "cache");
-	si.SetStringValue("Folders", "Textures", "textures");
-	si.SetStringValue("Folders", "InputProfiles", "inputprofiles");
-	si.SetStringValue("Folders", "Videos", "videos");
+	si.SetStringValue("Folders", "Cache", "/userdata/system/cache/ps2");
+	si.SetStringValue("Folders", "Textures", "/userdata/system/configs/PCSX2/textures");
+	si.SetStringValue("Folders", "InputProfiles", "/userdata/system/configs/PCSX2/inputprofiles");
+	si.SetStringValue("Folders", "Videos", "/userdata/system/configs/PCSX2/videos");
 	si.SetStringValue("Folders", "DebuggerLayouts", "debuggerlayouts");
 	si.SetStringValue("Folders", "DebuggerSettings", "debuggersettings");
 }
@@ -2279,20 +2279,20 @@ static std::string LoadPathFromSettings(SettingsInterface& si, const std::string
 
 void EmuFolders::LoadConfig(SettingsInterface& si)
 {
-	Bios = LoadPathFromSettings(si, DataRoot, "Bios", "bios");
-	Snapshots = LoadPathFromSettings(si, DataRoot, "Snapshots", "snaps");
-	Savestates = LoadPathFromSettings(si, DataRoot, "Savestates", "sstates");
-	MemoryCards = LoadPathFromSettings(si, DataRoot, "MemoryCards", "memcards");
-	Logs = LoadPathFromSettings(si, DataRoot, "Logs", "logs");
-	Cheats = LoadPathFromSettings(si, DataRoot, "Cheats", "cheats");
-	Patches = LoadPathFromSettings(si, DataRoot, "Patches", "patches");
-	Covers = LoadPathFromSettings(si, DataRoot, "Covers", "covers");
+	Bios = LoadPathFromSettings(si, DataRoot, "Bios", "/userdata/bios/ps2");
+	Snapshots = LoadPathFromSettings(si, DataRoot, "Snapshots", "/userdata/screenshots/ps2");
+	Savestates = LoadPathFromSettings(si, DataRoot, "Savestates", "/userdata/saves/ps2");
+	MemoryCards = LoadPathFromSettings(si, DataRoot, "MemoryCards", "/userdata/saves/ps2");
+	Logs = LoadPathFromSettings(si, DataRoot, "Logs", "/userdata/system/logs");
+	Cheats = LoadPathFromSettings(si, DataRoot, "Cheats", "/userdata/cheats/ps2/cheats");
+	Patches = LoadPathFromSettings(si, DataRoot, "Patches", "/usr/bin/pcsx2/resources");
+	Covers = LoadPathFromSettings(si, DataRoot, "Covers", "/userdata/system/configs/PCSX2/covers");
 	GameSettings = LoadPathFromSettings(si, DataRoot, "GameSettings", "gamesettings");
 	UserResources = LoadPathFromSettings(si, DataRoot, "UserResources", "resources");
-	Cache = LoadPathFromSettings(si, DataRoot, "Cache", "cache");
-	Textures = LoadPathFromSettings(si, DataRoot, "Textures", "textures");
-	InputProfiles = LoadPathFromSettings(si, DataRoot, "InputProfiles", "inputprofiles");
-	Videos = LoadPathFromSettings(si, DataRoot, "Videos", "videos");
+	Cache = LoadPathFromSettings(si, DataRoot, "Cache", "/userdata/system/cache/ps2");
+	Textures = LoadPathFromSettings(si, DataRoot, "Textures", "/userdata/system/configs/PCSX2/textures");
+	InputProfiles = LoadPathFromSettings(si, DataRoot, "InputProfiles", "/userdata/system/configs/PCSX2/inputprofiles");
+	Videos = LoadPathFromSettings(si, DataRoot, "Videos", "/userdata/system/configs/PCSX2/videos");
 	DebuggerLayouts = LoadPathFromSettings(si, Settings, "DebuggerLayouts", "debuggerlayouts");
 	DebuggerSettings = LoadPathFromSettings(si, Settings, "DebuggerSettings", "debuggersettings");
 
