diff --git a/pcsx2/CDVD/CDVD.cpp b/pcsx2/CDVD/CDVD.cpp
index 9d0e8ed5d..e7cd37f77 100644
--- a/pcsx2/CDVD/CDVD.cpp
+++ b/pcsx2/CDVD/CDVD.cpp
@@ -35,7 +35,7 @@
 #include "CDVD_internal.h"
 #include "IsoFileFormats.h"
 
-#include "GS.h" // for gsVideoMode
+#include "../GS.h" // for gsVideoMode
 #include "Elfheader.h"
 #include "ps2/BiosTools.h"
 #include "Recording/InputRecording.h"
diff --git a/pcsx2/DebugTools/DebugInterface.cpp b/pcsx2/DebugTools/DebugInterface.cpp
index a7f7d250a..688325810 100644
--- a/pcsx2/DebugTools/DebugInterface.cpp
+++ b/pcsx2/DebugTools/DebugInterface.cpp
@@ -21,7 +21,7 @@
 #include "R5900OpcodeTables.h"
 #include "Debug.h"
 #include "VU.h"
-#include "GS.h" // Required for gsNonMirroredRead()
+#include "../GS.h" // Required for gsNonMirroredRead()
 #include "Counters.h"
 
 #include "R3000A.h"
