diff --git a/pcsx2/Input/SDLInputSource.cpp b/pcsx2/Input/SDLInputSource.cpp
index 236e52a56..177253bc4 100644
--- a/pcsx2/Input/SDLInputSource.cpp
+++ b/pcsx2/Input/SDLInputSource.cpp
@@ -226,7 +226,7 @@ bool SDLInputSource::LoadControllerDB()
 {
 	Error error;
 	std::FILE* fp =
-		FileSystem::OpenCFile(Path::Combine(EmuFolders::Resources, CONTROLLER_DB_FILENAME).c_str(), "rb", &error);
+		FileSystem::OpenCFile(Path::Combine("/userdata/system/configs/PCSX2/", CONTROLLER_DB_FILENAME).c_str(), "rb", &error);
 	if (!fp)
 	{
 		Console.Error(fmt::format("SDLInputSource: Failed to open controller database: {}", error.GetDescription()));
