diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index c8a278ec8..98ce9bb40 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -94,7 +94,7 @@ if("${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "x86_64" OR "${CMAKE_HOST_SYSTEM_PR
 		else()
 			# Can't use march=native on Apple Silicon.
 			if(NOT APPLE OR "${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
-				add_compile_options("-march=native")
+				set(ARCH_FLAG "-march=x86-64-v2 -mtune=znver2")
 			endif()
 		endif()
 	endif()
diff --git a/pcsx2/CMakeLists.txt b/pcsx2/CMakeLists.txt
index 475a3d3cf..503806942 100644
--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -736,9 +736,9 @@ if(DISABLE_ADVANCE_SIMD)
 		set(compile_options_avx  -msse4.1 -mavx)
 		set(compile_options_sse4 -msse4.1)
 	else()
-		set(compile_options_avx2 -march=haswell -mtune=haswell)
-		set(compile_options_avx  -march=sandybridge -mtune=sandybridge)
-		set(compile_options_sse4 -msse4.1 -mtune=nehalem)
+		set(compile_options_avx2 -march=x86-64-v2 -mtune=znver2)
+		set(compile_options_avx  -march=x86-64-v2 -mtune=znver2)
+		set(compile_options_sse4 -march=x86-64-v2 -mtune=znver2)
 	endif()
 	# ODR violation time!
 	# Everything would be fine if we only defined things in cpp files, but C++ tends to like inline functions (STL anyone?)
