diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index 94d3baae1..0537b11aa 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -92,7 +92,7 @@ if("${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "x86_64" OR "${CMAKE_HOST_SYSTEM_PR
 		if (DISABLE_ADVANCE_SIMD)
 			add_compile_options("-msse" "-msse2" "-msse4.1" "-mfxsr")
 		else()
-			add_compile_options("-march=native")
+			set(ARCH_FLAG "-march=x86-64-v2 -mtune=znver2")
 		endif()
 	endif()
 else()
diff --git a/pcsx2/CMakeLists.txt b/pcsx2/CMakeLists.txt
index 214583881..a36d0532f 100644
--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -736,9 +736,9 @@ if(DISABLE_ADVANCE_SIMD)
 		set(compile_options_avx  -msse4.1 -mavx)
 		set(compile_options_sse4 -msse4.1)
 	else()
-		set(compile_options_avx2 -march=haswell -mtune=haswell)
-		set(compile_options_avx  -march=sandybridge -mtune=sandybridge)
-		set(compile_options_sse4 -msse4.1 -mtune=nehalem)
+		set(compile_options_avx2 -march=x86-64-v2 -mtune=znver2)
+		set(compile_options_avx  -march=x86-64-v2 -mtune=znver2)
+		set(compile_options_sse4 -march=x86-64-v2 -mtune=znver2)
 	endif()
 	# ODR violation time!
 	# Everything would be fine if we only defined things in cpp files, but C++ tends to like inline functions (STL anyone?)
