diff --git a/pcsx2/Input/SDLInputSource.cpp b/pcsx2/Input/SDLInputSource.cpp
index 6c647c530..2c560f4eb 100644
--- a/pcsx2/Input/SDLInputSource.cpp
+++ b/pcsx2/Input/SDLInputSource.cpp
@@ -1430,9 +1430,12 @@ bool SDLInputSource::CloseDevice(SDL_JoystickID joystick_index)
 
 	{
 		std::lock_guard lock(m_controllers_key_mutex);
-		InputManager::OnInputDeviceDisconnected(
-			{InputBindingKey{.source_type = InputSourceType::SDL, .source_index = static_cast<u32>(it->player_id)}},
-			fmt::format("SDL-{}", it->player_id));
+	// Properly initialize InputBindingKey
+	InputBindingKey key = {InputSourceType::SDL, static_cast<u32>(it->player_id)};
+	
+    InputManager::OnInputDeviceDisconnected(
+        key,
+        fmt::format("SDL-{}", it->player_id));
 
 		if (it->haptic)
 			SDL_CloseHaptic(it->haptic);
