diff --git a/pcsx2-qt/QtHost.cpp b/pcsx2-qt/QtHost.cpp
index 04eb78176..e335b6813 100644
--- a/pcsx2-qt/QtHost.cpp
+++ b/pcsx2-qt/QtHost.cpp
@@ -2131,6 +2131,11 @@ bool QtHost::ParseCommandLineOptions(const QStringList& args, std::shared_ptr<VM
 				AutoBoot(autoboot)->save_state = (++it)->toStdString();
 				continue;
 			}
+			else if (CHECK_ARG_PARAM(QStringLiteral("-stateindex")))
+			  {
+				AutoBoot(autoboot)->state_start_index = (++it)->toInt();
+				continue;
+			}
 			else if (CHECK_ARG_PARAM(QStringLiteral("-elf")))
 			{
 				AutoBoot(autoboot)->elf_override = (++it)->toStdString();
diff --git a/pcsx2/VMManager.cpp b/pcsx2/VMManager.cpp
index b247d662d..0457f9b74 100644
--- a/pcsx2/VMManager.cpp
+++ b/pcsx2/VMManager.cpp
@@ -1567,6 +1567,12 @@ bool VMManager::Initialize(VMBootParameters boot_params)
 
 	SetEmuThreadAffinities();
 
+	if (boot_params.state_start_index.has_value()) {
+	  for(int i=1; i<boot_params.state_start_index.value(); i++) {
+	    SaveStateSelectorUI::SelectNextSlot(false);
+	  }
+	}
+
 	// do we want to load state?
 	if (!GSDumpReplayer::IsReplayingDump() && !state_to_load.empty())
 	{
diff --git a/pcsx2/VMManager.h b/pcsx2/VMManager.h
index 11af8cd51..0fc8a5498 100644
--- a/pcsx2/VMManager.h
+++ b/pcsx2/VMManager.h
@@ -32,6 +32,7 @@ struct VMBootParameters
 	std::string elf_override;
 	std::string save_state;
 	std::optional<s32> state_index;
+  	std::optional<s32> state_start_index;
 	std::optional<CDVD_SourceType> source_type;
 
 	std::optional<bool> fast_boot;
