diff --git a/pcsx2/Config.h b/pcsx2/Config.h
index 14776ae8d..715b52ae7 100644
--- a/pcsx2/Config.h
+++ b/pcsx2/Config.h
@@ -1316,6 +1316,7 @@ struct Pcsx2Config
 		InhibitScreensaver : 1,
 		BackupSavestate : 1,
 		McdFolderAutoManage : 1,
+		AutoIncrementSlot : 1,
 		ManuallySetRealTimeClock : 1, // passes user-set real-time clock information to cdvd at startup
 		UseSystemLocaleFormat : 1, // presents OS time format instead of yyyy-MM-dd HH:mm:ss for manual RTC
 
diff --git a/pcsx2/Pcsx2Config.cpp b/pcsx2/Pcsx2Config.cpp
index b25ea49d3..7a62ba8de 100644
--- a/pcsx2/Pcsx2Config.cpp
+++ b/pcsx2/Pcsx2Config.cpp
@@ -1925,6 +1925,7 @@ Pcsx2Config::Pcsx2Config()
 	EnableDiscordPresence = false;
 	ManuallySetRealTimeClock = false;
 	UseSystemLocaleFormat = false;
+	AutoIncrementSlot = false;
 
 	// To be moved to FileMemoryCard pluign (someday)
 	for (uint slot = 0; slot < 8; ++slot)
@@ -1967,6 +1968,7 @@ void Pcsx2Config::LoadSaveCore(SettingsWrapper& wrap)
 	SettingsWrapBitBool(EnableDiscordPresence);
 	SettingsWrapBitBool(InhibitScreensaver);
 	SettingsWrapBitBool(HostFs);
+	SettingsWrapBitBool(AutoIncrementSlot);
 
 	SettingsWrapBitBool(BackupSavestate);
 	SettingsWrapBitBool(McdFolderAutoManage);
diff --git a/pcsx2/VMManager.cpp b/pcsx2/VMManager.cpp
index 5a218fbad..8d9a298a7 100644
--- a/pcsx2/VMManager.cpp
+++ b/pcsx2/VMManager.cpp
@@ -2053,6 +2053,12 @@ bool VMManager::SaveState(const char* filename, bool zip_on_thread, bool backup_
 
 bool VMManager::SaveStateToSlot(s32 slot, bool zip_on_thread)
 {
+	//increment the slot before saving
+	if(EmuConfig.AutoIncrementSlot) {
+	  SaveStateSelectorUI::SelectNextSlot(false);
+	  slot = SaveStateSelectorUI::GetCurrentSlot();
+	}
+
 	const std::string filename(GetCurrentSaveStateFileName(slot));
 	if (filename.empty())
 		return false;
