diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index 603e89fd9..32c08cbba 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -19,7 +19,7 @@ find_package(LZ4 REQUIRED)
 find_package(WebP REQUIRED) # v1.3.2, spews an error on Linux because no pkg-config.
 find_package(SDL3 3.2.6 REQUIRED)
 find_package(Freetype 2.11.1 REQUIRED)
-find_package(plutovg 0.0.13 REQUIRED)
+find_package(plutovg 1.0.0 REQUIRED)
 find_package(plutosvg 0.0.6 REQUIRED)
 
 if(USE_VULKAN)
