diff --git a/configure b/configure
index ffa407d53d..cbbbc2c8cd 100755
--- a/configure
+++ b/configure
@@ -4813,6 +4813,7 @@ cparser_flags(){
     for flag; do
         case $flag in
             -Wno-switch)             echo -Wno-switch-enum ;;
+            -Wincompatible-pointer-types)     echo -Wincompatible-pointer-types ;;
             -Wno-format-zero-length) ;;
             -Wdisabled-optimization) ;;
             -Wno-pointer-sign)       echo -Wno-other ;;
@@ -7444,6 +7445,7 @@ check_disable_warning -Wno-pointer-sign
 check_disable_warning -Wno-unused-const-variable
 check_disable_warning -Wno-bool-operation
 check_disable_warning -Wno-char-subscripts
+check_disable_warning -Wno-incompatible-pointer-types
 
 check_disable_warning_headers(){
     warning_flag=-W${1#-Wno-}
@@ -7630,6 +7632,7 @@ elif enabled clang; then
 elif enabled cparser; then
     add_cflags -Wno-missing-variable-declarations
     add_cflags -Wno-empty-statement
+    add_cflags -Wno-incompatible-pointer-types
 elif enabled armcc; then
     add_cflags -W${armcc_opt},--diag_suppress=4343 # hardfp compat
     add_cflags -W${armcc_opt},--diag_suppress=3036 # using . as system include dir
