diff --git a/Config/Supermodel.ini b/Config/Supermodel.ini
index 1889a8d..10a3a51 100644
--- a/Config/Supermodel.ini
+++ b/Config/Supermodel.ini
@@ -259,175 +259,421 @@ InputFishingReel = "KEY_SPACE,JOY1_ZAXIS_POS"
 InputFishingCast = "KEY_Z,JOY1_BUTTON1"
 InputFishingSelect = "KEY_X,JOY1_BUTTON2"
 InputFishingTension = "KEY_T,JOY1_ZAXIS_NEG"
- 
- ;Our non input settings...					   
+
+;Our non input settings...
 
 ; Game specific settings
+
 ;daytona 2 power edition
-    [ dayto2pe ]
+[dayto2pe]
 PowerPCFrequency = 90
 EmulateDSB = 1
+MusicVolume = 120
+SoundVolume = 100
+Balance = -10
+InputJoy1XSaturation = 155
+
 ;daytona 2: battle to the edge
-    [ daytona2 ]
+[daytona2]
 PowerPCFrequency = 90
 EmulateDSB = 1
-  ;Dirt Devils (Export, Revision A)  
-    [ dirtdvls ]
+MusicVolume = 120
+SoundVolume = 100
+Balance = -10
+InputJoy1XSaturation = 155
+
+;Dirt Devils (Export, Revision A)
+[dirtdvls]
 PowerPCFrequency = 60
+MusicVolume = 100
+SoundVolume = 130
+Balance = -20
+
 ;Dirt Devils (Australia, Revision A)
-    [ dirtdvlsa ]
+[dirtdvlsa]
 PowerPCFrequency = 60
+MusicVolume = 100
+SoundVolume = 130
+Balance = -20
+
 ;Dirt Devils (German)
-    [ dirtdvlsg ]
-PowerPCFrequency = 60     
-;Dirt Devils (Japan, Revision A)  
-    [ dirtdvlsj ]
+[dirtdvlsg]
+PowerPCFrequency = 60
+MusicVolume = 100
+SoundVolume = 130
+Balance = -20
+
+;Dirt Devils (Japan, Revision A)
+[dirtdvlsj]
 PowerPCFrequency = 60
-;Emergency Call Ambulance (Export)  - NOT WORKING PI
-    [ eca       ]
-;Emergency Call Ambulance (Japan)   - NOIT WORKING PI
-    [ ecaj      ]
-;Emergency Call Ambulance (US location test?)	- NOT WORKING PI
-    [ ecap      ]
-;Emergency Call Ambulance (USA)	- NOT WORKING PI
-    [ ecau      ]
-;Fighting Vipers 2 (Japan, Revision A)	
-    [ fvipers2 ]
-PowerPCFrequency = 55  
-;Fighting Vipers 2 (?)    
-    [ fvipers2o ]
+MusicVolume = 100
+SoundVolume = 130
+Balance = -20
+
+;Emergency Call Ambulance (Export) - NOT WORKING PI
+[eca]
+MusicVolume = 100
+SoundVolume = 100
+Balance = -35
+
+;Emergency Call Ambulance (Japan) - NOT WORKING PI
+[ecaj]
+MusicVolume = 100
+SoundVolume = 100
+Balance = -35
+
+;Emergency Call Ambulance (US location test?)- NOT WORKING PI
+[ecap]
+MusicVolume = 100
+SoundVolume = 100
+Balance = -35
+
+;Emergency Call Ambulance (USA)- NOT WORKING PI
+[ecau]
+MusicVolume = 100
+SoundVolume = 100
+Balance = -35
+
+;Fighting Vipers 2 (Japan, Revision A)
+[fvipers2]
+PowerPCFrequency = 55
+LegacySoundDSP = 1
+MusicVolume = 100
+SoundVolume = 170
+Balance = 10
+
+;Fighting Vipers 2 (?)
+[fvipers2o]
 PowerPCFrequency = 48
-;Get Bass (?)    
-    [ getbass ]
+LegacySoundDSP = 1
+MusicVolume = 100
+SoundVolume = 170
+Balance = 10
+
+;Get Bass (?)
+[getbass]
 PowerPCFrequency = 30
-;Harley-Davidson and L.A. Riders (Revision B)	- pi too slow
-    [ harley ] 
+WideScreen =0
+WideBackground =0
+MusicVolume = 100
+SoundVolume = 110
+Balance = 10
+
+;Harley-Davidson and L.A. Riders (Revision B)- pi too slow
+[harley]
 PowerPCFrequency = 166
+MusicVolume = 100
+SoundVolume = 100
+Balance = 10
+
 ;Harley-Davidson and L.A. Riders (Revision A)   - pi too slow
-    [ harleya ]
+[harleya]
 PowerPCFrequency = 166
-;L.A. Machineguns: Rage of the Machines (Japan)    
-    [ lamachin ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 10
+
+;L.A. Machineguns: Rage of the Machines (Japan)
+[lamachin]
 PowerPCFrequency = 40
-;Le Mans 24 (Revision B)	
-    [ lemans24 ]
- ;The Lost World (Japan, Revision A)   
-    [ lostwsga ];The Lost World (Original Revision)
-PowerPCFrequency = 33	
-    [ lostwsgo ]
+MusicVolume = 100
+SoundVolume = 115
+Balance = 0
+InputSystem=rawinput
+
+;Le Mans 24 (Revision B)
+[lemans24]
+MusicVolume = 100
+SoundVolume = 110
+Balance = 0
+InputJoy1XSaturation = 110
+
+;The Lost World (Japan, Revision A)
+[lostwsga]
+MusicVolume = 100
+SoundVolume = 105
+Balance = 20
+InputSystem=rawinput
+CrosshairStyle = bmp
+
+;The Lost World (Original Revision)
+PowerPCFrequency = 33
+[lostwsgo]
 PowerPCFrequency = 33
-;Magical Truck Adventure (Japan)    
-    [ magtruck ]
+MusicVolume = 100
+SoundVolume = 105
+Balance = 20
+InputSystem=rawinput
+CrosshairStyle = bmp
+
+;Magical Truck Adventure (Japan)
+[magtruck]
 PowerPCFrequency = 40
-;The Ocean Hunter - Default works fine on pi 
-    [ oceanhun ]
-MusicVolume = 200
-SoundVolume = 200
-    [ oceanhuna ]
-MusicVolume = 200
-SoundVolume = 200
-;Scud Race (Australia, Twin)	
-    [ scud ]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 15
+
+;The Ocean Hunter - Default works fine on pi
+[oceanhun]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+InputSystem=rawinput
+[oceanhuna]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+InputSystem=rawinput
+
+;Scud Race (Australia, Twin)
+[scud]
 PowerPCFrequency = 35
-MusicVolume = 200
 EmulateDSB = 1
-;Scud Race (Export, Twin)	
-    [ scuda ]
-PowerPCFrequency = 35
 MusicVolume = 200
-EmulateDSB = 1
-;Scud Race (Japan, Deluxe)	
-    [ scudj ]
+SoundVolume = 60
+Balance = 0
+InputJoy1XSaturation = 155
+
+;Scud Race (Export, Twin)
+[scuda]
 PowerPCFrequency = 35
-MusicVolume = 200
 EmulateDSB = 1
-;Scud Race Plus (Revision A)	
-    [ scudplus ]
-PowerPCFrequency = 35
 MusicVolume = 200
+SoundVolume = 60
+Balance = 0
+InputJoy1XSaturation = 155
+
+;Scud Race (Japan, Deluxe)
+[scudj]
+PowerPCFrequency = 35
 EmulateDSB = 1
-;Scud Race Plus (?)  
-    [ scudplusa ]
+MusicVolume = 200
+SoundVolume = 60
+Balance = 0
+InputJoy1XSaturation = 155
+
+;Scud Race Plus (Revision A)
+[scudplus]
 PowerPCFrequency = 35
+EmulateDSB = 1
 MusicVolume = 200
+SoundVolume = 60
+Balance = 0
+InputJoy1XSaturation = 155
+
+;Scud Race Plus (?)
+[scudplusa]
+PowerPCFrequency = 35
 EmulateDSB = 1
-;Ski Champ (Japan)	
-    [ skichamp  ]
-;PowerPCFrequency = 43
+MusicVolume = 200
+SoundVolume = 60
+Balance = 0
+InputJoy1XSaturation = 155
+
+;Ski Champ (Japan)
+[skichamp]
 PowerPCFrequency = 60
-;Spikeout Final Edition (?) - too slow on pi	
-    [ spikeofe  ]
+MusicVolume = 100
+SoundVolume = 200
+Balance = 10
+ForceFeedback=1
+
+;Spikeout Final Edition (?) - too slow on pi
+[spikeofe]
 PowerPCFrequency = 100
 EmulateDSB = 1
-;Spikeout (Revision C)	
-    [ spikeout  ]
+MusicVolume = 90
+SoundVolume = 90
+Balance = 0
+
+;Spikeout (Revision C)
+[spikeout]
+
 ;PowerPCFrequency = 35
 PowerPCFrequency = 55
 EmulateDSB = 1
-;Sega Rally 2 (?)	
-    [ srally2   ]
-EmulateDSB = 1 
-PowerPCFrequency = 60
-;Sega Rally 2 (Prototype)	
-    [ srally2p  ]
-PowerPCFrequency = 48
+MusicVolume = 90
+SoundVolume = 90
+Balance = 0
+
+;Sega Rally 2 (?)
+[srally2]
 EmulateDSB = 1
+SoundVolume = 155
+MusicVolume = 140
+Balance = 15
+PowerPCFrequency=75
+InputJoy1XSaturation = 150
+
+;Sega Rally 2 (Prototype)
+[srally2p]
+SoundVolume = 155
+MusicVolume = 140
+Balance = 15
+PowerPCFrequency=75
+InputJoy1XSaturation = 150
+
 ;Sega Rally 2 (Prototype Version A)
-    [ srally2pa ]
-PowerPCFrequency = 48
-EmulateDSB = 1
-;Sega Rally 2 (Deluxe)	
-    [ srally2x  ]  
+[srally2pa]
+SoundVolume = 155
+MusicVolume = 140
+Balance = 15
+PowerPCFrequency=75
+InputJoy1XSaturation = 150
+
+;Sega Rally 2 (Deluxe)
+[srally2x]
+
 ;Playable on pi but more slower down than srally2
-PowerPCFrequency = 100
-EmulateDSB = 1
-;Star Wars Trilogy Arcade - Default works fine on pi 
-    [ swtrilgy ]
+SoundVolume = 155
+MusicVolume = 140
+Balance = 15
+PowerPCFrequency=100
+InputJoy1XSaturation = 150
+
+;Star Wars Trilogy Arcade - Default works fine on pi
+[swtrilgy]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+InputSystem=rawinput
+
+;Star Wars Trilogy Arcade (?)
+[swtrilgya]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+InputSystem=rawinput
+
 ;Star Wars Trilogy Arcade (?)
-    [ swtrilgya ]
-;Star Wars Trilogy Arcade (?)   
-    [ swtrilgyp ]
+[swtrilgyp]
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+InputSystem=rawinput
+
 ;Virtua Fighter 3 (Revision D)
-    [ vf3       ]
+[vf3]
 PowerPCFrequency = 40
-;PowerPCFrequency = 100
-;Virtua Fighter 3 (Revision A)	
-    [ vf3a      ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -5
+
+;Virtua Fighter 3 (Revision A)
+[vf3a]
 PowerPCFrequency = 40
+MusicVolume = 100
+SoundVolume = 110
+Balance = -5
+
 ;Virtua Fighter 3 (Revision C)
-    [ vf3c      ]
-PowerPCFrequency = 40	
-;Virtua Fighter 3 Team Battle (?)	
-    [ vf3tb     ]
+[vf3c]
 PowerPCFrequency = 40
+MusicVolume = 100
+SoundVolume = 110
+Balance = -5
+
+;Virtua Fighter 3 Team Battle (?)
+[vf3tb]
+PowerPCFrequency = 40
+MusicVolume = 100
+SoundVolume = 110
+Balance = -5
+
 ;Virtual On 2: Oratorio Tangram (Revision B)
-    [ von2      ]
+[von2]
 PowerPCFrequency = 30
-;Virtual On 2: Oratorio Tangram (Ver 5.4g)	
-    [ von254g   ]
+
+;Virtual On 2: Oratorio Tangram (Ver 5.4g)
+[von254g]
 PowerPCFrequency = 30
-;Virtual On 2: Oratorio Tangram (Revision A)	
-    [ von2a     ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -5
+
+;Virtual On 2: Oratorio Tangram (Revision A)
+[von2a]
 PowerPCFrequency = 30
-;Virtual On 2: Oratorio Tangram (?)	
-    [ von2o     ]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -5
+
+;Virtual On 2: Oratorio Tangram (?)
+[von2o]
 PowerPCFrequency = 35
-;Virtua Striker 2 (Step 2.0) - no special settings    
-    [ vs2       ]
-;Virtua Striker 2 (Step 1.5) -  no special settings 
-    [ vs215     ]
-;Virtua Striker 2 (Step 1.5, older) -  no special settings 	
-    [ vs215o    ]
-;Virtua Striker 2 '98 (Step 2.0) -  no special settings 	
-    [ vs298     ]
-;Virtua Striker 2 '98 (Step 1.5) - no special settings 
-    [ vs29815   ]     
-;Virtua Striker 2 '99 (?)	
-    [ vs299     ]
-;Virtua Striker 2 '99 (Step 1.5)	
-    [ vs29915   ] 
-;Virtua Striker 2 '99 (Revision A)    
-    [ vs299a    ]
-;Virtua Striker 2 '99 (Revision B)	
-    [ vs299b    ]
-;Virtua Striker 2 '99.1 (Revision B)	
+MusicVolume = 100
+SoundVolume = 100
+Balance = 0
+
+;Virtua Striker 2 (Step 2.0) - no special settings
+[vs2]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+;Virtua Striker 2 (Step 1.5) -  no special settings
+[vs215]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+;Virtua Striker 2 (Step 1.5, older) -  no special settings
+[vs215o]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+;Virtua Striker 2 '98 (Step 2.0) -  no special settings
+[vs298]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+;Virtua Striker 2 '98 (Step 1.5) - no special settings
+[vs29815]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+;Virtua Striker 2 '99 (?)
+[vs299]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+;Virtua Striker 2 '99 (Step 1.5)
+[vs29915]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+;Virtua Striker 2 '99 (Revision A)
+[vs299a]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
+;Virtua Striker 2 '99 (Revision B)
+[vs299b]
+MusicVolume = 100
+SoundVolume = 110
+Balance = -10
+WideScreen =0
+WideBackground =0
+
