diff --git a/gzip.h b/gzip.h
index 7941e89..ef1b288 100644
--- a/gzip.h
+++ b/gzip.h
@@ -117,7 +117,7 @@ EXTERN(uch, window);         /* Sliding window and suffix table (unlzw) */
 #define tab_suffix window
 #ifndef MAXSEG_64K
 #  define tab_prefix prev    /* hash link (see deflate.c) */
-#  define head (prev+WSIZE)  /* hash head (see deflate.c) */
+//#  define head (prev+WSIZE)  /* hash head (see deflate.c) */
    EXTERN(ush, tab_prefix);  /* prefix code (see unlzw.c) */
 #else
 #  define tab_prefix0 prev
