diff --git a/src/input.c b/src/input.c
index cd75450f..83dd4f91 100644
--- a/src/input.c
+++ b/src/input.c
@@ -305,7 +305,8 @@ void Init_Joy (void)
   else
     joy = NULL;  /* signals that no yoystick is present */
 
-
+  SDL_JoystickEventState (SDL_DISABLE);
+  joy = NULL;  /* signals that no yoystick is present */
   return;
 }
 
diff --git a/src/sound.c b/src/sound.c
index 74b216a3..a5cbf373 100644
--- a/src/sound.c
+++ b/src/sound.c
@@ -126,7 +126,7 @@ Init_Audio(void)
   Uint16 audio_format = MIX_DEFAULT_FORMAT;
   int audio_channels = 2;	// = stereo
   //  int audio_buffers = 4096;
-  int audio_buffers = 100; // higher = less risk of sound skips, but more lag
+  int audio_buffers = 1024; // higher = less risk of sound skips, but more lag
 
   DebugPrintf(1, "\nInitializing SDL Audio Systems....\n");
 
