diff --git a/src/misc.c b/src/misc.c
index 0a9d07e5..b5f7e2c2 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -161,7 +161,7 @@ LoadGameConfig (void)
   else
     DebugPrintf (0, "found environment HOME = '%s'\n", homedir );
 
-  sprintf (ConfigDir, "%s/.freedroidClassic", homedir);
+  sprintf (ConfigDir, "%s/freedroidClassic", "/userdata/system/configs/");
 
   if (stat(ConfigDir, &statbuf) == -1) {
     DebugPrintf (0, "Couldn't stat Config-dir %s, I'll try to create it...", ConfigDir);
