diff --git a/src/main/rom.c b/src/main/rom.c
index 325c5b6c..900b2cf2 100644
--- a/src/main/rom.c
+++ b/src/main/rom.c
@@ -587,7 +587,8 @@ void romdatabase_open(void)
 
     int counter, value, lineno;
     unsigned char index;
-    const char *pathname = ConfigGetSharedDataFilepath("mupen64plus.ini");
+    //const char *pathname = ConfigGetSharedDataFilepath("mupen64plus.ini");
+    const char *pathname = "/usr/share/mupen64plus/mupen64plus.ini";
 
     if(g_romdatabase.have_database)
         return;
