diff --git a/src/main/savestates.c b/src/main/savestates.c
index 14b881b0..1f4ded9b 100644
--- a/src/main/savestates.c
+++ b/src/main/savestates.c
@@ -1546,9 +1546,6 @@ static int savestates_save_m64p(const struct device* dev, char *filepath)
 
     save->filepath = strdup(filepath);
 
-    if(autoinc_save_slot)
-        savestates_inc_slot();
-
     save_eventqueue_infos(&dev->r4300.cp0, queue);
 
     // Allocate memory for the save state data
@@ -2186,6 +2183,9 @@ int savestates_save(void)
     else if (fname == NULL) // Always save slots in M64P format
         type = savestates_type_m64p;
 
+    if(autoinc_save_slot)
+      savestates_inc_slot();
+
     filepath = savestates_generate_path(type);
     if (filepath != NULL)
     {
