diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index dbb9f1b8a..b6c2b9fac 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -123,7 +123,7 @@ if(${PCSX2_TARGET_ARCHITECTURES} MATCHES "x86_64")
 			endif()
 		else()
 			#set(ARCH_FLAG "-march=native -fabi-version=6")
-			set(ARCH_FLAG "-march=native")
+			set(ARCH_FLAG "-march=znver2")
 		endif()
 	elseif(NOT DEFINED ARCH_FLAG AND USE_CLANG_CL)
 		set(ARCH_FLAG "-msse4.1")
diff --git a/pcsx2/CMakeLists.txt b/pcsx2/CMakeLists.txt
index 15d88e7ce..989212f01 100644
--- a/pcsx2/CMakeLists.txt
+++ b/pcsx2/CMakeLists.txt
@@ -708,9 +708,9 @@ if(DISABLE_ADVANCE_SIMD)
 		set(compile_options_avx  -msse4.1 -mavx)
 		set(compile_options_sse4 -msse4.1)
 	else()
-		set(compile_options_avx2 -march=haswell -mtune=haswell)
-		set(compile_options_avx  -march=sandybridge -mtune=sandybridge)
-		set(compile_options_sse4 -msse4.1 -mtune=nehalem)
+		set(compile_options_avx2 -march=znver2 -mtune=znver2)
+		set(compile_options_avx  -march=znver2 -mtune=znver2)
+		set(compile_options_sse4 -march=znver2 -mtune=znver2)
 	endif()
 	# ODR violation time!
 	# Everything would be fine if we only defined things in cpp files, but C++ tends to like inline functions (STL anyone?)
