diff --git a/apps/ymir-sdl3/CMakeLists.txt b/apps/ymir-sdl3/CMakeLists.txt
index c0b3802e..4334a866 100644
--- a/apps/ymir-sdl3/CMakeLists.txt
+++ b/apps/ymir-sdl3/CMakeLists.txt
@@ -350,7 +350,7 @@ find_package(CURL REQUIRED)
 find_package(fmt CONFIG REQUIRED)
 find_package(rtmidi CONFIG REQUIRED)
 find_package(SDL3 CONFIG REQUIRED)
-find_package(Stb REQUIRED)
+find_package(stb REQUIRED)
 find_package(tomlplusplus CONFIG REQUIRED)
 
 target_include_directories(ymir-sdl3
@@ -371,7 +371,7 @@ target_link_libraries(ymir-sdl3 PRIVATE
     cereal::cereal
     lz4::lz4
     RtMidi::rtmidi
-    CURL::libcurl_static
+    CURL::libcurl
 )
 if (WIN32)
     target_link_libraries(ymir-sdl3 PRIVATE
diff --git a/apps/ymir-sdl3/src/app/app.cpp b/apps/ymir-sdl3/src/app/app.cpp
index 91f1f56f..b755c240 100644
--- a/apps/ymir-sdl3/src/app/app.cpp
+++ b/apps/ymir-sdl3/src/app/app.cpp
@@ -126,8 +126,8 @@
 
 #include <cmrc/cmrc.hpp>
 
-#include <stb_image.h>
-#include <stb_image_write.h>
+#include <stb/stb_image.h>
+#include <stb/stb_image_write.h>
 
 #include <fmt/chrono.h>
 #include <fmt/std.h>
diff --git a/apps/ymir-sdl3/src/app/stb_implementations.cpp b/apps/ymir-sdl3/src/app/stb_implementations.cpp
index aa7c528a..c4e57f6e 100644
--- a/apps/ymir-sdl3/src/app/stb_implementations.cpp
+++ b/apps/ymir-sdl3/src/app/stb_implementations.cpp
@@ -1,4 +1,4 @@
 #define STB_IMAGE_IMPLEMENTATION
 #define STB_IMAGE_WRITE_IMPLEMENTATION
-#include <stb_image.h>
-#include <stb_image_write.h>
+#include <stb/stb_image.h>
+#include <stb/stb_image_write.h>
diff --git a/apps/ymir-sdl3/src/app/ui/windows/about_window.cpp b/apps/ymir-sdl3/src/app/ui/windows/about_window.cpp
index f4245e34..2cd4a984 100644
--- a/apps/ymir-sdl3/src/app/ui/windows/about_window.cpp
+++ b/apps/ymir-sdl3/src/app/ui/windows/about_window.cpp
@@ -15,12 +15,12 @@
 #include <fmt/format.h>
 #include <lz4.h>
 #include <nghttp2/nghttp2ver.h>
-#include <nghttp3/version.h>
-#include <ngtcp2/version.h>
+//#include <nghttp3/version.h>
+//#include <ngtcp2/version.h>
 #include <nlohmann/json.hpp>
 #include <openssl/opensslv.h>
 #include <rtmidi/RtMidi.h>
-#include <semver.hpp>
+//#include <semver.hpp>
 #include <toml++/toml.hpp>
 #include <xxhash.h>
 
@@ -109,8 +109,8 @@ static const struct {
     {.name = "moodycamel::\nConcurrentQueue", .version = "\n" MC_CONCQUEUE_VERSION,  .license = licenseBSD2,          .repoURL = "https://github.com/cameron314/concurrentqueue",  .licenseURL = "https://github.com/cameron314/concurrentqueue/blob/master/LICENSE.md"},
     {.name = "Neargye/semver",                .version = SEMVER_VERSION,             .license = licenseMIT,           .repoURL = "https://github.com/Neargye/semver",              .licenseURL = "https://github.com/Neargye/semver/blob/master/LICENSE"},
     {.name = "nghttp2",                       .version = NGHTTP2_VERSION,            .license = licenseMIT,           .repoURL = "https://github.com/nghttp2/nghttp2",             .licenseURL = "https://github.com/nghttp2/nghttp2/blob/master/COPYING",                 .homeURL = "https://nghttp2.org/"},
-    {.name = "nghttp3",                       .version = NGHTTP3_VERSION,            .license = licenseMIT,           .repoURL = "https://github.com/ngtcp2/nghttp3",              .licenseURL = "https://github.com/ngtcp2/nghttp3/blob/main/COPYING",                    .homeURL = "https://nghttp2.org/nghttp3/"},
-    {.name = "ngtcp2",                        .version = NGTCP2_VERSION,             .license = licenseMIT,           .repoURL = "https://github.com/ngtcp2/ngtcp2",               .licenseURL = "https://github.com/ngtcp2/ngtcp2/blob/main/COPYING",                     .homeURL = "https://nghttp2.org/ngtcp2/"},
+//    {.name = "nghttp3",                       .version = NGHTTP3_VERSION,            .license = licenseMIT,           .repoURL = "https://github.com/ngtcp2/nghttp3",              .licenseURL = "https://github.com/ngtcp2/nghttp3/blob/main/COPYING",                    .homeURL = "https://nghttp2.org/nghttp3/"},
+//    {.name = "ngtcp2",                        .version = NGTCP2_VERSION,             .license = licenseMIT,           .repoURL = "https://github.com/ngtcp2/ngtcp2",               .licenseURL = "https://github.com/ngtcp2/ngtcp2/blob/main/COPYING",                     .homeURL = "https://nghttp2.org/ngtcp2/"},
     {.name = "nlohmann/json",                 .version = NLOHMANN_JSON_VERSION,      .license = licenseMIT,           .repoURL = "https://github.com/nlohmann/json",               .licenseURL = "https://github.com/nlohmann/json/blob/develop/LICENSE.MIT",              .homeURL = "https://json.nlohmann.me/"},
     {.name = "OpenSSL",                       .version = OPENSSL_FULL_VERSION_STR,   .license = licenseApache2_0,     .repoURL = "https://github.com/openssl/openssl",             .licenseURL = "https://github.com/openssl/openssl/blob/master/LICENSE.txt",             .homeURL = "https://www.openssl.org/"},
     {.name = "RtMidi",                        .version = RTMIDI_VERSION,             .license = licenseMITRtMidi,     .repoURL = "https://github.com/thestk/rtmidi",               .licenseURL = "https://github.com/thestk/rtmidi/blob/master/LICENSE"},
