diff --git a/apps/ymir-sdl3/CMakeLists.txt b/apps/ymir-sdl3/CMakeLists.txt
index 4a1f8c10..4ae7ed99 100644
--- a/apps/ymir-sdl3/CMakeLists.txt
+++ b/apps/ymir-sdl3/CMakeLists.txt
@@ -431,7 +431,7 @@ find_package(OpenSSL CONFIG REQUIRED)
 find_package(CURL REQUIRED)
 find_package(rtmidi CONFIG REQUIRED)
 find_package(SDL3 CONFIG REQUIRED)
-find_package(Stb REQUIRED)
+#find_package(Stb REQUIRED)
 find_package(tomlplusplus CONFIG REQUIRED)
 
 target_include_directories(ymir-sdl3
@@ -453,7 +453,7 @@ target_link_libraries(ymir-sdl3 PRIVATE
     cereal::cereal
     lz4::lz4
     RtMidi::rtmidi
-    CURL::libcurl_static
+    ${CURL_LIBRARIES}
 )
 if (WIN32)
     target_link_libraries(ymir-sdl3 PRIVATE
