diff --git a/libretro/libretro-core.c b/libretro/libretro-core.c
index e0f8b5a..8f66736 100644
--- a/libretro/libretro-core.c
+++ b/libretro/libretro-core.c
@@ -105,7 +105,7 @@ extern uae_u32 natmem_size;
 char full_path[RETRO_PATH_MAX] = {0};
 static char *uae_argv[] = { "puae" };
 static int restart_pending = 0;
-static struct puae_cart_info puae_carts[RETRO_NUM_CORE_OPTION_VALUES_MAX] = {0};
+static struct puae2021_cart_info puae2021_carts[RETRO_NUM_CORE_OPTION_VALUES_MAX] = {0};
 
 static long retro_now = 0;
 float retro_refresh = 0;
@@ -677,20 +677,20 @@ static void retro_set_paths(void)
             retro_system_directory, DIR_SEP_STR, "uae");
 }
 
-static void free_puae_carts(void)
+static void free_puae2021_carts(void)
 {
    size_t i;
    for (i = 0; i < RETRO_NUM_CORE_OPTION_VALUES_MAX; i++)
    {
-      if (puae_carts[i].value)
+      if (puae2021_carts[i].value)
       {
-         free(puae_carts[i].value);
-         puae_carts[i].value = NULL;
+         free(puae2021_carts[i].value);
+         puae2021_carts[i].value = NULL;
       }
-      if (puae_carts[i].label)
+      if (puae2021_carts[i].label)
       {
-         free(puae_carts[i].label);
-         puae_carts[i].label = NULL;
+         free(puae2021_carts[i].label);
+         puae2021_carts[i].label = NULL;
       }
    }
 }
@@ -747,7 +747,7 @@ static void retro_set_core_options()
    static struct retro_core_option_v2_definition option_defs_us[] =
    {
       {
-         "puae_model",
+         "puae2021_model",
          "System > Model",
          "Model",
          "'Automatic' defaults to 'A500' with floppy disks, 'A1200' with hard drives and 'CD32' with compact discs. 'Automatic' can be overridden with file path tags.\nCore restart required.",
@@ -773,7 +773,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_model_options_display",
+         "puae2021_model_options_display",
          "System > Show Automatic Model Options",
          "Show Automatic Model Options",
          "Show/hide default model options (Floppy/HD/CD) for 'Automatic' model.",
@@ -787,7 +787,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_model_fd",
+         "puae2021_model_fd",
          "Model > Automatic Floppy",
          "Automatic Floppy",
          "Default model when floppies are launched with 'Automatic' model.\nCore restart required.",
@@ -809,7 +809,7 @@ static void retro_set_core_options()
          "A500"
       },
       {
-         "puae_model_hd",
+         "puae2021_model_hd",
          "Model > Automatic HD",
          "Automatic HD",
          "Default model when HD interface is used with 'Automatic' model. Affects WHDLoad installs and other hard drive images.\nCore restart required.",
@@ -827,7 +827,7 @@ static void retro_set_core_options()
          "A1200"
       },
       {
-         "puae_model_cd",
+         "puae2021_model_cd",
          "Model > Automatic CD",
          "Automatic CD",
          "Default model when compact discs are launched with 'Automatic' model.\nCore restart required.",
@@ -842,7 +842,7 @@ static void retro_set_core_options()
          "CD32"
       },
       {
-         "puae_kickstart",
+         "puae2021_kickstart",
          "System > Kickstart ROM",
          "Kickstart ROM",
          "'Automatic' defaults to the most compatible version for the model. 'AROS' is a built-in replacement with fair compatibility.\nCore restart required.",
@@ -865,7 +865,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_chipmem_size",
+         "puae2021_chipmem_size",
          "System > Chip RAM",
          "Chip RAM",
          "'Automatic' defaults to the current preset model.\nCore restart required.",
@@ -882,7 +882,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_bogomem_size",
+         "puae2021_bogomem_size",
          "System > Slow RAM",
          "Slow RAM",
          "'Automatic' defaults to the current preset model.\nCore restart required.",
@@ -900,7 +900,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_fastmem_size",
+         "puae2021_fastmem_size",
          "System > Z2 Fast RAM",
          "Z2 Fast RAM",
          "'Automatic' defaults to the current preset model.\nCore restart required.",
@@ -918,7 +918,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_z3mem_size",
+         "puae2021_z3mem_size",
          "System > Z3 Fast RAM",
          "Z3 Fast RAM",
          "'Automatic' defaults to the current preset model.\nCore restart required.",
@@ -942,7 +942,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_cpu_model",
+         "puae2021_cpu_model",
          "System > CPU Model",
          "CPU Model",
          "'Automatic' defaults to the current preset model.\nCore restart required.",
@@ -960,7 +960,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_fpu_model",
+         "puae2021_fpu_model",
          "System > FPU Model",
          "FPU Model",
          "'Automatic' defaults to the current preset model.\nCore restart required.",
@@ -977,7 +977,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_cpu_throttle",
+         "puae2021_cpu_throttle",
          "System > CPU Speed",
          "CPU Speed",
          "Ignored with 'Cycle-exact'.",
@@ -1009,7 +1009,7 @@ static void retro_set_core_options()
          "0.0"
       },
       {
-         "puae_cpu_multiplier",
+         "puae2021_cpu_multiplier",
          "System > CPU Cycle-exact Speed",
          "CPU Cycle-exact Speed",
          "Applies only with 'Cycle-exact'.",
@@ -1029,7 +1029,7 @@ static void retro_set_core_options()
          "0"
       },
       {
-         "puae_cpu_compatibility",
+         "puae2021_cpu_compatibility",
          "System > CPU Compatibility",
          "CPU Compatibility",
          "Some games have graphic and/or speed issues without 'Cycle-exact'. 'Cycle-exact' can be forced with '(CE)' file path tag.",
@@ -1048,7 +1048,7 @@ static void retro_set_core_options()
 #endif
       },
       {
-         "puae_autoloadfastforward",
+         "puae2021_autoloadfastforward",
          "Media > Automatic Load Fast-Forward",
          "Automatic Load Fast-Forward",
          "Toggle frontend fast-forward during media access if there is no audio output. Mutes 'Floppy Sound Emulation'.",
@@ -1065,7 +1065,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_floppy_speed",
+         "puae2021_floppy_speed",
          "Media > Floppy Speed",
          "Floppy Speed",
          "Default speed is 300RPM. 'Turbo' removes disk rotation emulation.",
@@ -1082,7 +1082,7 @@ static void retro_set_core_options()
          "100"
       },
       {
-         "puae_floppy_multidrive",
+         "puae2021_floppy_multidrive",
          "Media > Floppy MultiDrive",
          "Floppy MultiDrive",
          "Insert each disk in different drives. Can be forced with '(MD)' file path tag. Maximum is 4 disks due to external drive limit! Not all games support external drives!\nCore restart required.",
@@ -1096,7 +1096,7 @@ static void retro_set_core_options()
          "enabled"
       },
       {
-         "puae_floppy_write_protection",
+         "puae2021_floppy_write_protection",
          "Media > Floppy Write Protection",
          "Floppy Write Protection",
          "Set all drives read only. Changing this while emulation is running ejects and reinserts all disks. IPF images are always read-only!",
@@ -1110,7 +1110,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_floppy_write_redirect",
+         "puae2021_floppy_write_redirect",
          "Media > Floppy Write Redirect",
          "Floppy Write Redirect",
          "Writes to a substitute disk under 'saves' instead of original disks. Works also with IPF images.",
@@ -1124,7 +1124,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_cd_speed",
+         "puae2021_cd_speed",
          "Media > CD Speed",
          "CD Speed",
          "Transfer rate in CD32 is 300KB/s (double-speed), CDTV is 150KB/s (single-speed). 'Turbo' removes seek delay emulation.",
@@ -1138,7 +1138,7 @@ static void retro_set_core_options()
          "100"
       },
       {
-         "puae_cd_startup_delayed_insert",
+         "puae2021_cd_startup_delayed_insert",
          "Media > CD Startup Delayed Insert",
          "CD Startup Delayed Insert",
          "Some games fail to load if CD32/CDTV is powered on with CD inserted. 'ON' inserts CD during boot animation.",
@@ -1152,7 +1152,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_shared_nvram",
+         "puae2021_shared_nvram",
          "Media > CD32/CDTV Shared NVRAM",
          "CD32/CDTV Shared NVRAM",
          "'OFF' saves separate files per content. Starting without content uses the shared file. CD32 and CDTV use separate shared files.\nCore restart required.",
@@ -1166,7 +1166,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_use_whdload",
+         "puae2021_use_whdload",
          "Media > WHDLoad Support",
          "WHDLoad Support",
          "Enable launching pre-installed WHDLoad installs. Creates a helper boot image for loading content and an empty image for saving. Legacy 'HDFs' mode is not recommended!\nCore restart required.\n- 'Files' creates data in directories\n- 'HDFs' creates data in images\n- 'OFF' boots hard drive images directly",
@@ -1181,7 +1181,7 @@ static void retro_set_core_options()
          "files"
       },
       {
-         "puae_use_whdload_theme",
+         "puae2021_use_whdload_theme",
          "Media > WHDLoad Theme",
          "WHDLoad Theme",
          "AmigaOS 'system-configuration' color prefs in WHDLoad helper image. Available only with 'Files' mode.\nCore restart required.\n- 'Default' = Black/White/DarkGray/LightGray\n- 'Native' = Gray/Black/White/LightBlue",
@@ -1195,7 +1195,7 @@ static void retro_set_core_options()
          "default"
       },
       {
-         "puae_use_whdload_prefs",
+         "puae2021_use_whdload_prefs",
          "Media > WHDLoad Splash Screen",
          "WHDLoad Splash Screen",
          "Space/Enter/Fire works as WHDLoad Start-button. Core restart required.\nOverride with buttons while booting:\n- 'Config': Hold 2nd fire / Blue\n- 'Splash': Hold LMB\n- 'Config + Splash': Hold RMB\n- ReadMe + MkCustom: Hold Red+Blue",
@@ -1211,7 +1211,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_use_boot_hd",
+         "puae2021_use_boot_hd",
          "Media > Global Boot HD",
          "Global Boot HD",
          "Attach a hard disk meant for Workbench, not for WHDLoad! Enabling forces a model with HD interface. Changing HDF size will not replace or edit the existing HDF.\nCore restart required.",
@@ -1232,7 +1232,7 @@ static void retro_set_core_options()
       },
       /* Sublabel and options filled dynamically in retro_set_environment() */
       {
-         "puae_cart_file",
+         "puae2021_cart_file",
          "Media > Cartridge",
          "Cartridge",
          "",
@@ -1244,7 +1244,7 @@ static void retro_set_core_options()
          NULL
       },
       {
-         "puae_video_options_display",
+         "puae2021_video_options_display",
          "Show Video Options",
          NULL,
          "",
@@ -1258,7 +1258,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_video_allow_hz_change",
+         "puae2021_video_allow_hz_change",
          "Video > Allow Hz Change",
          "Allow Hz Change",
          "Let Amiga decide the exact refresh rate when interlace mode or PAL/NTSC changes. 'Locked' changes only when video standard changes.\nCore restart required.",
@@ -1273,7 +1273,7 @@ static void retro_set_core_options()
          "locked"
       },
       {
-         "puae_video_standard",
+         "puae2021_video_standard",
          "Video > Standard",
          "Standard",
          "Output Hz & height:\n- 'PAL': 50Hz - 288px / 576px\n- 'NTSC': 60Hz - 240px / 480px\n- 'Automatic' switches region per file path tags.",
@@ -1289,7 +1289,7 @@ static void retro_set_core_options()
          "PAL auto"
       },
       {
-         "puae_video_resolution",
+         "puae2021_video_resolution",
          "Video > Resolution",
          "Resolution",
          "Output width:\n- 'Automatic' defaults to 'High' and switches to 'Super-High' when needed.",
@@ -1305,7 +1305,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_video_vresolution",
+         "puae2021_video_vresolution",
          "Video > Line Mode",
          "Line Mode",
          "Output height:\n- 'Automatic' defaults to 'Single Line' and switches to 'Double Line' on interlaced screens.",
@@ -1320,7 +1320,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_video_aspect",
+         "puae2021_video_aspect",
          "Video > Pixel Aspect Ratio",
          "Pixel Aspect Ratio",
          "Hotkey toggling disables this option until core restart.\n- 'PAL': 1/1 = 1.000\n- 'NTSC': 44/52 = 0.846",
@@ -1335,7 +1335,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_crop",
+         "puae2021_crop",
          "Video > Crop",
          "Crop",
          "Remove borders according to 'Crop Mode'.",
@@ -1356,7 +1356,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_zoom_mode",
+         "puae2021_zoom_mode",
          "Video > Zoom Mode",
          "Zoom Mode",
          "Hidden placeholder for backwards compatibility.",
@@ -1378,7 +1378,7 @@ static void retro_set_core_options()
          "deprecated"
       },
       {
-         "puae_crop_mode",
+         "puae2021_crop_mode",
          "Video > Crop Mode",
          "Crop Mode",
          "'Horizontal + Vertical' & 'Automatic' removes borders completely.",
@@ -1397,7 +1397,7 @@ static void retro_set_core_options()
          "both"
       },
       {
-         "puae_zoom_mode_crop",
+         "puae2021_zoom_mode_crop",
          "Video > Zoom Mode Crop",
          "Zoom Mode Crop",
          "Hidden placeholder for backwards compatibility.",
@@ -1417,7 +1417,7 @@ static void retro_set_core_options()
          "deprecated"
       },
       {
-         "puae_vertical_pos",
+         "puae2021_vertical_pos",
          "Video > Vertical Position",
          "Vertical Position",
          "'Automatic' keeps only cropped screens centered. Positive values move upward and negative values move downward.",
@@ -1476,7 +1476,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_horizontal_pos",
+         "puae2021_horizontal_pos",
          "Video > Horizontal Position",
          "Horizontal Position",
          "'Automatic' keeps screen centered. Positive values move right and negative values move left.",
@@ -1530,7 +1530,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_immediate_blits",
+         "puae2021_immediate_blits",
          "Video > Immediate/Waiting Blits",
          "Immediate/Waiting Blits",
          "- 'Immediate Blitter': Faster but less compatible blitter emulation.\n- 'Wait for Blitter': Compatibility hack for programs that don't wait for the blitter correctly.",
@@ -1545,7 +1545,7 @@ static void retro_set_core_options()
          "waiting"
       },
       {
-         "puae_collision_level",
+         "puae2021_collision_level",
          "Video > Collision Level",
          "Collision Level",
          "'Sprites and Playfields' is recommended.",
@@ -1561,7 +1561,7 @@ static void retro_set_core_options()
          "playfields"
       },
       {
-         "puae_gfx_flickerfixer",
+         "puae2021_gfx_flickerfixer",
          "Video > Remove Interlace Artifacts",
          "Remove Interlace Artifacts",
          "Best suited for still screens, Workbench etc.",
@@ -1575,7 +1575,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_gfx_framerate",
+         "puae2021_gfx_framerate",
          "Video > Frameskip",
          "Frameskip",
          "Not compatible with 'Cycle-exact'.",
@@ -1590,7 +1590,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_gfx_gamma",
+         "puae2021_gfx_gamma",
          "Video > Color Gamma",
          "Color Gamma",
          "Adjust color gamma.",
@@ -1613,7 +1613,7 @@ static void retro_set_core_options()
          "0"
       },
       {
-         "puae_gfx_colors",
+         "puae2021_gfx_colors",
          "Video > Color Depth",
          "Color Depth",
          "'24-bit' is slower and not available on all platforms. Full restart required.",
@@ -1627,7 +1627,7 @@ static void retro_set_core_options()
          "16bit"
       },
       {
-         "puae_vkbd_theme",
+         "puae2021_vkbd_theme",
          "OSD > Virtual KBD Theme",
          "Virtual KBD Theme",
          "The keyboard comes up with RetroPad Select by default.",
@@ -1649,7 +1649,7 @@ static void retro_set_core_options()
          "auto"
       },
       {
-         "puae_vkbd_transparency",
+         "puae2021_vkbd_transparency",
          "OSD > Virtual KBD Transparency",
          "Virtual KBD Transparency",
          "Keyboard transparency can be toggled with RetroPad A.",
@@ -1666,7 +1666,7 @@ static void retro_set_core_options()
          "25%"
       },
       {
-         "puae_vkbd_dimming",
+         "puae2021_vkbd_dimming",
          "OSD > Virtual KBD Dimming",
          "Virtual KBD Dimming",
          "Dimming level of the surrounding area.",
@@ -1683,7 +1683,7 @@ static void retro_set_core_options()
          "25%"
       },
       {
-         "puae_statusbar",
+         "puae2021_statusbar",
          "OSD > Statusbar Mode",
          "Statusbar Mode",
          "- 'Full': Joyports + Messages + LEDs\n- 'Basic': Messages + LEDs\n- 'Minimal': LED colors only",
@@ -1703,7 +1703,7 @@ static void retro_set_core_options()
          "bottom"
       },
       {
-         "puae_statusbar_messages",
+         "puae2021_statusbar_messages",
          "OSD > Statusbar Messages",
          "Statusbar Messages",
          "Show messages when statusbar is hidden.",
@@ -1717,7 +1717,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_audio_options_display",
+         "puae2021_audio_options_display",
          "Show Audio Options",
          NULL,
          "",
@@ -1731,7 +1731,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_sound_stereo_separation",
+         "puae2021_sound_stereo_separation",
          "Audio > Stereo Separation",
          "Stereo Separation",
          "Paula sound chip channel panning. Does not affect CD audio.",
@@ -1754,7 +1754,7 @@ static void retro_set_core_options()
          "100%"
       },
       {
-         "puae_sound_interpol",
+         "puae2021_sound_interpol",
          "Audio > Interpolation",
          "Interpolation",
          "Paula sound chip interpolation type.",
@@ -1771,7 +1771,7 @@ static void retro_set_core_options()
          "anti"
       },
       {
-         "puae_sound_filter",
+         "puae2021_sound_filter",
          "Audio > Filter",
          "Filter",
          "'Emulated' allows states between ON/OFF.",
@@ -1786,7 +1786,7 @@ static void retro_set_core_options()
          "emulated"
       },
       {
-         "puae_sound_filter_type",
+         "puae2021_sound_filter_type",
          "Audio > Filter Type",
          "Filter Type",
          "'Automatic' picks the filter type for the hardware.",
@@ -1801,7 +1801,7 @@ static void retro_set_core_options()
          "auto",
       },
       {
-         "puae_floppy_sound",
+         "puae2021_floppy_sound",
          "Audio > Floppy Sound Emulation",
          "Floppy Sound Emulation",
          "Floppy volume in percent.",
@@ -1834,7 +1834,7 @@ static void retro_set_core_options()
          "80"
       },
       {
-         "puae_floppy_sound_empty_mute",
+         "puae2021_floppy_sound_empty_mute",
          "Audio > Floppy Sound Mute Ejected",
          "Floppy Sound Mute Ejected",
          "Mute drive head clicking when floppy is not inserted.",
@@ -1848,7 +1848,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_floppy_sound_type",
+         "puae2021_floppy_sound_type",
          "Audio > Floppy Sound Type",
          "Floppy Sound Type",
          "External files go in 'system/uae/' or 'system/uae_data/'.",
@@ -1863,7 +1863,7 @@ static void retro_set_core_options()
          "internal"
       },
       {
-         "puae_sound_volume_cd",
+         "puae2021_sound_volume_cd",
          "Audio > CD Audio Volume",
          "CD Audio Volume",
          "CD volume in percent.",
@@ -1896,7 +1896,7 @@ static void retro_set_core_options()
          "100%"
       },
       {
-         "puae_analogmouse",
+         "puae2021_analogmouse",
          "Input > Analog Stick Mouse",
          "Analog Stick Mouse",
          "Default mouse control stick when remappings are empty.",
@@ -1912,7 +1912,7 @@ static void retro_set_core_options()
          "both"
       },
       {
-         "puae_analogmouse_deadzone",
+         "puae2021_analogmouse_deadzone",
          "Input > Analog Stick Mouse Deadzone",
          "Analog Stick Mouse Deadzone",
          "Required distance from stick center to register input.",
@@ -1935,7 +1935,7 @@ static void retro_set_core_options()
          "20"
       },
       {
-         "puae_analogmouse_speed",
+         "puae2021_analogmouse_speed",
          "Input > Analog Stick Mouse Speed",
          "Analog Stick Mouse Speed",
          "Mouse movement speed multiplier for analog stick.",
@@ -1977,7 +1977,7 @@ static void retro_set_core_options()
          "1.0"
       },
       {
-         "puae_dpadmouse_speed",
+         "puae2021_dpadmouse_speed",
          "Input > D-Pad Mouse Speed",
          "D-Pad Mouse Speed",
          "Mouse movement speed multiplier for directional pad.",
@@ -2007,7 +2007,7 @@ static void retro_set_core_options()
          "6"
       },
       {
-         "puae_mouse_speed",
+         "puae2021_mouse_speed",
          "Input > Mouse Speed",
          "Mouse Speed",
          "Global mouse speed.",
@@ -2049,7 +2049,7 @@ static void retro_set_core_options()
          "100"
       },
       {
-         "puae_physicalmouse",
+         "puae2021_physicalmouse",
          "Input > Physical Mouse",
          "Physical Mouse",
          "'Double' requirements: raw/udev input driver and proper mouse index per port.\nDoes not affect RetroPad emulated mice.",
@@ -2064,7 +2064,7 @@ static void retro_set_core_options()
          "enabled"
       },
       {
-         "puae_physical_keyboard_pass_through",
+         "puae2021_physical_keyboard_pass_through",
          "Input > Keyboard Pass-through",
          "Keyboard Pass-through",
          "'ON' passes all physical keyboard events to the core. 'OFF' prevents RetroPad keys from generating keyboard events.",
@@ -2078,7 +2078,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_keyrah_keypad_mappings",
+         "puae2021_keyrah_keypad_mappings",
          "Input > Keyrah Keypad Mappings",
          "Keyrah Keypad Mappings",
          "Hardcoded keypad to joyport mappings for Keyrah hardware.",
@@ -2092,7 +2092,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_mapping_options_display",
+         "puae2021_mapping_options_display",
          "Show Mapping Options",
          NULL,
          "",
@@ -2107,7 +2107,7 @@ static void retro_set_core_options()
       },
       /* Hotkeys */
       {
-         "puae_mapper_vkbd",
+         "puae2021_mapper_vkbd",
          "Hotkey > Toggle Virtual Keyboard",
          "Toggle Virtual Keyboard",
          "Press the mapped key to toggle the virtual keyboard.",
@@ -2117,7 +2117,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_statusbar",
+         "puae2021_mapper_statusbar",
          "Hotkey > Toggle Statusbar",
          "Toggle Statusbar",
          "Press the mapped key to toggle the statusbar.",
@@ -2127,7 +2127,7 @@ static void retro_set_core_options()
          "RETROK_F12"
       },
       {
-         "puae_mapper_mouse_toggle",
+         "puae2021_mapper_mouse_toggle",
          "Hotkey > Switch Joystick/Mouse",
          "Switch Joystick/Mouse",
          "Press the mapped key to switch between joystick and mouse control.",
@@ -2137,7 +2137,7 @@ static void retro_set_core_options()
          "RETROK_RCTRL"
       },
       {
-         "puae_mapper_turbo_fire_toggle",
+         "puae2021_mapper_turbo_fire_toggle",
          "Hotkey > Toggle Turbo Fire",
          "Toggle Turbo Fire",
          "Press the mapped key to toggle turbo fire.",
@@ -2147,7 +2147,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_save_disk_toggle",
+         "puae2021_mapper_save_disk_toggle",
          "Hotkey > Toggle Save Disk",
          "Toggle Save Disk",
          "Press the mapped key to create/insert/eject a save disk.",
@@ -2157,7 +2157,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_reset",
+         "puae2021_mapper_reset",
          "Hotkey > Reset",
          "Reset",
          "Press the mapped key to trigger soft reset (Ctrl-Amiga-Amiga).",
@@ -2167,7 +2167,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_aspect_ratio_toggle",
+         "puae2021_mapper_aspect_ratio_toggle",
          "Hotkey > Toggle Aspect Ratio",
          "Toggle Aspect Ratio",
          "Press the mapped key to toggle between PAL/NTSC pixel aspect ratio.",
@@ -2177,7 +2177,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_crop_toggle",
+         "puae2021_mapper_crop_toggle",
          "Hotkey > Toggle Crop",
          "Toggle Crop",
          "Press the mapped key to toggle crop.",
@@ -2187,7 +2187,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_zoom_mode_toggle",
+         "puae2021_mapper_zoom_mode_toggle",
          "Hotkey > Toggle Zoom Mode",
          "Toggle Zoom Mode",
          "Hidden placeholder for backwards compatibility.",
@@ -2198,7 +2198,7 @@ static void retro_set_core_options()
       },
       /* Button mappings */
       {
-         "puae_mapper_up",
+         "puae2021_mapper_up",
          "RetroPad > Up",
          "Up",
          "Unmapped defaults to joystick up.",
@@ -2208,7 +2208,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_down",
+         "puae2021_mapper_down",
          "RetroPad > Down",
          "Down",
          "Unmapped defaults to joystick down.",
@@ -2218,7 +2218,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_left",
+         "puae2021_mapper_left",
          "RetroPad > Left",
          "Left",
          "Unmapped defaults to joystick left.",
@@ -2228,7 +2228,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_right",
+         "puae2021_mapper_right",
          "RetroPad > Right",
          "Right",
          "Unmapped defaults to joystick right.",
@@ -2238,7 +2238,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_a",
+         "puae2021_mapper_a",
          "RetroPad > A",
          "A",
          "Unmapped defaults to 2nd fire button.\nVKBD: Toggle transparency. Remapping to non-keyboard keys overrides VKBD function!",
@@ -2248,7 +2248,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_b",
+         "puae2021_mapper_b",
          "RetroPad > B",
          "B",
          "Unmapped defaults to fire button.\nVKBD: Press selected key.",
@@ -2258,7 +2258,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_x",
+         "puae2021_mapper_x",
          "RetroPad > X",
          "X",
          "VKBD: Press 'Space'. Remapping to non-keyboard keys overrides VKBD function!",
@@ -2268,7 +2268,7 @@ static void retro_set_core_options()
          "RETROK_SPACE"
       },
       {
-         "puae_mapper_y",
+         "puae2021_mapper_y",
          "RetroPad > Y",
          "Y",
          "VKBD: Toggle 'CapsLock'. Remapping to non-keyboard keys overrides VKBD function!",
@@ -2278,7 +2278,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_select",
+         "puae2021_mapper_select",
          "RetroPad > Select",
          "Select",
          "VKBD comes up with RetroPad Select by default.",
@@ -2288,7 +2288,7 @@ static void retro_set_core_options()
          "TOGGLE_VKBD"
       },
       {
-         "puae_mapper_start",
+         "puae2021_mapper_start",
          "RetroPad > Start",
          "Start",
          "VKBD: Press 'Return'. Remapping to non-keyboard keys overrides VKBD function!",
@@ -2298,7 +2298,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_l",
+         "puae2021_mapper_l",
          "RetroPad > L",
          "L",
          "",
@@ -2308,7 +2308,7 @@ static void retro_set_core_options()
          ""
       },
       {
-         "puae_mapper_r",
+         "puae2021_mapper_r",
          "RetroPad > R",
          "R",
          "",
@@ -2318,7 +2318,7 @@ static void retro_set_core_options()
          ""
       },
       {
-         "puae_mapper_l2",
+         "puae2021_mapper_l2",
          "RetroPad > L2",
          "L2",
          "",
@@ -2328,7 +2328,7 @@ static void retro_set_core_options()
          "MOUSE_LEFT_BUTTON"
       },
       {
-         "puae_mapper_r2",
+         "puae2021_mapper_r2",
          "RetroPad > R2",
          "R2",
          "",
@@ -2338,7 +2338,7 @@ static void retro_set_core_options()
          "MOUSE_RIGHT_BUTTON"
       },
       {
-         "puae_mapper_l3",
+         "puae2021_mapper_l3",
          "RetroPad > L3",
          "L3",
          "",
@@ -2348,7 +2348,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_r3",
+         "puae2021_mapper_r3",
          "RetroPad > R3",
          "R3",
          "",
@@ -2359,7 +2359,7 @@ static void retro_set_core_options()
       },
       /* Left Stick */
       {
-         "puae_mapper_lu",
+         "puae2021_mapper_lu",
          "RetroPad > Left Analog > Up",
          "Left Analog > Up",
          "",
@@ -2369,7 +2369,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_ld",
+         "puae2021_mapper_ld",
          "RetroPad > Left Analog > Down",
          "Left Analog > Down",
          "",
@@ -2379,7 +2379,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_ll",
+         "puae2021_mapper_ll",
          "RetroPad > Left Analog > Left",
          "Left Analog > Left",
          "",
@@ -2389,7 +2389,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_lr",
+         "puae2021_mapper_lr",
          "RetroPad > Left Analog > Right",
          "Left Analog > Right",
          "",
@@ -2400,7 +2400,7 @@ static void retro_set_core_options()
       },
       /* Right Stick */
       {
-         "puae_mapper_ru",
+         "puae2021_mapper_ru",
          "RetroPad > Right Analog > Up",
          "Right Analog > Up",
          "",
@@ -2410,7 +2410,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_rd",
+         "puae2021_mapper_rd",
          "RetroPad > Right Analog > Down",
          "Right Analog > Down",
          "",
@@ -2420,7 +2420,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_rl",
+         "puae2021_mapper_rl",
          "RetroPad > Right Analog > Left",
          "Right Analog > Left",
          "",
@@ -2430,7 +2430,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_mapper_rr",
+         "puae2021_mapper_rr",
          "RetroPad > Right Analog > Right",
          "Right Analog > Right",
          "",
@@ -2440,7 +2440,7 @@ static void retro_set_core_options()
          "---"
       },
       {
-         "puae_turbo_fire",
+         "puae2021_turbo_fire",
          "RetroPad > Turbo Fire",
          "Turbo Fire",
          "Hotkey toggling disables this option until core restart.",
@@ -2454,7 +2454,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_turbo_fire_button",
+         "puae2021_turbo_fire_button",
          "RetroPad > Turbo Button",
          "Turbo Button",
          "Replace the mapped button with turbo fire button.",
@@ -2474,7 +2474,7 @@ static void retro_set_core_options()
          "B"
       },
       {
-         "puae_turbo_pulse",
+         "puae2021_turbo_pulse",
          "RetroPad > Turbo Pulse",
          "Turbo Pulse",
          "Frames in a button cycle.\n- '2' = 1 frame down, 1 frame up\n- '4' = 2 frames down, 2 frames up\n- '6' = 3 frames down, 3 frames up\netc.",
@@ -2492,7 +2492,7 @@ static void retro_set_core_options()
          "6"
       },
       {
-         "puae_joyport",
+         "puae2021_joyport",
          "RetroPad > Joystick/Mouse",
          "Joystick/Mouse",
          "Change D-Pad control between joyports. Hotkey toggling disables this option until core restart.",
@@ -2506,7 +2506,7 @@ static void retro_set_core_options()
          "Joystick"
       },
       {
-         "puae_joyport_order",
+         "puae2021_joyport_order",
          "RetroPad > Joystick Port Order",
          "Joystick Port Order",
          "Plug RetroPads in different ports. Useful for Arcadia system and games that use the 4-player adapter.",
@@ -2522,7 +2522,7 @@ static void retro_set_core_options()
          "1234"
       },
       {
-         "puae_retropad_options",
+         "puae2021_retropad_options",
          "RetroPad > Face Button Options",
          "RetroPad Face Button Options",
          "Rotate face buttons clockwise and/or make 2nd fire press up.",
@@ -2538,7 +2538,7 @@ static void retro_set_core_options()
          "disabled"
       },
       {
-         "puae_cd32pad_options",
+         "puae2021_cd32pad_options",
          "CD32 Pad > Face Button Options",
          "CD32 Pad Face Button Options",
          "Rotate face buttons clockwise and/or make blue button press up.",
@@ -2556,7 +2556,7 @@ static void retro_set_core_options()
       { NULL, NULL, NULL, NULL, NULL, NULL, {{0}}, NULL },
    };
 
-   free_puae_carts();
+   free_puae2021_carts();
 
    /* Fill in the values for all the mappers */
    int i = 0;
@@ -2572,17 +2572,17 @@ static void retro_set_core_options()
    }
    while (option_defs_us[i].key)
    {
-      if (strstr(option_defs_us[i].key, "puae_mapper_"))
+      if (strstr(option_defs_us[i].key, "puae2021_mapper_"))
       {
          /* Show different key list for hotkeys (special negatives removed) */
-         if (  strstr(option_defs_us[i].key, "puae_mapper_vkbd")
-            || strstr(option_defs_us[i].key, "puae_mapper_statusbar")
-            || strstr(option_defs_us[i].key, "puae_mapper_mouse_toggle")
-            || strstr(option_defs_us[i].key, "puae_mapper_reset")
-            || strstr(option_defs_us[i].key, "puae_mapper_aspect_ratio_toggle")
-            || strstr(option_defs_us[i].key, "puae_mapper_crop_toggle")
-            || strstr(option_defs_us[i].key, "puae_mapper_turbo_fire_toggle")
-            || strstr(option_defs_us[i].key, "puae_mapper_save_disk_toggle"))
+         if (  strstr(option_defs_us[i].key, "puae2021_mapper_vkbd")
+            || strstr(option_defs_us[i].key, "puae2021_mapper_statusbar")
+            || strstr(option_defs_us[i].key, "puae2021_mapper_mouse_toggle")
+            || strstr(option_defs_us[i].key, "puae2021_mapper_reset")
+            || strstr(option_defs_us[i].key, "puae2021_mapper_aspect_ratio_toggle")
+            || strstr(option_defs_us[i].key, "puae2021_mapper_crop_toggle")
+            || strstr(option_defs_us[i].key, "puae2021_mapper_turbo_fire_toggle")
+            || strstr(option_defs_us[i].key, "puae2021_mapper_save_disk_toggle"))
             hotkey = 1;
          else
             hotkey = 0;
@@ -2617,7 +2617,7 @@ static void retro_set_core_options()
          option_defs_us[i].values[j].value = NULL;
          option_defs_us[i].values[j].label = NULL;
       }
-      else if (!strcmp(option_defs_us[i].key, "puae_cart_file"))
+      else if (!strcmp(option_defs_us[i].key, "puae2021_cart_file"))
       {
          j = 0;
          option_defs_us[i].values[j].value = "none";
@@ -2640,16 +2640,16 @@ static void retro_set_core_options()
                   snprintf(cart_value, sizeof(cart_value), "%s", cart_dirp->d_name);
                   snprintf(cart_label, sizeof(cart_label), "%s", path_remove_extension(cart_dirp->d_name));
 
-                  puae_carts[j].value = strdup(cart_value);
-                  puae_carts[j].label = strdup(cart_label);
+                  puae2021_carts[j].value = strdup(cart_value);
+                  puae2021_carts[j].label = strdup(cart_label);
 
-                  option_defs_us[i].values[j].value = puae_carts[j].value;
-                  option_defs_us[i].values[j].label = puae_carts[j].label;
+                  option_defs_us[i].values[j].value = puae2021_carts[j].value;
+                  option_defs_us[i].values[j].label = puae2021_carts[j].label;
                   ++j;
                }
 
-               puae_carts[j].value = NULL;
-               puae_carts[j].label = NULL;
+               puae2021_carts[j].value = NULL;
+               puae2021_carts[j].label = NULL;
             }
             closedir(cart_dir);
          }
@@ -2982,11 +2982,11 @@ static void retro_set_options_display(void)
 
    /* Legacy zoom always hidden */
    option_display.visible = false;
-   option_display.key = "puae_zoom_mode";
+   option_display.key = "puae2021_zoom_mode";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_zoom_mode_crop";
+   option_display.key = "puae2021_zoom_mode_crop";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_zoom_mode_toggle";
+   option_display.key = "puae2021_mapper_zoom_mode_toggle";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
 
    /*** Options display ***/
@@ -2994,16 +2994,16 @@ static void retro_set_options_display(void)
    {
       option_display.visible = false;
 
-      option_display.key = "puae_mapping_options_display";
+      option_display.key = "puae2021_mapping_options_display";
       environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
 
-      option_display.key = "puae_video_options_display";
+      option_display.key = "puae2021_video_options_display";
       environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
 
-      option_display.key = "puae_audio_options_display";
+      option_display.key = "puae2021_audio_options_display";
       environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
 
-      option_display.key = "puae_model_options_display";
+      option_display.key = "puae2021_model_options_display";
       environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
       return;
    }
@@ -3011,143 +3011,143 @@ static void retro_set_options_display(void)
    /* Model options */
    option_display.visible = opt_model_options_display;
 
-   option_display.key = "puae_model_fd";
+   option_display.key = "puae2021_model_fd";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_model_hd";
+   option_display.key = "puae2021_model_hd";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_model_cd";
+   option_display.key = "puae2021_model_cd";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
 
    /* Video options */
    option_display.visible = opt_video_options_display;
 
-   option_display.key = "puae_video_resolution";
+   option_display.key = "puae2021_video_resolution";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_video_vresolution";
+   option_display.key = "puae2021_video_vresolution";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_video_allow_hz_change";
+   option_display.key = "puae2021_video_allow_hz_change";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_video_standard";
+   option_display.key = "puae2021_video_standard";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_video_aspect";
+   option_display.key = "puae2021_video_aspect";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_crop";
+   option_display.key = "puae2021_crop";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_crop_mode";
+   option_display.key = "puae2021_crop_mode";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_vertical_pos";
+   option_display.key = "puae2021_vertical_pos";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_horizontal_pos";
+   option_display.key = "puae2021_horizontal_pos";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_collision_level";
+   option_display.key = "puae2021_collision_level";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_immediate_blits";
+   option_display.key = "puae2021_immediate_blits";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_gfx_flickerfixer";
+   option_display.key = "puae2021_gfx_flickerfixer";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_gfx_framerate";
+   option_display.key = "puae2021_gfx_framerate";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_gfx_colors";
+   option_display.key = "puae2021_gfx_colors";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_gfx_gamma";
+   option_display.key = "puae2021_gfx_gamma";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_statusbar";
+   option_display.key = "puae2021_statusbar";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_statusbar_messages";
+   option_display.key = "puae2021_statusbar_messages";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_vkbd_theme";
+   option_display.key = "puae2021_vkbd_theme";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_vkbd_transparency";
+   option_display.key = "puae2021_vkbd_transparency";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_vkbd_dimming";
+   option_display.key = "puae2021_vkbd_dimming";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
 
    /* Audio options */
    option_display.visible = opt_audio_options_display;
 
-   option_display.key = "puae_sound_stereo_separation";
+   option_display.key = "puae2021_sound_stereo_separation";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_sound_interpol";
+   option_display.key = "puae2021_sound_interpol";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_sound_filter";
+   option_display.key = "puae2021_sound_filter";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_sound_filter_type";
+   option_display.key = "puae2021_sound_filter_type";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_sound_volume_cd";
+   option_display.key = "puae2021_sound_volume_cd";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_floppy_sound";
+   option_display.key = "puae2021_floppy_sound";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_floppy_sound_empty_mute";
+   option_display.key = "puae2021_floppy_sound_empty_mute";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_floppy_sound_type";
+   option_display.key = "puae2021_floppy_sound_type";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
 
    /* Mapping options */
    option_display.visible = opt_mapping_options_display;
 
-   option_display.key = "puae_mapper_up";
+   option_display.key = "puae2021_mapper_up";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_down";
+   option_display.key = "puae2021_mapper_down";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_left";
+   option_display.key = "puae2021_mapper_left";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_right";
+   option_display.key = "puae2021_mapper_right";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_select";
+   option_display.key = "puae2021_mapper_select";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_start";
+   option_display.key = "puae2021_mapper_start";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_b";
+   option_display.key = "puae2021_mapper_b";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_a";
+   option_display.key = "puae2021_mapper_a";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_y";
+   option_display.key = "puae2021_mapper_y";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_x";
+   option_display.key = "puae2021_mapper_x";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_l";
+   option_display.key = "puae2021_mapper_l";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_r";
+   option_display.key = "puae2021_mapper_r";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_l2";
+   option_display.key = "puae2021_mapper_l2";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_r2";
+   option_display.key = "puae2021_mapper_r2";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_l3";
+   option_display.key = "puae2021_mapper_l3";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_r3";
+   option_display.key = "puae2021_mapper_r3";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_lu";
+   option_display.key = "puae2021_mapper_lu";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_ld";
+   option_display.key = "puae2021_mapper_ld";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_ll";
+   option_display.key = "puae2021_mapper_ll";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_lr";
+   option_display.key = "puae2021_mapper_lr";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_ru";
+   option_display.key = "puae2021_mapper_ru";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_rd";
+   option_display.key = "puae2021_mapper_rd";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_rl";
+   option_display.key = "puae2021_mapper_rl";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_rr";
+   option_display.key = "puae2021_mapper_rr";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_vkbd";
+   option_display.key = "puae2021_mapper_vkbd";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_statusbar";
+   option_display.key = "puae2021_mapper_statusbar";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_mouse_toggle";
+   option_display.key = "puae2021_mapper_mouse_toggle";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_reset";
+   option_display.key = "puae2021_mapper_reset";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_aspect_ratio_toggle";
+   option_display.key = "puae2021_mapper_aspect_ratio_toggle";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_crop_toggle";
+   option_display.key = "puae2021_mapper_crop_toggle";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_turbo_fire_toggle";
+   option_display.key = "puae2021_mapper_turbo_fire_toggle";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
-   option_display.key = "puae_mapper_save_disk_toggle";
+   option_display.key = "puae2021_mapper_save_disk_toggle";
    environ_cb(RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY, &option_display);
 }
 
@@ -3230,42 +3230,42 @@ static void update_variables(void)
    uae_model[0]  = '\0';
    uae_config[0] = '\0';
 
-   var.key = "puae_model";
+   var.key = "puae2021_model";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       strlcpy(opt_model, var.value, sizeof(opt_model));
    }
 
-   var.key = "puae_model_fd";
+   var.key = "puae2021_model_fd";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       strlcpy(opt_model_fd, var.value, sizeof(opt_model_fd));
    }
 
-   var.key = "puae_model_hd";
+   var.key = "puae2021_model_hd";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       strlcpy(opt_model_hd, var.value, sizeof(opt_model_hd));
    }
 
-   var.key = "puae_model_cd";
+   var.key = "puae2021_model_cd";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       strlcpy(opt_model_cd, var.value, sizeof(opt_model_cd));
    }
 
-   var.key = "puae_kickstart";
+   var.key = "puae2021_kickstart";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       strlcpy(opt_kickstart, var.value, sizeof(opt_kickstart));
    }
 
-   var.key = "puae_cart_file";
+   var.key = "puae2021_cart_file";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3285,7 +3285,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_video_standard";
+   var.key = "puae2021_video_standard";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3324,7 +3324,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_video_aspect";
+   var.key = "puae2021_video_aspect";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3339,7 +3339,7 @@ static void update_variables(void)
          video_config_aspect = video_config_aspect_prev;
    }
 
-   var.key = "puae_video_allow_hz_change";
+   var.key = "puae2021_video_allow_hz_change";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3361,7 +3361,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_video_resolution";
+   var.key = "puae2021_video_resolution";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3423,7 +3423,7 @@ static void update_variables(void)
          request_init_custom_timer = 2;
    }
 
-   var.key = "puae_video_vresolution";
+   var.key = "puae2021_video_vresolution";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3472,7 +3472,7 @@ static void update_variables(void)
          request_init_custom_timer = 2;
    }
 
-   var.key = "puae_statusbar";
+   var.key = "puae2021_statusbar";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3498,7 +3498,7 @@ static void update_variables(void)
       opt_statusbar_position_old = opt_statusbar_position;
    }
 
-   var.key = "puae_statusbar_messages";
+   var.key = "puae2021_statusbar_messages";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3506,7 +3506,7 @@ static void update_variables(void)
          opt_statusbar |= STATUSBAR_MESSAGES;
    }
 
-   var.key = "puae_vkbd_theme";
+   var.key = "puae2021_vkbd_theme";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3519,7 +3519,7 @@ static void update_variables(void)
       if      (strstr(var.value, "outline")) opt_vkbd_theme |= 0x80;
    }
 
-   var.key = "puae_vkbd_transparency";
+   var.key = "puae2021_vkbd_transparency";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3530,7 +3530,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "100%")) opt_vkbd_alpha = GRAPH_ALPHA_0;
    }
 
-   var.key = "puae_vkbd_dimming";
+   var.key = "puae2021_vkbd_dimming";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3541,7 +3541,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "100%")) opt_vkbd_dim_alpha = GRAPH_ALPHA_100;
    }
 
-   var.key = "puae_chipmem_size";
+   var.key = "puae2021_chipmem_size";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3556,7 +3556,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_bogomem_size";
+   var.key = "puae2021_bogomem_size";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3571,7 +3571,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_fastmem_size";
+   var.key = "puae2021_fastmem_size";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3586,7 +3586,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_z3mem_size";
+   var.key = "puae2021_z3mem_size";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3603,7 +3603,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_cpu_model";
+   var.key = "puae2021_cpu_model";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3633,7 +3633,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_fpu_model";
+   var.key = "puae2021_fpu_model";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3659,7 +3659,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_cpu_compatibility";
+   var.key = "puae2021_cpu_compatibility";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3702,7 +3702,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_cpu_throttle";
+   var.key = "puae2021_cpu_throttle";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3714,7 +3714,7 @@ static void update_variables(void)
          changed_prefs.m68k_speed_throttle = atof(var.value);
    }
 
-   var.key = "puae_cpu_multiplier";
+   var.key = "puae2021_cpu_multiplier";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3726,7 +3726,7 @@ static void update_variables(void)
          changed_prefs.cpu_clock_multiplier = atoi(var.value) * 256;
    }
 
-   var.key = "puae_sound_stereo_separation";
+   var.key = "puae2021_sound_stereo_separation";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3742,7 +3742,7 @@ static void update_variables(void)
          changed_prefs.sound_stereo_separation = val;
    }
 
-   var.key = "puae_sound_interpol";
+   var.key = "puae2021_sound_interpol";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3760,7 +3760,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_sound_filter";
+   var.key = "puae2021_sound_filter";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3776,7 +3776,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_sound_filter_type";
+   var.key = "puae2021_sound_filter_type";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3796,7 +3796,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_sound_volume_cd";
+   var.key = "puae2021_sound_volume_cd";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3813,7 +3813,7 @@ static void update_variables(void)
          changed_prefs.sound_volume_cd = val;
    }
 
-   var.key = "puae_autoloadfastforward";
+   var.key = "puae2021_autoloadfastforward";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3826,7 +3826,7 @@ static void update_variables(void)
                                                                          AUTOLOADFASTFORWARD_CD;
    }
 
-   var.key = "puae_cd_speed";
+   var.key = "puae2021_cd_speed";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3838,7 +3838,7 @@ static void update_variables(void)
          changed_prefs.cd_speed = atoi(var.value);
    }
 
-   var.key = "puae_floppy_speed";
+   var.key = "puae2021_floppy_speed";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3850,7 +3850,7 @@ static void update_variables(void)
          changed_prefs.floppy_speed = atoi(var.value);
    }
 
-   var.key = "puae_floppy_multidrive";
+   var.key = "puae2021_floppy_multidrive";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3858,7 +3858,7 @@ static void update_variables(void)
       else                                opt_floppy_multidrive = true;
    }
 
-   var.key = "puae_floppy_write_protection";
+   var.key = "puae2021_floppy_write_protection";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3891,7 +3891,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_floppy_write_redirect";
+   var.key = "puae2021_floppy_write_redirect";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3929,7 +3929,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_floppy_sound";
+   var.key = "puae2021_floppy_sound";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3951,7 +3951,7 @@ static void update_variables(void)
          changed_prefs.dfxclickvolume = atoi(val);
    }
 
-   var.key = "puae_floppy_sound_empty_mute";
+   var.key = "puae2021_floppy_sound_empty_mute";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -3959,7 +3959,7 @@ static void update_variables(void)
       else                                opt_floppy_sound_empty_mute = true;
    }
 
-   var.key = "puae_floppy_sound_type";
+   var.key = "puae2021_floppy_sound_type";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4008,7 +4008,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_mouse_speed";
+   var.key = "puae2021_mouse_speed";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4024,7 +4024,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_immediate_blits";
+   var.key = "puae2021_immediate_blits";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4064,7 +4064,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_collision_level";
+   var.key = "puae2021_collision_level";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4081,7 +4081,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_gfx_framerate";
+   var.key = "puae2021_gfx_framerate";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4103,7 +4103,7 @@ static void update_variables(void)
          changed_prefs.gfx_framerate = val;
    }
 
-   var.key = "puae_gfx_colors";
+   var.key = "puae2021_gfx_colors";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4115,7 +4115,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_gfx_gamma";
+   var.key = "puae2021_gfx_gamma";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4137,7 +4137,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_gfx_flickerfixer";
+   var.key = "puae2021_gfx_flickerfixer";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4157,7 +4157,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_crop";
+   var.key = "puae2021_crop";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4174,7 +4174,7 @@ static void update_variables(void)
       opt_crop_id = crop_id;
    }
 
-   var.key = "puae_crop_mode";
+   var.key = "puae2021_crop_mode";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4187,7 +4187,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "5:4"))        crop_mode_id = CROP_MODE_5_4;
    }
 
-   var.key = "puae_vertical_pos";
+   var.key = "puae2021_vertical_pos";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4210,7 +4210,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_horizontal_pos";
+   var.key = "puae2021_horizontal_pos";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4231,7 +4231,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_use_whdload";
+   var.key = "puae2021_use_whdload";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4240,7 +4240,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "hdfs"))     opt_use_whdload = 2;
    }
 
-   var.key = "puae_use_whdload_theme";
+   var.key = "puae2021_use_whdload_theme";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4248,7 +4248,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "native"))   opt_use_whdload_theme = 1;
    }
 
-   var.key = "puae_use_whdload_prefs";
+   var.key = "puae2021_use_whdload_prefs";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4258,7 +4258,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "both"))     opt_use_whdload_prefs = 3;
    }
 
-   var.key = "puae_cd_startup_delayed_insert";
+   var.key = "puae2021_cd_startup_delayed_insert";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4266,7 +4266,7 @@ static void update_variables(void)
       else                                opt_cd_startup_delayed_insert = true;
    }
 
-   var.key = "puae_shared_nvram";
+   var.key = "puae2021_shared_nvram";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4274,7 +4274,7 @@ static void update_variables(void)
       else                                opt_shared_nvram = true;
    }
 
-   var.key = "puae_use_boot_hd";
+   var.key = "puae2021_use_boot_hd";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4288,7 +4288,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "hdf512"))   opt_use_boot_hd = 7;
    }
 
-   var.key = "puae_analogmouse";
+   var.key = "puae2021_analogmouse";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4298,28 +4298,28 @@ static void update_variables(void)
       else if (!strcmp(var.value, "both"))     opt_analogmouse = 3;
    }
 
-   var.key = "puae_analogmouse_deadzone";
+   var.key = "puae2021_analogmouse_deadzone";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       opt_analogmouse_deadzone = atoi(var.value);
    }
 
-   var.key = "puae_analogmouse_speed";
+   var.key = "puae2021_analogmouse_speed";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       opt_analogmouse_speed = atof(var.value);
    }
 
-   var.key = "puae_dpadmouse_speed";
+   var.key = "puae2021_dpadmouse_speed";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       opt_dpadmouse_speed = atoi(var.value);
    }
 
-   var.key = "puae_physicalmouse";
+   var.key = "puae2021_physicalmouse";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4328,7 +4328,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "double"))   opt_physicalmouse = 2;
    }
 
-   var.key = "puae_keyrah_keypad_mappings";
+   var.key = "puae2021_keyrah_keypad_mappings";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4336,7 +4336,7 @@ static void update_variables(void)
       else                                opt_keyrah_keypad = true;
    }
 
-   var.key = "puae_physical_keyboard_pass_through";
+   var.key = "puae2021_physical_keyboard_pass_through";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4344,7 +4344,7 @@ static void update_variables(void)
       else                                opt_keyboard_pass_through = true;
    }
 
-   var.key = "puae_model_options_display";
+   var.key = "puae2021_model_options_display";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4352,7 +4352,7 @@ static void update_variables(void)
       else                                opt_model_options_display = 1;
    }
 
-   var.key = "puae_mapping_options_display";
+   var.key = "puae2021_mapping_options_display";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4360,7 +4360,7 @@ static void update_variables(void)
       else                                opt_mapping_options_display = 1;
    }
 
-   var.key = "puae_video_options_display";
+   var.key = "puae2021_video_options_display";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4368,7 +4368,7 @@ static void update_variables(void)
       else                                opt_video_options_display = 1;
    }
 
-   var.key = "puae_audio_options_display";
+   var.key = "puae2021_audio_options_display";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4376,7 +4376,7 @@ static void update_variables(void)
       else                                opt_audio_options_display = 1;
    }
 
-   var.key = "puae_joyport";
+   var.key = "puae2021_joyport";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4387,14 +4387,14 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_joyport_order";
+   var.key = "puae2021_joyport_order";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       snprintf(opt_joyport_order, sizeof(opt_joyport_order), "%s", var.value);
    }
 
-   var.key = "puae_retropad_options";
+   var.key = "puae2021_retropad_options";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4404,7 +4404,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "rotate_jump")) opt_retropad_options = RETROPAD_OPTIONS_ROTATE_JUMP;
    }
 
-   var.key = "puae_cd32pad_options";
+   var.key = "puae2021_cd32pad_options";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4414,7 +4414,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "rotate_jump")) opt_cd32pad_options = RETROPAD_OPTIONS_ROTATE_JUMP;
    }
 
-   var.key = "puae_turbo_fire";
+   var.key = "puae2021_turbo_fire";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4425,7 +4425,7 @@ static void update_variables(void)
       }
    }
 
-   var.key = "puae_turbo_fire_button";
+   var.key = "puae2021_turbo_fire_button";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4439,7 +4439,7 @@ static void update_variables(void)
       else if (!strcmp(var.value, "R2")) turbo_fire_button = RETRO_DEVICE_ID_JOYPAD_R2;
    }
 
-   var.key = "puae_turbo_pulse";
+   var.key = "puae2021_turbo_pulse";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4447,168 +4447,168 @@ static void update_variables(void)
    }
 
    /* Mapper */
-   var.key = "puae_mapper_up";
+   var.key = "puae2021_mapper_up";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_UP] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_down";
+   var.key = "puae2021_mapper_down";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_DOWN] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_left";
+   var.key = "puae2021_mapper_left";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_LEFT] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_right";
+   var.key = "puae2021_mapper_right";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_RIGHT] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_select";
+   var.key = "puae2021_mapper_select";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_SELECT] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_start";
+   var.key = "puae2021_mapper_start";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_START] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_b";
+   var.key = "puae2021_mapper_b";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_B] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_a";
+   var.key = "puae2021_mapper_a";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_A] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_y";
+   var.key = "puae2021_mapper_y";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_Y] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_x";
+   var.key = "puae2021_mapper_x";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_X] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_l";
+   var.key = "puae2021_mapper_l";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_L] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_r";
+   var.key = "puae2021_mapper_r";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_R] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_l2";
+   var.key = "puae2021_mapper_l2";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_L2] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_r2";
+   var.key = "puae2021_mapper_r2";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_R2] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_l3";
+   var.key = "puae2021_mapper_l3";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_L3] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_r3";
+   var.key = "puae2021_mapper_r3";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_R3] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_lr";
+   var.key = "puae2021_mapper_lr";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_LR] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_ll";
+   var.key = "puae2021_mapper_ll";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_LL] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_ld";
+   var.key = "puae2021_mapper_ld";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_LD] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_lu";
+   var.key = "puae2021_mapper_lu";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_LU] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_rr";
+   var.key = "puae2021_mapper_rr";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_RR] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_rl";
+   var.key = "puae2021_mapper_rl";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_RL] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_rd";
+   var.key = "puae2021_mapper_rd";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_DEVICE_ID_JOYPAD_RD] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_ru";
+   var.key = "puae2021_mapper_ru";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4616,56 +4616,56 @@ static void update_variables(void)
    }
 
    /* Mapper hotkeys */
-   var.key = "puae_mapper_vkbd";
+   var.key = "puae2021_mapper_vkbd";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_MAPPER_VKBD] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_statusbar";
+   var.key = "puae2021_mapper_statusbar";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_MAPPER_STATUSBAR] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_mouse_toggle";
+   var.key = "puae2021_mapper_mouse_toggle";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_MAPPER_JOYMOUSE] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_reset";
+   var.key = "puae2021_mapper_reset";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_MAPPER_RESET] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_aspect_ratio_toggle";
+   var.key = "puae2021_mapper_aspect_ratio_toggle";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_MAPPER_ASPECT_RATIO] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_crop_toggle";
+   var.key = "puae2021_mapper_crop_toggle";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_MAPPER_CROP] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_turbo_fire_toggle";
+   var.key = "puae2021_mapper_turbo_fire_toggle";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       mapper_keys[RETRO_MAPPER_TURBO_FIRE] = retro_keymap_id(var.value);
    }
 
-   var.key = "puae_mapper_save_disk_toggle";
+   var.key = "puae2021_mapper_save_disk_toggle";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4759,7 +4759,7 @@ static void update_variables(void)
    char legacy_zoom_crop_string[20]   = {0};
    char legacy_zoom_toggle_string[20] = {0};
 
-   var.key = "puae_zoom_mode";
+   var.key = "puae2021_zoom_mode";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4776,7 +4776,7 @@ static void update_variables(void)
       strlcpy(legacy_zoom_string, var.value, sizeof(legacy_zoom_string));
    }
 
-   var.key = "puae_zoom_mode_crop";
+   var.key = "puae2021_zoom_mode_crop";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4791,7 +4791,7 @@ static void update_variables(void)
       strlcpy(legacy_zoom_crop_string, var.value, sizeof(legacy_zoom_crop_string));
    }
 
-   var.key = "puae_mapper_zoom_mode_toggle";
+   var.key = "puae2021_mapper_zoom_mode_toggle";
    var.value = NULL;
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
@@ -4802,32 +4802,32 @@ static void update_variables(void)
    if (legacy_zoom > CROP_NONE)
    {
       log_cb(RETRO_LOG_INFO, "Migrating 'zoom_mode' to 'crop'..\n");
-      set_variable("puae_crop", legacy_zoom_string);
+      set_variable("puae2021_crop", legacy_zoom_string);
       request_update_variables = true;
    }
 
    if (legacy_zoom_crop > 0)
    {
       log_cb(RETRO_LOG_INFO, "Migrating 'zoom_mode_crop' to 'crop_mode'..\n");
-      set_variable("puae_crop_mode", legacy_zoom_crop_string);
+      set_variable("puae2021_crop_mode", legacy_zoom_crop_string);
       request_update_variables = true;
    }
 
    if (legacy_zoom_toggle > 0)
    {
       log_cb(RETRO_LOG_INFO, "Migrating 'mapper_zoom_mode_toggle' to 'mapper_crop_toggle'..\n");
-      set_variable("puae_mapper_crop_toggle", legacy_zoom_toggle_string);
+      set_variable("puae2021_mapper_crop_toggle", legacy_zoom_toggle_string);
       request_update_variables = true;
    }
 
    if (strcmp(legacy_zoom_string, "deprecated"))
-      set_variable("puae_zoom_mode", "deprecated");
+      set_variable("puae2021_zoom_mode", "deprecated");
 
    if (strcmp(legacy_zoom_crop_string, "deprecated"))
-      set_variable("puae_zoom_mode_crop", "deprecated");
+      set_variable("puae2021_zoom_mode_crop", "deprecated");
 
    if (strcmp(legacy_zoom_toggle_string, "---"))
-      set_variable("puae_mapper_zoom_mode_toggle", "---");
+      set_variable("puae2021_mapper_zoom_mode_toggle", "---");
 
    updating_variables = false;
    if (request_update_variables)
@@ -5163,7 +5163,7 @@ void retro_deinit(void)
       dc_free(dc);
 
    /* Clean dynamic cartridge info */
-   free_puae_carts();
+   free_puae2021_carts();
 
    /* Clean ZIP temp */
    if (!string_is_empty(retro_temp_directory) && path_is_directory(retro_temp_directory))
diff --git a/libretro/libretro-core.h b/libretro/libretro-core.h
index e9a8bc1..ad3b39d 100644
--- a/libretro/libretro-core.h
+++ b/libretro/libretro-core.h
@@ -223,14 +223,14 @@ static retro_kickstarts uae_kickstarts[15] =
 };
 
 /* Dynamic cartridge core option info */
-struct puae_cart_info
+struct puae2021_cart_info
 {
    char *value;
    char *label;
 };
 
 /* Support files */
-#define LIBRETRO_PUAE_PREFIX    "puae_libretro"
+#define LIBRETRO_PUAE_PREFIX    "puae2021_libretro"
 
 /* Video */
 #define PUAE_VIDEO_PAL          0x01
diff --git a/retrodep/sysconfig.h b/retrodep/sysconfig.h
index 2cbea46..479da11 100644
--- a/retrodep/sysconfig.h
+++ b/retrodep/sysconfig.h
@@ -366,7 +366,7 @@
 #define HAVE__BOOL 1
 
 /* Name of package */
-#define PACKAGE "puae"
+#define PACKAGE "puae2021"
 
 /* Define to the address where bug reports for this package should be sent. */
 #define PACKAGE_BUGREPORT ""
@@ -375,7 +375,7 @@
 #define PACKAGE_COMMIT "0186c1b16f7181ffa02d73e6920d3180ce457c46"
 
 /* Define to the full name of this package. */
-#define PACKAGE_NAME "PUAE"
+#define PACKAGE_NAME "PUAE2021"
 
 /* Define to the full name and version of this package. */
 #define PACKAGE_STRING "PUAE 2.6.1"
