diff --git a/libretro/libretro-core.h b/libretro/libretro-core.h
index 0680699..9e8cffa 100644
--- a/libretro/libretro-core.h
+++ b/libretro/libretro-core.h
@@ -3,6 +3,7 @@
 
 #include <stdint.h>
 #include <string.h>
+#undef __USE_ISOC99
 #include <math.h>
 #include <stdlib.h>
 #include <stdio.h>
diff --git a/sources/src/audio.c b/sources/src/audio.c
index a312e4c..c82c2f7 100644
--- a/sources/src/audio.c
+++ b/sources/src/audio.c
@@ -43,6 +43,7 @@
 #endif
 #include "threaddep/thread.h"
 
+#undef __USE_ISOC99
 #include <math.h>
 
 #define DEBUG_AUDIO 0
diff --git a/sources/src/custom.c b/sources/src/custom.c
index f8046bc..5253cdf 100644
--- a/sources/src/custom.c
+++ b/sources/src/custom.c
@@ -13,6 +13,7 @@
 
 #include <ctype.h>
 #include <assert.h>
+#undef __USE_ISOC99
 #include <math.h>
 
 #include "cfgfile.h"
diff --git a/sources/src/gfxutil.c b/sources/src/gfxutil.c
index 8907d90..5d2bbcb 100644
--- a/sources/src/gfxutil.c
+++ b/sources/src/gfxutil.c
@@ -6,6 +6,7 @@
  * (c) 1996 Bernd Schmidt, Ed Hanway, Samuel Devulder
  */
 
+#undef __USE_ISOC99
 #include <math.h>
 
 #include "sysconfig.h"
