diff --git a/retrodep/sysconfig.h b/retrodep/sysconfig.h
index 97ab551..65c2dd5 100644
--- a/retrodep/sysconfig.h
+++ b/retrodep/sysconfig.h
@@ -1,3 +1,4 @@
+#include <math.h>
 #ifndef UAE_SYSCONFIG_H
 #define UAE_SYSCONFIG_H
 
diff --git a/sources/src/archivers/lha/slide.c b/sources/src/archivers/lha/slide.c
index ddd0a03..083ac03 100644
--- a/sources/src/archivers/lha/slide.c
+++ b/sources/src/archivers/lha/slide.c
@@ -108,7 +108,7 @@ static unsigned int hval;
 static int matchlen;
 static unsigned int matchpos;
 static unsigned int pos;
-static unsigned int remainder;
+//static unsigned int remainder;
 
 #if 0
 /* ------------------------------------------------------------------------ */
