diff --git a/Makefile.common b/Makefile.common
index 33a472e..5731513 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -189,9 +189,10 @@ SOURCES_C += \
 endif
 
 ifeq ($(HAVE_CHD), 1)
-	CFLAGS += -DWITH_CHD
+	CFLAGS += -DWITH_CHD -DZSTD_DISABLE_ASM
 	INCFLAGS += \
-		-I$(DEPS_DIR)/libchdr/include
+		-I$(DEPS_DIR)/libchdr/include \
+		-I$(DEPS_DIR)/zstd/lib \
 
 	SOURCES_C += \
 		$(DEPS_DIR)/7zip/LzFind.c \
diff --git a/sources/src/include/sysdeps.h b/sources/src/include/sysdeps.h
index 23a68be..2e54684 100644
--- a/sources/src/include/sysdeps.h
+++ b/sources/src/include/sysdeps.h
@@ -550,6 +550,16 @@ typedef int64_t off64_t;
 typedef uint32_t uint32;
 typedef uint8_t uint8;
 
+typedef uint64_t UINT64;
+typedef uint32_t UINT32;
+typedef uint16_t UINT16;
+typedef uint8_t UINT8;
+
+typedef int64_t INT64;
+typedef int32_t INT32;
+typedef int16_t INT16;
+typedef int8_t INT8;
+
 #ifndef _WIN32
 #define TCHAR char
 #endif
