diff --git a/sources/src/include/custom.h b/sources/src/include/custom.h
index 0b8b86a..b521f66 100644
--- a/sources/src/include/custom.h
+++ b/sources/src/include/custom.h
@@ -11,6 +11,10 @@
 
 #include "machdep/rpt.h"
 
+#define BEAMCON0_HARDDIS	0x4000
+#define BEAMCON0_CSCBEN	0x0400
+#define BEAMCON0_VARVSYEN	0x0200
+
 /* These are the masks that are ORed together in the chipset_mask option.
  * If CSMASK_AGA is set, the ECS bits are guaranteed to be set as well.  */
 #define CSMASK_ECS_AGNUS 1
