diff --git a/src/emucore/CartELF.cxx b/src/emucore/CartELF.cxx
index 57c5a5aa1..327907646 100644
--- a/src/emucore/CartELF.cxx
+++ b/src/emucore/CartELF.cxx
@@ -32,6 +32,7 @@
 #endif
 
 #include "CartELF.hxx"
+#include <iomanip>
 
 // NOLINTBEGIN(bugprone-unchecked-optional-access)
 
diff --git a/src/emucore/M6502.hxx b/src/emucore/M6502.hxx
index 002953f0d..2a603dd5f 100644
--- a/src/emucore/M6502.hxx
+++ b/src/emucore/M6502.hxx
@@ -37,6 +37,7 @@ class DispatchResult;
 #include "bspf.hxx"
 #include "Device.hxx"
 #include "Serializable.hxx"
+#include <climits>
 
 /**
   The 6502 is an 8-bit microprocessor that has a 64K addressing space.
