diff --git a/Src/OSD/DefaultConfigFile.h b/Src/OSD/DefaultConfigFile.h
index 6202eb9..c5e17a5 100644
--- a/Src/OSD/DefaultConfigFile.h
+++ b/Src/OSD/DefaultConfigFile.h
@@ -244,10 +244,10 @@
     "InputAnalogGunRight2 = \"NONE\"\n"
     "InputAnalogGunUp2 = \"NONE\"\n"
     "InputAnalogGunDown2 = \"NONE\"\n"
-    "InputAnalogGunX2 = \"NONE\"\n"
-    "InputAnalogGunY2 = \"NONE\"\n"
-    "InputAnalogTriggerLeft2 = \"NONE\"\n"
-    "InputAnalogTriggerRight2 = \"NONE\"\n"
+    "InputAnalogGunX2 = \"JOY2_XAXIS\"\n"
+    "InputAnalogGunY2 = \"JOY2_YAXIS\"\n"
+    "InputAnalogTriggerLeft2 = \"JOY2_BUTTON1\"\n"
+    "InputAnalogTriggerRight2 = \"JOY2_BUTTON2\"\n"
     "\n"
     "; Ski Champ controls\n"
     "InputSkiLeft = \"KEY_LEFT\"\n"
