diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 8ebba1535..9000266f4 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -97,7 +97,8 @@ AddReg=\
     Tapi,\
     ThemeManager,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.ntamd64]
 RegisterDlls=RegisterDllsSection
@@ -123,7 +124,8 @@ AddReg=\
     Tapi,\
     ThemeManager,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.ntarm64]
 RegisterDlls=RegisterDllsSection
@@ -148,7 +150,8 @@ AddReg=\
     Tapi,\
     ThemeManager,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [Wow64Install.ntx86]
 RegisterDlls=RegisterDllsSection
@@ -180,6 +183,10 @@ AddReg=\
     Tapi,\
     VersionInfo,\
     LicenseInformation
+    
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
 
 [DefaultInstall.Services]
 AddService=BITS,0,BITSService
