diff --git a/elf/Makefile b/elf/Makefile
index c00e2ccfc5..29357678ad 100644
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -395,7 +395,6 @@ tests += \
   tst-big-note \
   tst-debug1 \
   tst-deep1 \
-  tst-dl-is_dso \
   tst-dlclose-lazy \
   tst-dlmodcount \
   tst-dlmopen-dlerror \
diff --git a/elf/ldconfig.c b/elf/ldconfig.c
index d26eef1fb4..55d71b0f0e 100644
--- a/elf/ldconfig.c
+++ b/elf/ldconfig.c
@@ -42,7 +42,6 @@
 #include <ldconfig.h>
 #include <dl-cache.h>
 #include <dl-hwcaps.h>
-#include <dl-is_dso.h>
 
 #include <dl-procinfo.h>
 
@@ -706,7 +705,9 @@ search_dir (const struct dir_entry *entry)
 	continue;
       /* Does this file look like a shared library?  The dynamic linker
 	 is also considered as shared library.  */
-      if (!_dl_is_dso (direntry->d_name)
+      if (((strncmp (direntry->d_name, "lib", 3) != 0
+       && strncmp (direntry->d_name, "ld-", 3) != 0)
+      || strstr (direntry->d_name, ".so") == NULL)
 	  && (direntry->d_type == DT_REG || entry->hwcaps == NULL))
 	continue;
 
