diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 1f4019cc417..12c5a5bb474 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -99,6 +99,7 @@ AddReg=\
     LicenseInformation,\
     NVIDIANGX, \
     ProtonOverrides,\
+    BatoceraOverrides,\
     SteamClient
 
 [DefaultInstall.ntamd64]
@@ -128,6 +129,7 @@ AddReg=\
     LicenseInformation,\
     NVIDIANGX, \
     ProtonOverrides,\
+    BatoceraOverrides,\
     SteamClient.ntamd64
 
 [DefaultInstall.ntarm64]
@@ -173,6 +175,7 @@ AddReg=\
     LicenseInformation,\
     NVIDIANGX, \
     ProtonOverrides,\
+    BatoceraOverrides,\
     SteamClient.ntamd64
 
 [Wow64Install.ntarm]
@@ -2904,3 +2907,8 @@ HKCU,Software\Wine\AppDefaults\Spider-Man.exe\DllOverrides,"atiadlxx",,"builtin"
 HKCU,Software\Wine\AppDefaults\RiftApart.exe\DllOverrides,"atiadlxx",,"builtin"
 HKLM,Software\Wow6432Node\lucasarts entertainment company llc\Star Wars: Episode I Racer\v1.0,"Display Height",0x10001,480
 HKLM,Software\Wow6432Node\lucasarts entertainment company llc\Star Wars: Episode I Racer\v1.0,"Display Width",0x10001,640
+
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+
