diff --git a/def.h b/def.h
index 231e052..d55c097 100755
--- a/def.h
+++ b/def.h
@@ -5,12 +5,12 @@
 #define INHIBIT(X) /* X */
 
 #ifndef FONTS
-#define FONTS {"/usr/share/fonts/truetype/dejavu/DejaVuSansCondensed.ttf",10},{"FreeSans.ttf",10},{"DroidSansFallback.ttf",9}
+#define FONTS {"/usr/share/fonts/dejavu/DejaVuSansCondensed.ttf",10},{"/usr/share/fonts/truetype/nanum/NanumMyeongjo.ttf",10},{"/usr/share/fonts/truetype/droid/DroidSansFallback.ttf",9}
 #endif
 
 // Font stack to use on screens with low DPI.
 #ifndef LOW_DPI_FONTS
-#define LOW_DPI_FONTS {"Fiery_Turk.ttf",8},{"/usr/share/fonts/truetype/dejavu/DejaVuSansCondensed.ttf",10},{"FreeSans.ttf",10},{"DroidSansFallback.ttf",9}
+#define LOW_DPI_FONTS {"Fiery_Turk.ttf",8},{"/usr/share/fonts/dejavu/DejaVuSansCondensed.ttf",10},{"/usr/share/fonts/truetype/nanum/NanumMyeongjo.ttf",10},{"/usr/share/fonts/truetype/droid/DroidSansFallback.ttf",9}
 #endif
 
 // Panel
