diff --git a/bauh/view/core/config.py b/bauh/view/core/config.py
index e9c7c5d8..5f0b2b7c 100644
--- a/bauh/view/core/config.py
+++ b/bauh/view/core/config.py
@@ -30,7 +30,7 @@ class CoreConfigManager(YAMLConfigManager):
             },
             'suggestions': {
                 'enabled': True,
-                'by_type': 10
+                'by_type': 200
             },
             'ui': {
                 'table': {
