diff --git a/src/Cafe/CMakeLists.txt b/src/Cafe/CMakeLists.txt
index 1b0def8..8032494 100644
--- a/src/Cafe/CMakeLists.txt
+++ b/src/Cafe/CMakeLists.txt
@@ -574,7 +574,7 @@ target_link_libraries(CemuCafe PRIVATE
 	OpenSSL::Crypto
 	OpenSSL::SSL
 	PNG::PNG
-	pugixml::pugixml
+	pugixml
 	ZArchive::zarchive
 	ZLIB::ZLIB
 	zstd::zstd
diff --git a/src/Cemu/CMakeLists.txt b/src/Cemu/CMakeLists.txt
index 1fce8ec..bb09cf7 100644
--- a/src/Cemu/CMakeLists.txt
+++ b/src/Cemu/CMakeLists.txt
@@ -46,7 +46,7 @@ target_link_libraries(CemuComponents PRIVATE
 	CURL::libcurl
 	OpenSSL::Crypto
 	OpenSSL::SSL
-	pugixml::pugixml
+	pugixml
 	ZLIB::ZLIB
 )
 
diff --git a/src/config/CMakeLists.txt b/src/config/CMakeLists.txt
index abcacc6..0882ef5 100644
--- a/src/config/CMakeLists.txt
+++ b/src/config/CMakeLists.txt
@@ -22,5 +22,5 @@ target_link_libraries(CemuConfig PRIVATE
 	CemuUtil
 	Boost::headers
 	Boost::program_options
-	pugixml::pugixml
+	pugixml
 )
diff --git a/src/gui/wxgui/CMakeLists.txt b/src/gui/wxgui/CMakeLists.txt
index 921ba05..4690ee0 100644
--- a/src/gui/wxgui/CMakeLists.txt
+++ b/src/gui/wxgui/CMakeLists.txt
@@ -154,7 +154,7 @@ target_link_libraries(CemuWxGui PRIVATE
 	CURL::libcurl
 	libzip::zip
 	OpenSSL::Crypto
-	pugixml::pugixml
+	pugixml
 	ZArchive::zarchive
 )
 
diff --git a/src/input/CMakeLists.txt b/src/input/CMakeLists.txt
index 3158ce9..6a5b9a8 100644
--- a/src/input/CMakeLists.txt
+++ b/src/input/CMakeLists.txt
@@ -96,7 +96,7 @@ target_link_libraries(CemuInput PRIVATE
 	Boost::headers
 	Boost::program_options
 	glm::glm
-	pugixml::pugixml
+	pugixml
 	SDL2::SDL2
 )
 
