diff --git a/src/Cafe/CMakeLists.txt b/src/Cafe/CMakeLists.txt
index 7033014..9b27a34 100644
--- a/src/Cafe/CMakeLists.txt
+++ b/src/Cafe/CMakeLists.txt
@@ -563,7 +563,7 @@ target_link_libraries(CemuCafe PRIVATE
   CemuGui
   ZArchive::zarchive
   imguiImpl
-  pugixml::pugixml
+  pugixml
   ZLIB::ZLIB
   CURL::libcurl
   ih264d
diff --git a/src/gui/wxgui/CMakeLists.txt b/src/gui/wxgui/CMakeLists.txt
index 9e24a59..5aa85c2 100644
--- a/src/gui/wxgui/CMakeLists.txt
+++ b/src/gui/wxgui/CMakeLists.txt
@@ -132,7 +132,7 @@ target_link_libraries(CemuWxGui PRIVATE
   ZArchive::zarchive
   CemuComponents
   SDL2::SDL2
-  pugixml::pugixml
+  pugixml
   CemuCafe
   PUBLIC
   CURL::libcurl
