diff --git a/src/Cafe/CMakeLists.txt b/src/Cafe/CMakeLists.txt
index 0fb7a44..035c62e 100644
--- a/src/Cafe/CMakeLists.txt
+++ b/src/Cafe/CMakeLists.txt
@@ -548,7 +548,7 @@ target_link_libraries(CemuCafe PRIVATE
 	OpenSSL::Crypto
 	OpenSSL::SSL
 	PNG::PNG
-	pugixml::pugixml
+	pugixml
 	ZArchive::zarchive
 	ZLIB::ZLIB
 	zstd::zstd
diff --git a/src/Cemu/CMakeLists.txt b/src/Cemu/CMakeLists.txt
index 1fce8ec..bb09cf7 100644
--- a/src/Cemu/CMakeLists.txt
+++ b/src/Cemu/CMakeLists.txt
@@ -46,7 +46,7 @@ target_link_libraries(CemuComponents PRIVATE
 	CURL::libcurl
 	OpenSSL::Crypto
 	OpenSSL::SSL
-	pugixml::pugixml
+	pugixml
 	ZLIB::ZLIB
 )
 
diff --git a/src/config/CMakeLists.txt b/src/config/CMakeLists.txt
index d53e857..5c76d60 100644
--- a/src/config/CMakeLists.txt
+++ b/src/config/CMakeLists.txt
@@ -21,7 +21,7 @@ target_link_libraries(CemuConfig PRIVATE
 	CemuUtil
 	Boost::headers
 	Boost::program_options
-	pugixml::pugixml
+	pugixml
 )
 
 if (ENABLE_WXWIDGETS)
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 02f96a9..35771c7 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -149,7 +149,7 @@ target_link_libraries(CemuGui PRIVATE
 	CURL::libcurl
 	libzip::zip
 	OpenSSL::Crypto
-	pugixml::pugixml
+	pugixml
 	ZArchive::zarchive
 )
 
diff --git a/src/input/CMakeLists.txt b/src/input/CMakeLists.txt
index 9f7873a..4636f0e 100644
--- a/src/input/CMakeLists.txt
+++ b/src/input/CMakeLists.txt
@@ -87,7 +87,7 @@ target_link_libraries(CemuInput PRIVATE
 	Boost::headers
 	Boost::program_options
 	glm::glm
-	pugixml::pugixml
+	pugixml
 	SDL2::SDL2
 )
 if (ENABLE_HIDAPI)
