diff --git a/src/Cafe/CMakeLists.txt b/src/Cafe/CMakeLists.txt
index 61b4815..51e65ad 100644
--- a/src/Cafe/CMakeLists.txt
+++ b/src/Cafe/CMakeLists.txt
@@ -496,7 +496,7 @@ target_link_libraries(CemuCafe PRIVATE
 	OpenSSL::Crypto
 	OpenSSL::SSL
 	PNG::PNG
-	pugixml::pugixml
+	pugixml
 	ZArchive::zarchive
 	ZLIB::ZLIB
 	zstd::zstd
diff --git a/src/Cemu/CMakeLists.txt b/src/Cemu/CMakeLists.txt
index 1fce8ec..bb09cf7 100644
--- a/src/Cemu/CMakeLists.txt
+++ b/src/Cemu/CMakeLists.txt
@@ -46,7 +46,7 @@ target_link_libraries(CemuComponents PRIVATE
 	CURL::libcurl
 	OpenSSL::Crypto
 	OpenSSL::SSL
-	pugixml::pugixml
+	pugixml
 	ZLIB::ZLIB
 )
 
diff --git a/src/Common/CMakeLists.txt b/src/Common/CMakeLists.txt
index 7e5825f..cca5e04 100644
--- a/src/Common/CMakeLists.txt
+++ b/src/Common/CMakeLists.txt
@@ -61,7 +61,7 @@ target_link_libraries(CemuCommon PRIVATE
 	CemuComponents
 	Boost::nowide
 	Boost::filesystem
-	glm::glm
+	glm
 )
 
 if (UNIX AND NOT APPLE)
diff --git a/src/config/CMakeLists.txt b/src/config/CMakeLists.txt
index f02b95d..88be2a3 100644
--- a/src/config/CMakeLists.txt
+++ b/src/config/CMakeLists.txt
@@ -25,7 +25,7 @@ target_link_libraries(CemuConfig PRIVATE
 	CemuUtil
 	Boost::headers
 	Boost::program_options
-	pugixml::pugixml
+	pugixml
 )
 
 if (ENABLE_WXWIDGETS)
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index a08891d..7b7c09d 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -143,7 +143,7 @@ target_link_libraries(CemuGui PRIVATE
 	CURL::libcurl
 	libzip::zip
 	OpenSSL::Crypto
-	pugixml::pugixml
+	pugixml
 	ZArchive::zarchive
 )
 
diff --git a/src/input/CMakeLists.txt b/src/input/CMakeLists.txt
index ecb88cd..16bb9d7 100644
--- a/src/input/CMakeLists.txt
+++ b/src/input/CMakeLists.txt
@@ -83,8 +83,8 @@ target_link_libraries(CemuInput PRIVATE
 	CemuUtil
 	Boost::headers
 	Boost::program_options
-	glm::glm
-	pugixml::pugixml
+	glm
+	pugixml
 	SDL2::SDL2
 )
 
