diff --git a/src/config/CemuConfig.cpp b/src/config/CemuConfig.cpp
index e51dfbe..7eaeb2d 100644
--- a/src/config/CemuConfig.cpp
+++ b/src/config/CemuConfig.cpp
@@ -8,7 +8,7 @@
 #include "PermanentConfig.h"
 #include "ActiveSettings.h"
 
-XMLCemuConfig_t g_config(L"settings.xml");
+XMLCemuConfig_t g_config(L"/userdata/system/configs/cemu/settings.xml");
 
 void CemuConfig::SetMLCPath(fs::path path, bool save)
 {
