diff --git a/src/Cafe/Filesystem/FST/KeyCache.cpp b/src/Cafe/Filesystem/FST/KeyCache.cpp
index f85d2b5..896fd21d 100644
--- a/src/Cafe/Filesystem/FST/KeyCache.cpp
+++ b/src/Cafe/Filesystem/FST/KeyCache.cpp
@@ -60,7 +60,18 @@ void KeyCache_Prepare()
 	sKeyCachePrepared = true;
 	g_keyCache.clear();
 	// load keys
-	auto keysPath = ActiveSettings::GetUserDataPath("keys.txt");
+	// Use /userdata/bios/cemu for the keys.txt directory, to match batocera
+	// documentation at https://wiki.batocera.org/systems:wiiu#keystxt
+
+	std::string keysDir = "/userdata/bios/cemu";
+	// Let's first make sure the parent directory exists.
+	std::error_code err;
+	fs::create_directories(keysDir, err);
+	if (err) {
+		WindowSystem::ShowErrorDialog(_tr("Unable to create file keys.txt\nThis can happen if Cemu does not have write permission to its own directory, the disk is full or if anti-virus software is blocking Cemu."), _tr("Error"), WindowSystem::ErrorCategory::KEYS_TXT_CREATION);
+	}
+
+	auto keysPath = keysDir + "/keys.txt";
 	FileStream* fs_keys = FileStream::openFile2(keysPath);
 	if( !fs_keys )
 	{
