diff --git a/src/proto.h b/src/proto.h
index c28dd7d..b60bfd2 100644
--- a/src/proto.h
+++ b/src/proto.h
@@ -564,7 +564,7 @@ void free_chtab(chtab_type *chtab_ptr);
 image_type* decode_image(image_data_type* image_data, dat_pal_type* palette);
 image_type*load_image(int index, dat_pal_type* palette);
 void draw_image_transp(image_type* image,image_type* mask,int xpos,int ypos);
-int set_joy_mode(void);
+int set_joy_mode(int joynum);
 surface_type *make_offscreen_buffer(const rect_type* rect);
 void free_surface(surface_type *surface);
 void free_peel(peel_type *peel_ptr);
diff --git a/src/seg000.c b/src/seg000.c
index 2165e5e..953c1ac 100644
--- a/src/seg000.c
+++ b/src/seg000.c
@@ -32,6 +32,8 @@ dat_type * dathandle;
 // data:4C08
 word need_redraw_because_flipped;
 
+int joynum = 0;
+
 void fix_sound_priorities(void);
 
 // seg000:0000
@@ -52,6 +54,11 @@ void pop_main() {
 		seed_was_init = 1;
 	}
 
+	const char* joynumstr = check_param("joynum=");
+	if (joynumstr != NULL) {
+		joynum = atoi(joynumstr+7);
+	}
+
 	// debug only: check that the sequence table deobfuscation did not mess things up
 	#ifdef CHECK_SEQTABLE_MATCHES_ORIGINAL
 	check_seqtable_matches_original();
@@ -106,7 +113,7 @@ void pop_main() {
 	parse_cmdline_sound();
 
 	show_loading();
-	set_joy_mode();
+	set_joy_mode(joynum);
 	cheats_enabled = check_param("megahit") != NULL;
 #ifdef USE_DEBUG_CHEATS
 	debug_cheats_enabled = check_param("debug") != NULL;
@@ -619,7 +626,7 @@ int process_key() {
 			if ((sound_flags & sfDigi) && sound_mode == smTandy) {
 				answer_text = "JOYSTICK UNAVAILABLE";
 			} else {
-				if (set_joy_mode()) {
+				if (set_joy_mode(joynum)) {
 					answer_text = "JOYSTICK MODE";
 				} else {
 					answer_text = "JOYSTICK NOT FOUND";
diff --git a/src/seg009.c b/src/seg009.c
index e668cbe..9fc3fad 100644
--- a/src/seg009.c
+++ b/src/seg009.c
@@ -875,7 +875,7 @@ void draw_image_transp(image_type* image,image_type* mask,int xpos,int ypos) {
 }
 
 // seg009:157E
-int set_joy_mode() {
+int set_joy_mode(int joynum) {
 	// stub
 	if (SDL_NumJoysticks() < 1) {
 		is_joyst_mode = 0;
@@ -884,8 +884,8 @@ int set_joy_mode() {
 			SDL_GameControllerAddMappingsFromFile(gamecontrollerdb_file);
 		}
 
-		if (SDL_IsGameController(0)) {
-			sdl_controller_ = SDL_GameControllerOpen(0);
+		if (SDL_IsGameController(joynum)) {
+			sdl_controller_ = SDL_GameControllerOpen(joynum);
 			if (sdl_controller_ == NULL) {
 				is_joyst_mode = 0;
 			} else {
@@ -895,13 +895,13 @@ int set_joy_mode() {
 		// We have a joystick connected, but it's NOT compatible with the SDL_GameController
 		// interface, so we resort to the classic SDL_Joystick interface instead
 		else {
-			sdl_joystick_ = SDL_JoystickOpen(0);
+			sdl_joystick_ = SDL_JoystickOpen(joynum);
 			is_joyst_mode = 1;
 			using_sdl_joystick_interface = 1;
 		}
 	}
 	if (enable_controller_rumble && is_joyst_mode) {
-		sdl_haptic = SDL_HapticOpen(0);
+		sdl_haptic = SDL_HapticOpen(joynum);
 		SDL_HapticRumbleInit(sdl_haptic); // initialize the device for simple rumble
 	} else {
 		sdl_haptic = NULL;
