diff --git a/src/menu.c b/src/menu.c
index 374f4ea..fe4b0be 100644
--- a/src/menu.c
+++ b/src/menu.c
@@ -2026,15 +2026,15 @@ void process_additional_menu_input() {
 	have_mouse_input = (mouse_moved || mouse_clicked || mouse_button_clicked_right || menu_control_scroll_y);
 
 	dword flags = SDL_GetWindowFlags(window_);
-	if (flags & SDL_WINDOW_FULLSCREEN_DESKTOP) {
-		if (have_mouse_input) {
-			SDL_ShowCursor(SDL_ENABLE);
-		} else if (have_keyboard_or_controller_input) {
+//	if (flags & SDL_WINDOW_FULLSCREEN_DESKTOP) {
+//		if (have_mouse_input) {
+//			SDL_ShowCursor(SDL_ENABLE);
+//		} else if (have_keyboard_or_controller_input) {
 			SDL_ShowCursor(SDL_DISABLE);
-		}
-	} else {
-		SDL_ShowCursor(SDL_ENABLE);
-	}
+//		}
+//	} else {
+//		SDL_ShowCursor(SDL_ENABLE);
+//	}
 }
 
 bool joy_ABXY_buttons_released;
@@ -2302,11 +2302,11 @@ void menu_was_closed(void) {
 	}
 	// In fullscreen mode, hide the mouse cursor (because it is only needed in the menu).
 	dword flags = SDL_GetWindowFlags(window_);
-	if (flags & SDL_WINDOW_FULLSCREEN_DESKTOP) {
+//	if (flags & SDL_WINDOW_FULLSCREEN_DESKTOP) {
 		SDL_ShowCursor(SDL_DISABLE);
-	} else {
-		SDL_ShowCursor(SDL_ENABLE);
-	}
+//	} else {
+//		SDL_ShowCursor(SDL_ENABLE);
+//	}
 }
 
 
diff --git a/src/seg009.c b/src/seg009.c
index 03505bf..1b1d569 100644
--- a/src/seg009.c
+++ b/src/seg009.c
@@ -2567,9 +2567,9 @@ void set_gr_mode(byte grmode) {
 	}
 	init_overlay();
 	init_scaling();
-	if (start_fullscreen) {
+//	if (start_fullscreen) {
 		SDL_ShowCursor(SDL_DISABLE);
-	}
+//	}
 
 
 	//SDL_WM_SetCaption(WINDOW_TITLE, NULL);
