diff --git a/src/seg000.c b/src/seg000.c
index adab27d..bb6d32c 100644
--- a/src/seg000.c
+++ b/src/seg000.c
@@ -2292,16 +2292,12 @@ const rect_type splash_text_2_rect = {50, 0, 200, 320};
 const char* splash_text_1 = "SDLPoP " SDLPOP_VERSION;
 const char* splash_text_2 =
 #ifdef USE_QUICKSAVE
-		"To quick save/load, press F6/F9 in-game.\n"
+		"To quick save/load, press Hotkey + Y/X.\n"
 		"\n"
 #endif
-#ifdef USE_REPLAY
-		"To record replays, press Ctrl+Tab in-game.\n"
-		"To view replays, press Tab on the title screen.\n"
-		"\n"
-#endif
-		"Edit SDLPoP.ini to customize SDLPoP.\n"
-		"Mods also work with SDLPoP.\n"
+		"To take a screenshot, press Hotkey + Left Bumper.\n"
+		"To restart the level, press Hotkey + Right Bumper.\n"
+		"To return to the main menu, press Hotkey + A.\n"
 		"\n"
 		"For more information, read doc/Readme.txt.\n"
 		"Questions? Visit https://forum.princed.org\n"
