diff --git a/src/3d_main.cpp b/src/3d_main.cpp
index eba632e3..52cb3b88 100644
--- a/src/3d_main.cpp
+++ b/src/3d_main.cpp
@@ -10727,7 +10727,7 @@ const std::string& get_profile_dir()
 {
 	static std::string profile_dir;
 	static auto is_initialized = false;
-
+	profile_dir = "/userdata/saves/bstone";
 	if (!is_initialized)
 	{
 		is_initialized = true;
@@ -10760,8 +10760,9 @@ const std::string& get_screenshot_dir()
 {
 	static auto is_initialized = false;
 	static auto screenshot_dir = std::string{};
+	screenshot_dir = "/userdata/screenshots/bstone";
 
-	if (!is_initialized)
+	/*if (!is_initialized)
 	{
 		is_initialized = true;
 
@@ -10772,7 +10773,7 @@ const std::string& get_screenshot_dir()
 		{
 			screenshot_dir = get_profile_dir();
 		}
-	}
+	}*/
 
 	return screenshot_dir;
 }
