diff --git a/src/bstone/src/3d_main.cpp b/src/bstone/src/3d_main.cpp
index 5b52122a..73c44a8c 100644
--- a/src/bstone/src/3d_main.cpp
+++ b/src/bstone/src/3d_main.cpp
@@ -10778,7 +10778,7 @@ const std::string& get_profile_dir()
 {
 	static std::string profile_dir;
 	static auto is_initialized = false;
-
+	profile_dir = "/userdata/saves/bstone";
 	if (!is_initialized)
 	{
 		is_initialized = true;
@@ -10818,8 +10818,9 @@ const std::string& get_screenshot_dir()
 {
 	static auto is_initialized = false;
 	static auto screenshot_dir = std::string{};
+	screenshot_dir = "/userdata/screenshots/bstone";
 
-	if (!is_initialized)
+	/*if (!is_initialized)
 	{
 		is_initialized = true;
 
@@ -10830,7 +10831,7 @@ const std::string& get_screenshot_dir()
 		{
 			screenshot_dir = get_profile_dir();
 		}
-	}
+	}*/
 
 	return screenshot_dir;
 }
