diff --git a/src/platform/libretro/main.cpp b/src/platform/libretro/main.cpp
index 38a752a..a8c62fa 100644
--- a/src/platform/libretro/main.cpp
+++ b/src/platform/libretro/main.cpp
@@ -631,12 +631,12 @@ if (!path_is_absolute(info->path))
       return false;
    }
 
-   char basedir[1024] = {0};
+   char basedir[1024] = "/userdata/roms/ports/openlara/level/";
    fill_pathname_basedir(basedir, info->path, sizeof(basedir));
 
    // contentDir acts as the current working directory in OpenLara
    strcpy(contentDir, basedir);
-   path_parent_dir(contentDir);
+   //path_parent_dir(contentDir);
    fill_pathname_parent_dir_name(basedir, contentDir, sizeof(basedir));
 
    if (strcmp(basedir, "level") == 0)
