diff --git a/src/yuzu/main.cpp b/src/yuzu/main.cpp
index 901a39cc9f..be86e1bd48 100644
--- a/src/yuzu/main.cpp
+++ b/src/yuzu/main.cpp
@@ -22,7 +22,7 @@
 #include <csignal>
 #include <sys/socket.h>
 #endif
-#ifdef __linux__
+#ifdef __gamemode__
 #include "common/linux/gamemode.h"
 #endif
 
@@ -419,7 +419,7 @@ GMainWindow::GMainWindow(bool has_broken_vulkan)
     SetupSigInterrupts();
 #endif
 
-#ifdef __linux__
+#ifdef __gamemode__
     SetGamemodeEnabled(Settings::values.enable_gamemode.GetValue());
 #endif
 
@@ -2264,7 +2264,7 @@ void GMainWindow::OnEmulationStopped() {
 
     discord_rpc->Update();
 
-#ifdef __linux__
+#ifdef __gamemode__
     Common::Linux::StopGamemode();
 #endif
 
@@ -3106,7 +3106,7 @@ void GMainWindow::OnStartGame() {
 
     discord_rpc->Update();
 
-#ifdef __linux__
+#ifdef __gamemode__
     Common::Linux::StartGamemode();
 #endif
 }
@@ -3130,7 +3130,7 @@ void GMainWindow::OnPauseGame() {
     UpdateMenuState();
     AllowOSSleep();
 
-#ifdef __linux__
+#ifdef __gamemode__
     Common::Linux::StopGamemode();
 #endif
 }
@@ -3427,7 +3427,7 @@ void GMainWindow::OnConfigure() {
     const auto old_theme = UISettings::values.theme;
     const bool old_discord_presence = UISettings::values.enable_discord_presence.GetValue();
     const auto old_language_index = Settings::values.language_index.GetValue();
-#ifdef __linux__
+#ifdef __gamemode__
     const bool old_gamemode = Settings::values.enable_gamemode.GetValue();
 #endif
 
@@ -3489,7 +3489,7 @@ void GMainWindow::OnConfigure() {
     if (UISettings::values.enable_discord_presence.GetValue() != old_discord_presence) {
         SetDiscordEnabled(UISettings::values.enable_discord_presence.GetValue());
     }
-#ifdef __linux__
+#ifdef __gamemode__
     if (Settings::values.enable_gamemode.GetValue() != old_gamemode) {
         SetGamemodeEnabled(Settings::values.enable_gamemode.GetValue());
     }
@@ -4777,7 +4777,7 @@ void GMainWindow::SetDiscordEnabled([[maybe_unused]] bool state) {
     discord_rpc->Update();
 }
 
-#ifdef __linux__
+#ifdef __gamemode__
 void GMainWindow::SetGamemodeEnabled(bool state) {
     if (emulation_running) {
         Common::Linux::SetGamemodeState(state);
diff --git a/src/yuzu_cmd/yuzu.cpp b/src/yuzu_cmd/yuzu.cpp
index 4a99f34861..d0aeb9b191 100644
--- a/src/yuzu_cmd/yuzu.cpp
+++ b/src/yuzu_cmd/yuzu.cpp
@@ -63,7 +63,7 @@ __declspec(dllexport) int AmdPowerXpressRequestHighPerformance = 1;
 }
 #endif
 
-#ifdef __linux__
+#ifdef __gamemode__
 #include "common/linux/gamemode.h"
 #endif
 
@@ -433,7 +433,7 @@ int main(int argc, char** argv) {
         exit(0);
     });
 
-#ifdef __linux__
+#ifdef __gamemode__
     Common::Linux::StartGamemode();
 #endif
 
@@ -448,7 +448,7 @@ int main(int argc, char** argv) {
     void(system.Pause());
     system.ShutdownMainProcess();
 
-#ifdef __linux__
+#ifdef __gamemode__
     Common::Linux::StopGamemode();
 #endif
 
