diff --git a/externals/nx_tzdb/CMakeLists.txt b/externals/nx_tzdb/CMakeLists.txt
index a51939f7c8..1ffa141f7f 100644
--- a/externals/nx_tzdb/CMakeLists.txt
+++ b/externals/nx_tzdb/CMakeLists.txt
@@ -14,7 +14,7 @@ find_program(GIT git)
 find_program(GNU_MAKE make)
 find_program(DATE_PROG date)
 
-set(CAN_BUILD_NX_TZDB true)
+set(CAN_BUILD_NX_TZDB false)
 
 if (NOT (GIT AND GNU_MAKE AND DATE_PROG) OR CMAKE_SYSTEM_NAME STREQUAL "Windows" OR ANDROID)
     # tzdb_to_nx currently requires a posix-compliant host
