diff --git a/CMakeLists.txt b/CMakeLists.txt
index f436c0a183..c60172b34d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -507,7 +507,7 @@ else()
 
     # wow
     if (PLATFORM_LINUX)
-        find_package(Boost 1.57.0 CONFIG REQUIRED headers context system fiber)
+        find_package(Boost 1.57.0 CONFIG REQUIRED headers context fiber)
     else()
         find_package(Boost 1.57.0 CONFIG REQUIRED)
     endif()
