--- a/CMakeLists.txt	2025-08-21 12:27:30.151114862 +0200
+++ b/CMakeLists.txt	2025-08-21 12:27:38.196362658 +0200
@@ -925,7 +925,7 @@
     endif()
 
     if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
-        find_package(Qt6 REQUIRED COMPONENTS DBus OPTIONAL_COMPONENTS GuiPrivate)
+        find_package(Qt6 REQUIRED COMPONENTS DBus Gui)
     elseif (UNIX AND NOT APPLE)
         find_package(Qt6 REQUIRED COMPONENTS DBus Gui)
     endif()
