--- a/CMakeLists.txt	2025-08-21 12:36:51.631052132 +0200
+++ b/CMakeLists.txt	2025-08-21 12:37:11.676437409 +0200
@@ -228,6 +228,9 @@
     file(READ "${PROJECT_SOURCE_DIR}/.gitmodules" gitmodules)
     string(REGEX MATCHALL "path *= *[^ \t\r\n]*" gitmodules ${gitmodules})
     foreach(module ${gitmodules})
+    	if("${module}" STREQUAL "libusb")
+        	continue() # skip
+    	endif()
         string(REGEX REPLACE "path *= *" "" module ${module})
 
         file(GLOB RESULT "${PROJECT_SOURCE_DIR}/${module}/*")
