diff --git a/meson.build b/meson.build
index becd754..db701f6 100644
--- a/meson.build
+++ b/meson.build
@@ -110,9 +110,9 @@ if dep_wayland_client.found()
   vulkan_wsi_deps += dep_wayland_client
 endif
 
-if is_unixy and not dep_x11.found() and not dep_wayland_client.found()
-  error('At least one of "with_x11" and "with_wayland" should be enabled')
-endif
+#if is_unixy and not dep_x11.found() and not dep_wayland_client.found()
+#  error('At least one of "with_x11" and "with_wayland" should be enabled')
+#endif
 
 inc_common = [
     include_directories('include'),
diff --git a/src/keybinds.h b/src/keybinds.h
index cbf095b..d606771 100644
--- a/src/keybinds.h
+++ b/src/keybinds.h
@@ -72,6 +72,10 @@ static inline bool keys_are_pressed(const std::vector<KeySym>& keys) {
 
     return false;
 }
+#else
+static inline bool keys_are_pressed(const std::vector<KeySym>& keys) {
+  return false;
+}
 #endif
 
 #endif //MANGOHUD_KEYBINDS_H
