diff --git a/src/cpu.cpp b/src/cpu.cpp
index f46faba..2e97710 100644
--- a/src/cpu.cpp
+++ b/src/cpu.cpp
@@ -615,6 +615,10 @@ bool CPUStats::GetCpuFile() {
         } else if (name == "apm_xgene") {
             find_input(path, "temp", input, "SoC Temperature");
             break;
+        } else if (name == "cpu0_thermal") { // Batocera RPMini sm8250
+            find_input(path, "temp", input, "temp1");
+            break;
+
         } else {
             path.clear();
         }
