diff --git a/src/cpu.cpp b/src/cpu.cpp
index 093a55d..d1d7046 100644
--- a/src/cpu.cpp
+++ b/src/cpu.cpp
@@ -606,6 +606,10 @@ bool CPUStats::GetCpuFile() {
         } else if (name == "apm_xgene") {
             find_input(path, "temp", input, "SoC Temperature");
             break;
+        } else if (name == "cpu0_thermal") { // Batocera RPMini sm8250
+            find_input(path, "temp", input, "temp1");
+            break;
+
         } else {
             path.clear();
         }
