diff --git a/src/battery.cpp b/src/battery.cpp
index e5d311b..d82f3ba 100644
--- a/src/battery.cpp
+++ b/src/battery.cpp
@@ -13,7 +13,7 @@ void BatteryStats::numBattery() {
     fs::path path("/sys/class/power_supply/");
     for (auto& p : fs::directory_iterator(path)) {
         string fileName = p.path().filename();
-        if (fileName.find("BAT") != std::string::npos) {
+	if (fileName.find("BAT") != std::string::npos || fileName.find("qcom-battmgr-bat") != std::string::npos) {
             battPath[batteryCount] = p.path();
             batteryCount += 1;
         }
