diff --git a/src/gl/inject_egl.cpp b/src/gl/inject_egl.cpp
index 2b6a522..11cd90e 100644
--- a/src/gl/inject_egl.cpp
+++ b/src/gl/inject_egl.cpp
@@ -167,12 +167,12 @@ EXPORT_C_(int) eglTerminate(void *display)
         pfn_eglTerminate = reinterpret_cast<decltype(pfn_eglTerminate)>(get_egl_proc_address("eglTerminate"));
 
     ret = pfn_eglTerminate(display);
-
+#ifdef HAVE_WAYLAND
     if (ret)
     {
         wayland_data_unref(NULL, display);
     }
-
+#endif
     return ret;
 }
 
