diff --git a/src/osd/libretro/libretro-internal/libretro.cpp b/src/osd/libretro/libretro-internal/libretro.cpp
index 9b56a7f02f0..bf4714cf324 100755
--- a/src/osd/libretro/libretro-internal/libretro.cpp
+++ b/src/osd/libretro/libretro-internal/libretro.cpp
@@ -8,6 +8,7 @@
 #include "osdepend.h"
 
 #include "emu.h"
+#include "video.h"
 #include "emuopts.h"
 #include "render.h"
 #include "ui/uimain.h"
diff --git a/src/osd/libretro/window.cpp b/src/osd/libretro/window.cpp
index 77cb3c4f9b2..1c809cc4fb2 100644
--- a/src/osd/libretro/window.cpp
+++ b/src/osd/libretro/window.cpp
@@ -24,6 +24,7 @@
 
 // MAME headers
 #include "emu.h"
+#include "video.h"
 #include "emuopts.h"
 #include "render.h"
 #include "rendlay.h"
@@ -256,7 +257,7 @@ int retro_window_info::window_init()
 
 	// store more reasonable geometry for SVG screens
 	const screen_device *screen = screen_device_enumerator(machine().root_device()).byindex(index());
-	if ((screen != nullptr) && (screen->screen_type() == SCREEN_TYPE_SVG))
+	if ((screen != nullptr) && screen->is_vector())
 	{
 		const rectangle &visarea = screen->visible_area();
 		svg_width  = render_round_nearest(visarea.width());
@@ -492,7 +493,7 @@ void retro_window_info::update()
 
 			// Check whether window has vector screens
 
-			if ((screen != nullptr) && (screen->screen_type() == SCREEN_TYPE_VECTOR))
+			if ((screen != nullptr) && screen->is_vector())
 				renderer().set_flags(osd_renderer::FLAG_HAS_VECTOR_SCREEN);
 			else
 				renderer().clear_flags(osd_renderer::FLAG_HAS_VECTOR_SCREEN);
