diff --git a/src/emu/screen.cpp b/src/emu/screen.cpp
index 12dc53b82b9..90864afab29 100644
--- a/src/emu/screen.cpp
+++ b/src/emu/screen.cpp
@@ -490,14 +490,14 @@ void screen_device::configure(int width, int height, const rectangle &visarea, a
 	else
 		m_vblank_period = m_scantime * (height - visarea.height());
 
-#ifdef __LIBRETRO__
-	/* Performance hack fix for "pong" and "breakout" */
+/*#ifdef __LIBRETRO__
+	// Performance hack fix for "pong" and "breakout"
 	if (screen_configured > 10
 			&& width == m_width
 			&& height == m_height
 			&& floorf(ATTOSECONDS_TO_HZ(frame_period)) == floorf(ATTOSECONDS_TO_HZ(m_frame_period)))
 		return;
-#endif
+#endif*/
 
 	// we are now fully configured with the new parameters
 	// and can safely call time_until_pos(), etc.
