diff --git a/scripts/genie.lua b/scripts/genie.lua
index 6487dcc3ec0..737e0e1f22f 100644
--- a/scripts/genie.lua
+++ b/scripts/genie.lua
@@ -853,7 +853,7 @@ if _OPTIONS["OPTIMIZE"] then
 	if _OPTIONS["LTO"]=="1" then
 		buildoptions {
 -- windows native mingw GCC 5.2 fails with -flto=x with x > 1. bug unfixed as of this commit
-			"-flto=1",
+			"-flto=28",
 -- if ld fails, just buy more RAM or uncomment this!
 --          "-Wl,-no-keep-memory",
 			"-Wl,-v",
@@ -871,12 +871,12 @@ if _OPTIONS["OPTIMIZE"] then
 --          "-fmem-report-wpa","-fmem-report","-fpre-ipa-mem-report","-fpost-ipa-mem-report","-flto-report-wpa","-fmem-report",
 -- this six flag combo lets MAME compile with LTO=1 on linux with no errors and ~2% speed boost, but compile time is much longer
 -- if you are going to wait on lto, you might as well enable these for GCC
---          "-fdevirtualize-at-ltrans","-fgcse-sm","-fgcse-las",
---          "-fipa-pta","-fipa-icf","-fvariable-expansion-in-unroller",
+			"-fdevirtualize-at-ltrans","-fgcse-sm","-fgcse-las",
+			"-fipa-pta","-fipa-icf","-fvariable-expansion-in-unroller",
 		}
 -- same flags are needed by linker
 		linkoptions {
-			"-flto=1",
+			"-flto=28",
 --          "-Wl,-no-keep-memory",
 			"-Wl,-v",
 			"-Wl,-allow-multiple-definition",
@@ -889,8 +889,8 @@ if _OPTIONS["OPTIMIZE"] then
 --          "-fmem-report-wpa","-fmem-report","-fpre-ipa-mem-report","-fpost-ipa-mem-report","-flto-report-wpa","-fmem-report",
 -- this six flag combo lets MAME compile with LTO=1 on linux with no errors and ~2% speed boost, but compile time is much longer
 -- if you are going to wait on lto, you might as well enable these for GCC
---          "-fdevirtualize-at-ltrans","-fgcse-sm","-fgcse-las",
---          "-fipa-pta","-fipa-icf","-fvariable-expansion-in-unroller",
+			"-fdevirtualize-at-ltrans","-fgcse-sm","-fgcse-las",
+			"-fipa-pta","-fipa-icf","-fvariable-expansion-in-unroller",
 
 		}
 
