diff --git a/src/devices/machine/macseconds.cpp b/src/devices/machine/macseconds.cpp
index 7ed18bc69f4..3fc4f429a63 100644
--- a/src/devices/machine/macseconds.cpp
+++ b/src/devices/machine/macseconds.cpp
@@ -7,10 +7,9 @@
     This is a mix-in for anything wanting to convert MAME time to the classic Mac time format
     (seconds since 1/1/1904 at midnight).
 */
-
+#include "emu.h"
 #include "dirtc.h"
 
-#include "emu.h"
 #include "macseconds.h"
 
 macseconds_interface::macseconds_interface()
diff --git a/src/devices/video/serialvfd.cpp b/src/devices/video/serialvfd.cpp
index a6ac758bc04..5aa7508ab97 100644
--- a/src/devices/video/serialvfd.cpp
+++ b/src/devices/video/serialvfd.cpp
@@ -1,8 +1,8 @@
 // license:BSD-3-Clause
 // copyright-holders: NaokiS
 
-#include "serialvfd.h"
 #include "emu.h"
+#include "serialvfd.h"
 
 #define LOG_VFD     (1U << 1)
 //#define VERBOSE     ( LOG_VFD )
