diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index a653adf8a..06ca8410d 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -102,7 +102,7 @@ if(${PCSX2_TARGET_ARCHITECTURES} MATCHES "x86_64")
 		else()
 			# Can't use march=native on Apple Silicon.
 			if(NOT APPLE OR "${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "x86_64")
-				set(ARCH_FLAG "-march=native")
+				set(ARCH_FLAG "-msse -msse2 -msse4.1 -mfxsr")
 			endif()
 		endif()
 	elseif(NOT DEFINED ARCH_FLAG AND USE_CLANG_CL)
