diff --git a/libretro/main.cpp b/libretro/main.cpp
index 15c6efdf8..da5b9e802 100644
--- a/libretro/main.cpp
+++ b/libretro/main.cpp
@@ -313,7 +313,7 @@ void retro_init(void)
 
 	// get the BIOS available and fill the option
 
-	bios_dir = Path::Combine(system, "pcsx2/bios");
+	bios_dir = Path::Combine(system, "");
 	wxDir::GetAllFiles(bios_dir.GetFullPath(), &bios_list, L"*.*", wxDIR_FILES);
 	for (wxString bios_file : bios_list)
 	{
