diff --git a/src/sdlport/setup.cpp b/src/sdlport/setup.cpp
index 751a52c..1f5eb88 100644
--- a/src/sdlport/setup.cpp
+++ b/src/sdlport/setup.cpp
@@ -413,7 +413,7 @@ void setup( int argc, char **argv )
     if( (homedir = getenv( "HOME" )) != NULL )
     {
         savedir = (char *)malloc( strlen( homedir ) + 9 );
-        sprintf( savedir, "%s/.abuse/", homedir );
+        sprintf( savedir, "/userdata/system/configs/abuse/", homedir );
         // Check if we already have a savegame directory
         if( (fd = fopen( savedir, "r" )) == NULL )
         {
