diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..7aaa1f1
--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.0)
+project("prototype")
+add_executable(prototype
+    AnimGenerator.cpp  Bug.h             Editor_EntityProperties.cpp  Fire.cpp        LadyBird.cpp     PowerDroid.h       Rocket.cpp      SpaceTrash.h      stdafx.h             UTIL_bitop.h
+    AnimGenerator.h    Bullet.cpp        Engine.cpp                   Fire.h          LadyBird.h       PowerUp.cpp        Rocket.h        Sperm.cpp         SuperTurret.cpp      UTIL_Fade.cpp
+    Beam.cpp           Bullet.h          Engine_Generators.cpp        Font.cpp        Log.cpp          PowerUp.h          SDL_KeyMap.h    Sperm.h           SuperTurret.h        UTIL_Fade.h
+    Beam.h             Button.cpp        Engine.h                     Font.h          Log.h            Profiler.cpp       Serializer.cpp  Sprite.cpp        Tile.h               UTIL_Misc.cpp
+    Boss.cpp           Button.h          Entity.cpp                   FrontEnd.cpp    Map.cpp          Profiler.h         Serializer.h    SpriteEffect.cpp  Timer.cpp            UTIL_Misc.h
+    Boss.h             CheckPoint.cpp    Entity.h                     Game.cpp        MidBoss.cpp      ProtoType.cpp      Serpent.cpp     SpriteEffect.h    Timer.h              UTIL_openGL.cpp
+    BrainBoss.cpp      CheckPoint.h      Explosion.cpp                Generator.cpp   MidBoss.h        RandExplosion.cpp  Serpent.h       Sprite.h          TriggerSlowDown.cpp  UTIL_openGL.h
+    BrainBoss.h        Debris.cpp        Explosion.h                  Generator.h     Orb.cpp          RandExplosion.h    SolarFlare.cpp  SpriteSheet.cpp   TriggerSlowDown.h    UTIL_SDL.cpp
+    BrainCell.cpp      Debris.h          Fighter.cpp                  Geo2D.cpp       Orb.h            resource1.h        SolarFlare.h    SpriteSheet.h     TriggerSpeedUp.cpp   UTIL_SDL.h
+    BrainCell.h        Demo.cpp          Fighter.h                    Geo2D_Draw.cpp  Player.cpp       resource2.h        Sound.cpp       StarBurst.cpp     TriggerSpeedUp.h     Vector.h
+    BreakScenery.cpp   Demo.h            FileIO.h                     Geo2D_Draw.h    Player.h         resource.h         Sound.h         StarBurst.h       Turret.cpp
+    BreakScenery.h     Editor_Anims.cpp  firebug.cpp                  Geo2D.h         PostEffects.cpp  Rhino.cpp          Sound_Macros.h  stb_image.h       Turret.h
+    Bug.cpp            Editor.cpp        fireBug.h                    HUD.cpp         PowerDroid.cpp   Rhino.h            SpaceTrash.cpp  stdafx.cpp        UTIL_bitop.cpp
+)
+
+target_compile_options(prototype PRIVATE -g -O -MMD -Wall -Wpedantic -DUSE_SDL2 -DNO_FMOD)
+set_property(TARGET prototype PROPERTY CXX_STANDARD 11)
+
+include(FindPkgConfig)
+
+pkg_search_module(SDL2 REQUIRED IMPORTED_TARGET sdl2)
+pkg_search_module(SDL2MIXER REQUIRED IMPORTED_TARGET SDL2_mixer)
+
+find_library(GL OpenGL)
+find_library(ZLIB z)
+
+target_include_directories(prototype PUBLIC PkgConfig::SDL2)
+target_link_libraries(prototype PkgConfig::SDL2 PkgConfig::SDL2MIXER ${GL} ${ZLIB})
