diff -Naur FreeImage.orig/Makefile.gnu FreeImage/Makefile.gnu
--- FreeImage.orig/Makefile.gnu	2015-03-10 09:04:00.000000000 +0000
+++ FreeImage/Makefile.gnu	2021-07-24 13:02:19.292229932 +0100
@@ -15,7 +15,7 @@
 
 MODULES = $(SRCS:.c=.o)
 MODULES := $(MODULES:.cpp=.o)
-CFLAGS ?= -O3 -fPIC -fexceptions -fvisibility=hidden
+CFLAGS ?= -O3 -fPIC -fexceptions -fvisibility=hidden -std=gnu++98
 # OpenJPEG
 CFLAGS += -DOPJ_STATIC
 # LibRaw
@@ -25,7 +25,7 @@
 CFLAGS += $(INCLUDE)
 CXXFLAGS ?= -O3 -fPIC -fexceptions -fvisibility=hidden -Wno-ctor-dtor-privacy
 # LibJXR
-CXXFLAGS += -D__ANSI__
+CXXFLAGS += -D__ANSI__ -std=gnu++98
 CXXFLAGS += $(INCLUDE)
 
 ifeq ($(shell sh -c 'uname -m 2>/dev/null || echo not'),x86_64)
