diff -Naur bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83.orig/projects/cmake/bgdc/CMakeLists.txt bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83/projects/cmake/bgdc/CMakeLists.txt
--- bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83.orig/projects/cmake/bgdc/CMakeLists.txt	2021-10-06 04:57:50.000000000 +0100
+++ bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83/projects/cmake/bgdc/CMakeLists.txt	2021-10-20 23:06:56.298658793 +0100
@@ -82,7 +82,7 @@
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
 
 # Include the compiler flags from the external include file
-include(compiler_flags)
+#include(compiler_flags)
 
 # Link the executable to the libraries
 target_link_libraries(${PROJECT_NAME} ${ZLIB_LIBRARIES} ${EXTRA_LIBS})
@@ -96,4 +96,4 @@
     set(INSTALL_BIN_DIR "${CMAKE_INSTALL_PREFIX}/bin/gnulinux32" CACHE PATH "Installation directory for executables")
 endif()
 install(TARGETS ${PROJECT_NAME}
-        RUNTIME DESTINATION "${INSTALL_BIN_DIR}")
\ No newline at end of file
+        RUNTIME DESTINATION "${INSTALL_BIN_DIR}")
diff -Naur bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83.orig/projects/cmake/bgdi/bgdrtm/CMakeLists.txt bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83/projects/cmake/bgdi/bgdrtm/CMakeLists.txt
--- bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83.orig/projects/cmake/bgdi/bgdrtm/CMakeLists.txt	2021-10-06 04:57:50.000000000 +0100
+++ bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83/projects/cmake/bgdi/bgdrtm/CMakeLists.txt	2021-10-20 23:11:43.560191747 +0100
@@ -151,5 +151,5 @@
 add_library(${PROJECT_NAME} STATIC ${SOURCE_FILES})
 
 # Include the compiler flags from the external include file
-include(compiler_flags)
+#include(compiler_flags)
 
diff -Naur bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83.orig/projects/cmake/bgdi/CMakeLists.txt bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83/projects/cmake/bgdi/CMakeLists.txt
--- bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83.orig/projects/cmake/bgdi/CMakeLists.txt	2021-10-06 04:57:50.000000000 +0100
+++ bennugd-monolithic-60ee3389efcf9b402d66035e87f33d17d70cbd83/projects/cmake/bgdi/CMakeLists.txt	2021-10-20 23:07:24.751000609 +0100
@@ -36,7 +36,7 @@
 add_executable(${PROJECT_NAME} ${SOURCE_FILES})
 
 # Include the compiler flags from the external include file
-include(compiler_flags)
+#include(compiler_flags)
 
 # Link the executable to the libraries
 target_link_libraries(${PROJECT_NAME}
