diff --git a/Makefile b/Makefile
index 293ff75..e2a61a5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,19 +1,9 @@
 obj-m += hid-nx.o
+KVERSION := `uname -r`
+gamecon_id-nx.o := -O3 -Wno-implicit-fallthrough
 
-KERNEL_RELEASE := $(shell uname -r)
-KDIR           := /lib/modules/$(KERNEL_RELEASE)/build
+all:
+	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) modules
 
-OBJ_FILE       := $(obj-m)
-SRC_FILE       := $(OBJ_FILE:.o=.c)
-CMD_FILE       := .$(OBJ_FILE).cmd
-MODNAME        := $(OBJ_FILE:.o=)
-
-all default: modules
-install: modules_install
-
-$(OBJ_FILE) $(MODNAME).ko: $(SRC_FILE)
-
-modules modules_install clean $(OBJ_FILE) $(MODNAME).ko:
-	$(MAKE) -C $(KDIR) M=$(CURDIR) $@
-
-.PHONY: all default install modules modules_install clean
+clean:
+	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) clean
