diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 1508239..1580620 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -29,7 +29,7 @@ FetchContent_Declare(cmake_git_version_tracking
   GIT_TAG 6c0cb87edd029ddfb403a8e24577c144a03605a6
 )
 
-FetchContent_MakeAvailable(cmake_git_version_tracking)
+#FetchContent_MakeAvailable(cmake_git_version_tracking)
 
 add_library(core OBJECT ${COMMON_SRC} ${BGDRTM_SRC})
 
@@ -64,6 +64,5 @@ target_include_directories(core PRIVATE )
 target_link_libraries(core
     ${OPENSSL_CRYPTO_LIBRARY}
     ZLIB::ZLIB
-    SDL::SDL
-    cmake_git_version_tracking)
+    SDL::SDL)
 
diff --git a/core/common/bgd_version.c b/core/common/bgd_version.c
index 24bdd6e..591377a 100644
--- a/core/common/bgd_version.c
+++ b/core/common/bgd_version.c
@@ -7,7 +7,7 @@ const char* bgd_getversion()
 {
     if (*buffer==0)
     {
-        snprintf(buffer, sizeof(buffer), "1.0.0-BennuGD_libretro-%s-%s", git_Branch(), git_CommitSHA1() );
+        snprintf(buffer, sizeof(buffer), "1.0.0-BennuGD_libretro-%s-%s", git_Branch, git_CommitSHA1 );
     }
 
     return buffer;
diff --git a/core/include/git.h b/core/include/git.h
new file mode 100644
index 0000000..895a432
--- /dev/null
+++ b/core/include/git.h
@@ -0,0 +1,4 @@
+#include <stdint.h>
+
+extern char git_Branch[] = "master";
+extern char git_CommitSHA1[] = "1234abc";
