diff --git a/src/trx/av/video.c b/src/trx/av/video.c
index b167a0bdf..672002e0f 100644
--- a/src/trx/av/video.c
+++ b/src/trx/av/video.c
@@ -1758,7 +1758,9 @@ static int M_ReadThread(void *arg)
     is->ic = ic;
 
     avformat_find_stream_info(ic, nullptr);
-    av_format_inject_global_side_data(ic);
+#if (LIBAVFORMAT_VERSION_MAJOR < 61) || ((LIBAVFORMAT_VERSION_MAJOR == 61) && (LIBAVFORMAT_VERSION_MINOR < 8))
+    av_format_inject_global_side_data(&ic);
+#endif 
 
     if (ic->pb) {
         ic->pb->eof_reached = 0;
