diff --git a/ayaneo-platform.c b/ayaneo-platform.c
index 15a31e9..27e9465 100644
--- a/ayaneo-platform.c
+++ b/ayaneo-platform.c
@@ -914,7 +914,9 @@ static void suspend_mode_register_attr(void)
         }
 }
 
-ATTRIBUTE_GROUPS(ayaneo_led_mc);
+static struct attribute_group ayaneo_led_mc_group = {
+      .attrs = ayaneo_led_mc_attrs,
+};
 
 struct mc_subled ayaneo_led_mc_subled_info[] = {
         {
@@ -939,7 +941,7 @@ struct mc_subled ayaneo_led_mc_subled_info[] = {
 
 struct led_classdev_mc ayaneo_led_mc = {
         .led_cdev = {
-                .name = "multicolor:chassis",
+                .name = "ayn:rgb:joystick_rings",
                 .brightness = 0,
                 .max_brightness = 255,
                 .brightness_set = ayaneo_led_mc_brightness_set,
@@ -1013,7 +1015,7 @@ static int ayaneo_platform_probe(struct platform_device *pdev)
         if (ret)
                 return ret;
 
-        ret = devm_device_add_group(ayaneo_led_mc.led_cdev.dev, *ayaneo_led_mc_groups);
+        ret = devm_device_add_group(ayaneo_led_mc.led_cdev.dev, &ayaneo_led_mc_group);
         return ret;
 }
 
@@ -1037,7 +1039,7 @@ static struct platform_driver ayaneo_platform_driver = {
         .resume = ayaneo_platform_resume,
         .suspend = ayaneo_platform_suspend,
         .shutdown = ayaneo_platform_shutdown,
-        .remove_new = ayaneo_platform_remove,
+        .remove = ayaneo_platform_remove,
 };
 
 static struct platform_device *ayaneo_platform_device;
