diff --git a/bin/windows/vpx/src/backtrace.c b/bin/windows/vpx/src/backtrace.c
index 4bbe88bbe..5e20d114a 100644
--- a/bin/windows/vpx/src/backtrace.c
+++ b/bin/windows/vpx/src/backtrace.c
@@ -531,7 +531,7 @@ exception_filter(LPEXCEPTION_POINTERS info)
         }
     }
 
-    logfd = open(crashlogfilename, O_APPEND | O_CREAT | O_RDWR, S_IRUSR | S_IWUSR);
+    logfd = open("/userdata/system/logs/gzdoom.crash.log", O_APPEND | O_CREAT | O_RDWR, S_IRUSR | S_IWUSR);
 
     if (logfd) {
         time_t curtime;
diff --git a/src/common/platform/posix/sdl/i_main.cpp b/src/common/platform/posix/sdl/i_main.cpp
index a8ee3bf80..a83fc9db3 100644
--- a/src/common/platform/posix/sdl/i_main.cpp
+++ b/src/common/platform/posix/sdl/i_main.cpp
@@ -184,23 +184,7 @@ int main (int argc, char **argv)
 
 	Args = new FArgs(argc, argv);
 
-#ifdef PROGDIR
-	progdir = PROGDIR;
-#else
-	char program[PATH_MAX];
-	if (realpath (argv[0], program) == NULL)
-		strcpy (program, argv[0]);
-	char *slash = strrchr (program, '/');
-	if (slash != NULL)
-	{
-		*(slash + 1) = '\0';
-		progdir = program;
-	}
-	else
-	{
-		progdir = "./";
-	}
-#endif
+    progdir = "/usr/share/gzdoom/";
 
 	I_StartupJoysticks();
 
diff --git a/src/common/platform/posix/unix/i_specialpaths.cpp b/src/common/platform/posix/unix/i_specialpaths.cpp
index c06e27b67..268d83e42 100644
--- a/src/common/platform/posix/unix/i_specialpaths.cpp
+++ b/src/common/platform/posix/unix/i_specialpaths.cpp
@@ -47,59 +47,7 @@ extern bool netgame;
 
 FString GetUserFile (const char *file)
 {
-	FString path;
-	struct stat info;
-
-	path = NicePath("$HOME/" GAME_DIR "/");
-
-	if (stat (path.GetChars(), &info) == -1)
-	{
-		struct stat extrainfo;
-
-		// Sanity check for $HOME/.config
-		FString configPath = NicePath("$HOME/.config/");
-		if (stat (configPath.GetChars(), &extrainfo) == -1)
-		{
-			if (mkdir (configPath.GetChars(), S_IRUSR | S_IWUSR | S_IXUSR) == -1)
-			{
-				I_FatalError ("Failed to create $HOME/.config directory:\n%s", strerror(errno));
-			}
-		}
-		else if (!S_ISDIR(extrainfo.st_mode))
-		{
-			I_FatalError ("$HOME/.config must be a directory");
-		}
-
-		// This can be removed after a release or two
-		// Transfer the old zdoom directory to the new location
-		bool moved = false;
-		FString oldpath = NicePath("$HOME/." GAMENAMELOWERCASE "/");
-		if (stat (oldpath.GetChars(), &extrainfo) != -1)
-		{
-			if (rename(oldpath.GetChars(), path.GetChars()) == -1)
-			{
-				I_Error ("Failed to move old " GAMENAMELOWERCASE " directory (%s) to new location (%s).",
-					oldpath.GetChars(), path.GetChars());
-			}
-			else
-				moved = true;
-		}
-
-		if (!moved && mkdir (path.GetChars(), S_IRUSR | S_IWUSR | S_IXUSR) == -1)
-		{
-			I_FatalError ("Failed to create %s directory:\n%s",
-				path.GetChars(), strerror (errno));
-		}
-	}
-	else
-	{
-		if (!S_ISDIR(info.st_mode))
-		{
-			I_FatalError ("%s must be a directory", path.GetChars());
-		}
-	}
-	path += file;
-	return path;
+    return FString("/userdata/saves/gzdoom/") + file;
 }
 
 //===========================================================================
@@ -114,7 +62,7 @@ FString M_GetAppDataPath(bool create)
 {
 	// Don't use GAME_DIR and such so that ZDoom and its child ports can
 	// share the node cache.
-	FString path = NicePath("$HOME/.config/" GAMENAMELOWERCASE);
+	FString path = NicePath("/userdata/saves/gzdoom");
 	if (create)
 	{
 		CreatePath(path.GetChars());
@@ -134,7 +82,7 @@ FString M_GetCachePath(bool create)
 {
 	// Don't use GAME_DIR and such so that ZDoom and its child ports can
 	// share the node cache.
-	FString path = NicePath("$HOME/.config/zdoom/cache");
+	FString path = NicePath("/userdata/system/cache/gzdoom");
 	if (create)
 	{
 		CreatePath(path.GetChars());
@@ -152,7 +100,7 @@ FString M_GetCachePath(bool create)
 
 FString M_GetAutoexecPath()
 {
-	return GetUserFile("autoexec.cfg");
+	return FString("/userdata/system/configs/gzdoom/autoexec.cfg");
 }
 
 //===========================================================================
@@ -167,7 +115,7 @@ FString M_GetAutoexecPath()
 
 FString M_GetConfigPath(bool for_reading)
 {
-	return GetUserFile(GAMENAMELOWERCASE ".ini");
+	return FString("/userdata/system/configs/gzdoom/gzdoom.ini");
 }
 
 //===========================================================================
@@ -180,7 +128,7 @@ FString M_GetConfigPath(bool for_reading)
 
 FString M_GetScreenshotsPath()
 {
-	return NicePath("$HOME/" GAME_DIR "/screenshots/");
+	return FString("/userdata/screenshots/");
 }
 
 //===========================================================================
@@ -209,7 +157,7 @@ FString M_GetSavegamesPath()
 
 FString M_GetDocumentsPath()
 {
-	return NicePath("$HOME/" GAME_DIR "/");
+	return FString("/userdata/saves/gzdoom/");
 }
 
 //===========================================================================
diff --git a/src/gameconfigfile.cpp b/src/gameconfigfile.cpp
index 3b259e8f5..6ee1792c1 100644
--- a/src/gameconfigfile.cpp
+++ b/src/gameconfigfile.cpp
@@ -156,7 +156,7 @@ FGameConfigFile::FGameConfigFile ()
 		SetValueForKey ("Path", SHARE_DIR "/games/doom", true);
 		SetValueForKey ("Path", "/usr/local/share/doom", true);
 		SetValueForKey ("Path", "/usr/local/share/games/doom", true);
-		SetValueForKey ("Path", "/usr/share/doom", true);
+		SetValueForKey ("Path", "/usr/share/gzdoom", true);
 		SetValueForKey ("Path", "/usr/share/games/doom", true);
 #endif
 		SetValueForKey ("Path", "$DOOMWADDIR", true);
@@ -187,8 +187,10 @@ FGameConfigFile::FGameConfigFile ()
 		SetValueForKey("Path", "/usr/local/share/doom/fm_banks", true);
 		SetValueForKey("Path", "/usr/local/share/games/doom/soundfonts", true);
 		SetValueForKey("Path", "/usr/local/share/games/doom/fm_banks", true);
-		SetValueForKey("Path", "/usr/share/doom/soundfonts", true);
-		SetValueForKey("Path", "/usr/share/doom/fm_banks", true);
+		SetValueForKey("Path", "/usr/share/gzdoom/soundfonts", true);
+		SetValueForKey("Path", "/usr/share/gzdoom/fm_banks", true);
+		SetValueForKey("Path", "/userdata/system/configs/gzdoom/soundfonts", true);
+		SetValueForKey("Path", "/userdata/system/configs/gzdoom/fm_banks", true);
 		SetValueForKey("Path", "/usr/share/games/doom/soundfonts", true);
 		SetValueForKey("Path", "/usr/share/games/doom/fm_banks", true);
 		SetValueForKey("Path", SHARE_DIR "/doom/soundfonts", true);
