diff --git a/CMakeLists.txt b/CMakeLists.txt
index 77bf8aa7e..9af66316b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -217,7 +217,6 @@ endmacro()
 option( NO_OPENAL "Disable OpenAL sound support" OFF )
 
 find_package( BZip2 )
-find_package( VPX )
 if (NOT FORCE_INTERNAL_CPPDAP)
 	find_package( cppdap CONFIG )
 endif()
@@ -233,7 +232,6 @@ if( ${TARGET_ARCHITECTURE} MATCHES "x86_64" )
 	set( HAVE_VM_JIT ON )
 endif()
 if(NOT TOOLS_ONLY)
-	find_package( JPEG )
 	find_package( VPX )
 endif()
 
@@ -430,42 +428,6 @@ if(NOT TOOLS_ONLY)
 	set( DRPC_LIBRARIES discord-rpc )
 	set( DRPC_LIBRARY discord-rpc )
 
-	if( HAVE_VM_JIT AND UNIX )
-		check_symbol_exists( "backtrace" "execinfo.h" HAVE_BACKTRACE )
-		if( NOT HAVE_BACKTRACE )
-			set( CMAKE_REQUIRED_FLAGS "-lexecinfo" )
-			check_symbol_exists( "backtrace" "execinfo.h" HAVE_LIBEXECINFO )
-			if( HAVE_LIBEXECINFO )
-				set( ALL_C_FLAGS "${ALL_C_FLAGS} -lexecinfo" )
-			else( HAVE_LIBEXECINFO )
-				set( HAVE_VM_JIT NO )
-			endif( HAVE_LIBEXECINFO )
-			set( CMAKE_REQUIRED_FLAGS )
-		endif( NOT HAVE_BACKTRACE )
-	endif( HAVE_VM_JIT AND UNIX )
-
-	if( ${HAVE_VM_JIT} )
-		if( ASMJIT_FOUND AND NOT FORCE_INTERNAL_ASMJIT )
-			message( STATUS "Using system asmjit, includes found at ${ASMJIT_INCLUDE_DIR}" )
-		else()
-			message( STATUS "Using internal asmjit" )
-			set( SKIP_INSTALL_ALL TRUE ) # Avoid installing asmjit alongside zdoom
-			add_subdirectory( libraries/asmjit )
-			set( ASMJIT_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/libraries/asmjit )
-			set( ASMJIT_LIBRARIES asmjit )
-			set( ASMJIT_LIBRARY asmjit )
-		endif()
-	endif()
-
-	if( JPEG_FOUND AND NOT FORCE_INTERNAL_JPEG )
-		message( STATUS "Using system jpeg library, includes found at ${JPEG_INCLUDE_DIR}" )
-	else()
-		message( STATUS "Using internal jpeg library" )
-		add_subdirectory( libraries/jpeg )
-		set( JPEG_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/libraries/jpeg )
-		set( JPEG_LIBRARIES jpeg )
-		set( JPEG_LIBRARY jpeg )
-	endif()
 endif()
 
 
