diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ac374e55..4e720ca21 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -348,33 +348,6 @@ set( DRPC_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/libraries/discordrpc/include"
 set( DRPC_LIBRARIES discord-rpc )
 set( DRPC_LIBRARY discord-rpc )
 
-if( HAVE_VM_JIT AND UNIX )
-	check_symbol_exists( "backtrace" "execinfo.h" HAVE_BACKTRACE )
-	if( NOT HAVE_BACKTRACE )
-		set( CMAKE_REQUIRED_FLAGS "-lexecinfo" )
-		check_symbol_exists( "backtrace" "execinfo.h" HAVE_LIBEXECINFO )
-		if( HAVE_LIBEXECINFO )
-			set( ALL_C_FLAGS "${ALL_C_FLAGS} -lexecinfo" )
-		else( HAVE_LIBEXECINFO )
-			set( HAVE_VM_JIT NO )
-		endif( HAVE_LIBEXECINFO )
-		set( CMAKE_REQUIRED_FLAGS )
-	endif( NOT HAVE_BACKTRACE )
-endif( HAVE_VM_JIT AND UNIX )
-
-if( ${HAVE_VM_JIT} )
-	if( ASMJIT_FOUND AND NOT FORCE_INTERNAL_ASMJIT )
-		message( STATUS "Using system asmjit, includes found at ${ASMJIT_INCLUDE_DIR}" )
-	else()
-		message( STATUS "Using internal asmjit" )
-		set( SKIP_INSTALL_ALL TRUE ) # Avoid installing asmjit alongside zdoom
-		add_subdirectory( libraries/asmjit )
-		set( ASMJIT_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/libraries/asmjit )
-		set( ASMJIT_LIBRARIES asmjit )
-		set( ASMJIT_LIBRARY asmjit )
-	endif()
-endif()
-
 if( BZIP2_FOUND AND NOT FORCE_INTERNAL_BZIP2 )
 	message( STATUS "Using system bzip2 library, includes found at ${BZIP2_INCLUDE_DIR}" )
 else()
