diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 82733be96..d1baff833 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1416,7 +1416,7 @@ add_custom_command(TARGET zdoom POST_BUILD
 	${CMAKE_SOURCE_DIR}/fm_banks/gs-by-papiezak-and-sneakernets.wopn $<TARGET_FILE_DIR:zdoom>/fm_banks/gs-by-papiezak-and-sneakernets.wopn
 )
 
-if ( NOT APPLE )
+if ( APPLE )
 	# bring zmusic in
 	add_custom_command(TARGET zdoom POST_BUILD
 		COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:zmusic> $<TARGET_FILE_DIR:zdoom>
