diff --git a/src/xenia/base/premake5.lua b/src/xenia/base/premake5.lua
index d53c34351..f1d64fc1b 100644
--- a/src/xenia/base/premake5.lua
+++ b/src/xenia/base/premake5.lua
@@ -17,4 +17,3 @@ project("xenia-base")
     "debug_visualizers.natvis",
   })
 
-include("testing")
diff --git a/src/xenia/cpu/premake5.lua b/src/xenia/cpu/premake5.lua
index 14a423126..d1bfd94a5 100644
--- a/src/xenia/cpu/premake5.lua
+++ b/src/xenia/cpu/premake5.lua
@@ -21,8 +21,6 @@ project("xenia-cpu")
   local_platform_files("hir")
   local_platform_files("ppc")
 
-include("testing")
-include("ppc/testing")
 --  filter({"configurations:Release", "platforms:Windows"})
 --  buildoptions({
 --    "/O1",
diff --git a/src/xenia/vfs/premake5.lua b/src/xenia/vfs/premake5.lua
index 214f18864..7095ac36b 100644
--- a/src/xenia/vfs/premake5.lua
+++ b/src/xenia/vfs/premake5.lua
@@ -38,4 +38,3 @@ project("xenia-vfs-dump")
   resincludedirs({
     project_root,
   })
-include("testing")
diff --git a/tools/build/premake5.lua b/tools/build/premake5.lua
index 6f99284e4..6019a6dc4 100644
--- a/tools/build/premake5.lua
+++ b/tools/build/premake5.lua
@@ -6,4 +6,3 @@ include("scripts/force_compile_as_cc.lua")
 include("scripts/pkg_config.lua")
 include("scripts/platform_files.lua")
 include("scripts/single_library.lua")
-include("scripts/test_suite.lua")
diff --git a/xenia-build b/xenia-build
index 3e3c75dea..c1f19411a 100755
--- a/xenia-build
+++ b/xenia-build
@@ -513,7 +513,6 @@ def run_premake(target_os, action, cc=None):
         os.path.join('tools', 'build', 'premake'),
         '--file=premake5.lua',
         '--os=%s' % target_os,
-        '--test-suite-mode=combined',
         '--verbose',
         action,
     ]
