diff --git a/CMakeLists.txt b/CMakeLists.txt
index 115ffab45d..4704b72cd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -309,3 +309,7 @@ set(CMAKE_FOLDER "")
 
 # Set default startup project for Visual Studio
 set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT xenia-app)
+
+# Generate version.h
+include(cmake/version_h.cmake)
+generate_version_h()
diff --git a/cmake/version_h.cmake b/cmake/version_h.cmake
new file mode 100644
index 0000000000..341ec60108
--- /dev/null
+++ b/cmake/version_h.cmake
@@ -0,0 +1,46 @@
+function(generate_version_h)
+    message(STATUS "Generating version.h...")
+    set(XE_BUILD_BRANCH "detached")
+    set(XE_BUILD_COMMIT "unknown")
+    set(XE_BUILD_COMMIT_SHORT "unknown")
+
+    find_package(Git)
+    if(${Git_FOUND})
+        execute_process(
+            COMMAND ${GIT_EXECUTABLE} symbolic-ref --short -q HEAD
+            WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+            OUTPUT_VARIABLE XE_BUILD_BRANCH
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+        if(NOT XE_BUILD_BRANCH)
+            set(XE_BUILD_BRANCH "detached")
+        endif()
+        message(STATUS "  Build branch: ${XE_BUILD_BRANCH}")
+
+        execute_process(
+            COMMAND ${GIT_EXECUTABLE} log -1 --format=%H
+            WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+            OUTPUT_VARIABLE XE_BUILD_COMMIT
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+        message(STATUS "  Git hash: ${XE_BUILD_COMMIT}")
+
+        execute_process(
+            COMMAND ${GIT_EXECUTABLE} log -1 --format=%h
+            WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
+            OUTPUT_VARIABLE XE_BUILD_COMMIT_SHORT
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+        message(STATUS "  Git hash (short): ${XE_BUILD_COMMIT_SHORT}")
+    else()
+        message(WARNING "Failed to find git; cannot determine branch and commit!")
+    endif()
+
+    if($ENV{GITHUB_REF} MATCHES "refs\/pull\/.*")
+        set(XE_BUILD_IS_PR TRUE)
+        string(REGEX REPLACE "refs\/pull\/([0-9]+)\/.*" "\\1" XE_BUILD_PR_NUMBER $ENV{GITHUB_REF})
+        message(STATUS "  PR number: ${XE_BUILD_PR_NUMBER}")
+    endif()
+
+    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/version.h.in ${CMAKE_CURRENT_SOURCE_DIR}/src/version.h @ONLY)
+endfunction()
diff --git a/src/version.h.in b/src/version.h.in
new file mode 100644
index 0000000000..37290b7ec9
--- /dev/null
+++ b/src/version.h.in
@@ -0,0 +1,9 @@
+#ifndef GENERATED_VERSION_H_
+#define GENERATED_VERSION_H_
+#define XE_BUILD_BRANCH "@XE_BUILD_BRANCH@"
+#define XE_BUILD_COMMIT "@XE_BUILD_COMMIT@"
+#define XE_BUILD_COMMIT_SHORT "@XE_BUILD_COMMIT_SHORT@"
+#define XE_BUILD_DATE __DATE__
+#cmakedefine XE_BUILD_IS_PR
+#cmakedefine XE_BUILD_PR_NUMBER "@XE_BUILD_PR_NUMBER@"
+#endif  // GENERATED_VERSION_H_
