From 02161d025409a5dc7b74fb60e87760e9023fff98 Mon Sep 17 00:00:00 2001
From: Shoegzer <scriptis@hotmail.com>
Date: Thu, 29 Dec 2022 16:22:52 -0500
Subject: [PATCH] Add include to fix compiling

---
 src/xenia/ui/imgui_drawer.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/xenia/ui/imgui_drawer.h b/src/xenia/ui/imgui_drawer.h
index 6009ebdf17..f5057f4033 100644
--- a/src/xenia/ui/imgui_drawer.h
+++ b/src/xenia/ui/imgui_drawer.h
@@ -13,6 +13,7 @@
 #include <cstddef>
 #include <cstdint>
 #include <memory>
+#include <optional>
 #include <vector>
 
 #include "xenia/ui/immediate_drawer.h"
