diff --git a/base/internal_version.h b/base/internal_version.h
index 8e0d74a1e1f..a8a039dc2cb 100644
--- a/base/internal_version.h
+++ b/base/internal_version.h
@@ -16,7 +16,7 @@
 #define SCUMMVM_REVISION
 #endif
 
-#define SCUMMVM_VERSION "2.5.1pre"
+#define SCUMMVM_VERSION "2.5.1 (bd2bf31)"
 
 #define SCUMMVM_VER_MAJOR 2
 #define SCUMMVM_VER_MINOR 5
diff --git a/configure b/configure
index c2b6c757e4d..461cc030fd4 100755
--- a/configure
+++ b/configure
@@ -1490,9 +1490,6 @@ for ac_option in $@; do
 			engine_disable ${engine_name}
 		done
 		;;
-	*)
-		option_error
-		;;
 	esac;
 done;
 
@@ -3292,13 +3289,13 @@ if test -n "$_host"; then
 			# but Raspbian does.
 			# Be careful as it's the linker (LDFLAGS) which must know about sysroot.
 			# These are needed to build against Raspbian's libSDL.
-			append_var LDFLAGS "--sysroot=$RPI_ROOT"
-			append_var LDFLAGS "-B$RPI_ROOT/usr/lib/arm-linux-gnueabihf"
-			append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/usr/lib/arm-linux-gnueabihf/pulseaudio"
-			append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/usr/lib/arm-linux-gnueabihf"
-			append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/lib/arm-linux-gnueabihf"
-			append_var CXXFLAGS "-isystem $RPI_ROOT/usr/include/arm-linux-gnueabihf"
-			append_var CXXFLAGS "-I$RPI_ROOT/usr/include"
+			#append_var LDFLAGS "--sysroot=$RPI_ROOT"
+			#append_var LDFLAGS "-B$RPI_ROOT/usr/lib/arm-linux-gnueabihf"
+			#append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/usr/lib/arm-linux-gnueabihf/pulseaudio"
+			#append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/usr/lib/arm-linux-gnueabihf"
+			#append_var LDFLAGS "-Xlinker --rpath-link=$RPI_ROOT/lib/arm-linux-gnueabihf"
+			#append_var CXXFLAGS "-isystem $RPI_ROOT/usr/include/arm-linux-gnueabihf"
+			#append_var CXXFLAGS "-I$RPI_ROOT/usr/include"
 			# We prefer SDL2 on the Raspberry Pi: acceleration now depends on it
 			# since SDL2 manages dispmanx/GLES2 very well internally.
 			# SDL1 is bit-rotten on this platform.
