diff -Naur scummvm-80b593cdbae9035dd043f8726915ad582ecda5b0.orig/backends/events/sdl/sdl-events.cpp scummvm-80b593cdbae9035dd043f8726915ad582ecda5b0/backends/events/sdl/sdl-events.cpp
--- scummvm-80b593cdbae9035dd043f8726915ad582ecda5b0.orig/backends/events/sdl/sdl-events.cpp	2021-09-11 23:21:13.000000000 +0100
+++ scummvm-80b593cdbae9035dd043f8726915ad582ecda5b0/backends/events/sdl/sdl-events.cpp	2021-10-10 07:36:38.202061248 +0100
@@ -34,7 +34,7 @@
 #include "gui/gui-manager.h"
 
 #if SDL_VERSION_ATLEAST(2, 0, 0)
-#define GAMECONTROLLERDB_FILE "gamecontrollerdb.txt"
+#define GAMECONTROLLERDB_FILE "/usr/share/gamecontrollerdb.txt"
 
 static uint32 convUTF8ToUTF32(const char *src) {
 	if (!src || src[0] == 0)
