diff --git a/CMakeLists.txt b/CMakeLists.txt
index f14a4db2..646b5e29 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,6 +44,7 @@ if(SUPPORTS_WAYLAND)
 endif()
 if((LINUX OR FREEBSD) OR ANDROID)
   option(USE_EGL "Support EGL OpenGL context creation" ON)
+  option(USE_MALI "Link with libmali for EGL support" OFF)
 endif()
 if((LINUX OR FREEBSD) AND NOT ANDROID)
   option(USE_DRMKMS "Support DRM/KMS OpenGL contexts" OFF)
diff --git a/CMakeModules/FindEGL.cmake b/CMakeModules/FindEGL.cmake
index 48e51728..c1c947b1 100644
--- a/CMakeModules/FindEGL.cmake
+++ b/CMakeModules/FindEGL.cmake
@@ -86,6 +86,17 @@ find_library(EGL_LIBRARY
         ${PKG_EGL_LIBRARY_DIRS}
 )
 
+if(USE_MALI)
+pkg_check_modules(PKG_MALI QUIET mali)
+find_library(MALI_LIBRARY
+    NAMES
+        mali
+    HINTS
+        ${PKG_MALI_LIBRARY_DIRS}
+)
+endif()
+
+
 # NB: We do *not* use the version information from pkg-config, as that
 #     is the implementation version (eg: the Mesa version)
 if(EGL_INCLUDE_DIR)
@@ -117,7 +128,11 @@ endif()
 
 cmake_push_check_state(RESET)
 list(APPEND CMAKE_REQUIRED_LIBRARIES "${EGL_LIBRARY}")
+if(USE_MALI)
+list(APPEND CMAKE_REQUIRED_LIBRARIES "${MALI_LIBRARY}")
+endif()
 list(APPEND CMAKE_REQUIRED_INCLUDES "${EGL_INCLUDE_DIR}")
+list(APPEND CMAKE_REQUIRED_DEFINITIONS "-DEGL_NO_X11")
 
 check_cxx_source_compiles("
 #include <EGL/egl.h>
@@ -161,7 +176,11 @@ endif()
 mark_as_advanced(EGL_LIBRARY EGL_INCLUDE_DIR HAVE_EGL)
 
 # compatibility variables
+if (USE_MALI)
+set(EGL_LIBRARIES ${EGL_LIBRARY} ${MALI_LIBRARY})
+else()
 set(EGL_LIBRARIES ${EGL_LIBRARY})
+endif()
 set(EGL_INCLUDE_DIRS ${EGL_INCLUDE_DIR})
 set(EGL_VERSION_STRING ${EGL_VERSION})
 
diff --git a/dep/glad/CMakeLists.txt b/dep/glad/CMakeLists.txt
index 24a4fb45..59eb66e5 100644
--- a/dep/glad/CMakeLists.txt
+++ b/dep/glad/CMakeLists.txt
@@ -17,8 +17,9 @@ if(WIN32)
   target_sources(glad PRIVATE src/glad_wgl.c)
 else()
   if(USE_EGL)
+    target_compile_definitions(glad PRIVATE -DEGL_NO_X11)
     target_sources(glad PRIVATE src/glad_egl.c)
-    target_link_libraries(glad PRIVATE EGL::EGL)
+    target_link_libraries(glad PRIVATE EGL)
   endif()
   if(USE_X11)
     target_sources(glad PRIVATE src/glad_glx.c)
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index ea086d19..da426ec9 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -169,12 +169,13 @@ if(ENABLE_OPENGL)
       )
     endif()
     if(USE_DRMKMS)
+      target_compile_definitions(common PRIVATE "-DEGL_NO_X11=1")
       target_compile_definitions(common PRIVATE "-DUSE_GBM=1")
       target_sources(common PRIVATE
         gl/context_egl_gbm.cpp
         gl/context_egl_gbm.h
       )
-      target_link_libraries(common PUBLIC GBM::GBM)
+      target_link_libraries(common PUBLIC gbm EGL)
     endif()
     if(USE_FBDEV)
       target_compile_definitions(common PRIVATE "-DUSE_FBDEV=1")
diff --git a/src/duckstation-nogui/CMakeLists.txt b/src/duckstation-nogui/CMakeLists.txt
index 17301849..fbce73e1 100644
--- a/src/duckstation-nogui/CMakeLists.txt
+++ b/src/duckstation-nogui/CMakeLists.txt
@@ -74,4 +74,8 @@ if(USE_DRMKMS AND USE_EVDEV)
   )
   target_include_directories(duckstation-nogui PRIVATE ${LIBEVDEV_INCLUDE_DIRS})
   target_link_libraries(duckstation-nogui PRIVATE ${LIBEVDEV_LIBRARIES})
+if (USE_MALI)
+  target_link_libraries(duckstation-nogui PRIVATE ${MALI_LIBRARY})
+endif()
+
 endif()
