diff --git a/src/common/file_system.cpp b/src/common/file_system.cpp
index 2ed7686b..fcec78f9 100644
--- a/src/common/file_system.cpp
+++ b/src/common/file_system.cpp
@@ -1553,6 +1553,8 @@ bool FileSystem::DeleteDirectory(const char* path)
 
 std::string FileSystem::GetProgramPath()
 {
+  return "/usr/share/duckstation";
+
   std::wstring buffer;
   buffer.resize(MAX_PATH);
 
@@ -2041,6 +2043,8 @@ bool FileSystem::DeleteDirectory(const char* path)
 
 std::string FileSystem::GetProgramPath()
 {
+  return "/usr/share/duckstation";
+
 #if defined(__linux__)
   static const char* exeFileName = "/proc/self/exe";
 
