diff --git a/CMakeModules/DuckStationDependencies.cmake b/CMakeModules/DuckStationDependencies.cmake
index 0d2419c8c..c92df0405 100644
--- a/CMakeModules/DuckStationDependencies.cmake
+++ b/CMakeModules/DuckStationDependencies.cmake
@@ -39,7 +39,7 @@ find_package(Freetype 2.13.3 REQUIRED)
 find_package(harfbuzz REQUIRED)
 find_package(plutosvg 0.0.6 REQUIRED)
 find_package(cpuinfo REQUIRED)
-find_package(DiscordRPC 3.4.0 REQUIRED)
+#find_package(DiscordRPC 3.4.0 REQUIRED)
 find_package(SoundTouch 2.3.3 REQUIRED)
 find_package(libzip 1.11.4 REQUIRED)
 find_package(Shaderc REQUIRED)
diff --git a/src/core/system.cpp b/src/core/system.cpp
index ff0eff710..7f1cbbd33 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -96,7 +96,7 @@ LOG_CHANNEL(System);
 #endif
 
 #ifndef __ANDROID__
-#define ENABLE_DISCORD_PRESENCE 1
+//#define ENABLE_DISCORD_PRESENCE 1
 #define ENABLE_GDB_SERVER 1
 #define ENABLE_SOCKET_MULTIPLEXER 1
 #include "gdb_server.h"
