diff --git a/src/core/system.cpp b/src/core/system.cpp
index 0c59ea075..7e637c3f7 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -5134,12 +5134,12 @@ void System::WarnAboutUnsafeSettings()
     if (g_settings.cpu_execution_mode == CPUExecutionMode::CachedInterpreter)
       append(TRANSLATE_SV("System", "Cached interpreter is being used, this may be incompatible with some games."));
 
-    // Always display TC warning.
+    /* Always display TC warning.
     if (g_settings.gpu_texture_cache)
     {
       append(TRANSLATE_SV(
         "System", "Texture cache is enabled. This feature is experimental, some games may not render correctly."));
-    }
+    }*/
 
     // Potential performance issues.
     if (g_settings.cpu_fastmem_mode != Settings::DEFAULT_CPU_FASTMEM_MODE)
diff --git a/src/duckstation-mini/mini_host.cpp b/src/duckstation-mini/mini_host.cpp
index c68ba6183..69d9f410e 100644
--- a/src/duckstation-mini/mini_host.cpp
+++ b/src/duckstation-mini/mini_host.cpp
@@ -165,7 +165,7 @@ bool MiniHost::EarlyProcessStartup()
     return false;
   }
 
-#if !__has_include("scmversion/tag.h")
+#if __has_include("scmversion/tag.h")
   //
   // To those distributing their own builds or packages of DuckStation, and seeing this message:
   //
@@ -989,7 +989,7 @@ void MiniHost::CoreThreadEntryPoint()
   // start the fullscreen UI and get it going
   if (GPUThread::StartFullscreenUI(s_state.start_fullscreen_ui_fullscreen, &error))
   {
-    WarnAboutInterface();
+    //WarnAboutInterface();
 
     // kick a game list refresh if we're not in batch mode
     if (!s_state.batch_mode)
diff --git a/src/duckstation-qt/qthost.cpp b/src/duckstation-qt/qthost.cpp
index 3c27f6631..0a0498c82 100644
--- a/src/duckstation-qt/qthost.cpp
+++ b/src/duckstation-qt/qthost.cpp
@@ -3438,7 +3438,7 @@ int main(int argc, char* argv[])
   QtHost::UpdateApplicationTheme();
 
   // Build warning.
-  AutoUpdaterDialog::warnAboutUnofficialBuild();
+  //AutoUpdaterDialog::warnAboutUnofficialBuild();
 
   // Start logging early.
   LogWindow::updateSettings();
@@ -3475,7 +3475,7 @@ int main(int argc, char* argv[])
     s_state.start_fullscreen_ui_fullscreen = false;
 
   // Always kick off update check. It'll take over if the user is booting a game fullscreen.
-  g_main_window->startupUpdateCheck();
+  //g_main_window->startupUpdateCheck();
 
   if (autoboot)
     g_core_thread->bootSystem(std::move(autoboot));
