diff --git a/src/core/system.cpp b/src/core/system.cpp
index ff0eff710..0599d2b23 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -4977,12 +4977,12 @@ void System::WarnAboutUnsafeSettings()
     if (g_settings.cpu_execution_mode == CPUExecutionMode::CachedInterpreter)
       append(TRANSLATE_SV("System", "Cached interpreter is being used, this may be incompatible with some games."));
 
-    // Always display TC warning.
+    /* Always display TC warning.
     if (g_settings.gpu_texture_cache)
     {
       append(TRANSLATE_SV(
         "System", "Texture cache is enabled. This feature is experimental, some games may not render correctly."));
-    }
+    }*/
 
     // Potential performance issues.
     if (g_settings.cpu_fastmem_mode != Settings::DEFAULT_CPU_FASTMEM_MODE)
diff --git a/src/duckstation-mini/mini_host.cpp b/src/duckstation-mini/mini_host.cpp
index 122811c4d..b51ded2c9 100644
--- a/src/duckstation-mini/mini_host.cpp
+++ b/src/duckstation-mini/mini_host.cpp
@@ -159,7 +159,7 @@ bool MiniHost::EarlyProcessStartup()
     return false;
   }
 
-#if !__has_include("scmversion/tag.h")
+#if __has_include("scmversion/tag.h")
   //
   // To those distributing their own builds or packages of DuckStation, and seeing this message:
   //
@@ -975,7 +975,7 @@ void MiniHost::CPUThreadEntryPoint()
   // start the fullscreen UI and get it going
   if (GPUThread::StartFullscreenUI(s_state.start_fullscreen_ui_fullscreen, &error))
   {
-    WarnAboutInterface();
+    //WarnAboutInterface();
 
     // kick a game list refresh if we're not in batch mode
     if (!s_state.batch_mode)
diff --git a/src/duckstation-qt/qthost.cpp b/src/duckstation-qt/qthost.cpp
index 993cd9ade..6970ec101 100644
--- a/src/duckstation-qt/qthost.cpp
+++ b/src/duckstation-qt/qthost.cpp
@@ -3344,7 +3344,7 @@ int main(int argc, char* argv[])
   QtHost::UpdateApplicationTheme();
 
   // Build warning.
-  AutoUpdaterDialog::warnAboutUnofficialBuild();
+  //AutoUpdaterDialog::warnAboutUnofficialBuild();
 
   // Start logging early.
   LogWindow::updateSettings();
@@ -3383,7 +3383,7 @@ int main(int argc, char* argv[])
     s_state.start_fullscreen_ui_fullscreen = false;
 
   // Always kick off update check. It'll take over if the user is booting a game fullscreen.
-  g_main_window->startupUpdateCheck();
+  //g_main_window->startupUpdateCheck();
 
   if (autoboot)
     g_emu_thread->bootSystem(std::move(autoboot));
