diff --git a/src/core/system.cpp b/src/core/system.cpp
index 99702abe4..5bbc098bb 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -4904,14 +4904,14 @@ void System::WarnAboutUnsafeSettings()
              TRANSLATE_SV("System", "Cached interpreter is being used, this may be incompatible with some games."));
     }
 
-    // Always display TC warning.
+    /* Always display TC warning.
     if (g_settings.gpu_texture_cache)
     {
       append(
         ICON_FA_PAINT_ROLLER,
         TRANSLATE_SV("System",
                      "Texture cache is enabled. This feature is experimental, some games may not render correctly."));
-    }
+    }*/
 
     // Potential performance issues.
     if (g_settings.cpu_fastmem_mode != Settings::DEFAULT_CPU_FASTMEM_MODE)
diff --git a/src/duckstation-mini/mini_host.cpp b/src/duckstation-mini/mini_host.cpp
index 400081188..e5ffcc46d 100644
--- a/src/duckstation-mini/mini_host.cpp
+++ b/src/duckstation-mini/mini_host.cpp
@@ -158,7 +158,7 @@ bool MiniHost::EarlyProcessStartup()
     return false;
   }
 
-#if !__has_include("scmversion/tag.h")
+#if __has_include("scmversion/tag.h")
   //
   // To those distributing their own builds or packages of DuckStation, and seeing this message:
   //
@@ -1002,7 +1002,7 @@ void MiniHost::CPUThreadEntryPoint()
   // start the fullscreen UI and get it going
   if (GPUThread::StartFullscreenUI(s_state.start_fullscreen_ui_fullscreen, &error))
   {
-    WarnAboutInterface();
+    //WarnAboutInterface();
 
     // kick a game list refresh if we're not in batch mode
     if (!s_state.batch_mode)
diff --git a/src/duckstation-qt/qthost.cpp b/src/duckstation-qt/qthost.cpp
index 65f0bb964..dcfaf232b 100644
--- a/src/duckstation-qt/qthost.cpp
+++ b/src/duckstation-qt/qthost.cpp
@@ -3353,7 +3353,7 @@ int main(int argc, char* argv[])
   QtHost::UpdateApplicationTheme();
 
   // Build warning.
-  AutoUpdaterWindow::warnAboutUnofficialBuild();
+  // AutoUpdaterWindow::warnAboutUnofficialBuild();
 
   // Start logging early.
   LogWindow::updateSettings();
