diff --git a/dep/CMakeLists.txt b/dep/CMakeLists.txt
index 633267c66..df9087917 100644
--- a/dep/CMakeLists.txt
+++ b/dep/CMakeLists.txt
@@ -2,8 +2,8 @@ set(_saved_CMAKE_MESSAGE_LOG_LEVEL ${CMAKE_MESSAGE_LOG_LEVEL})
 set(CMAKE_MESSAGE_LOG_LEVEL NOTICE)
 
 set(FMT_INSTALL OFF CACHE BOOL "")
-add_subdirectory(fmt EXCLUDE_FROM_ALL)
-disable_compiler_warnings_for_target(fmt)
+#add_subdirectory(fmt EXCLUDE_FROM_ALL)
+#disable_compiler_warnings_for_target(fmt)
 add_subdirectory(minizip EXCLUDE_FROM_ALL)
 disable_compiler_warnings_for_target(minizip)
 add_subdirectory(lzma EXCLUDE_FROM_ALL)
