diff --git a/src/core/system.cpp b/src/core/system.cpp
index 09cb8c1eb..ad5bd837b 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -5198,12 +5198,12 @@ void System::WarnAboutUnsafeSettings()
     if (g_settings.cpu_execution_mode == CPUExecutionMode::CachedInterpreter)
       append(TRANSLATE_SV("System", "Cached interpreter is being used, this may be incompatible with some games."));
 
-    // Always display TC warning.
+    /* Always display TC warning.
     if (g_settings.gpu_texture_cache)
     {
       append(TRANSLATE_SV(
         "System", "Texture cache is enabled. This feature is experimental, some games may not render correctly."));
-    }
+    }*/
 
     // Potential performance issues.
     if (g_settings.cpu_fastmem_mode != Settings::DEFAULT_CPU_FASTMEM_MODE)
diff --git a/src/duckstation-mini/mini_host.cpp b/src/duckstation-mini/mini_host.cpp
index 1f1c4d8a7..e8f7f6a70 100644
--- a/src/duckstation-mini/mini_host.cpp
+++ b/src/duckstation-mini/mini_host.cpp
@@ -179,7 +179,7 @@ bool MiniHost::EarlyProcessStartup()
   }
 #endif
 
-#if !__has_include("scmversion/tag.h")
+#if __has_include("scmversion/tag.h")
   //
   // To those distributing their own builds or packages of DuckStation, and seeing this message:
   //
@@ -965,7 +965,7 @@ void MiniHost::CoreThreadEntryPoint()
   // start the fullscreen UI and get it going
   if (VideoThread::StartFullscreenUI(s_state.start_fullscreen_ui_fullscreen, &error))
   {
-    WarnAboutInterface();
+    //WarnAboutInterface();
 
     // kick a game list refresh if we're not in batch mode
     if (!s_state.batch_mode)
diff --git a/src/duckstation-qt/qthost.cpp b/src/duckstation-qt/qthost.cpp
index ea1953acd..dfe463737 100644
--- a/src/duckstation-qt/qthost.cpp
+++ b/src/duckstation-qt/qthost.cpp
@@ -3637,7 +3637,7 @@ int main(int argc, char* argv[])
   QtHost::UpdateApplicationTheme();
 
   // Build warning.
-  AutoUpdaterDialog::warnAboutUnofficialBuild();
+  //AutoUpdaterDialog::warnAboutUnofficialBuild();
 
   // Create core thread object, but don't start it yet. That way the main window can connect to it,
   // and ensures that no signals are lost. Then we create and connect the main window.
@@ -3694,7 +3694,7 @@ int main(int argc, char* argv[])
     QtHost::s_state.start_fullscreen_ui_fullscreen = false;
 
   // Always kick off update check. It'll take over if the user is booting a game fullscreen.
-  g_main_window->startupUpdateCheck();
+  //g_main_window->startupUpdateCheck();
 
   if (autoboot)
     g_core_thread->bootSystem(std::move(autoboot));
