diff --git a/src/util/imgui_manager.cpp b/src/util/imgui_manager.cpp
index f4311732b..6ec181442 100644
--- a/src/util/imgui_manager.cpp
+++ b/src/util/imgui_manager.cpp
@@ -192,7 +192,7 @@ static constexpr const std::array FIXED_FONT_FILENAMES = {
 #undef ENUMERATE_FIXED_FONT_NAMES
 
 static constexpr const char* DEFAULT_FIXED_FONT_NAME = FIXED_FONT_NAMES[0];
-static constexpr const char* FA_FONT_NAME = FONT_ICON_FILE_NAME_FAS;
+static constexpr const char* FA_FONT_NAME = "fa-solid-900.ttf";
 static constexpr const char* PF_FONT_NAME = "promptfont.otf";
 static constexpr const char* EMOJI_FONT_NAME = "TwitterColorEmoji-SVGinOT.ttf";
 
