diff --git a/CMakeModules/DuckStationDependencies.cmake b/CMakeModules/DuckStationDependencies.cmake
index ee4213ae1..f8f5bcc8d 100644
--- a/CMakeModules/DuckStationDependencies.cmake
+++ b/CMakeModules/DuckStationDependencies.cmake
@@ -84,7 +84,7 @@ if(BUILD_QT_FRONTEND)
 
   # Have to verify it down here, don't want users using unpatched Qt.
   if(NOT Qt6_DIR MATCHES "^${CMAKE_PREFIX_PATH}")
-    message(FATAL_ERROR "Using incorrect Qt library. Check your dependencies.")
+    #message(FATAL_ERROR "Using incorrect Qt library. Check your dependencies.")
   endif()
 endif()
 
