diff --git a/dep/reshadefx/include/effect_module.hpp b/dep/reshadefx/include/effect_module.hpp
index ccf2cbe2..4af2d23c 100644
--- a/dep/reshadefx/include/effect_module.hpp
+++ b/dep/reshadefx/include/effect_module.hpp
@@ -7,6 +7,7 @@
 
 #include "effect_expression.hpp"
 #include <unordered_set>
+#include <cstdint>
 
 namespace reshadefx
 {
diff --git a/dep/reshadefx/include/effect_token.hpp b/dep/reshadefx/include/effect_token.hpp
index 072d4398..2753c77f 100644
--- a/dep/reshadefx/include/effect_token.hpp
+++ b/dep/reshadefx/include/effect_token.hpp
@@ -7,6 +7,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace reshadefx
 {
diff --git a/dep/reshadefx/src/effect_symbol_table.cpp b/dep/reshadefx/src/effect_symbol_table.cpp
index 8ba800f5..ce7a74fe 100644
--- a/dep/reshadefx/src/effect_symbol_table.cpp
+++ b/dep/reshadefx/src/effect_symbol_table.cpp
@@ -12,6 +12,7 @@
 #endif
 #include <algorithm> // std::upper_bound, std::sort
 #include <functional> // std::greater
+#include <cstdint>
 
 enum class intrinsic_id : uint32_t
 {
