diff --git a/CMakeModules/DuckStationDependencies.cmake b/CMakeModules/DuckStationDependencies.cmake
index 79713d5b..c6662dcb 100644
--- a/CMakeModules/DuckStationDependencies.cmake
+++ b/CMakeModules/DuckStationDependencies.cmake
@@ -17,7 +17,7 @@ find_package(PNG 1.6.40 REQUIRED)
 find_package(JPEG REQUIRED) # No version because flatpak uses libjpeg-turbo.
 find_package(Freetype 2.11.1 REQUIRED)
 find_package(cpuinfo REQUIRED)
-find_package(DiscordRPC 3.4.0 REQUIRED)
+#find_package(DiscordRPC 3.4.0 REQUIRED)
 find_package(SoundTouch 2.3.3 REQUIRED)
 
 if(NOT WIN32)
diff --git a/src/core/system.cpp b/src/core/system.cpp
index a6e1f14c..13aec426 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -83,7 +83,7 @@ Log_SetChannel(System);
 #endif
 
 #ifndef __ANDROID__
-#define ENABLE_DISCORD_PRESENCE 1
+//#define ENABLE_DISCORD_PRESENCE 1
 #define ENABLE_PINE_SERVER 1
 #define ENABLE_GDB_SERVER 1
 #define ENABLE_SOCKET_MULTIPLEXER 1
