diff --git a/src/core/system.cpp b/src/core/system.cpp
index d7bda692..64e79f57 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -3729,13 +3729,13 @@ void System::CheckForSettingsChanges(const Settings& old_settings)
 
   FullscreenUI::CheckForConfigChanges(old_settings);
 
-  if (g_settings.enable_discord_presence != old_settings.enable_discord_presence)
+  /*if (g_settings.enable_discord_presence != old_settings.enable_discord_presence)
   {
     if (g_settings.enable_discord_presence)
       InitializeDiscordPresence();
     else
       ShutdownDiscordPresence();
-  }
+  }*/
 
   if (g_settings.log_level != old_settings.log_level || g_settings.log_filter != old_settings.log_filter ||
       g_settings.log_to_console != old_settings.log_to_console ||
diff --git a/src/duckstation-nogui/wayland_nogui_platform.cpp b/src/duckstation-nogui/wayland_nogui_platform.cpp
index 52bf9672..b276dafb 100644
--- a/src/duckstation-nogui/wayland_nogui_platform.cpp
+++ b/src/duckstation-nogui/wayland_nogui_platform.cpp
@@ -7,7 +7,7 @@
 #include "common/string_util.h"
 #include "common/threading.h"
 #include "core/host.h"
-#include "core/host_settings.h"
+#include "core/settings.h"
 #include "nogui_host.h"
 #include "nogui_platform.h"
 
diff --git a/src/duckstation-nogui/x11_nogui_platform.h b/src/duckstation-nogui/x11_nogui_platform.h
index 395353e7..73ca495d 100644
--- a/src/duckstation-nogui/x11_nogui_platform.h
+++ b/src/duckstation-nogui/x11_nogui_platform.h
@@ -12,7 +12,7 @@
 #include "common/string_util.h"
 #include "common/threading.h"
 #include "core/host.h"
-#include "core/host_settings.h"
+#include "core/settings.h"
 #include "nogui_host.h"
 #include "nogui_platform.h"
 
