diff --git a/src/duckstation-nogui/nogui_host.cpp b/src/duckstation-nogui/nogui_host.cpp
index 2d9b7b6b..126ac4de 100644
--- a/src/duckstation-nogui/nogui_host.cpp
+++ b/src/duckstation-nogui/nogui_host.cpp
@@ -724,7 +724,7 @@ void NoGUIHost::CPUThreadMainLoop()
     Host::PumpMessagesOnCPUThread();
     System::Internal::IdlePollUpdate();
     System::PresentDisplay(false);
-    if (!g_gpu_device->IsVsyncEnabled())
+    if (!g_gpu_device->IsVSyncActive())
       g_gpu_device->ThrottlePresentation();
   }
 }
