diff --git a/backends/platform/libretro/Makefile b/backends/platform/libretro/Makefile
index 301ad0f791b..2b5b8f7be5b 100644
--- a/backends/platform/libretro/Makefile
+++ b/backends/platform/libretro/Makefile
@@ -88,6 +88,16 @@ else ifeq ($(platform), rpi4_64)
    HAVE_OPENGLES2 := 1
    HAVE_NEON := 1
 
+# webOS (32 bit)
+else ifeq ($(platform), webos)
+   TARGET = $(TARGET_NAME)_libretro.so
+   DEFINES += -fPIC -D_ARM_ASSEM_ -DUSE_CXX11 -marm -DARM
+   LDFLAGS += -shared -Wl,--version-script=$(ROOT_PATH)/link.T -fPIC -static-libgcc -static-libstdc++
+   CFLAGS += -fPIC -mcpu=cortex-a9 -mtune=cortex-a53 -mfloat-abi=softfp -fomit-frame-pointer -ffast-math
+   CXXFLAGS = $(CFLAGS) -frtti -std=c++11
+   HAVE_OPENGLES2 := 1
+   HAVE_NEON := 1
+
 # iOS
 else ifneq (,$(findstring ios,$(platform)))
    TARGET  := $(TARGET_NAME)_libretro_ios.dylib
@@ -169,6 +179,8 @@ else ifneq (,$(filter $(platform), ps3 psl1ght))
    DEFINES := -DPLAYSTATION3 -D__PS3__ -DRETRO_IS_BIG_ENDIAN=1 -DRETRO_IS_LITTLE_ENDIAN=0 -DWORDS_BIGENDIAN=1 -mcpu=cell -mno-fp-in-toc -I$(PS3DEV)/ppu/include
    ifeq ($(platform), psl1ght)
       DEFINES += -D__PSL1GHT__
+      CFLAGS   += -mvsx -mvsx-timode
+      CXXFLAGS += -mvsx -mvsx-timode
       HAVE_OPENGL := 1
    endif
    STATIC_LINKING = 1
@@ -194,7 +206,7 @@ else ifeq ($(platform), libnx)
     DEFINES   += -g -O3 -fPIE -I$(LIBNX)/include/ -ffunction-sections -fdata-sections -ftls-model=local-exec -ffast-math -mcpu=cortex-a57+crc+fp+simd -march=armv8-a -mtune=cortex-a57 -mtp=soft
     DEFINES += $(INCDIRS)
     DEFINES += -D__SWITCH__ -DHAVE_LIBNX -D__arm64__ -D__ARM_NEON__
-    CXXFLAGS := $(ASFLAGS) -std=gnu++11 -fpermissive
+    CXXFLAGS += $(ASFLAGS) -std=gnu++11 -fpermissive
     STATIC_LINKING = 1
     HAVE_OPENGL := 1
     HAVE_NEON := 1
@@ -375,7 +387,6 @@ else ifeq ($(platform), emscripten)
    CXXFLAGS += -std=c++11
    STATIC_LINKING = 1
    USE_LIBCO  = 0
-   USE_CLOUD = 0
 
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
@@ -605,6 +616,8 @@ CFLAGS += $(DEFINES) $(INCLUDES)
 # Include the build instructions for all modules
 include $(addprefix $(SCUMMVM_PATH)/, $(addsuffix /module.mk,$(MODULES)))
 
+include $(ROOT_PATH)/overrides.mk
+
 # Depdir information
 DEPDIRS := $(addsuffix $(DEPDIR),$(MODULE_PATHS))
 
diff --git a/backends/platform/libretro/Makefile.common b/backends/platform/libretro/Makefile.common
index df89acc079c..7f533ae2748 100644
--- a/backends/platform/libretro/Makefile.common
+++ b/backends/platform/libretro/Makefile.common
@@ -105,7 +105,7 @@ USE_JPEG          = 1
 
 USE_FREETYPE2     = 1
 
-USE_CLOUD        ?= 0
+USE_CLOUD        ?=
 
 # Components (defines handled by engines.awk)
 USE_THEORADEC    ?= 1
@@ -136,7 +136,7 @@ else
 DEFINES += -DUSE_IMGUI
 endif
 
-USE_MPEG2         ?= 0
+USE_MPEG2         ?= 1
 UNAVAILABLE_DEPS += $(call check_deps_availability, $(USE_MPEG2), component_mpeg2)
 ifneq ($(USE_MPEG2),1)
 USE_MPEG2=
@@ -144,7 +144,7 @@ else
 DEFINES += -DUSE_MPEG2
 endif
 
-USE_GIF         ?= 0
+USE_GIF         ?= 1
 UNAVAILABLE_DEPS += $(call check_deps_availability, $(USE_GIF), component_gif)
 ifneq ($(USE_GIF),1)
 USE_GIF=
diff --git a/backends/platform/libretro/dependencies.mk b/backends/platform/libretro/dependencies.mk
index 616af1bbff6..715f8f333f8 100644
--- a/backends/platform/libretro/dependencies.mk
+++ b/backends/platform/libretro/dependencies.mk
@@ -7,7 +7,7 @@ DEPS_SUBMODULES             := libretro-deps libretro-common
 
 DEPS_FOLDER_libretro-deps   := libretro-deps
 DEPS_URL_libretro-deps      := https://github.com/libretro/libretro-deps
-DEPS_COMMIT_libretro-deps   := abf5246b016569759e7d1b0ea91bb98c2e34d160
+DEPS_COMMIT_libretro-deps   := 7e6e34f0319f4c7448d72f0e949e76265ccf55a1
 
 DEPS_FOLDER_libretro-common := libretro-common
 DEPS_URL_libretro-common    := https://github.com/libretro/libretro-common
@@ -580,6 +580,81 @@ OBJS_DEPS += $(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/fribidi/fribidi-arabic.o
 endif
 endif
 
+######################################################################
+# libmpeg2 settings
+######################################################################
+
+ifeq ($(USE_MPEG2), 1)
+DEFINES += -DUSE_MPEG2
+this_lib_subpath :=
+this_lib_header  := mpeg2dec/mpeg2.h
+this_lib_flags   := -lmpeg2
+include $(ROOT_PATH)/sharedlib_test.mk
+ifneq ($(this_lib_available), yes)
+INCLUDES += \
+	-I$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/include \
+	-I$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/include/mpeg2dec \
+	-I$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2
+OBJS_DEPS += \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/alloc.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/header.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/decode.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/slice.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/motion_comp.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/idct.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/cpu_accel.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/cpu_state.o
+
+# --- Optional accelerations --------------
+# x86/x64 -> MMX
+ifneq ($(findstring x86,$(platform))$(findstring x64,$(platform)),)
+OBJS_DEPS += \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/motion_comp_mmx.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/idct_mmx.o
+endif
+
+# PowerPC (PS3/PSL1GHT) -> AltiVec/VMX
+ifneq ($(findstring ps3,$(platform))$(findstring psl1ght,$(platform)),)
+OBJS_DEPS += \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/motion_comp_altivec.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/idct_altivec.o
+endif
+
+# ARM families (rpi*, libnx, ios/tvos, vita, ctr, miyoo/miyoomini)
+ifneq ($(findstring rpi,$(platform))$(findstring armv7,$(platform))$(findstring armv8,$(platform))$(findstring libnx,$(platform))$(findstring ios,$(platform))$(findstring tvos,$(platform))$(findstring vita,$(platform))$(findstring ctr,$(platform))$(findstring miyoo,$(platform))$(findstring miyoomini,$(platform)),)
+OBJS_DEPS += \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/libmpeg2/libmpeg2/motion_comp_arm.o
+endif
+endif
+endif
+
+######################################################################
+# giflib settings
+######################################################################
+
+ifeq ($(USE_GIF), 1)
+DEFINES += -DUSE_GIF
+this_lib_subpath :=
+this_lib_header  := gif_lib.h
+this_lib_flags   := -lgif
+include $(ROOT_PATH)/sharedlib_test.mk
+ifneq ($(this_lib_available), yes)
+INCLUDES += \
+  -I$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib
+OBJS_DEPS += \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/dgif_lib.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/egif_lib.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/gifalloc.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/gif_err.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/gif_hash.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/quantize.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/gif_font.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/openbsd-reallocarray.o \
+	$(DEPS_PATH)/$(DEPS_FOLDER_libretro-deps)/giflib/qprintf.o
+
+endif
+endif
+
 ######################################################################
 # libcurl settings
 ######################################################################
diff --git a/backends/platform/libretro/include/libretro-core-options-intl.h b/backends/platform/libretro/include/libretro-core-options-intl.h
index 7564752a3fd..b5ef01b7388 100644
--- a/backends/platform/libretro/include/libretro-core-options-intl.h
+++ b/backends/platform/libretro/include/libretro-core-options-intl.h
@@ -487,7 +487,7 @@ struct retro_core_option_v2_definition option_defs_it[] = {
 	},
 	{
 		"scummvm_mapper_rr",
-		"RetroPad > Leva Analogica Destra > Leva Analogica Destra",
+		"RetroPad > Leva Analogica Destra > Destra",
 		"Leva Analogica Destra > Destra",
 		NULL,
 		NULL,
diff --git a/backends/platform/libretro/overrides.mk b/backends/platform/libretro/overrides.mk
new file mode 100644
index 00000000000..7abe5fc376f
--- /dev/null
+++ b/backends/platform/libretro/overrides.mk
@@ -0,0 +1,20 @@
+# Specific platform overrides, to be dropped asap...
+
+# -DEMSCRIPTEN flag should not be needed for emscripten build of libretro core, as it is a specific Project flag
+# needed for emscripten backend. Anyways in several places, both in ScummVM and libretro-common,
+# it is used as platform flag instead of __EMSCRIPTEN__, hence we need to keep EMSCRIPTEN flag for now, and apply
+# the following workaround to solve relevant conflicts of the libretro core with emscripten backend at build time.
+ifeq ($(platform), emscripten)
+backends/fs/emscripten/%.o:
+	@:
+backends/libbackends.a:
+	@:
+MODULE_OBJS-backends := $(filter-out backends/fs/emscripten/%,$(MODULE_OBJS-backends))
+backends/libbackends.a: $(MODULE_OBJS-backends)
+	$(AR) $@ $(MODULE_OBJS-backends)
+	$(RANLIB) $@
+
+$(SCUMMVM_PATH)/gui/textviewer.o: CXXFLAGS += -UEMSCRIPTEN
+$(SCUMMVM_PATH)/gui/textviewer.o: CFLAGS   += -UEMSCRIPTEN
+$(SCUMMVM_PATH)/gui/textviewer.o: CPPFLAGS += -UEMSCRIPTEN
+endif
diff --git a/backends/platform/libretro/scripts/configure_engines.sh b/backends/platform/libretro/scripts/configure_engines.sh
index b04e2520c31..0f8574bca71 100755
--- a/backends/platform/libretro/scripts/configure_engines.sh
+++ b/backends/platform/libretro/scripts/configure_engines.sh
@@ -108,7 +108,7 @@ for comp in $(get_var _components); do
 done
 
 # Create needed engines build files
-awk -f "engines.awk" < /dev/null > /dev/null 2>&1
+awk -f "./engines.awk" < /dev/null > /dev/null 2>&1
 
 mkdir -p "engines"
 
diff --git a/backends/platform/libretro/src/libretro-core.cpp b/backends/platform/libretro/src/libretro-core.cpp
index 5b11d73fa04..0371b91d82f 100644
--- a/backends/platform/libretro/src/libretro-core.cpp
+++ b/backends/platform/libretro/src/libretro-core.cpp
@@ -613,17 +613,25 @@ uint16 retro_setting_get_sample_rate(void) {
 	return sample_rate;
 }
 
+
+static uint32 next_pow2(uint32 x) {
+	if (x <= 1) return 1;
+	x--;
+	x |= x >> 1; x |= x >> 2; x |= x >> 4;
+	x |= x >> 8; x |= x >> 16;
+	return x + 1;
+}
+
 uint16 retro_setting_get_audio_samples_buffer_size(void) {
 	/* ScummVM audio buffer size is normally between 512 and 8192, but the value
 	must be one of: 256, 512, 1024, 2048, 4096, 8192, 16384, or 32768. */
-	uint16 v = audio_samples_per_frame--;
-	v |= v >> 1;
-	v |= v >> 2;
-	v |= v >> 4;
-	v |= v >> 8;
-	v |= v >> 16;
-
-	return ++v;
+	static const uint16 allowed[] = {256,512,1024,2048,4096,8192,16384,32768};
+	uint32 target = (uint32)(audio_samples_per_frame * 2.0f + 0.5f); // stereo
+	uint32 pow2   = next_pow2(target);
+	for (uint16 v : allowed) {
+		if (pow2 <= v) return v;
+	}
+	return allowed[sizeof(allowed)/sizeof(allowed[0])];
 }
 
 void init_command_params(void) {
diff --git a/backends/platform/libretro/src/libretro-fs-factory.cpp b/backends/platform/libretro/src/libretro-fs-factory.cpp
index d123a03dcfc..02ee0e7a80c 100644
--- a/backends/platform/libretro/src/libretro-fs-factory.cpp
+++ b/backends/platform/libretro/src/libretro-fs-factory.cpp
@@ -39,8 +39,11 @@ AbstractFSNode *LibRetroFilesystemFactory::makeCurrentDirectoryFileNode() const
 #ifdef PLAYSTATION3
 	return new LibRetroFilesystemNode("/");
 #else
-	char buf[MAXPATHLEN];
-	return getcwd(buf, MAXPATHLEN) ? new LibRetroFilesystemNode(buf) : NULL;
+	char *cwd = getcwd(NULL, 0);
+	AbstractFSNode *node = cwd ? new LibRetroFilesystemNode(Common::String(cwd)) : NULL;
+	if (cwd)
+		free(cwd);
+	return node;
 #endif
 }
 
diff --git a/backends/platform/libretro/src/libretro-fs.cpp b/backends/platform/libretro/src/libretro-fs.cpp
index 8518e847c32..1934c21995f 100644
--- a/backends/platform/libretro/src/libretro-fs.cpp
+++ b/backends/platform/libretro/src/libretro-fs.cpp
@@ -242,16 +242,22 @@ bool assureDirectoryExists(const Common::String &dir, const char *prefix) {
 
 Common::String LibRetroFilesystemNode::getHomeDir(void) {
 	Common::String path;
-#if defined(__WIN32)
-	const char *c_homeDriveDir = getenv("HOMEDRIVE");
-	const char *c_homePathDir = getenv("HOMEPATH");
-	char c_homeDir[strlen(c_homeDriveDir) + strlen(c_homePathDir) + 1] = {0};
-	strcat(strcat(c_homeDir, c_homeDriveDir), c_homePathDir);
+	const char *home = nullptr;
+
+#ifdef _WIN32
+	const char *drv = getenv("HOMEDRIVE");
+	const char *pth = getenv("HOMEPATH");
+	if (drv && *drv && pth && *pth) {
+		Common::String s = Common::String(drv);
+		s += pth;
+		return s;
+	}
 #else
-	const char *c_homeDir = getenv("HOME");
+	home = getenv("HOME");
 #endif
-	if (c_homeDir && *c_homeDir)
-		path = c_homeDir;
+
+	if (home && *home)
+		path = home;
 
 	return path;
 }
diff --git a/backends/platform/libretro/src/libretro-options-widget.cpp b/backends/platform/libretro/src/libretro-options-widget.cpp
index 39ad3905c03..bb9880c406b 100644
--- a/backends/platform/libretro/src/libretro-options-widget.cpp
+++ b/backends/platform/libretro/src/libretro-options-widget.cpp
@@ -21,8 +21,8 @@
 #include "backends/platform/libretro/include/libretro-fs.h"
 #include "backends/platform/libretro/include/libretro-os.h"
 #include "gui/launcher.h"
-
 #include "common/translation.h"
+#include "common/formats/json.h"
 
 enum {
 	kPlaylistPathCmd = 'pchp',
@@ -142,11 +142,14 @@ bool LibretroOptionsWidget::generatePlaylist(Common::String playlistPath) {
 	Common::String hookFilePath;
 	Common::String title;
 
+	Common::JSONObject root;
+	Common::JSONArray  items;
+
 	/* Create playlist file */
 	RFILE *playlistFile = filestream_open(Common::String(playlistPath + "/" + CORE_NAME + ".lpl").c_str(), RETRO_VFS_FILE_ACCESS_WRITE, RETRO_VFS_FILE_ACCESS_HINT_NONE);
 	if (!playlistFile) {
 		_playlistStatus->setLabel(_("Failed, can't access playlist file"));
-		retro_log_cb(RETRO_LOG_ERROR, "Failed to access playlst file at '%s'.\n", Common::String(playlistPath + CORE_NAME + ".lpl").c_str());
+		retro_log_cb(RETRO_LOG_ERROR, "Failed to access playlst file at '%s'.\n", Common::String(playlistPath +  "/" + CORE_NAME + ".lpl").c_str());
 		return false;
 	} else
 		retro_log_cb(RETRO_LOG_INFO, "Playlist file created in '%s'.\n", Common::String(playlistPath + CORE_NAME + ".lpl").c_str());
@@ -166,9 +169,14 @@ bool LibretroOptionsWidget::generatePlaylist(Common::String playlistPath) {
 		LibRetroFilesystemNode corePath(".");
 		if (cCorePath)
 			corePath = LibRetroFilesystemNode(Common::String(cCorePath));
-		filestream_printf(playlistFile, "{\n  \"version\": \"1.5\",\n  \"default_core_path\": \"%s\",\n  \"default_core_name\": \"ScummVM\",\n  \"label_display_mode\": 0,\n  \"right_thumbnail_mode\": 2,\n  \"left_thumbnail_mode\": 3,\n  \"sort_mode\": 0,\n  \"items\": [", corePath.exists() ? corePath.getPath().c_str() : "");
-		playlistElement = "%s\n    {\n      \"path\": \"%s\",\n      \"label\": \"%s\",\n      \"core_path\": \"DETECT\",\n      \"core_name\": \"DETECT\",\n      \"crc32\": \"DETECT\",\n      \"db_name\": \"" CORE_NAME ".lpl\"\n    }";
-		playlistFooter = "\n  ]\n}";
+
+		root["version"]			= new Common::JSONValue(Common::String("1.5"));
+		root["default_core_path"]  = new Common::JSONValue(corePath.exists() ? corePath.getPath() : Common::String(""));
+		root["default_core_name"]  = new Common::JSONValue(Common::String("ScummVM"));
+		root["label_display_mode"] = new Common::JSONValue((long long)0);
+		root["right_thumbnail_mode"] = new Common::JSONValue((long long)2);
+		root["left_thumbnail_mode"]  = new Common::JSONValue((long long)3);
+		root["sort_mode"]			= new Common::JSONValue((long long)0);
 	} else
 		playlistElement = "%s%s\n%s\nDETECT\nDETECT\nDETECT\nScummVM.lpl\n";
 
@@ -193,11 +201,22 @@ bool LibretroOptionsWidget::generatePlaylist(Common::String playlistPath) {
 		iter->_value.tryGetVal("description", title);
 		hookFilePath = hookPath + + "/" + iter->_key.c_str() + "." + CORE_EXTENSIONS;
 
-		filestream_printf(playlistFile, playlistElement.c_str(), separator, hookFilePath.c_str(), title.c_str());
-
-		if (isFirstEntry && ConfMan.getInt("libretro_playlist_version", _domain) != kPlaylistFormat6lines) {
-			*separator = ',';
-			isFirstEntry = false;
+		if (ConfMan.getInt("libretro_playlist_version", _domain) != kPlaylistFormat6lines) {
+			Common::JSONObject item;
+			item["path"]	  = new Common::JSONValue(hookFilePath);
+			item["label"]	 = new Common::JSONValue(title);
+			item["core_path"] = new Common::JSONValue(Common::String("DETECT"));
+			item["core_name"] = new Common::JSONValue(Common::String("DETECT"));
+			item["crc32"]	 = new Common::JSONValue(Common::String("DETECT"));
+			item["db_name"]   = new Common::JSONValue(Common::String(CORE_NAME ".lpl"));
+
+			items.push_back(new Common::JSONValue(item));
+		} else {
+			filestream_printf(playlistFile, playlistElement.c_str(), separator, hookFilePath.c_str(), title.c_str());
+			if (isFirstEntry) {
+				*separator = ',';
+				isFirstEntry = false;
+			}
 		}
 
 		/* Create hook file */
@@ -214,7 +233,15 @@ bool LibretroOptionsWidget::generatePlaylist(Common::String playlistPath) {
 		filestream_close(hookFile);
 	}
 
-	filestream_printf(playlistFile, playlistFooter.c_str());
+	if (ConfMan.getInt("libretro_playlist_version", _domain) != kPlaylistFormat6lines) {
+		root["items"] = new Common::JSONValue(items);
+		Common::JSONValue *rootVal = new Common::JSONValue(root);
+		Common::String out = Common::JSON::stringify(rootVal);
+		filestream_write(playlistFile, out.c_str(), out.size());
+		delete rootVal;
+	} else {
+		filestream_printf(playlistFile, playlistFooter.c_str());
+	}
 	filestream_close(playlistFile);
 
 	Common::String response;
diff --git a/backends/platform/libretro/src/libretro-os-inputs.cpp b/backends/platform/libretro/src/libretro-os-inputs.cpp
index 745ace09d01..c4815db8bf9 100644
--- a/backends/platform/libretro/src/libretro-os-inputs.cpp
+++ b/backends/platform/libretro/src/libretro-os-inputs.cpp
@@ -55,7 +55,7 @@ void OSystem_libretro::updateMouseXY(float deltaAcc, float *cumulativeXYAcc, int
 		// Set mouse position
 		*mouseXY += cumulativeXYAcc_int;
 		*mouseXY = (*mouseXY < 0) ? 0 : *mouseXY;
-		*mouseXY = (*mouseXY >= screen_wh) ? screen_wh : *mouseXY;
+		*mouseXY = (*mouseXY >= screen_wh) ? (screen_wh ? screen_wh - 1 : 0) : *mouseXY;
 		// Update accumulator
 		*cumulativeXYAcc -= (float)cumulativeXYAcc_int;
 	}
diff --git a/backends/platform/libretro/src/libretro-os-utils.cpp b/backends/platform/libretro/src/libretro-os-utils.cpp
index e4d7c4ed855..ae86dde4f07 100644
--- a/backends/platform/libretro/src/libretro-os-utils.cpp
+++ b/backends/platform/libretro/src/libretro-os-utils.cpp
@@ -280,4 +280,7 @@ Common::String OSystem_libretro::getSaveDir(void) {
 void OSystem_libretro::addSysArchivesToSearchSet(Common::SearchSet &s, int priority) {
 	if (!s_systemDir.empty())
 		s.add("systemDir", new Common::FSDirectory(Common::FSNode(Common::Path(s_systemDir))), priority);
+	// Add the current dir as a very last resort (cf. bug #3984).
+	// TODO: check if it's really needed
+	s.addDirectory(".", ".", priority - 1);
 }
