diff --git a/SamTFE/Sources/SeriousSam/SeriousSam.cpp b/SamTFE/Sources/SeriousSam/SeriousSam.cpp
index 9437bc4..80414e6 100644
--- a/SamTFE/Sources/SeriousSam/SeriousSam.cpp
+++ b/SamTFE/Sources/SeriousSam/SeriousSam.cpp
@@ -102,7 +102,7 @@ __extern INDEX sam_bWideScreen = FALSE;
 __extern FLOAT sam_fPlayerOffset = 0.0f;
 
 // display mode settings
-__extern INDEX sam_bFullScreenActive = FALSE;
+__extern INDEX sam_bFullScreenActive = TRUE;
 __extern INDEX sam_bBorderLessActive = FALSE;
 __extern INDEX sam_iScreenSizeI = 1024;  // current size of the window
 __extern INDEX sam_iScreenSizeJ = 768;  // current size of the window
diff --git a/SamTSE/Sources/SeriousSam/SeriousSam.cpp b/SamTSE/Sources/SeriousSam/SeriousSam.cpp
index 9437bc4..80414e6 100644
--- a/SamTSE/Sources/SeriousSam/SeriousSam.cpp
+++ b/SamTSE/Sources/SeriousSam/SeriousSam.cpp
@@ -102,7 +102,7 @@ __extern INDEX sam_bWideScreen = FALSE;
 __extern FLOAT sam_fPlayerOffset = 0.0f;
 
 // display mode settings
-__extern INDEX sam_bFullScreenActive = FALSE;
+__extern INDEX sam_bFullScreenActive = TRUE;
 __extern INDEX sam_bBorderLessActive = FALSE;
 __extern INDEX sam_iScreenSizeI = 1024;  // current size of the window
 __extern INDEX sam_iScreenSizeJ = 768;  // current size of the window
