diff --git a/CorsixTH/Lua/config_finder.lua b/CorsixTH/Lua/config_finder.lua
index 73815c6e..e2880401 100644
--- a/CorsixTH/Lua/config_finder.lua
+++ b/CorsixTH/Lua/config_finder.lua
@@ -132,7 +132,7 @@ local config_defaults = {
   midi_api = nil,
   midi_port = nil,
   midi_sysex_master_volume = false,
-  theme_hospital_install = [[X:\ThemeHospital\hospital]],
+  theme_hospital_install = [[/userdata/roms/ports/themehospital]],
   debug = false,
   track_fps = false,
   zoom_speed = 80,
@@ -436,7 +436,7 @@ unicode_font = nil -- [[X:\ThemeHospital\font.ttc]]
 -- uncomment the following line, and point it to a directory which exists and
 -- is more suitable.
 --
-savegames = nil -- [[X:\ThemeHospital\Saves]]
+savegames = nil -- [[/userdata/saves/themehospital]]
 
 -------------------------------------------------------------------------------
 -- Levels and Campaigns. By default, the "Levels" and "Campaigns" directory next to
@@ -469,7 +469,7 @@ new_graphics_folder = nil -- [[X:\ThemeHospital\Graphics]]
 -- uncomment the following line, and point it to a directory which exists and
 -- is more suitable.
 --
-screenshots = nil -- [[X:\ThemeHospital\Screenshots]]
+screenshots = nil -- [[/userdata/screenshots]]
 
 -------------------------------------------------------------------------------
 -- If you want to listen to non-Theme-Hospital music, then follow these steps:
@@ -480,7 +480,7 @@ screenshots = nil -- [[X:\ThemeHospital\Screenshots]]
 --     "names.txt" and write the file name on one row, followed by the desired
 --     ingame name on the next row.
 --
-audio_music = nil -- [[X:\ThemeHospital\Music]]
+audio_music = nil -- [[/userdata/music]]
 
 -------------------------------------------------------------------------------
 -- SoundFont: CorsixTH uses the FluidR3 SoundFont by default for playing MIDI music.
