diff --git a/CorsixTH/Src/th_lua_gfx.cpp b/CorsixTH/Src/th_lua_gfx.cpp
index 8216eee3..eddce65a 100644
--- a/CorsixTH/Src/th_lua_gfx.cpp
+++ b/CorsixTH/Src/th_lua_gfx.cpp
@@ -763,7 +763,7 @@ render_target_creation_params l_surface_creation_params(lua_State* L,
   params.min_width = static_cast<int>(luaL_checkinteger(L, iArgStart + 2));
   params.min_height = static_cast<int>(luaL_checkinteger(L, iArgStart + 3));
 
-  params.fullscreen = false;
+  params.fullscreen = true;
   params.present_immediate = false;
   params.direct_zoom = false;
 
