diff --git a/CorsixTH/Lua/app.lua b/CorsixTH/Lua/app.lua
index b3178f94..e2ad541b 100644
--- a/CorsixTH/Lua/app.lua
+++ b/CorsixTH/Lua/app.lua
@@ -1467,6 +1467,7 @@ function App:checkInstallFolder()
       if win_home_dir == user_dir then win_home_dir = nil; end
     end
     local possible_locations = {
+      "/userdata/roms/ports",
       user_dir,
       user_dir and (user_dir .. pathsep .. "Documents"),
       win_home_dir,
@@ -1480,7 +1481,7 @@ function App:checkInstallFolder()
       os.getenv("ProgramFiles"),
       os.getenv("ProgramFiles(x86)"),
       [[C:]], [[D:]], [[E:]], [[F:]], [[G:]], [[H:]] }
-    local possible_folders = { "ThemeHospital", "Theme Hospital", "HOSP", "TH97",
+    local possible_folders = { "themehospital", "Theme Hospital", "HOSP", "TH97",
       [[GOG Galaxy\Games\Theme Hospital]], [[GOG.com\Theme Hospital]],
       [[GOG Games\Theme Hospital]], [[Origin Games\Theme Hospital\data\Game]],
       [[EA Games\Theme Hospital\data\Game]]
