diff --git a/CorsixTH/Src/th_lua_gfx.cpp b/CorsixTH/Src/th_lua_gfx.cpp
index 7af59ba0..7eb53a1e 100644
--- a/CorsixTH/Src/th_lua_gfx.cpp
+++ b/CorsixTH/Src/th_lua_gfx.cpp
@@ -555,7 +555,7 @@ render_target_creation_params l_surface_creation_params(lua_State* L,
   oParams.width = static_cast<int>(luaL_checkinteger(L, iArgStart));
   oParams.height = static_cast<int>(luaL_checkinteger(L, iArgStart + 1));
 
-  oParams.fullscreen = false;
+  oParams.fullscreen = true;
   oParams.present_immediate = false;
   oParams.direct_zoom = false;
 
