diff --git a/cmake/FetchDependencies.cmake b/cmake/FetchDependencies.cmake
index 9b380eb..6496548 100644
--- a/cmake/FetchDependencies.cmake
+++ b/cmake/FetchDependencies.cmake
@@ -43,7 +43,7 @@ endfunction()
 
 fetch_dependency(melonDS "https://github.com/JesseTG/melonDS" "f6692df")
 fetch_dependency(libretro-common "https://github.com/JesseTG/libretro-common" "8e2b884")
-fetch_dependency("embed-binaries" "https://github.com/andoalon/embed-binaries.git" "21f28ca")
+fetch_dependency("embed-binaries" "https://github.com/andoalon/embed-binaries.git" "078b62b")
 fetch_dependency(glm "https://github.com/g-truc/glm" "33b4a62")
 fetch_dependency(libslirp "https://github.com/JesseTG/libslirp-mirror" "e61dbd4")
 fetch_dependency(pntr "https://github.com/robloach/pntr" "650237a")
@@ -82,4 +82,4 @@ endif()
 set_target_properties(example minigzip PROPERTIES EXCLUDE_FROM_ALL TRUE)
 if(HAVE_OFF64_T)
     set_target_properties(example64 minigzip64 PROPERTIES EXCLUDE_FROM_ALL TRUE)
-endif()
\ No newline at end of file
+endif()
