diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index 98749175..ac29eb49 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -38,10 +38,9 @@ namespace Config
 using namespace melonDS;
 
 
-const char* kConfigFile = "melonDS.toml";
+const char* kConfigFile = "/userdata/system/configs/melonDS/melonDS.toml";
 
-const char* kLegacyConfigFile = "melonDS.ini";
-const char* kLegacyUniqueConfigFile = "melonDS.%d.ini";
+const char* kLegacyConfigFile = "/userdata/system/configs/melonDS/melonDS.ini";
 
 toml::value RootTable;
 
