diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index d6d01825..94b34393 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -165,8 +165,7 @@ bool GdbARM9BreakOnStartup;
 CameraConfig Camera[2];
 
 
-const char* kConfigFile = "melonDS.ini";
-const char* kUniqueConfigFile = "melonDS.%d.ini";
+const char* kConfigFile = "/userdata/system/configs/melonDS/melonDS.ini";
 
 ConfigEntry ConfigFile[] =
 {
