diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index 2fdfc3ba..610b8778 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -163,8 +163,7 @@ bool GdbARM9BreakOnStartup;
 CameraConfig Camera[2];
 
 
-const char* kConfigFile = "melonDS.ini";
-const char* kUniqueConfigFile = "melonDS.%d.ini";
+const char* kConfigFile = "/userdata/system/configs/melonDS/melonDS.ini";
 
 ConfigEntry ConfigFile[] =
 {
