diff --git a/src/frontend/qt_sdl/Config.cpp b/src/frontend/qt_sdl/Config.cpp
index 898e4a16..7ca28935 100644
--- a/src/frontend/qt_sdl/Config.cpp
+++ b/src/frontend/qt_sdl/Config.cpp
@@ -355,14 +355,7 @@ ConfigEntry ConfigFile[] =
 void LoadFile(int inst)
 {
     Platform::FileHandle* f;
-    if (inst > 0)
-    {
-        char name[100] = {0};
-        snprintf(name, 99, kUniqueConfigFile, inst+1);
-        f = Platform::OpenLocalFile(name, Platform::FileMode::ReadText);
-    }
-    else
-        f = Platform::OpenLocalFile(kConfigFile, Platform::FileMode::ReadText);
+    f = Platform::OpenLocalFile(kConfigFile, "r");
 
     if (!f) return;
 
@@ -382,9 +375,6 @@ void LoadFile(int inst)
         {
             if (!strncmp(entry->Name, entryname, 32))
             {
-                if ((inst > 0) && (!entry->InstanceUnique))
-                    break;
-
                 switch (entry->Type)
                 {
                 case 0: *(int*)entry->Value = strtol(entryval, NULL, 10); break;
@@ -414,10 +404,6 @@ void Load()
     }
 
     LoadFile(0);
-
-    int inst = Platform::InstanceID();
-    if (inst > 0)
-        LoadFile(inst);
 }
 
 void Save()
@@ -425,22 +411,12 @@ void Save()
     int inst = Platform::InstanceID();
 
     Platform::FileHandle* f;
-    if (inst > 0)
-    {
-        char name[100] = {0};
-        snprintf(name, 99, kUniqueConfigFile, inst+1);
-        f = Platform::OpenLocalFile(name, Platform::FileMode::WriteText);
-    }
-    else
-        f = Platform::OpenLocalFile(kConfigFile, Platform::FileMode::WriteText);
+    f = Platform::OpenLocalFile(kConfigFile, "w");
 
     if (!f) return;
 
     for (ConfigEntry* entry = &ConfigFile[0]; entry->Value; entry++)
     {
-        if ((inst > 0) && (!entry->InstanceUnique))
-            continue;
-
         switch (entry->Type)
         {
         case 0: Platform::FileWriteFormatted(f, "%s=%d\r\n", entry->Name, *(int*)entry->Value); break;
