diff --git a/SDL/OSD.cpp b/SDL/OSD.cpp
index 806df6e..71bb413 100644
--- a/SDL/OSD.cpp
+++ b/SDL/OSD.cpp
@@ -69,7 +69,7 @@ std::string OSD::MakeFilePath(PathType type, const std::string& filename)
 #elif defined(__AMIGAOS4__)
     base_path = "PROGDIR:";
 #else
-    base_path = getenv("HOME");
+    base_path = "/userdata/system/configs";
 #endif
 
     switch (type)
@@ -82,7 +82,7 @@ std::string OSD::MakeFilePath(PathType type, const std::string& filename)
 #elif defined(__AMIGAOS4__)
         path = base_path;
 #else
-        path = base_path / ".simcoupe";
+        path = base_path / "simcoupe";
 #endif
         break;
 
