diff -Naur batocera-emulationstation.orig/es-app/src/guis/GuiMenu.cpp batocera-emulationstation/es-app/src/guis/GuiMenu.cpp
--- batocera-emulationstation.orig/es-app/src/guis/GuiMenu.cpp	2022-02-06 16:31:06.884840515 +0000
+++ batocera-emulationstation/es-app/src/guis/GuiMenu.cpp	2022-02-06 16:38:09.825015511 +0000
@@ -182,7 +182,7 @@
 		if (ApiSystem::getInstance()->isScriptingSupported(ApiSystem::BATOCERASTORE) || ApiSystem::getInstance()->isScriptingSupported(ApiSystem::THEMESDOWNLOADER) ||
 			(ApiSystem::getInstance()->isScriptingSupported(ApiSystem::THEBEZELPROJECT) && ApiSystem::getInstance()->isScriptingSupported(ApiSystem::DECORATIONS)) ||
 			ApiSystem::getInstance()->isScriptingSupported(ApiSystem::UPGRADE))
-			addEntry(_("UPDATES & DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
+			addEntry(_("DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
 
 		addEntry(_("SYSTEM SETTINGS").c_str(), true, [this] { openSystemSettings(); }, "iconSystem");
 	}
@@ -794,7 +794,7 @@
 
 void GuiMenu::openUpdatesSettings()
 {
-	GuiSettings *updateGui = new GuiSettings(mWindow, _("UPDATES & DOWNLOADS").c_str());
+	GuiSettings *updateGui = new GuiSettings(mWindow, _("DOWNLOADS").c_str());
 
 	updateGui->addGroup(_("DOWNLOADS"));
 
@@ -835,7 +835,7 @@
 
 	if (ApiSystem::getInstance()->isScriptingSupported(ApiSystem::UPGRADE))
 	{
-		updateGui->addGroup(_("SOFTWARE UPDATES"));
+		/*updateGui->addGroup(_("SOFTWARE UPDATES"));
 
 		// Enable updates
 		updateGui->addSwitch(_("CHECK FOR UPDATES"), "updates.enabled", false);
@@ -876,7 +876,7 @@
 
 				mWindow->pushGui(new GuiUpdate(mWindow));
 			}
-		});
+		});*/
 	}
 
 	mWindow->pushGui(updateGui);
