diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d6b3e11..c02ada88 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1321,6 +1321,7 @@ endif()
 
 if(THEXTECH_BUILD_GL_ES_MODERN)
     target_compile_definitions(thextech PRIVATE -DTHEXTECH_BUILD_GL_ES_MODERN)
+    target_link_libraries(thextech PRIVATE EGL GLESv2)
 endif()
 
 if(THEXTECH_BUILD_GL_DESKTOP_MODERN OR THEXTECH_BUILD_GL_ES_MODERN)
@@ -1333,6 +1334,7 @@ endif()
 
 if(THEXTECH_BUILD_GL_ES_LEGACY)
     target_compile_definitions(thextech PRIVATE -DTHEXTECH_BUILD_GL_ES_LEGACY)
+    target_link_libraries(thextech PRIVATE EGL GLESv1_CM)
 endif()
 
 if(THEXTECH_ENABLE_NEW_USER_PATHS)
