diff --git a/CMakeLists.txt b/CMakeLists.txt
index c0c6bfd2..89a41a38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1300,6 +1300,7 @@ endif()
 
 if(THEXTECH_BUILD_GL_ES_MODERN)
     target_compile_definitions(thextech PRIVATE -DTHEXTECH_BUILD_GL_ES_MODERN)
+    target_link_libraries(thextech PRIVATE EGL GLESv2)
 endif()
 
 if(THEXTECH_BUILD_GL_DESKTOP_MODERN OR THEXTECH_BUILD_GL_ES_MODERN)
@@ -1312,6 +1313,7 @@ endif()
 
 if(THEXTECH_BUILD_GL_ES_LEGACY)
     target_compile_definitions(thextech PRIVATE -DTHEXTECH_BUILD_GL_ES_LEGACY)
+    target_link_libraries(thextech PRIVATE EGL GLESv1_CM)
 endif()
 
 if(THEXTECH_ENABLE_NEW_USER_PATHS)
