diff --git a/src/control/touchscreen.cpp b/src/control/touchscreen.cpp
index ec82af04..560fd9f9 100644
--- a/src/control/touchscreen.cpp
+++ b/src/control/touchscreen.cpp
@@ -1003,8 +1003,8 @@ void TouchScreenController::scanTouchDevices()
             d.id = SDL_GetTouchDevice(i);
 
 // A reminder to don't forget change the condition after SDL2 gets updated
-#if SDL_VERSION_ATLEAST(2, 30, 7)
-#   error PLEASE CHANGE THE CONDITION FOR SDL2 VERSION TO THE VERSION 2.30.8!
+#if SDL_VERSION_ATLEAST(2, 30, 8)
+#   error PLEASE CHANGE THE CONDITION FOR SDL2 VERSION TO THE VERSION 2.30.9!
 #endif
 
 #if !defined(VITA) || SDL_VERSION_ATLEAST(2, 30, 6) // FIXME: Enable this tweak after SDL2 side will be fixed
