From 0ed9a528b4c3ce63158e43699c4d30c21d6face8 Mon Sep 17 00:00:00 2001
From: Markus Kalkbrenner <git@kalki.de>
Date: Sat, 3 Jan 2026 15:40:25 +0100
Subject: [PATCH] install headers

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e71fa08..ed33f20 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -123,6 +123,7 @@ if(BUILD_SHARED)
    install(FILES
       src/serum.h
       src/serum-decode.h
+      src/TimeUtils.h
       DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 
    if(PLATFORM STREQUAL "win" OR PLATFORM STREQUAL "macos" OR PLATFORM STREQUAL "linux")
@@ -155,6 +156,7 @@ if(BUILD_STATIC)
    install(FILES
       src/serum.h
       src/serum-decode.h
+      src/TimeUtils.h
       DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
 
    if(PLATFORM STREQUAL "win" OR PLATFORM STREQUAL "macos" OR PLATFORM STREQUAL "linux")
