diff --git a/src/gallium/targets/dri/dri_target.c b/src/gallium/targets/dri/dri_target.c
index 6604afe99f5..9d4d757993b 100644
--- a/src/gallium/targets/dri/dri_target.c
+++ b/src/gallium/targets/dri/dri_target.c
@@ -101,6 +101,9 @@ DEFINE_LOADER_DRM_ENTRYPOINT(tegra);
 
 #if defined(GALLIUM_KMSRO)
 DEFINE_LOADER_DRM_ENTRYPOINT(armada_drm)
+DEFINE_LOADER_DRM_ENTRYPOINT(drm_rp1_dpi)
+DEFINE_LOADER_DRM_ENTRYPOINT(drm_rp1_dsi)
+DEFINE_LOADER_DRM_ENTRYPOINT(drm_rp1_vec)
 DEFINE_LOADER_DRM_ENTRYPOINT(exynos)
 DEFINE_LOADER_DRM_ENTRYPOINT(gm12u320)
 DEFINE_LOADER_DRM_ENTRYPOINT(hdlcd)
diff --git a/src/gallium/targets/dril/meson.build b/src/gallium/targets/dril/meson.build
index 3adf97b24a7..3726992df87 100644
--- a/src/gallium/targets/dril/meson.build
+++ b/src/gallium/targets/dril/meson.build
@@ -64,6 +64,9 @@ dril_dri = shared_library(
 
 foreach d : [[with_gallium_kmsro, [
                'armada-drm_dri.so',
+               'drm-rp1-dpi_dri.so',
+               'drm-rp1-dsi_dri.so',
+               'drm-rp1-vec_dri.so',
                'exynos_dri.so',
                'gm12u320_dri.so',
                'hdlcd_dri.so',
