diff --git a/modules/mod_sound/mod_sound.c b/modules/mod_sound/mod_sound.c
index ea800db..73ece80 100644
--- a/modules/mod_sound/mod_sound.c
+++ b/modules/mod_sound/mod_sound.c
@@ -125,6 +125,11 @@ static int SDLCALL __modsound_close_cb( SDL_RWops *context )
     return( 0 );
 }
 
+static Sint64 SDLCALL __modsound_size_cb( SDL_RWops *context )
+{
+    return (Sint64)file_size( context->hidden.unknown.data1 );
+}
+
 static SDL_RWops *SDL_RWFromBGDFP( file *fp )
 {
     SDL_RWops *rwops = SDL_AllocRW();
@@ -134,6 +139,7 @@ static SDL_RWops *SDL_RWFromBGDFP( file *fp )
         rwops->read = __modsound_read_cb;
         rwops->write = __modsound_write_cb;
         rwops->close = __modsound_close_cb;
+        rwops->size = __modsound_size_cb;
         rwops->hidden.unknown.data1 = fp;
     }
     return( rwops );
