diff --git a/externals/CMakeLists.txt b/externals/CMakeLists.txt
index 2317724a2..4c5798b1e 100644
--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -427,9 +427,9 @@ if (ENABLE_VULKAN)
         add_library(glslang INTERFACE)
         add_library(SPIRV INTERFACE)
         target_link_libraries(glslang INTERFACE glslang::glslang)
-        target_link_libraries(SPIRV INTERFACE glslang::SPIRV)
+        target_link_libraries(SPIRV INTERFACE glslang::glslang)
         # System include path is different from submodule include path
-        get_target_property(GLSLANG_PREFIX glslang::SPIRV INTERFACE_INCLUDE_DIRECTORIES)
+        get_target_property(GLSLANG_PREFIX glslang::glslang INTERFACE_INCLUDE_DIRECTORIES)
         target_include_directories(SPIRV SYSTEM INTERFACE "${GLSLANG_PREFIX}/glslang")
     else()
         set(SKIP_GLSLANG_INSTALL ON CACHE BOOL "")
