diff --git a/src/core/hle/service/http/http_c.cpp b/src/core/hle/service/http/http_c.cpp
index f3583982c..251792512 100644
--- a/src/core/hle/service/http/http_c.cpp
+++ b/src/core/hle/service/http/http_c.cpp
@@ -519,11 +519,11 @@ void HTTP_C::GetResponseStatusCodeImpl(Kernel::HLERequestContext& ctx, bool time
         itr->second.request_future.wait();
     }
 
-    const u32 response_code = itr->second.response.status;
+    //const u32 response_code = itr->second.response.status;
 
     IPC::RequestBuilder rb = rp.MakeBuilder(2, 0);
     rb.Push(RESULT_SUCCESS);
-    rb.Push(response_code);
+    //rb.Push(response_code);
 }
 
 void HTTP_C::SetClientCertContext(Kernel::HLERequestContext& ctx) {
@@ -759,13 +759,13 @@ void HTTP_C::GetDownloadSizeState(Kernel::HLERequestContext& ctx) {
     u32 content_length = 0;
     const bool is_complete = itr->second.request_future.wait_for(std::chrono::milliseconds(0)) ==
                              std::future_status::ready;
-    if (is_complete) {
+    /*if (is_complete) {
         const auto& headers = itr->second.response.headers;
         const auto& it = headers.find("Content-Length");
         if (it != headers.end()) {
             content_length = std::stoi(it->second);
         }
-    }
+    }*/
 
     IPC::RequestBuilder rb = rp.MakeBuilder(3, 0);
     rb.Push(RESULT_SUCCESS);
