diff --git a/ext/x265/gstx265enc.c b/ext/x265/gstx265enc.c
index e6aa973ac7cc..ee72389720ef 100644
--- a/ext/x265/gstx265enc.c
+++ b/ext/x265/gstx265enc.c
@@ -948,8 +948,13 @@ gst_x265_enc_init_encoder_locked (GstX265Enc * encoder)
        * HEVC uses gbr order
        * See spec D.3.28 display_primaries_x and display_primaries_y
        */
-      encoder->x265param.masteringDisplayColorVolume =
-          g_strdup_printf ("G(%hu,%hu)B(%hu,%hu)R(%hu,%hu)WP(%hu,%hu)L(%u,%u)",
+#if X265_BUILD < 214
+      encoder->x265param.masteringDisplayColorVolume = g_strdup_printf (
+#else
+      snprintf (encoder->x265param.masteringDisplayColorVolume,
+          X265_MAX_STRING_SIZE,
+#endif
+          "G(%hu,%hu)B(%hu,%hu)R(%hu,%hu)WP(%hu,%hu)L(%u,%u)",
           minfo.display_primaries[1].x, minfo.display_primaries[1].y,
           minfo.display_primaries[2].x, minfo.display_primaries[2].y,
           minfo.display_primaries[0].x, minfo.display_primaries[0].y,
