diff --git a/Quake/Makefile b/Quake/Makefile
index ff2bb9c..c2c5e09 100644
--- a/Quake/Makefile
+++ b/Quake/Makefile
@@ -5,21 +5,21 @@
 # "make DO_USERDIRS=1" to enable user directories support
 
 # Enable/Disable user directories support
-DO_USERDIRS=0
+DO_USERDIRS=1
 
 ### Enable/Disable SDL2
-USE_SDL2=0
+USE_SDL2=1
 
 ### Enable/Disable codecs for streaming music support
 USE_CODEC_WAVE=1
-USE_CODEC_FLAC=0
+USE_CODEC_FLAC=1
 USE_CODEC_MP3=1
 USE_CODEC_VORBIS=1
 USE_CODEC_OPUS=0
 # either xmp or mikmod (or modplug)
 USE_CODEC_MIKMOD=0
-USE_CODEC_XMP=0
-USE_CODEC_MODPLUG=0
+USE_CODEC_XMP=1
+USE_CODEC_MODPLUG=1
 USE_CODEC_UMX=0
 
 # which library to use for mp3 decoding: mad or mpg123
@@ -102,8 +102,8 @@ SDL_CONFIG ?= sdl2-config
 else
 SDL_CONFIG ?= sdl-config
 endif
-SDL_CFLAGS := $(shell $(SDL_CONFIG) --cflags)
-SDL_LIBS   := $(shell $(SDL_CONFIG) --libs)
+SDL_CFLAGS := `pkg-config sdl2 --cflags`
+SDL_LIBS   := `pkg-config sdl2 --libs`
 
 ifeq ($(HOST_OS),sunos)
 NET_LIBS   :=-lsocket -lnsl -lresolv
