diff --git a/softmmu/vl.c b/softmmu/vl.c
index 2eecce1cd7..fa3c1b5e01 100644
--- a/softmmu/vl.c
+++ b/softmmu/vl.c
@@ -2789,19 +2789,19 @@ static const char *get_eeprom_path(void)
     int size = get_image_size(path);
 
     if (size < 0) {
-        char *msg = g_strdup_printf("Failed to open EEPROM file '%s'.\n\n"
-                                    "Please check machine settings.", path);
-        xemu_queue_error_message(msg);
-        g_free(msg);
+        //char *msg = g_strdup_printf("Failed to open EEPROM file '%s'.\n\n"
+        //                            "Please check machine settings.", path);
+        //xemu_queue_error_message(msg);
+        //g_free(msg);
         return NULL;
     }
 
     if (size != 256) {
-        char *msg = g_strdup_printf(
-            "Invalid EEPROM file '%s' size of %d; should be 256 bytes.\n\n"
-            "Please check machine settings.", path, size);
-        xemu_queue_error_message(msg);
-        g_free(msg);
+       // char *msg = g_strdup_printf(
+       //     "Invalid EEPROM file '%s' size of %d; should be 256 bytes.\n\n"
+       //     "Please check machine settings.", path, size);
+       // xemu_queue_error_message(msg);
+       // g_free(msg);
         return NULL;
     }
 
