diff --git a/CMakeLists.txt b/CMakeLists.txt
index 40ca8b149..f0f532e45 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -171,7 +171,7 @@ if (YUZU_TESTS)
     find_package(Catch2 2.13.7 REQUIRED CONFIG)
 endif()
 
-find_package(Boost 1.73.0 COMPONENTS context)
+find_package(Boost 1.78.0 COMPONENTS context)
 if (Boost_FOUND)
     set(Boost_LIBRARIES Boost::boost)
     # Conditionally add Boost::context only if the found Boost package provides it
