diff --git a/src/lang/pl.h b/src/lang/pl.h
index fb98dbf..f5fb7dd 100644
--- a/src/lang/pl.h
+++ b/src/lang/pl.h
@@ -273,7 +273,7 @@ const char *STR_PL[] = { ""
     /* 40 */ , "No dobra. To jest bardzo du*zy skok.@Zr)ob skok z rozbiegiem tak samo jak przed chwil|a,@ale dodatkowo w powietrzu, wci)snij i trzymaj Akcj|e@abym mog}la si|e chwyci)c kraw|edzi."
     /* 41 */ , "Nie)zle."
     /* 42 */ , "Spr)obuj tu wskoczy)c na g)or|e.@Wci)snij Prz)od i trzymaj Akcj|e."
-    /* 43 */ , "Nie mog|e wspi|a)c si|e na g)or|e, poniewa*z szczelina jest zbyt ma}la.@Ale naci)snij w prawo, a ja b|ed|e kiwa}la si|e na boki@a*z b|edzie miejsce, a potem naci)snij do przodu.""
+    /* 43 */ , "Nie mog|e wspi|a)c si|e na g)or|e, poniewa*z szczelina jest zbyt ma}la.@Ale naci)snij w prawo, a ja b|ed|e kiwa}la si|e na boki@a*z b|edzie miejsce, a potem naci)snij do przodu."
     /* 44 */ , ")Swietnie! Je)sli jest d}lugi spadek i nie chc|e@ zrobi)c sobie krzywdy zeskakuj|ac z niego,@mog|e ostro*znie opu)sci)c si|e w d)o}l."
     /* 45 */ , "Naci)snij klawisz do ty}lu, a odskocz|e do ty}lu.@Natychmiast naci)snij i przytrzymaj przycisk akcji,@a z}lapi|e si|e gzymsu w drodze na d)o}l."
     /* 46 */ , "Nast|epnie pu)s)c."
