diff --git a/src/configuration.c b/src/configuration.c
index de6006d0..1a53ae4a 100644
--- a/src/configuration.c
+++ b/src/configuration.c
@@ -736,7 +736,7 @@ void Configuration_SetDefault(void)
 	/* Set defaults for Rom */
 	File_MakePathBuf(ConfigureParams.Rom.szTosImageFileName,
 	                 sizeof(ConfigureParams.Rom.szTosImageFileName),
-	                 Paths_GetDataDir(), "tos", "img");
+	                 "/userdata/bios", "tos", "img");
 	ConfigureParams.Rom.bPatchTos = true;
 	strcpy(ConfigureParams.Rom.szCartridgeImageFileName, "");
 
