diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9e6affdb..ebd14456 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -16,7 +16,7 @@ if(EMSCRIPTEN)
 endif()
 
 # Disk image code is shared with the hmsa tool, so we put it into a library:
-add_library(Floppy createBlankImage.c dim.c msa.c st.c zip.c utils.c)
+add_library(Floppy STATIC createBlankImage.c dim.c msa.c st.c zip.c utils.c)
 
 # When building for macOS, define specific sources for gui and resources
 if(ENABLE_OSX_BUNDLE)
diff --git a/src/cpu/CMakeLists.txt b/src/cpu/CMakeLists.txt
index 087e1b0d..deead2c3 100644
--- a/src/cpu/CMakeLists.txt
+++ b/src/cpu/CMakeLists.txt
@@ -95,7 +95,7 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
 		PROPERTIES COMPILE_FLAGS ${CPUMAIN_CFLAGS})
 endif()
 
-add_library(UaeCpu ${CPUEMU_SRCS} ${WINUAE_SRCS} custom.c events.c memory.c
+add_library(UaeCpu STATIC ${CPUEMU_SRCS} ${WINUAE_SRCS} custom.c events.c memory.c
 		   hatari-glue.c)
 
 target_link_libraries(UaeCpu PRIVATE ${SDL2_LIBRARIES})
diff --git a/src/debug/CMakeLists.txt b/src/debug/CMakeLists.txt
index 9c23a7fd..eacc958c 100644
--- a/src/debug/CMakeLists.txt
+++ b/src/debug/CMakeLists.txt
@@ -7,7 +7,7 @@ if(ENABLE_DSP_EMU)
 	set(DSPDBG_C debugdsp.c)
 endif(ENABLE_DSP_EMU)
 
-add_library(Debug
+add_library(Debug STATIC
 	    log.c debugui.c breakcond.c debugcpu.c debugInfo.c
 	    ${DSPDBG_C} evaluate.c history.c symbols.c vars.c
 	    profile.c profilecpu.c profiledsp.c
diff --git a/src/falcon/CMakeLists.txt b/src/falcon/CMakeLists.txt
index b7423013..70478653 100644
--- a/src/falcon/CMakeLists.txt
+++ b/src/falcon/CMakeLists.txt
@@ -3,7 +3,6 @@ if(ENABLE_DSP_EMU)
 	set(DSP_SOURCES dsp_core.c dsp_cpu.c dsp_disasm.c)
 endif(ENABLE_DSP_EMU)
 
-add_library(Falcon
-	    crossbar.c microphone.c nvram.c videl.c dsp.c ${DSP_SOURCES})
+add_library(Falcon STATIC crossbar.c microphone.c nvram.c videl.c dsp.c ${DSP_SOURCES})
 
 target_link_libraries(Falcon PRIVATE ${SDL2_LIBRARIES})
diff --git a/src/gui-sdl/CMakeLists.txt b/src/gui-sdl/CMakeLists.txt
index 4f822ba0..a9e06935 100644
--- a/src/gui-sdl/CMakeLists.txt
+++ b/src/gui-sdl/CMakeLists.txt
@@ -5,7 +5,7 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-write-strings")
 endif()
 
-add_library(GuiSdl
+add_library(GuiSdl STATIC
 	dlgAbout.c dlgAlert.c dlgCpu.c dlgDevice.c dlgFileSelect.c dlgFloppy.c
 	dlgHalt.c dlgHardDisk.c dlgJoystick.c dlgKeyboard.c dlgMain.c
 	dlgMemory.c dlgNewDisk.c dlgRom.c dlgScreen.c dlgSound.c dlgSystem.c
