diff --git a/cheevos/cheevos.c b/cheevos/cheevos.c
index 3c827f78d3..03d1cb716c 100644
--- a/cheevos/cheevos.c
+++ b/cheevos/cheevos.c
@@ -475,9 +475,10 @@ static void rcheevos_award_achievement(const rc_client_achievement_t* cheevo)
       if (shotname)
       {
          const char *path_directory_screenshot = settings->paths.directory_screenshot;
-         snprintf(shotname, shotname_len, "%s/%s-cheevo-%u",
+         snprintf(shotname, shotname_len, "%s/%s - %s (%u)",
             path_directory_screenshot,
             path_basename(path_get(RARCH_PATH_BASENAME)),
+            cheevo->title,
             (unsigned)cheevo->id);
          shotname[shotname_len - 1] = '\0';
 
