diff --git a/cheevos/cheevos.c b/cheevos/cheevos.c
index 68acaaac56..17eeb957a3 100644
--- a/cheevos/cheevos.c
+++ b/cheevos/cheevos.c
@@ -345,9 +345,10 @@ static void rcheevos_award_achievement(const rc_client_achievement_t* cheevo)
       if (shotname)
       {
          video_driver_state_t* video_st = video_state_get_ptr();;
-         snprintf(shotname, shotname_len, "%s/%s-cheevo-%u",
+         snprintf(shotname, shotname_len, "%s/%s - %s (%u)",
             settings->paths.directory_screenshot,
             path_basename(path_get(RARCH_PATH_BASENAME)),
+            cheevo->title,
             (unsigned)cheevo->id);
          shotname[shotname_len - 1] = '\0';
 
