diff --git a/cheevos/cheevos.c b/cheevos/cheevos.c
index a83a4eb177..47f4cbc0ed 100644
--- a/cheevos/cheevos.c
+++ b/cheevos/cheevos.c
@@ -844,9 +844,10 @@ void rcheevos_award_achievement(rcheevos_locals_t* locals,
       if (shotname)
       {
          video_driver_state_t *video_st  = video_state_get_ptr();;
-         snprintf(shotname, shotname_len, "%s/%s-cheevo-%u",
+         snprintf(shotname, shotname_len, "%s/%s - %s (%u)",
                settings->paths.directory_screenshot,
                path_basename(path_get(RARCH_PATH_BASENAME)),
+               cheevo->title,
                cheevo->id);
          shotname[shotname_len - 1] = '\0';
 
