diff --git a/cheevos/cheevos.c b/cheevos/cheevos.c
index ac5a648bf6..9f9f0a13f8 100644
--- a/cheevos/cheevos.c
+++ b/cheevos/cheevos.c
@@ -375,9 +375,10 @@ void rcheevos_award_achievement(rcheevos_locals_t* locals,
 
       if (shotname)
       {
-         snprintf(shotname, shotname_len, "%s/%s-cheevo-%u",
+         snprintf(shotname, shotname_len, "%s/%s - %s (%u)",
                settings->paths.directory_screenshot,
                path_basename(path_get(RARCH_PATH_BASENAME)),
+               cheevo->title,
                cheevo->id);
          shotname[shotname_len - 1] = '\0';
 
