diff --git a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
index 29b3cd5654..551cdc5339 100644
--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -44,7 +44,7 @@
 #if defined(__linux__) && !defined(ANDROID) && __STDC_VERSION__ >= 199901L \
       && (!defined(HAVE_CONFIG_H) || defined(HAVE_GAMEMODE))
 #include "../../deps/feralgamemode/gamemode_client.h"
-#define FERAL_GAMEMODE
+//#define FERAL_GAMEMODE
 #endif
 
 #ifdef ANDROID
