diff --git a/frontend/drivers/platform_unix.c b/frontend/drivers/platform_unix.c
index f2da086600..be7639f32e 100644
--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -32,7 +32,7 @@
 #include <linux/version.h>
 #if __STDC_VERSION__ >= 199901L && !defined(ANDROID)
 #include "../../deps/feralgamemode/gamemode_client.h"
-#define FERAL_GAMEMODE
+//#define FERAL_GAMEMODE
 #endif
 #endif
 
