diff --git a/runloop.c b/runloop.c
index ef45c7f448..7f6fe23a4e 100644
--- a/runloop.c
+++ b/runloop.c
@@ -7552,7 +7552,9 @@ bool runloop_get_entry_state_path(char *s, size_t len, int slot)
       return false;
 
    _len = strlcpy(s, name_savestate, len);
-   snprintf(s + _len, len - _len, "%d.entry", slot);
+   
+   if (slot > 0)
+     snprintf(s + _len, len - _len, "%d", slot);
 
    return true;
 }
