diff --git a/runloop.c b/runloop.c
index c16e76f595..eb7964c04d 100644
--- a/runloop.c
+++ b/runloop.c
@@ -7337,7 +7337,9 @@ bool runloop_get_entry_state_path(char *s, size_t len, unsigned slot)
       return false;
 
    _len = strlcpy(s, name_savestate, len);
-   snprintf(s + _len, len - _len, "%d.entry", slot);
+   
+   if (slot > 0)
+     snprintf(s + _len, len - _len, "%d", slot);
 
    return true;
 }
