diff --git a/libretro-common/vfs/vfs_implementation.c b/libretro-common/vfs/vfs_implementation.c
index 6d24a6b621..5f2f71705e 100644
--- a/libretro-common/vfs/vfs_implementation.c
+++ b/libretro-common/vfs/vfs_implementation.c
@@ -1361,7 +1361,7 @@ int retro_vfs_file_punch_hole_impl(libretro_vfs_implementation_file *stream,
          return -1;
       return 0;
    }
-#elif defined(__linux__) && defined(FALLOC_FL_PUNCH_HOLE)
+#elif defined(__linux__) && defined(FALLOC_FL_PUNCH_HOLE1)
    {
       int fd = stream->fd;
 
