diff --git a/ports/gmloader/main.c b/ports/gmloader/main.c
index 72193f5..52d201c 100644
--- a/ports/gmloader/main.c
+++ b/ports/gmloader/main.c
@@ -6,7 +6,6 @@
 #include <errno.h>
 #include <getopt.h>
 
-#include "io_util.h"
 #include "platform.h"
 #include "so_util.h"
 #include "fake_jni.h"
@@ -48,7 +47,6 @@ int main(int argc, char *argv[])
 {
     int disp_w = -1, disp_h = -1;
     uintptr_t addr_yoyo = 0x40000000;
-    uintptr_t addr_stdcpp = addr_yoyo - (0x200000 + 0x10000);
     static struct option long_options[] = {
         {"width", required_argument, 0, 'w'},
         {"height", required_argument, 0, 'h'},
@@ -103,43 +101,22 @@ int main(int argc, char *argv[])
     }
 
     /* Load elf module */
-    void *yoyo_inflated_ptr = NULL;
-    void *stdcpp_inflated_ptr = NULL;
+    void *inflated_ptr = NULL;
 #if ARCH_ARMV6
-    char *libyoyo = "lib/armeabi/libyoyo.so";
-    char *libcpp = "lib/armeabi/libc++_shared.so";
+    char *lib = "lib/armeabi/libyoyo.so";
 #else
-    char *libyoyo = "lib/armeabi-v7a/libyoyo.so";
-    char *libcpp = "lib/armeabi-v7a/libc++_shared.so";
+    char *lib = "lib/armeabi-v7a/libyoyo.so";
 #endif
-    warning("Inflating %s...\n", libcpp);
+    warning("Inflating %s...\n", lib);
     size_t inflated_bytes = 0;
-    if (zip_inflate_buf(apk, libcpp, &inflated_bytes, &stdcpp_inflated_ptr) == 0) {
-        int fd;
-        if (!io_buffer_from_file("libc++_shared.so", &fd, &stdcpp_inflated_ptr, &inflated_bytes, 0)) {
-            fatal_error("Failed to acquire libc++_shared.so, exiting.\n");
-            return -1;
-        }
-    }
-
-    warning("Loading runner elf %s (%p, %d bytes)...\n", libcpp, stdcpp_inflated_ptr, inflated_bytes);
-    so_module cpp = {};
-    int ret = so_load(&cpp, libcpp, addr_stdcpp, stdcpp_inflated_ptr, inflated_bytes);
-    if (ret != 0) {
-        fatal_error("Unable to load library!\n");
-        return -1;
-    }
-
-    warning("Inflating %s...\n", libyoyo);
-    inflated_bytes = 0;
-    if (zip_inflate_buf(apk, libyoyo, &inflated_bytes, &yoyo_inflated_ptr) == 0) {
+    if (zip_inflate_buf(apk, lib, &inflated_bytes, &inflated_ptr) == 0) {
         fatal_error("Failed to acquire shared library, exiting.\n");
         return -1;
     }
 
-    warning("Loading runner elf %s (%p, %d bytes)...\n", libyoyo, yoyo_inflated_ptr, inflated_bytes);
+    warning("Loading runner elf %s (%p, %d bytes)...\n", lib, inflated_ptr, inflated_bytes);
     so_module runner = {};
-    ret = so_load(&runner, libyoyo, addr_yoyo, yoyo_inflated_ptr, inflated_bytes);
+    int ret = so_load(&runner, lib, addr_yoyo, inflated_ptr, inflated_bytes);
     if (ret != 0) {
         fatal_error("Unable to load library!\n");
         return -1;
