diff --git a/bridges/misc_bridges.c b/bridges/misc_bridges.c
index 762d682..e4921b4 100644
--- a/bridges/misc_bridges.c
+++ b/bridges/misc_bridges.c
@@ -350,7 +350,7 @@ int nanosleep(const struct timespec *req, struct timespec *rem) {
 }
 #endif
 
-int usleep(long usec)
+/*int usleep(long usec)
 {
     struct timespec ts = {
         .tv_sec = (long int) (usec / 1000000),
@@ -358,7 +358,7 @@ int usleep(long usec)
     };
 
     return nanosleep(&ts, NULL);
-}
+}*/
 
 DynLibFunction symtable_misc[] = {
     {"_ZdaPv", (uintptr_t)&_ZdaPv},
@@ -465,4 +465,4 @@ DynLibFunction symtable_misc[] = {
     {"_ZTVSt12length_error", (uintptr_t)&_ZTVSt12length_error},
 
     {NULL, (uintptr_t)NULL}
-};
\ No newline at end of file
+};
