Submodule deps/CodeGen contains modified content
diff --git a/deps/CodeGen/include/Jitter_Symbol.h b/deps/CodeGen/include/Jitter_Symbol.h
index 4422d37..c3428e5 100644
--- a/deps/CodeGen/include/Jitter_Symbol.h
+++ b/deps/CodeGen/include/Jitter_Symbol.h
@@ -5,6 +5,7 @@
 #include <string>
 #include <cassert>
 #include <cstdlib>
+#include <cstdint>
 
 namespace Jitter
 {
diff --git a/deps/CodeGen/include/ObjectFile.h b/deps/CodeGen/include/ObjectFile.h
index 4ec1550..d6a0230 100644
--- a/deps/CodeGen/include/ObjectFile.h
+++ b/deps/CodeGen/include/ObjectFile.h
@@ -4,6 +4,7 @@
 #include <string>
 #include <vector>
 #include <memory>
+#include <cstdint>
 #include "Stream.h"
 
 namespace Jitter
