diff --git a/Source/ee/Ee_LibMc2.cpp b/Source/ee/Ee_LibMc2.cpp
index 83850203..2920d367 100644
--- a/Source/ee/Ee_LibMc2.cpp
+++ b/Source/ee/Ee_LibMc2.cpp
@@ -1,4 +1,5 @@
 #include "Ee_LibMc2.h"
+#include <cstring>
 #include <cassert>
 #include "zip/ZipArchiveWriter.h"
 #include "zip/ZipArchiveReader.h"
diff --git a/Source/iop/IopBios.cpp b/Source/iop/IopBios.cpp
index 6d336ac0..22f51da8 100644
--- a/Source/iop/IopBios.cpp
+++ b/Source/iop/IopBios.cpp
@@ -1,4 +1,5 @@
 #include <vector>
+#include <cstring>
 
 #include "string_format.h"
 #include "PtrStream.h"
diff --git a/Source/iop/Iop_McServ.cpp b/Source/iop/Iop_McServ.cpp
index 9b4cedda..6af70c89 100644
--- a/Source/iop/Iop_McServ.cpp
+++ b/Source/iop/Iop_McServ.cpp
@@ -1,5 +1,6 @@
 #include <assert.h>
 #include <stdio.h>
+#include <cstring>
 #include <algorithm>
 #include <states/XmlStateFile.h>
 #include <xml/Utils.h>
