diff --git a/Source/ui_qt/CMakeLists.txt b/Source/ui_qt/CMakeLists.txt
index acb51501..76d82cb4 100644
--- a/Source/ui_qt/CMakeLists.txt
+++ b/Source/ui_qt/CMakeLists.txt
@@ -245,21 +245,21 @@ set(QT_RESOURCES
 )
 
 set(QT_COMPONENTS Widgets Core)
-if(TARGET_PLATFORM_UNIX)
-	list(APPEND QT_COMPONENTS X11Extras)
-endif()
-find_package(Qt5 REQUIRED COMPONENTS ${QT_COMPONENTS})
+#if(TARGET_PLATFORM_UNIX)
+#	list(APPEND QT_COMPONENTS X11Extras)
+#endif()
+find_package(Qt6 REQUIRED COMPONENTS ${QT_COMPONENTS})
 
-list(APPEND PROJECT_LIBS Qt5::Widgets Qt5::Core)
-if(TARGET_PLATFORM_UNIX)
-	list(APPEND PROJECT_LIBS Qt5::X11Extras)
-endif()
+list(APPEND PROJECT_LIBS Qt6::Widgets Qt6::Core Qt6::GuiPrivate)
+#if(TARGET_PLATFORM_UNIX)
+#	list(APPEND PROJECT_LIBS Qt6::X11Extras)
+#endif()
 
 add_definitions(${QT_DEFINITIONS})
 
-QT5_ADD_RESOURCES(QT_RES_SOURCES ${QT_RESOURCES})
-QT5_WRAP_UI(QT_UI_HEADERS ${QT_UIS})
-QT5_WRAP_CPP(QT_MOC_SRCS ${QT_MOC_HEADERS})
+Qt6_ADD_RESOURCES(QT_RES_SOURCES ${QT_RESOURCES})
+Qt6_WRAP_UI(QT_UI_HEADERS ${QT_UIS})
+Qt6_WRAP_CPP(QT_MOC_SRCS ${QT_MOC_HEADERS})
 
 if(TARGET_PLATFORM_MACOS)
 	set(QT_SOURCES
diff --git a/Source/ui_qt/GSH_VulkanQt.cpp b/Source/ui_qt/GSH_VulkanQt.cpp
index d59270fe..ef9f36df 100644
--- a/Source/ui_qt/GSH_VulkanQt.cpp
+++ b/Source/ui_qt/GSH_VulkanQt.cpp
@@ -8,7 +8,8 @@
 #endif
 
 #ifdef __linux__
-#include <QX11Info>
+//#include <QX11Info>
+#include <QtGui/private/qtx11extras_p.h>
 #endif
 
 CGSH_VulkanQt::CGSH_VulkanQt(QWindow* renderWindow)
diff --git a/tools/ElfView/CMakeLists.txt b/tools/ElfView/CMakeLists.txt
index cf087e9a..3347eaa4 100644
--- a/tools/ElfView/CMakeLists.txt
+++ b/tools/ElfView/CMakeLists.txt
@@ -17,8 +17,8 @@ endif()
 list(APPEND PROJECT_LIBS PlayCore)
 
 set(QT_COMPONENTS Widgets Core)
-find_package(Qt5 REQUIRED COMPONENTS ${QT_COMPONENTS})
-list(APPEND PROJECT_LIBS Qt5::Widgets Qt5::Core)
+find_package(Qt6 REQUIRED COMPONENTS ${QT_COMPONENTS})
+list(APPEND PROJECT_LIBS Qt6::Widgets Qt6::Core)
 
 set(DEBUG_SUPPORT_ROOT ${CMAKE_CURRENT_SOURCE_DIR}/../../Source/ui_qt/DebugSupport)
 
@@ -63,9 +63,9 @@ set(QT_RESOURCES
 	qt_ui/resources.qrc
 )
 
-QT5_ADD_RESOURCES(QT_RES_SOURCES ${QT_RESOURCES})
-QT5_WRAP_UI(ELFVIEW_UI_HEADERS ${ELFVIEW_UI})
-QT5_WRAP_CPP(ELFVIEW_MOC_SRCS ${ELFVIEW_MOC_HEADERS})
+Qt6_ADD_RESOURCES(QT_RES_SOURCES ${QT_RESOURCES})
+Qt6_WRAP_UI(ELFVIEW_UI_HEADERS ${ELFVIEW_UI})
+Qt6_WRAP_CPP(ELFVIEW_MOC_SRCS ${ELFVIEW_MOC_HEADERS})
 
 add_executable(ElfView WIN32 ${ELFVIEW_SRC} ${ELFVIEW_MOC_SRCS} ${QT_RES_SOURCES} ${ELFVIEW_UI_HEADERS})
 target_link_libraries(ElfView PRIVATE ${PROJECT_LIBS})
