diff --git a/sway/ipc-server.c b/sway/ipc-server.c
index 00b01d7d..8f60cfbf 100644
--- a/sway/ipc-server.c
+++ b/sway/ipc-server.c
@@ -140,7 +140,7 @@ struct sockaddr_un *ipc_user_sockaddr(void) {
 		dir = "/tmp";
 	}
 	if (path_size <= snprintf(ipc_sockaddr->sun_path, path_size,
-			"%s/sway-ipc.%u.%i.sock", dir, getuid(), getpid())) {
+			"%s/sway-ipc.%u.sock", dir, getuid())) {
 		sway_abort("Socket path won't fit into ipc_sockaddr->sun_path");
 	}
 
diff --git a/sway/main.c b/sway/main.c
index 85bc2f1c..2cf38556 100644
--- a/sway/main.c
+++ b/sway/main.c
@@ -151,6 +151,11 @@ static void log_kernel(void) {
 }
 
 static bool detect_suid(void) {
+	if (getuid() == 0 || getgid() == 0) {
+		sway_log(SWAY_INFO, "Running sway as root user");
+		return true;
+	}
+
 	if (geteuid() != 0 && getegid() != 0) {
 		return false;
 	}
