From a56abbc14ab8223976726c1e25553ee65d1746a4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Santiago=20C=C3=A9zar?= <santiagocezar2013@gmail.com>
Date: Sat, 10 Sep 2022 12:36:53 -0300
Subject: [PATCH] makes the scale setting work correctly

---
 SonicMania/Objects/Menu/UIWinSize.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/SonicMania/Objects/Menu/UIWinSize.c b/SonicMania/Objects/Menu/UIWinSize.c
index 0e7faad6..82baac7c 100644
--- a/SonicMania/Objects/Menu/UIWinSize.c
+++ b/SonicMania/Objects/Menu/UIWinSize.c
@@ -132,9 +132,9 @@ void UIWinSize_SetupText(EntityUIWinSize *entityPtr)
 
         self->maxScale = height / SCREEN_YSIZE;
         if (self->selection < 1)
-            self->selection = self->maxScale - 1;
+            self->selection = self->maxScale;
 
-        if (self->selection >= self->maxScale)
+        if (self->selection > self->maxScale)
             self->selection = 1;
 
         char buffer[0x10];
