Submodule dependencies/RSDKv5 contains modified content
diff --git a/dependencies/RSDKv5/makefiles/Linux.cfg b/dependencies/RSDKv5/makefiles/Linux.cfg
index 95a5430..50071bb 100644
--- a/dependencies/RSDKv5/makefiles/Linux.cfg
+++ b/dependencies/RSDKv5/makefiles/Linux.cfg
@@ -30,6 +30,6 @@ ifeq ($(SUBSYSTEM),SDL2)
     RSDK_LIBS += `$(PKGCONFIG) --libs --static sdl2`
 endif
 
-RSDK_CFLAGS += `$(PKGCONFIG) --cflags --static theora theoradec zlib portaudio`
-RSDK_LIBS += `$(PKGCONFIG) --libs --static theora theoradec zlib portaudio`
+RSDK_CFLAGS += `$(PKGCONFIG) --cflags --static theora theoradec zlib`
+RSDK_LIBS += `$(PKGCONFIG) --libs --static theora theoradec zlib`
 
