diff --git a/libretro/libretro-core.h b/libretro/libretro-core.h
index 0680699..9e8cffa 100644
--- a/libretro/libretro-core.h
+++ b/libretro/libretro-core.h
@@ -3,6 +3,7 @@
 
 #include <stdint.h>
 #include <string.h>
+#undef __USE_ISOC99
 #include <math.h>
 #include <stdlib.h>
 #include <stdio.h>
diff --git a/sources/src/audio.c b/sources/src/audio.c
index b028760..f72f031 100644
--- a/sources/src/audio.c
+++ b/sources/src/audio.c
@@ -44,6 +44,7 @@
 #endif
 #include "threaddep/thread.h"
 
+#undef __USE_ISOC99
 #include <math.h>
 
 #define DEBUG_AUDIO 0
diff --git a/sources/src/custom.c b/sources/src/custom.c
index 4f7cc41..57ce01e 100644
--- a/sources/src/custom.c
+++ b/sources/src/custom.c
@@ -13,6 +13,7 @@
 
 #include <ctype.h>
 #include <assert.h>
+#undef __USE_ISOC99
 #include <math.h>
 
 #include "options.h"
diff --git a/sources/src/gfxutil.c b/sources/src/gfxutil.c
index 44b9720..aca0f0c 100644
--- a/sources/src/gfxutil.c
+++ b/sources/src/gfxutil.c
@@ -15,6 +15,7 @@
 #include "gfxfilter.h"
 #include "machdep/maccess.h"
 
+#undef __USE_ISOC99
 #include <math.h>
 
 float getvsyncrate(int monid, float hz, int *mult)
