diff -Naur libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6.orig/Makefile.libretro libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6/Makefile.libretro
--- libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6.orig/Makefile.libretro	2021-04-23 17:03:26.000000000 +0100
+++ libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6/Makefile.libretro	2021-06-12 14:20:44.905865177 +0100
@@ -75,16 +75,11 @@
 	endif
 	CC = gcc
 	CXX = gcc
-	LDFLAGS += -I/usr/include/SDL2 -L/usr/lib/x86_64-linux-gnu -lSDL2
-	CFLAGS += -I/usr/include/SDL2 `sdl2-config --cflags`
-	CXXFLAGS += -I/usr/include/SDL2 `sdl2-config --cflags`
 
 	# Raspberry Pi
 # ---FOR TEST ---
 	ifneq (,$(findstring rpi,$(platform)))
 		PLATFORM_DEFINES += -D_RASPBERRYPI
-		CFLAGS += -fomit-frame-pointer -ffast-math -marm
-		CXXFLAGS += $(CFLAGS)
 		ifneq (,$(findstring rpi0,$(platform)))
 			PLATFORM_DEFINES += -D_RASPBERRYPI0
 			PLATFORM_DEFINES += -D_RESOLUTION_AUTO_LOW
@@ -97,14 +92,20 @@
 			PLATFORM_DEFINES += -D_RASPBERRYPI2
 			PLATFORM_DEFINES += -D_RESOLUTION_AUTO_LOW
 			CFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
-		else ifneq (,$(findstring rpi3,$(platform)))
+        # Raspberry Pi 3 (64-bit)
+        else ifeq ($(platform), rpi3_64)
 			PLATFORM_DEFINES += -D_RASPBERRYPI3
 			PLATFORM_DEFINES += -D_RESOLUTION_AUTO_LOW
-			CFLAGS += -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
-		else ifneq (,$(findstring rpi4,$(platform)))
+			CFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53 -ffast-math
+			CFLAGS += -DARM
+		    CXXFLAGS += $(CFLAGS)
+        # Raspberry Pi 4 (64-bit)
+        else ifeq ($(platform), rpi4_64)
 			PLATFORM_DEFINES += -D_RASPBERRYPI4
 			PLATFORM_DEFINES += -D_RESOLUTION_AUTO_LOW
-			CFLAGS += -mcpu=cortex-a72 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+			CFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math
+			CFLAGS += -DARM
+		    CXXFLAGS += $(CFLAGS)
 		else
 			PLATFORM_DEFINES += -D_RASPBERRYPI
 			PLATFORM_DEFINES += -D_RESOLUTION_AUTO_MEDIUM
@@ -731,7 +732,6 @@
 rebuild: cls clean-all build-all
 
 cls:
-	@clear
 
 build-all: $(TARGET)
 
