diff -Naur libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6.orig/Makefile.libretro libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6/Makefile.libretro
--- libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6.orig/Makefile.libretro	2021-04-23 17:03:26.000000000 +0100
+++ libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6/Makefile.libretro	2021-06-19 14:35:26.201675025 +0100
@@ -75,9 +75,9 @@
 	endif
 	CC = gcc
 	CXX = gcc
-	LDFLAGS += -I/usr/include/SDL2 -L/usr/lib/x86_64-linux-gnu -lSDL2
-	CFLAGS += -I/usr/include/SDL2 `sdl2-config --cflags`
-	CXXFLAGS += -I/usr/include/SDL2 `sdl2-config --cflags`
+	LDFLAGS += `pkg-config sdl2 --libs`
+	CFLAGS += -I. `pkg-config sdl2 --cflags`
+	CXXFLAGS += -I. `pkg-config sdl2 --cflags`
 
 	# Raspberry Pi
 # ---FOR TEST ---
@@ -724,14 +724,11 @@
 	endif
 
 
-build: cls build-all
+build: build-all
 
-clean: cls clean-all
+clean: clean-all
 
-rebuild: cls clean-all build-all
-
-cls:
-	@clear
+rebuild: clean-all build-all
 
 build-all: $(TARGET)
 
