diff -Naur libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6.orig/src/emuscv.cpp libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6/src/emuscv.cpp
--- libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6.orig/src/emuscv.cpp	2021-04-23 17:03:26.000000000 +0100
+++ libretro-emuscv-eb14596d4f3cefef9ed178cf4eea7aca992e18a6/src/emuscv.cpp	2021-06-19 15:31:05.199882517 +0100
@@ -769,8 +769,8 @@
 		return FALSE;
 
 	// Create Recalbox logo SDL surface & renderer
-	if(!CreateRecalboxSurface())
-		return FALSE;
+	//if(!CreateRecalboxSurface())
+	//	return FALSE;
 
 	// reset the frame counter
 	retro_frame_counter = 0;
@@ -2189,16 +2189,16 @@
 //			RetroLogPrintf(RETRO_LOG_DEBUG, "[%s] SDL blit scaled secondary->main ok (emuSCV logo)\n", EMUSCV_NAME);
 
 		// Draw Recalbox logo
-		RectSrc.x = 0;
-		RectSrc.y = 0;
-		RectSrc.w = RectDst.w = recalbox_surface->w;
-		RectSrc.h = RectDst.h = recalbox_surface->h;
-		RectDst.x = config.window_space;
-		RectDst.y = config.window_height - config.window_space - RectDst.h;
-		SDL_SetSurfaceBlendMode(recalbox_surface, SDL_BLENDMODE_BLEND);
-		SDL_SetSurfaceAlphaMod(recalbox_surface, alpha);
-		if(SDL_BlitScaled(recalbox_surface, &RectSrc, frame_surface, &RectDst) != 0)
-			RetroLogPrintf(RETRO_LOG_ERROR, "[%s] SDL blit scaled secondary->main failed! (Recalbox logo) %s\n", EMUSCV_NAME, SDL_GetError());
+		//RectSrc.x = 0;
+		//RectSrc.y = 0;
+		//RectSrc.w = RectDst.w = recalbox_surface->w;
+		//RectSrc.h = RectDst.h = recalbox_surface->h;
+		//RectDst.x = config.window_space;
+		//RectDst.y = config.window_height - config.window_space - RectDst.h;
+		//SDL_SetSurfaceBlendMode(recalbox_surface, SDL_BLENDMODE_BLEND);
+		//SDL_SetSurfaceAlphaMod(recalbox_surface, alpha);
+		//if(SDL_BlitScaled(recalbox_surface, &RectSrc, frame_surface, &RectDst) != 0)
+		//	RetroLogPrintf(RETRO_LOG_ERROR, "[%s] SDL blit scaled secondary->main failed! (Recalbox logo) %s\n", EMUSCV_NAME, SDL_GetError());
 //		else
 //			RetroLogPrintf(RETRO_LOG_DEBUG, "[%s] SDL blit scaled secondary->main ok (Recalbox logo)\n", EMUSCV_NAME);
 	}
@@ -3307,7 +3307,7 @@
 	CreateLogoSurface();
 
 	// Create Recalbox logo SDL surface & renderer
-	CreateRecalboxSurface();
+	// CreateRecalboxSurface();
 
 //	RetroLogPrintf(RETRO_LOG_DEBUG, "[%s] apply_display_config()\n", EMUSCV_NAME);
 }
