diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1abb7d7cbf..c54cf65579 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -556,14 +556,7 @@ if (_M_X86)
   add_subdirectory(Externals/Bochs_disasm)
 endif()
 add_subdirectory(Externals/cpp-optparse)
-find_package(fmt 6.0)
-if(fmt_FOUND)
-  message(STATUS "Using shared fmt ${fmt_VERSION}")
-else()
-  check_vendoring_approved(fmt)
-  message(STATUS "Using static fmt from Externals")
-  add_subdirectory(Externals/fmt EXCLUDE_FROM_ALL)
-endif()
+add_subdirectory(Externals/fmt)
 add_subdirectory(Externals/glslang)
 add_subdirectory(Externals/imgui)
 
