diff --git a/Source/Core/Core/IOS/FS/FileSystemProxy.cpp b/Source/Core/Core/IOS/FS/FileSystemProxy.cpp
index 181a739748..1e8f2af7ba 100644
--- a/Source/Core/Core/IOS/FS/FileSystemProxy.cpp
+++ b/Source/Core/Core/IOS/FS/FileSystemProxy.cpp
@@ -66,7 +66,7 @@ template <typename T, typename... Args>
 static void LogResult(const Result<T>& result, std::string_view format, Args&&... args)
 {
   const auto result_code = result.Succeeded() ? ResultCode::Success : result.Error();
-  LogResult(result_code, format, std::forward<Args>(args)...);
+  //LogResult(result_code, format, std::forward<Args>(args)...);
 }
 
 enum class FileLookupMode
