diff --git a/startx.in b/startx.in
index 9b224af..c8cd327 100644
--- a/startx.in
+++ b/startx.in
@@ -115,7 +115,7 @@ if [ "$(uname -s)" = "Darwin" ] ; then
     fi
 
     if [ "$(defaults read $X11_PREFS_DOMAIN no_auth)" = 0 ] ; then
-        enable_xauth=1
+        enable_xauth=0
     else
         enable_xauth=0
     fi
@@ -139,7 +139,7 @@ if [ "$(uname -s)" = "Darwin" ] ; then
     fi
 
 else
-    enable_xauth=1
+    enable_xauth=0
 fi
 
 # Automatically determine an unused $DISPLAY
