diff --git a/scripts/toolchain.lua b/scripts/toolchain.lua
index a8e58d287b3..c0d00675705 100644
--- a/scripts/toolchain.lua
+++ b/scripts/toolchain.lua
@@ -614,6 +614,11 @@ function toolchain(_buildDir, _subDir)
 	configuration { "osx*", "arm64", "Debug" }
 		targetdir (_buildDir .. "osx_clang" .. "/bin/x64/Debug")
 
+	configuration { "linux-*" }
+		flags {
+			"NoPCH",
+		}
+
 	configuration {} -- reset configuration
 
 	return true
