diff --git a/scripts/toolchain.lua b/scripts/toolchain.lua
index ed200e58ba9..8d4348f8be8 100644
--- a/scripts/toolchain.lua
+++ b/scripts/toolchain.lua
@@ -580,6 +580,11 @@ function toolchain(_buildDir, _subDir)
 	configuration { "osx*", "arm64", "Debug" }
 		targetdir (_buildDir .. "osx_clang" .. "/bin/x64/Debug")
 
+	configuration { "linux-*" }
+		flags {
+			"NoPCH",
+		}
+
 	configuration {} -- reset configuration
 
 	return true
