diff --git a/scripts/toolchain.lua b/scripts/toolchain.lua
index ab6630698b9..9f01cc930b6 100644
--- a/scripts/toolchain.lua
+++ b/scripts/toolchain.lua
@@ -619,6 +619,11 @@ function toolchain(_buildDir, _subDir)
 	configuration { "osx*", "arm64", "Debug" }
 		targetdir (_buildDir .. "osx_clang" .. "/bin/x64/Debug")
 
+	configuration { "linux-*" }
+		flags {
+			"NoPCH",
+		}
+
 	configuration {} -- reset configuration
 
 	return true
