diff --git a/Makefile.libretro b/Makefile.libretro
index 06c51b6..5cd3d16 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -157,6 +157,13 @@ ifneq (,$(findstring unix,$(platform)))
       PLATFORM_DEFINES += -mfloat-abi=hard -mfpu=neon-fp-armv8
    endif
 
+   # Odroid-N2
+   ifneq (,$(findstring odin,$(platform)))
+      ENDIANNESS_DEFINES += -DALIGN_LONG
+      CFLAGS += -fomit-frame-pointer -ffast-math
+      PLATFORM_DEFINES += -DARM -march=armv8-a+crc -mcpu=cortex-a75 -mtune=cortex-a75.cortex-a55
+   endif
+
    # Raspberry Pi
    ifneq (,$(findstring rpi,$(platform)))
       ENDIANNESS_DEFINES += -DALIGN_LONG
