diff --git a/CMakeLists.txt b/CMakeLists.txt
index 32d22f4d..4e46b603 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -529,8 +529,7 @@ endif()
 
 if(ENABLE_VULKAN)
     find_package(
-        Vulkan REQUIRED
-        COMPONENTS glslang
+        glslang REQUIRED
     )
 
     set(RENDERER_VK_INCLUDE_FILES include/renderer_vk/renderer_vk.hpp
@@ -588,7 +587,7 @@ if(ENABLE_VULKAN)
 
     target_sources(AlberCore PRIVATE ${RENDERER_VK_SOURCE_FILES})
     target_compile_definitions(AlberCore PUBLIC "PANDA3DS_ENABLE_VULKAN=1")
-    target_link_libraries(AlberCore PRIVATE Vulkan::Vulkan resources_renderer_vk)
+    target_link_libraries(AlberCore PRIVATE glslang resources_renderer_vk)
 endif()
 
 if(ENABLE_METAL AND APPLE)
