diff --git a/src/core/renderer_vk/renderer_vk.cpp b/src/core/renderer_vk/renderer_vk.cpp
index 14bde2c8..57533bde 100644
--- a/src/core/renderer_vk/renderer_vk.cpp
+++ b/src/core/renderer_vk/renderer_vk.cpp
@@ -887,7 +887,7 @@ void RendererVK::display() {
 
 // DynamicLoader is in a different namespace in different versions of Vulkan-Hpp
 #if VK_HEADER_VERSION >= 301
-using VulkanDynamicLoader = vk::detail::DynamicLoader
+using VulkanDynamicLoader = vk::detail::DynamicLoader;
 #else
 using VulkanDynamicLoader = vk::DynamicLoader;
 #endif
