diff --git a/arch/Kconfig b/arch/Kconfig
index 9b087f9bb413..e623942af750 100644
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -14,7 +14,7 @@ config ARCH_CONFIGURES_CPU_MITIGATIONS
 
 if !ARCH_CONFIGURES_CPU_MITIGATIONS
 config CPU_MITIGATIONS
-	def_bool y
+	def_bool n
 endif
 
 #
diff --git a/kernel/cpu.c b/kernel/cpu.c
index ad755db29efd..fa16baed93d2 100644
--- a/kernel/cpu.c
+++ b/kernel/cpu.c
@@ -3217,11 +3217,11 @@ bool cpu_mitigations_auto_nosmt(void)
 	return cpu_mitigations == CPU_MITIGATIONS_AUTO_NOSMT;
 }
 EXPORT_SYMBOL_GPL(cpu_mitigations_auto_nosmt);
-#else
+/*#else
 static int __init mitigations_parse_cmdline(char *arg)
 {
 	pr_crit("Kernel compiled without mitigations, ignoring 'mitigations'; system may still be vulnerable\n");
 	return 0;
-}
+}*/
 #endif
 early_param("mitigations", mitigations_parse_cmdline);
