diff -Naur retro8-1e7d61b23a6d81bb85f0b79fc14881d643a02bb5.orig/src/common.h retro8-1e7d61b23a6d81bb85f0b79fc14881d643a02bb5/src/common.h
--- retro8-1e7d61b23a6d81bb85f0b79fc14881d643a02bb5.orig/src/common.h	2021-03-28 13:14:17.000000000 +0100
+++ retro8-1e7d61b23a6d81bb85f0b79fc14881d643a02bb5/src/common.h	2021-07-24 15:03:29.434272405 +0100
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <stdio.h>
+#include <stdlib.h>
 #include <cstdint>
 #include <type_traits>
 #include <vector>
