diff --git a/Src/OSD/Unix/FileSystemPath.cpp b/Src/OSD/Unix/FileSystemPath.cpp
index f30bb04..2d14742 100644
--- a/Src/OSD/Unix/FileSystemPath.cpp
+++ b/Src/OSD/Unix/FileSystemPath.cpp
@@ -56,30 +56,30 @@ namespace FileSystemPath
         switch (pathType)
         {
         case Analysis:
-            strPathType = "Analysis";
+            strPathType = "/userdata/system/configs/supermodel/analysis";
             break;
         case Config:
-            strPathType = "Config";
+            strPathType = "/userdata/system/configs/supermodel";
             break;
         case Log:
-            strPathType = "Log";
+            strPathType = "/userdata/system/logs";
             break;
         case NVRAM:
-            strPathType = "NVRAM";
+            strPathType = "/userdata/saves/supermodel/NVRAM";
             break;
         case Saves:
-            strPathType = "Saves";
+            strPathType = "/userdata/saves/supermodel";
             break;
         case Screenshots:
-            strPathType = "Screenshots";
+            strPathType = "/userdata/screenshots/supermodel";
             break;
         case Assets:
-            strPathType = "Assets/";
+            strPathType = "/userdata/system/configs/supermodel/assets";
             break;
         }
 
         // Get user's HOME directory
-        if (pwd)
+        /*if (pwd)
         {
             homePath = pwd->pw_dir;
         }
@@ -98,13 +98,13 @@ namespace FileSystemPath
             }
             else
             {
-                // If directory doesn't exist, create it
+                // If directory doesn't exist, create it*/
                 if (!FileSystemPath::PathExists(strPathType))
                 {
                     mkdir(strPathType.c_str(), 0775);
                 }
                 finalPath = strPathType;
-            }
+            /*}
         }
         // Check if $HOME/.supermodel exists
         else if (FileSystemPath::PathExists(Util::Format() << homePath << "/.supermodel"))
@@ -152,7 +152,7 @@ namespace FileSystemPath
                 }
             }
             
-        }
+        }*/
         
         if (finalPath != "")
         {
