diff --git a/src/xenia/app/xenia_main.cc b/src/xenia/app/xenia_main.cc
index 4bc271e9a..92b149405 100644
--- a/src/xenia/app/xenia_main.cc
+++ b/src/xenia/app/xenia_main.cc
@@ -583,7 +583,7 @@ bool EmulatorApp::OnInitialize() {
 #if XE_PLATFORM_ANDROID
       // TODO(Triang3l): Point to the app's external storage "files" directory.
 #else
-      storage_root = storage_root / "Xenia";
+      storage_root = "/userdata/system/configs/xenia_edge";
 #endif
     }
   }
