diff --git a/drivers/hid/hid-steam.c b/drivers/hid/hid-steam.c
index aa88cedff905..55164d0b31aa 100644
--- a/drivers/hid/hid-steam.c
+++ b/drivers/hid/hid-steam.c
@@ -971,7 +971,9 @@ static int steam_register(struct steam_device *steam)
 
 		hid_info(steam->hdev, "Steam Controller '%s' connected",
 				steam->serial_no);
-
+		/* start in gamepad mode */
+		steam->gamepad_mode = true;
+		steam_set_lizard_mode(steam, false);
 		/* ignore battery errors, we can live without it */
 		if (steam->quirks & STEAM_QUIRK_WIRELESS)
 			steam_battery_register(steam);
@@ -1610,10 +1612,10 @@ static void steam_do_deck_input_event(struct steam_device *steam,
 	b13 = data[13];
 	b14 = data[14];
 
-	if (!(b9 & BIT(6)) && steam->did_mode_switch) {
+	if (!(b9 & BIT(5)) && !(b14 & BIT(2)) && steam->did_mode_switch) {
 		steam->did_mode_switch = false;
 		cancel_delayed_work(&steam->mode_switch);
-	} else if (!steam->client_opened && (b9 & BIT(6)) && !steam->did_mode_switch) {
+	} else if (!steam->client_opened && (b9 & BIT(5)) && (b14 & BIT(2)) && !steam->did_mode_switch) {
 		steam->did_mode_switch = true;
 		schedule_delayed_work(&steam->mode_switch, 45 * HZ / 100);
 	}
