diff --git a/nanocoat/frontend/CMakeLists.txt b/nanocoat/frontend/CMakeLists.txt
index 9d63089f60..2d1ec0926f 100644
--- a/nanocoat/frontend/CMakeLists.txt
+++ b/nanocoat/frontend/CMakeLists.txt
@@ -7,7 +7,7 @@
 # ---------------------------------------------------------------------------
 
 # SquirrelJME Java SE Emulator Front-End
-if(NOT EMSCRIPTEN AND
+if(NOT EMSCRIPTEN AND NOT SQUIRRELJME_ENABLE_FRONTEND_LIBRETRO AND
 	NOT SQUIRRELJME_IS_CROSS_COMPILE AND
 	SQUIRRELJME_ENABLE_DYLIB)
 	squirreljme_include("jni.cmake")
