diff --git a/dlls/appwiz.cpl/appwiz.c b/dlls/appwiz.cpl/appwiz.c
index 6f95c23bf28..d5e1c6588ed 100644
--- a/dlls/appwiz.cpl/appwiz.c
+++ b/dlls/appwiz.cpl/appwiz.c
@@ -952,12 +952,10 @@ static LONG start_params(const WCHAR *params)
         return FALSE;
 
     if(!wcscmp(params, L"install_gecko")) {
-        install_addon(ADDON_GECKO);
         return TRUE;
     }
 
     if(!wcscmp(params, L"install_mono")) {
-        install_addon(ADDON_MONO);
         return TRUE;
     }
 
