diff --git a/yabause/src/libretro/libretro.c b/yabause/src/libretro/libretro.c
index 93dbe3b4c..38afb17b4 100644
--- a/yabause/src/libretro/libretro.c
+++ b/yabause/src/libretro/libretro.c
@@ -988,7 +988,7 @@ void check_variables(void)
       else if (strcmp(var.value, "4k") == 0)
       {
          resolution_mode = RES_NATIVE;
-         native_resolution_mode = N_RES_4k;
+         native_resolution_mode = N_RES_4X;
       }
       else if (strcmp(var.value, "8k") == 0)
       {
@@ -1141,7 +1141,6 @@ void retro_get_system_av_info(struct retro_system_av_info *info)
       {
          case RES_ORIGINAL:
          case RES_SD:
-         case RES_1X:
             window_width = 704;
             window_height = 512;
             break;
