diff --git a/src/r8168_n.c b/src/r8168_n.c
index 19ae62b..32e89fa 100755
--- a/src/r8168_n.c
+++ b/src/r8168_n.c
@@ -99,6 +99,10 @@
 #include <linux/seq_file.h>
 #endif
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 8, 0))
+#define strlcpy strscpy
+#endif
+
 #define FIRMWARE_8168D_1    "rtl_nic/rtl8168d-1.fw"
 #define FIRMWARE_8168D_2    "rtl_nic/rtl8168d-2.fw"
 #define FIRMWARE_8168E_1    "rtl_nic/rtl8168e-1.fw"
