diff --git a/src/common/fs/path_util.cpp b/src/common/fs/path_util.cpp
index a6f2593f0..f3335dc62 100644
--- a/src/common/fs/path_util.cpp
+++ b/src/common/fs/path_util.cpp
@@ -115,21 +115,21 @@ public:
         }
 #endif
 
-        GenerateCitronPath(CitronPath::CitronDir, citron_path);
-        GenerateCitronPath(CitronPath::AmiiboDir, citron_path / AMIIBO_DIR);
-        GenerateCitronPath(CitronPath::CacheDir, citron_path_cache);
-        GenerateCitronPath(CitronPath::ConfigDir, citron_path_config);
+        GenerateCitronPath(CitronPath::CitronDir, "/userdata/saves/switch/yuzu");
+        GenerateCitronPath(CitronPath::AmiiboDir, "/userdata/bios/switch/amiibo");
+        GenerateCitronPath(CitronPath::CacheDir, "/userdata/system/cache/yuzu");
+        GenerateCitronPath(CitronPath::ConfigDir, "/userdata/system/configs/yuzu");
         GenerateCitronPath(CitronPath::CrashDumpsDir, citron_path / CRASH_DUMPS_DIR);
-        GenerateCitronPath(CitronPath::DumpDir, citron_path / DUMP_DIR);
-        GenerateCitronPath(CitronPath::KeysDir, citron_path / KEYS_DIR);
-        GenerateCitronPath(CitronPath::LoadDir, citron_path / LOAD_DIR);
-        GenerateCitronPath(CitronPath::LogDir, citron_path / LOG_DIR);
-        GenerateCitronPath(CitronPath::NANDDir, citron_path / NAND_DIR);
+        GenerateCitronPath(CitronPath::DumpDir, "/userdata/saves/switch/yuzu/dump");
+        GenerateCitronPath(CitronPath::KeysDir, "/userdata/bios/switch/keys");
+        GenerateCitronPath(CitronPath::LoadDir, "/userdata/saves/switch/yuzu/load");
+        GenerateCitronPath(CitronPath::LogDir, "/userdata/system/logs");
+        GenerateCitronPath(CitronPath::NANDDir, "/userdata/bios/switch/nand");
         GenerateCitronPath(CitronPath::PlayTimeDir, citron_path / PLAY_TIME_DIR);
-        GenerateCitronPath(CitronPath::ScreenshotsDir, citron_path / SCREENSHOTS_DIR);
-        GenerateCitronPath(CitronPath::SDMCDir, citron_path / SDMC_DIR);
-        GenerateCitronPath(CitronPath::ShaderDir, citron_path / SHADER_DIR);
-        GenerateCitronPath(CitronPath::TASDir, citron_path / TAS_DIR);
+        GenerateCitronPath(CitronPath::ScreenshotsDir, "/userdata/screenshots/switch");
+        GenerateCitronPath(CitronPath::SDMCDir, "/userdata/bios/switch/sdmc");
+        GenerateCitronPath(CitronPath::ShaderDir, "/userdata/saves/switch/yuzu/shaders");
+        GenerateCitronPath(CitronPath::TASDir, "/userdata/saves/switch/yuzu/tas");
         GenerateCitronPath(CitronPath::IconsDir, citron_path / ICONS_DIR);
     }
 
