diff -Naur suyu.orig/src/input_common/drivers/udp_client.cpp suyu/src/input_common/drivers/udp_client.cpp
--- suyu.orig/src/input_common/drivers/udp_client.cpp	2024-12-20 19:59:21.320042088 +0000
+++ suyu/src/input_common/drivers/udp_client.cpp	2024-12-20 19:55:06.440481580 +0000
@@ -113,7 +113,7 @@
     }
 
     SocketCallback callback;
-    boost::asio::io_service io_service;
+    boost::asio::io_context io_service;
     boost::asio::basic_waitable_timer<clock> timer;
     udp::socket socket;
 
