diff --git a/src/core/opengl/render_gl_backend.cpp b/src/core/opengl/render_gl_backend.cpp
index 8b9fa50b..46f61f9f 100644
--- a/src/core/opengl/render_gl_backend.cpp
+++ b/src/core/opengl/render_gl_backend.cpp
@@ -37,6 +37,10 @@
 #endif
 
 
+#ifndef GL_STREAM_DRAW
+#define GL_STREAM_DRAW 0x88E0
+#endif
+
 // arguments
 constexpr bool s_prefer_fb_copyTex = false;
 static int s_num_pass = 2;
diff --git a/src/core/opengl/render_gl_frontend.cpp b/src/core/opengl/render_gl_frontend.cpp
index 7ee763ab..7a69f223 100644
--- a/src/core/opengl/render_gl_frontend.cpp
+++ b/src/core/opengl/render_gl_frontend.cpp
@@ -404,12 +404,13 @@ void RenderGL::close()
             m_vertex_buffer[i] = 0;
     }
 
+#ifdef RENDERGL_HAS_SHADERS
     if(m_light_ubo)
     {
         glDeleteBuffers(1, &m_light_ubo);
         m_light_ubo = 0;
     }
-
+#endif
 #endif
 
 #ifdef RENDERGL_HAS_VAO
