diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 567e35a7..d562c1d1 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -71,7 +71,8 @@ AddReg=\
     Printing,\
     Services, \
     SessionMgr,\
-    ThemeManager
+    ThemeManager,\
+    WineBatoOverrides
 
 [DefaultInstall]
 Needs=BaseInstall
@@ -120,7 +121,8 @@ AddReg=\
     LicenseInformation,\
     MCI,\
     Misc,\
-    VersionInfo
+    VersionInfo,\
+    WineBatoOverrides
 
 [Wow64Install.ntx86]
 Needs=BaseWow64Install
@@ -1581,3 +1583,9 @@ InfFiles  = 17
 NlsFiles  = 11
 SortFiles = 10,globalization\sorting
 WinmdFiles = 11,winmetadata
+
+[WineBatoOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+HKCU,Software\Xenia,"XEFLAGS",0x10001,"1"
+
