diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 3b1d3b415ff..53d24044075 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -99,7 +99,8 @@ AddReg=\
     ThemeManager,\
     VCRuntime.ntx86,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.ntamd64]
 RegisterDlls=RegisterDllsSection
@@ -126,7 +127,8 @@ AddReg=\
     ThemeManager,\
     VCRuntime.ntamd64,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.ntarm64]
 RegisterDlls=RegisterDllsSection
@@ -153,7 +155,8 @@ AddReg=\
     VCRuntime.ntamd64,\
     VCRuntime.ntarm64,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [Wow64Install.ntx86]
 RegisterDlls=RegisterDllsSection
@@ -187,6 +190,10 @@ AddReg=\
     Tapi,\
     VersionInfo,\
     LicenseInformation
+    
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
 
 [DefaultInstall.Services]
 AddService=BITS,0,BITSService
