diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 2578d64aa0d..764ab1d39cd 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -97,7 +97,8 @@ AddReg=\
     ThemeManager,\
     VCRuntime.ntx86,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.ntamd64]
 RegisterDlls=RegisterDllsSection
@@ -123,7 +124,8 @@ AddReg=\
     ThemeManager,\
     VCRuntime.ntamd64,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.ntarm64]
 RegisterDlls=RegisterDllsSection
@@ -150,7 +152,8 @@ AddReg=\
     VCRuntime.ntamd64,\
     VCRuntime.ntarm64,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [Wow64Install.ntx86]
 RegisterDlls=RegisterDllsSection
@@ -183,6 +186,10 @@ AddReg=\
     Tapi,\
     VersionInfo,\
     LicenseInformation
+    
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
 
 [DefaultInstall.Services]
 AddService=BITS,0,BITSService
