diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 68415d7977e..8a8e56d633a 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -84,13 +84,15 @@ Needs=BaseInstall
 WineFakeDlls=FakeDllsWin32
 AddReg=\
     VCRuntime.ntx86,\
-    VersionInfo
-
+    VersionInfo,\
+    BatoceraOverrides
+ 
 [DefaultInstall.ntarm]
 Needs=BaseInstall
 WineFakeDlls=FakeDllsWin32
 AddReg=\
-    VersionInfo
+    VersionInfo,\
+    BatoceraOverrides
 
 [DefaultInstall.ntamd64]
 Needs=BaseInstall
@@ -98,7 +100,8 @@ WineFakeDlls=FakeDllsWin64
 AddReg=\
     CurrentVersionWow64.ntamd64,\
     VCRuntime.ntamd64,\
-    VersionInfo
+    VersionInfo,\
+    BatoceraOverrides
 
 [DefaultInstall.ntarm64]
 Needs=BaseInstall
@@ -107,7 +110,13 @@ AddReg=\
     CurrentVersionWow64.ntarm64,\
     VCRuntime.ntamd64,\
     VCRuntime.ntarm64,\
-    VersionInfo
+    VersionInfo,\
+    BatoceraOverrides
+
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+HKCU,Software\Xenia,"XEFLAGS",0x10001,"1"
 
 [BaseWow64Install]
 RegisterDlls=RegisterDllsSection
