diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index b29a9790676..293f9b6905c 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -4921,8 +4921,7 @@ BOOL WINAPI __wine_needs_override_large_address_aware(void)
 
 static BOOL is_large_address_aware(void)
 {
-    return (main_image_info.ImageCharacteristics & IMAGE_FILE_LARGE_ADDRESS_AWARE)
-           || __wine_needs_override_large_address_aware();
+    return __wine_needs_override_large_address_aware();
 }
 
 /***********************************************************************
