diff --git a/CMakeLists.txt b/CMakeLists.txt
index df7da6311..fa2ec9a3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -510,11 +510,11 @@ endif()
 
 if(ENABLE_LAUNCHER OR ENABLE_EDITOR)
 	# Widgets finds its own dependencies (QtGui and QtCore).
-	find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Network)
-	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Network)
+	find_package(Qt6 REQUIRED COMPONENTS Widgets Network)
+	find_package(Qt6 REQUIRED COMPONENTS Widgets Network)
 
 	if(ENABLE_TRANSLATIONS)
-		find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS LinguistTools)
+		find_package(Qt6 REQUIRED COMPONENTS LinguistTools)
 		add_definitions(-DENABLE_QT_TRANSLATIONS)
 	endif()
 endif()
@@ -748,7 +748,7 @@ if(WIN32)
 	endif()
 
 	if(ENABLE_LAUNCHER OR ENABLE_EDITOR)
-		get_target_property(QtCore_location Qt${QT_VERSION_MAJOR}::Core LOCATION)
+		get_target_property(QtCore_location Qt6::Core LOCATION)
 		get_filename_component(Qtbin_folder ${QtCore_location} PATH)
 		file(GLOB dep_files
 			${dep_files}
