diff --git a/lib/VCMIDirs.cpp b/lib/VCMIDirs.cpp
index d8f630da2..8ac3bfe77 100644
--- a/lib/VCMIDirs.cpp
+++ b/lib/VCMIDirs.cpp
@@ -542,7 +542,7 @@ bfs::path VCMIDirsXDG::userDataPath() const
 	if((homeDir = getenv("XDG_DATA_HOME")))
 		return bfs::path(homeDir) / "vcmi";
 	else if((homeDir = getenv("HOME")))
-		return bfs::path(homeDir) / ".local" / "share" / "vcmi";
+		return "/userdata/roms/ports/vcmi";
 	else
 		return ".";
 }
@@ -564,7 +564,7 @@ bfs::path VCMIDirsXDG::userConfigPath() const
 	if ((tempResult = getenv("XDG_CONFIG_HOME")))
 		return bfs::path(tempResult) / "vcmi";
 	else if ((tempResult = getenv("HOME")))
-		return bfs::path(tempResult) / ".config" / "vcmi";
+		return "/userdata/roms/ports/vcmi";
 	else
 		return ".";
 }
