diff --git a/lib/VCMIDirs.cpp b/lib/VCMIDirs.cpp
index 1e8dc49ac..86f218412 100644
--- a/lib/VCMIDirs.cpp
+++ b/lib/VCMIDirs.cpp
@@ -600,7 +600,7 @@ bfs::path VCMIDirsXDG::userDataPath() const
 	if((homeDir = getenv("XDG_DATA_HOME")))
 		return bfs::path(homeDir) / "vcmi";
 	else if((homeDir = getenv("HOME")))
-		return bfs::path(homeDir) / ".local" / "share" / "vcmi";
+		return "/userdata/roms/ports/vcmi";
 	else
 		return ".";
 }
@@ -624,7 +624,7 @@ bfs::path VCMIDirsXDG::userConfigPath() const
 	
 	tempResult = getenv("HOME");
 	if (tempResult)
-		return bfs::path(tempResult) / ".config" / "vcmi";
+		return "/userdata/roms/ports/vcmi";
 
 	return ".";
 }
