diff --git a/lib/VCMIDirs.cpp b/lib/VCMIDirs.cpp
index b6e7aeb06..778f91e99 100644
--- a/lib/VCMIDirs.cpp
+++ b/lib/VCMIDirs.cpp
@@ -542,7 +542,7 @@ bfs::path VCMIDirsXDG::userDataPath() const
 	if((homeDir = getenv("XDG_DATA_HOME")))
 		return bfs::path(homeDir) / "vcmi";
 	else if((homeDir = getenv("HOME")))
-		return bfs::path(homeDir) / ".local" / "share" / "vcmi";
+		return "/userdata/roms/ports/vcmi";
 	else
 		return ".";
 }
@@ -566,7 +566,7 @@ bfs::path VCMIDirsXDG::userConfigPath() const
 	
 	tempResult = getenv("HOME");
 	if (tempResult)
-		return bfs::path(tempResult) / ".config" / "vcmi";
+		return "/userdata/roms/ports/vcmi";
 
 	return ".";
 }
