diff --git a/config/schemas/settings.json b/config/schemas/settings.json
index 2975c852f..5a1728134 100644
--- a/config/schemas/settings.json
+++ b/config/schemas/settings.json
@@ -87,7 +87,7 @@
 						"width"  : { "type" : "number" },
 						"height" : { "type" : "number" }
 					},
-					"default": {"width" : 800, "height": 600 }
+					"default": {"width" : 640, "height": 480 }
 				},
 				"bitsPerPixel" : {
 					"type" : "number",
@@ -95,11 +95,11 @@
 				},
 				"fullscreen": {
 					"type": "boolean",
-					"default": false
+					"default": true
 				},
 				"realFullscreen" : {
 					"type" : "boolean",
-					"default" : false
+					"default" : true
 				},
 				"showIntro" : {
 					"type" : "boolean",
diff --git a/lib/VCMIDirs.cpp b/lib/VCMIDirs.cpp
index d8f630da2..8ac3bfe77 100644
--- a/lib/VCMIDirs.cpp
+++ b/lib/VCMIDirs.cpp
@@ -542,7 +542,7 @@ bfs::path VCMIDirsXDG::userDataPath() const
 	if((homeDir = getenv("XDG_DATA_HOME")))
 		return bfs::path(homeDir) / "vcmi";
 	else if((homeDir = getenv("HOME")))
-		return bfs::path(homeDir) / ".local" / "share" / "vcmi";
+		return "/userdata/roms/ports/vcmi";
 	else
 		return ".";
 }
@@ -564,7 +564,7 @@ bfs::path VCMIDirsXDG::userConfigPath() const
 	if ((tempResult = getenv("XDG_CONFIG_HOME")))
 		return bfs::path(tempResult) / "vcmi";
 	else if ((tempResult = getenv("HOME")))
-		return bfs::path(tempResult) / ".config" / "vcmi";
+		return "/userdata/roms/ports/vcmi";
 	else
 		return ".";
 }
