diff --git a/src/emu/qt/CMakeLists.txt b/src/emu/qt/CMakeLists.txt
index 600bf6313..49859cd33 100644
--- a/src/emu/qt/CMakeLists.txt
+++ b/src/emu/qt/CMakeLists.txt
@@ -11,7 +11,7 @@ if(ANDROID)
     endif()
 endif()
 
-find_package(QT NAMES Qt6 Qt5 COMPONENTS Widgets LinguistTools Gui REQUIRED)
+find_package(QT NAMES Qt6 COMPONENTS Widgets LinguistTools Gui REQUIRED)
 
 if(${QT_VERSION_MAJOR} GREATER_EQUAL 6)
     set(EXCLUSIVE_QT_COMPONENTS OpenGLWidgets)
