diff --git a/SP/code/qcommon/q_shared.h b/SP/code/qcommon/q_shared.h
index c2888d36..a86e8227 100644
--- a/SP/code/qcommon/q_shared.h
+++ b/SP/code/qcommon/q_shared.h
@@ -62,7 +62,7 @@ If you have questions concerning this license or the applicable additional terms
 #ifdef USE_XDG
   #define HOMEPATH_NAME_UNIX		"iortcw"
 #else
-  #define HOMEPATH_NAME_UNIX		".wolf"
+  #define HOMEPATH_NAME_UNIX		"iortcw"
 #endif
   #define HOMEPATH_NAME_WIN		"RTCW"
   #define STEAMPATH_NAME		"Return To Castle Wolfenstein"
diff --git a/SP/code/sys/sys_unix.c b/SP/code/sys/sys_unix.c
index 32ba948a..d3191b4a 100644
--- a/SP/code/sys/sys_unix.c
+++ b/SP/code/sys/sys_unix.c
@@ -102,7 +102,7 @@ char *Sys_DefaultHomePath(void)
 				Q_strcat(homePath, sizeof(homePath), HOMEPATH_NAME_UNIX);
 		}
 #else
-		if( ( p1 = getenv( "HOME" ) ) != NULL )
+		if( ( p1 = "/userdata/system/configs/" ) != NULL )
 		{
 			Com_sprintf(homePath, sizeof(homePath), "%s%c", p1, PATH_SEP);
 
