diff --git a/Makefile b/Makefile
index 737701df..a886de9f 100644
--- a/Makefile
+++ b/Makefile
@@ -45,29 +45,29 @@ srcFiles=src/resources/defaultConfig.ini \
 
 # Windows 64-bit target
 Ikemen_GO.exe: ${srcFiles}
-	bash ./build.sh Win64
+	bash ./build/build.sh Win64
 
 # Windows 32-bit target
 Ikemen_GO_x86.exe: ${srcFiles}
-	bash ./build.sh Win32
+	bash ./build/build.sh Win32
 
 # Linux target
 Ikemen_GO_Linux: ${srcFiles}
-	./build.sh Linux
+	./build/build.sh Linux
 
 # Linux ARM target
 Ikemen_GO_LinuxARM: ${srcFiles}
-	./build.sh LinuxARM
+	./build/build.sh LinuxARM
 
 # MacOS x64 target
 Ikemen_GO_MacOS: ${srcFiles}
-	bash ./build.sh MacOS
+	bash ./build/build.sh MacOS
 	$(MAKE) clean_appbundle
 	$(MAKE) appbundle BINNAME=bin/Ikemen_GO_MacOS
 
 # MacOS Apple Silicon target
 Ikemen_GO_MacOSARM: ${srcFiles}
-	bash ./build.sh MacOSARM
+	bash ./build/build.sh MacOSARM
 	$(MAKE) clean_appbundle
 	$(MAKE) appbundle BINNAME=bin/Ikemen_GO_MacOSARM
 
