diff --git a/source/build/src/vfs.cpp b/source/build/src/vfs.cpp
index a6b36170f..f868f57fa 100644
--- a/source/build/src/vfs.cpp
+++ b/source/build/src/vfs.cpp
@@ -286,7 +286,7 @@ int32_t findfrompath(const char *fn, char **where)
     if (!fn || !fn[0])
         return -1;
 
-    if (pathsearchmode)
+    if (1)
     {
         // test unmolested filename first
         if (buildvfs_exists(fn))
diff --git a/source/duke3d/src/common.cpp b/source/duke3d/src/common.cpp
index 0c3376865..ce78a7639 100644
--- a/source/duke3d/src/common.cpp
+++ b/source/duke3d/src/common.cpp
@@ -271,15 +271,7 @@ void G_ExtPreInit(int32_t argc,char const * const * argv)
 
         if ((homedir = Bgethomedir()))
         {
-            Bsnprintf(cwd, ARRAY_SIZE(cwd), "%s/"
-#if defined(_WIN32)
-                APPNAME
-#elif defined(GEKKO)
-                "apps/" APPBASENAME
-#else
-                ".config/" APPBASENAME
-#endif
-                ,homedir);
+            Bsnprintf(cwd, ARRAY_SIZE(cwd), "/userdata/saves/" APPBASENAME);
             asperr = addsearchpath(cwd);
             if (asperr == -2)
             {
diff --git a/source/duke3d/src/game.cpp b/source/duke3d/src/game.cpp
index a3f46b528..eef726722 100644
--- a/source/duke3d/src/game.cpp
+++ b/source/duke3d/src/game.cpp
@@ -6599,7 +6599,7 @@ int app_main(int argc, char const* const* argv)
 
     G_ExtPreInit(argc, argv);
 
-    engineSetLogFile(APPBASENAME ".log", LOG_GAME_MAX);
+    engineSetLogFile("/userdata/system/logs/" APPBASENAME ".log", LOG_GAME_MAX);
     engineSetLogVerbosityCallback(dukeVerbosityCallback);
 
 #ifdef __APPLE__
@@ -6910,7 +6910,7 @@ int app_main(int argc, char const* const* argv)
 
     CONFIG_ReadSettings();
 
-    OSD_Exec("autoexec.cfg");
+    OSD_Exec("/userdata/system/configs/" APPBASENAME "/autoexec.cfg");
 
     CONFIG_SetDefaultKeys(keydefaults, true);
 
diff --git a/source/duke3d/src/startgtk.game.cpp b/source/duke3d/src/startgtk.game.cpp
index 24d8fb29a..761bccee9 100644
--- a/source/duke3d/src/startgtk.game.cpp
+++ b/source/duke3d/src/startgtk.game.cpp
@@ -287,7 +287,7 @@ static unsigned char GetModsDirNames(GtkListStore *list)
         if (buildvfs_exists("user_profiles_disabled"))
             buildvfs_getcwd(pdir, sizeof(pdir));
         else
-            Bsnprintf(pdir, sizeof(pdir), "%s/.config/" APPBASENAME, homedir);
+            Bsnprintf(pdir, sizeof(pdir), "/userdata/saves/" APPBASENAME);
 
         dirs = klistpath(pdir, "*", BUILDVFS_FIND_DIR);
         for (; dirs != NULL; dirs=dirs->next)
