diff --git a/Common.mak b/Common.mak
index e69d1fd06..d593c38a5 100644
--- a/Common.mak
+++ b/Common.mak
@@ -572,8 +572,8 @@ ifndef OPTOPT
         else
             ifeq ($(PLATFORM),DARWIN)
                 OPTOPT := -march=core2 -mmmx -msse -msse2 -msse3 -mssse3
-            else
-                OPTOPT := -march=nocona
+           # else
+           #     OPTOPT := -march=nocona
             endif
         endif
     endif
@@ -600,6 +600,9 @@ ifndef OPTOPT
     ifeq ($(PLATFORM),WII)
         OPTOPT := -mtune=750
     endif
+    ifneq (0,$(RPI4))
+        OPTOPT := -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math
+    endif
 endif
 
 ifeq ($(PACKAGE_REPOSITORY),0)
@@ -888,7 +891,7 @@ endif
 
 ifeq ($(RENDERTYPE),SDL)
     ifeq ($(SDL_TARGET),2)
-        SDLCONFIG := sdl2-config
+        SDLCONFIG := sdl2
         SDLNAME := SDL2
     else ifeq ($(SDL_TARGET),1)
         SDLCONFIG := sdl-config
@@ -937,7 +940,8 @@ ifeq ($(RENDERTYPE),SDL)
             SDLCONFIG_LIBS := $(strip $(subst -mwindows,,$(shell $(SDLCONFIG) --libs)))
 
             COMPILERFLAGS += $(SDLCONFIG_CFLAGS)
-            LIBS += $(SDLCONFIG_LIBS)
+            COMPILERFLAGS += `pkg-config sdl2 --cflags`
+            LIBS += `pkg-config sdl2 --libs`
         else
             ifeq ($(SDL_TARGET),1)
                 COMPILERFLAGS += -D_GNU_SOURCE=1
diff --git a/GNUmakefile b/GNUmakefile
index 43a059af2..7cd753669 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -573,7 +573,7 @@ duke3d_game_deps := audiolib mact
 duke3d_editor_deps := audiolib
 
 duke3d_game := eduke32
-duke3d_editor := mapster32
+duke3d_editor := 
 
 ifneq (,$(APPBASENAME))
     duke3d_game := $(APPBASENAME)
@@ -693,7 +693,7 @@ sw_game_deps := audiolib mact
 sw_editor_deps := audiolib
 
 sw_game := voidsw
-sw_editor := wangulator
+sw_editor :=
 
 sw_game_proper := VoidSW
 sw_editor_proper := Wangulator
@@ -879,7 +879,8 @@ endif
 
 #### Targets
 
-all: duke3d
+all: duke3d \
+     sw
 
 start:
 	$(BUILD_STARTED)
