diff -Naur eduke32.orig/Common.mak eduke32/Common.mak
--- eduke32.orig/Common.mak	2022-01-17 19:31:32.000000000 +0000
+++ eduke32/Common.mak	2022-01-26 21:21:07.054943997 +0000
@@ -575,8 +575,8 @@
         else
             ifeq ($(PLATFORM),DARWIN)
                 OPTOPT := -march=core2 -mmmx -msse -msse2 -msse3 -mssse3
-            else
-                OPTOPT := -march=nocona
+#            else
+#                OPTOPT := -march=nocona
             endif
         endif
     endif
@@ -603,6 +603,9 @@
     ifeq ($(PLATFORM),WII)
         OPTOPT := -mtune=750
     endif
+    ifneq (0,$(RPI4))
+        OPTOPT := -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math
+    endif
 endif
 
 ifeq ($(PACKAGE_REPOSITORY),0)
@@ -897,7 +900,7 @@
 
 ifeq ($(RENDERTYPE),SDL)
     ifeq ($(SDL_TARGET),2)
-        SDLCONFIG := sdl2-config
+        SDLCONFIG := sdl2
         SDLNAME := SDL2
     else ifeq ($(SDL_TARGET),1)
         SDLCONFIG := sdl-config
@@ -939,8 +942,8 @@
             SDLCONFIG_CFLAGS := $(strip $(subst -Dmain=SDL_main,,$(shell $(SDLCONFIG) --cflags)))
             SDLCONFIG_LIBS := $(strip $(subst -mwindows,,$(shell $(SDLCONFIG) --libs)))
 
-            COMPILERFLAGS += $(SDLCONFIG_CFLAGS)
-            LIBS += $(SDLCONFIG_LIBS)
+            COMPILERFLAGS += `pkg-config sdl2 --cflags`
+            LIBS += `pkg-config sdl2 --libs`
         else
             ifeq ($(SDL_TARGET),1)
                 COMPILERFLAGS += -D_GNU_SOURCE=1
diff -Naur eduke32.orig/GNUmakefile eduke32/GNUmakefile
--- eduke32.orig/GNUmakefile	2022-01-17 19:31:32.000000000 +0000
+++ eduke32/GNUmakefile	2022-01-26 21:25:51.021369025 +0000
@@ -653,7 +653,7 @@
 duke3d_editor_deps := audiolib
 
 duke3d_game := eduke32
-duke3d_editor := mapster32
+duke3d_editor :=
 
 ifneq (,$(APPBASENAME))
     duke3d_game := $(APPBASENAME)
@@ -1413,8 +1413,10 @@
 #### Targets
 
 all: \
+    duke3d \
     blood \
     rr \
+    sw \
     exhumed \
 
 start:
