diff --git a/Common.mak b/Common.mak
index 17589b174..d9c6a18e0 100644
--- a/Common.mak
+++ b/Common.mak
@@ -537,9 +537,6 @@ else ifeq ($(PLATFORM),WII)
     LIBDIRS += -L$(LIBOGC_LIB)
 else ifeq ($(PLATFORM),$(filter $(PLATFORM),DINGOO GCW))
     COMPILERFLAGS += -D__OPENDINGUX__
-else ifeq ($(SUBPLATFORM),LINUX)
-    # Locate .so files
-    LINKERFLAGS += -Wl,-rpath,'$$ORIGIN' -Wl,-z,origin
 endif
 ASFLAGS += -f $(ASFORMAT)
 
