diff --git a/source/sw/src/draw.cpp b/source/sw/src/draw.cpp
index ca7f5bf1f..4271ab8ac 100644
--- a/source/sw/src/draw.cpp
+++ b/source/sw/src/draw.cpp
@@ -1570,7 +1570,7 @@ int COVERsetgamemode(int mode, int xdim, int ydim, int bpp)
 {
     ud_setup.ScreenHeight = ydim;
     ud_setup.ScreenWidth  = xdim;
-    ud_setup.ScreenMode   = mode;
+    ud_setup.ScreenMode   = 1;
     ud_setup.ScreenBPP    = bpp;
 
     // [JM] Should I be using upscalefactor here, or some SW equivalent to Duke's ud.detail? !CHECKME!
diff --git a/source/sw/src/game.cpp b/source/sw/src/game.cpp
index 7ca209f51..9e0c9fedf 100644
--- a/source/sw/src/game.cpp
+++ b/source/sw/src/game.cpp
@@ -728,7 +728,7 @@ void Set_GameMode(void)
         LOG_F(ERROR, "Failure setting video mode %dx%dx%d %s! Attempting safer mode...",
                      ud_setup.ScreenWidth,ud_setup.ScreenHeight,ud_setup.ScreenBPP,
                      ud_setup.ScreenMode ? "fullscreen" : "windowed");
-        ud_setup.ScreenMode = 0;
+        ud_setup.ScreenMode = 1;
         ud_setup.ScreenWidth = 640;
         ud_setup.ScreenHeight = 480;
         ud_setup.ScreenBPP = 8;
