diff --git a/CMakeLists.txt b/CMakeLists.txt
index 85999a50..1d77a7fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,13 +65,13 @@ option(CURL_SUPPORT "cURL support" ON)
 option(OPENAL_SUPPORT "OpenAL support" ON)
 option(SYSTEMWIDE_SUPPORT "Enable systemwide installation of game assets" OFF)
 option(SDL3_SUPPORT "Build against SDL 3 instead of SDL2" ON)
-option(GL1_RENDERER "Build the GL1 renderer" ON)
-option(GL3_RENDERER "Build the GL3 renderer" ON)
-option(GL4_RENDERER "Build the GL4 renderer" ON)
+option(GL1_RENDERER "Build the GL1 renderer" OFF)
+option(GL3_RENDERER "Build the GL3 renderer" OFF)
+option(GL4_RENDERER "Build the GL4 renderer" OFF)
 option(GLES1_RENDERER "Build the GLES1 renderer" OFF)
-option(GLES3_RENDERER "Build the GLES3 renderer" ON)
+option(GLES3_RENDERER "Build the GLES3 renderer" OFF)
 option(SOFT_RENDERER "Build the software renderer" ON)
-option(VK_RENDERER "Build the Vulkan renderer" ON)
+option(VK_RENDERER "Build the Vulkan renderer" OFF)
 
 set(SYSTEMDIR "" CACHE STRING "Override the system default directory")
 
