diff --git a/Makefile b/Makefile
index 5604fc35..fc5b5ba1 100644
--- a/Makefile
+++ b/Makefile
@@ -345,10 +345,9 @@ endif
 
 # Extra CFLAGS for SDL.
 ifeq ($(WITH_SDL3),yes)
-SDLCFLAGS := $(shell $(PKG_CONFIG) --cflags sdl3)
 SDLCFLAGS += -DUSE_SDL3
 else
-SDLCFLAGS := $(shell sdl2-config --cflags)
+SDLCFLAGS :=
 endif
 
 ifdef NO_SDL_GYRO
@@ -435,17 +434,9 @@ endif
 
 # Extra LDFLAGS for SDL
 ifeq ($(WITH_SDL3),yes)
-ifeq ($(YQ2_OSTYPE), Darwin)
 SDLLDFLAGS := -lSDL3
 else
-SDLLDFLAGS := $(shell $(PKG_CONFIG) --libs sdl3)
-endif
-else
-ifeq ($(YQ2_OSTYPE), Darwin)
 SDLLDFLAGS := -lSDL2
-else
-SDLLDFLAGS := $(shell sdl2-config --libs)
-endif
 endif
 
 # The renderer libs don't need libSDL2main, libmingw32 or -mwindows.
