diff --git a/es-app/src/main.cpp b/es-app/src/main.cpp
index 4892972a8..9cf5e0204 100644
--- a/es-app/src/main.cpp
+++ b/es-app/src/main.cpp
@@ -747,8 +747,9 @@ int main(int argc, char* argv[])
 					}
 				}				
 
-				if (event.type == SDL_QUIT)
-					running = false;
+				if (Utils::Platform::ExitRequest::user == 1) {
+				  running = false;
+				}
 			} 
 			while(SDL_PollEvent(&event));
 
diff --git a/es-core/src/InputManager.cpp b/es-core/src/InputManager.cpp
index 469af25a8..7dafcd458 100644
--- a/es-core/src/InputManager.cpp
+++ b/es-core/src/InputManager.cpp
@@ -581,22 +581,16 @@ bool InputManager::parseEvent(const SDL_Event& ev, Window* window)
 		return true;
 
 	case SDL_KEYDOWN:
+	  if ((ev.key.keysym.mod & KMOD_SHIFT) != 0 && (ev.key.keysym.mod & KMOD_ALT) != 0 && ev.key.keysym.sym == SDLK_F4) {
+	    Utils::Platform::ExitRequest::user = 1;
+	  }
+
 		if (ev.key.keysym.sym == SDLK_BACKSPACE && SDL_IsTextInputActive())
 			window->textInput("\b");
 
 		if (ev.key.repeat)
 			return false;
 
-#if !WIN32
-		if (ev.key.keysym.sym == SDLK_F4)
-		{
-			SDL_Event* quit = new SDL_Event();
-			quit->type = SDL_QUIT;
-			SDL_PushEvent(quit);
-			return false;
-		}
-#endif
-
 		window->input(getInputConfigByDevice(DEVICE_KEYBOARD), Input(DEVICE_KEYBOARD, TYPE_KEY, ev.key.keysym.sym, 1, false));
 		return true;
 
diff --git a/es-core/src/utils/Platform.cpp b/es-core/src/utils/Platform.cpp
index 12d57b7ef..66221345c 100644
--- a/es-core/src/utils/Platform.cpp
+++ b/es-core/src/utils/Platform.cpp
@@ -31,6 +31,8 @@ namespace Utils
 {
 	namespace Platform
 	{
+	  int ExitRequest::user = 0;
+
 		ProcessStartInfo::ProcessStartInfo()
 		{
 			window = nullptr;
@@ -229,6 +231,8 @@ namespace Utils
 				break;
 			}
 
+			ExitRequest::user = 1;
+
 			SDL_Event* quit = new SDL_Event();
 			quit->type = SDL_QUIT;
 			SDL_PushEvent(quit);
diff --git a/es-core/src/utils/Platform.h b/es-core/src/utils/Platform.h
index 56c841191..fb1a41135 100644
--- a/es-core/src/utils/Platform.h
+++ b/es-core/src/utils/Platform.h
@@ -17,6 +17,11 @@ namespace Utils
 {
 	namespace Platform
 	{
+	  class ExitRequest {
+	  public:
+	    static int user;
+	  };
+
 		enum QuitMode
 		{
 			QUIT = 0,
