diff --git a/es-core/src/Settings.cpp b/es-core/src/Settings.cpp
index 2870d3fb..110fc04f 100644
--- a/es-core/src/Settings.cpp
+++ b/es-core/src/Settings.cpp
@@ -25,7 +25,7 @@ IMPLEMENT_STATIC_BOOL_SETTING(DrawFramerate, false)
 IMPLEMENT_STATIC_BOOL_SETTING(VolumePopup, true)
 IMPLEMENT_STATIC_BOOL_SETTING(BackgroundMusic, true)
 IMPLEMENT_STATIC_BOOL_SETTING(VSync, true)
-IMPLEMENT_STATIC_BOOL_SETTING(PreloadMedias, false)
+IMPLEMENT_STATIC_BOOL_SETTING(PreloadMedias, true)
 IMPLEMENT_STATIC_BOOL_SETTING(IgnoreLeadingArticles, false)
 IMPLEMENT_STATIC_INT_SETTING(ScreenSaverTime, 5 * 60 * 1000)
 
@@ -191,8 +191,8 @@ void Settings::setDefaults()
 	mIntMap["MaxVRAM"] = 100;
 #endif
 
-	mStringMap["TransitionStyle"] = "auto";
-	mStringMap["GameTransitionStyle"] = "auto";
+	mStringMap["TransitionStyle"] = "instant";
+	mStringMap["GameTransitionStyle"] = "instant";
 
 	mStringMap["ThemeSet"] = "";
 	mStringMap["ScreenSaverBehavior"] = "dim";
@@ -286,7 +286,7 @@ void Settings::setDefaults()
 
 	mBoolMap["ThreadedLoading"] = true;
 	mBoolMap["AsyncImages"] = true;
-	mBoolMap["PreloadUI"] = false;
+	mBoolMap["PreloadUI"] = true;
 	mBoolMap["PreloadMedias"] = Settings::_PreloadMedias;
 	mBoolMap["OptimizeVRAM"] = true;
 	mBoolMap["OptimizeVideo"] = true;
