diff --git a/es-app/src/ApiSystem.cpp b/es-app/src/ApiSystem.cpp
index d25e329ba..505f739cf 100644
--- a/es-app/src/ApiSystem.cpp
+++ b/es-app/src/ApiSystem.cpp
@@ -1081,7 +1081,7 @@ bool ApiSystem::downloadGitRepository(const std::string& url, const std::string&
 			}
 		}
 
-		std::this_thread::sleep_for(std::chrono::milliseconds(20));
+		std::this_thread::sleep_for(std::chrono::milliseconds(1));
 	}
 
 	if (httpreq.status() != HttpReq::REQ_SUCCESS)
