diff --git a/es-app/src/ApiSystem.cpp b/es-app/src/ApiSystem.cpp
index 94a6b9c7c..2ec94da6c 100644
--- a/es-app/src/ApiSystem.cpp
+++ b/es-app/src/ApiSystem.cpp
@@ -912,7 +912,7 @@ bool ApiSystem::downloadGitRepository(const std::string& url, const std::string&
 			}
 		}
 
-		std::this_thread::sleep_for(std::chrono::milliseconds(20));
+		std::this_thread::sleep_for(std::chrono::milliseconds(1));
 	}
 
 	if (httpreq.status() != HttpReq::REQ_SUCCESS)
