diff --git a/es-core/src/utils/Platform.cpp b/es-core/src/utils/Platform.cpp
index 66221345c..cb50add28 100644
--- a/es-core/src/utils/Platform.cpp
+++ b/es-core/src/utils/Platform.cpp
@@ -197,7 +197,7 @@ namespace Utils
 #ifdef WIN32 // windows
 			return system("shutdown -s -t 0");
 #else // osx / linux
-			return system("shutdown -h now");
+			return system("shutdown -t1 now -h");
 #endif
 		}
 
@@ -206,7 +206,7 @@ namespace Utils
 #ifdef WIN32 // windows
 			return system("shutdown -r -t 0");
 #else // osx / linux
-			return system("shutdown -r now");
+			return system("shutdown -t1 now -r");
 #endif
 		}
 
