diff --git a/es-app/src/guis/GuiMenu.cpp b/es-app/src/guis/GuiMenu.cpp
index a13443062..d81829323 100644
--- a/es-app/src/guis/GuiMenu.cpp
+++ b/es-app/src/guis/GuiMenu.cpp
@@ -815,7 +815,7 @@ void GuiMenu::openDeveloperSettings()
 		// support
 		s->addEntry(_("CREATE A SUPPORT FILE"), true, [window] 
 		{
-			window->pushGui(new GuiMsgBox(window, _("CREATE A SUPPORT FILE? THIS INCLUDES ALL DATA IN YOUR SYSTEM FOLDER."), _("YES"),
+			window->pushGui(new GuiMsgBox(window, _("CREATE A SUPPORT FILE? THIS INCLUDES ALL DATA IN YOUR SAVES FOLDER."), _("YES"),
 				[window] 
 				{
 					if (ApiSystem::getInstance()->generateSupportFile())
