diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3fa6d0a03..7382de7f7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,8 @@ option(BCM "BCM host" OFF)
 
 option(BATOCERA "Set to ON to enable BATOCERA specific code" OFF)
 option(RETROBAT "Set to ON to enable RETROBAT specific code" OFF)
+option(PIBOY "Set to ON to enable DMG Piboy compatibility" OFF)
+option(NOUPDATE "Set to ON to disable system updates" OFF)
 
 option(ENABLE_FILEMANAGER "Set to ON to enable f1 shortcut for filesystem" OFF)
 option(DISABLE_KODI "Set to ON to disable kodi in menu" OFF)
@@ -97,6 +99,16 @@ if(RETROBAT)
   add_definitions(-DRETROBAT)
 endif()
 
+if(PIBOY)
+  MESSAGE("Piboy build")
+  add_definitions(-DPIBOY)
+endif()
+
+if(NOUPDATE)
+  MESSAGE("Update disabled")
+  add_definitions(-DNOUPDATE)
+endif()
+
 # disable kodi menu
 if(DISABLE_KODI)
   MESSAGE("Kodi disabled")
diff --git a/es-app/src/guis/GuiMenu.cpp b/es-app/src/guis/GuiMenu.cpp
index 850da7246..24ef4ebd1 100644
--- a/es-app/src/guis/GuiMenu.cpp
+++ b/es-app/src/guis/GuiMenu.cpp
@@ -84,8 +84,15 @@
 #define fake_gettext_architecture _("Architecture")
 #define fake_gettext_diskformat   _("Disk format")
 #define fake_gettext_temperature  _("Temperature")
+#if PIBOY
+#define fake_gettext_fan          _("Fan speed")
+#endif
 #define fake_gettext_avail_memory _("Available memory")
 #define fake_gettext_battery      _("Battery")
+#if PIBOY
+#define fake_gettext_battery_volts   _("Battery voltage")
+#define fake_gettext_battery_amps    _("Battery current")
+#endif
 #define fake_gettext_model        _("Model")
 #define fake_gettext_cpu_model    _("Cpu model")
 #define fake_gettext_cpu_number   _("Cpu number")
@@ -204,9 +211,12 @@ GuiMenu::GuiMenu(Window *window, bool animate) : GuiComponent(window), mMenu(win
 		if (ApiSystem::getInstance()->isScriptingSupported(ApiSystem::BATOCERASTORE) || ApiSystem::getInstance()->isScriptingSupported(ApiSystem::THEMESDOWNLOADER) ||
 			(ApiSystem::getInstance()->isScriptingSupported(ApiSystem::THEBEZELPROJECT) && ApiSystem::getInstance()->isScriptingSupported(ApiSystem::DECORATIONS)) ||
 			ApiSystem::getInstance()->isScriptingSupported(ApiSystem::UPGRADE))
-			addEntry(_("UPDATES & DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
-
-		addEntry(_("SYSTEM SETTINGS").c_str(), true, [this] { openSystemSettings(); }, "iconSystem");
+			#if !NOUPDATE
+				addEntry(_("UPDATES & DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
+			#else
+				addEntry(_("DOWNLOADS"), true, [this] { openUpdatesSettings(); }, "iconUpdates");
+			#endif	
+			addEntry(_("SYSTEM SETTINGS").c_str(), true, [this] { openSystemSettings(); }, "iconSystem");
 	}
 	else
 	{
@@ -1180,7 +1190,11 @@ void GuiMenu::openDeveloperSettings()
 
 void GuiMenu::openUpdatesSettings()
 {
-	GuiSettings *updateGui = new GuiSettings(mWindow, _("UPDATES & DOWNLOADS").c_str());
+	#if !NOUPDATE
+		GuiSettings *updateGui = new GuiSettings(mWindow, _("UPDATES & DOWNLOADS").c_str());
+	#else
+		GuiSettings *updateGui = new GuiSettings(mWindow, _("DOWNLOADS").c_str());
+	#endif
 
 	updateGui->addGroup(_("DOWNLOADS"));
 
@@ -1221,6 +1235,7 @@ void GuiMenu::openUpdatesSettings()
 
 	if (ApiSystem::getInstance()->isScriptingSupported(ApiSystem::UPGRADE))
 	{
+#if !NOUPDATE
 		updateGui->addGroup(_("SOFTWARE UPDATES"));
 
 		// Enable updates
@@ -1236,11 +1251,11 @@ void GuiMenu::openUpdatesSettings()
 
 		std::string updatesType = SystemConf::getInstance()->get("updates.type");
 
-#if WIN32
+    #if WIN32
 		if (updatesType == "unstable")
 			updatesTypeList->add("unstable", "unstable", updatesType == "unstable");
 		else
-#endif
+    #endif
 			if (updatesType.empty() || updatesType != BETA_NAME)
 				updatesType = "stable";
 
@@ -1269,6 +1284,7 @@ void GuiMenu::openUpdatesSettings()
 				mWindow->pushGui(new GuiUpdate(mWindow));
 			}
 		});
+#endif
 	}
 
 	mWindow->pushGui(updateGui);
diff --git a/es-core/src/utils/Platform.cpp b/es-core/src/utils/Platform.cpp
index ca759f01c..12c539e12 100644
--- a/es-core/src/utils/Platform.cpp
+++ b/es-core/src/utils/Platform.cpp
@@ -23,6 +23,7 @@
 #include "Paths.h"
 #include <fstream>
 #include <string>
+#include <iostream>
 #include "renderers/Renderer.h"
 
 // #define DEVTEST
@@ -196,6 +197,8 @@ namespace Utils
 		{
 #ifdef WIN32 // windows
 			return system("shutdown -s -t 0");
+#elseif PIBOY
+return system("/sbin/rmmod xpi_gamecon ; /sbin/rmmod xpi_gamecon_xrs ; shutdown -t1 now -h");
 #else // osx / linux
 			return system("shutdown -t1 now -h");
 #endif
@@ -205,6 +208,8 @@ namespace Utils
 		{
 #ifdef WIN32 // windows
 			return system("shutdown -r -t 0");
+#elseif PIBOY
+	return system("reboot");
 #else // osx / linux
 			return system("shutdown -t1 now -r");
 #endif
@@ -431,9 +436,20 @@ namespace Utils
 				std::string fuelgaugeRootPath;
 				std::string chargerRootPath;
 
-				auto files = Utils::FileSystem::getDirContent("/sys/class/power_supply");
+	        #if PIBOY
+		        auto files = Utils::FileSystem::getDirContent("/sys/kernel");
+	        #else
+         		auto files = Utils::FileSystem::getDirContent("/sys/class/power_supply");
+	        #endif
 				for (auto file : files)
 				{
+    		#if PIBOY
+    			if (Utils::String::toLower(file).find("/xpi_gamecon") != std::string::npos)
+    			{
+    				batteryRootPath = file;
+    				break;
+    			}
+    		#else
 					if ((Utils::String::toLower(file).find("/bat") != std::string::npos) && (batteryRootPath.empty()))
 						batteryRootPath = file;
 
@@ -446,7 +462,8 @@ namespace Utils
 
 					if ((Utils::String::toLower(file).find("charger") != std::string::npos) && (chargerRootPath.empty()))
 						chargerRootPath = file;
-				}
+		    #endif
+		    }
 
 				// If there's no battery device, look for discrete charger and fuel gauge
 				if (batteryRootPath.empty())
@@ -470,8 +487,13 @@ namespace Utils
 				}
 				else
 				{
-					batteryStatusPath = batteryRootPath + "/status";
-					batteryCapacityPath = batteryRootPath + "/capacity";
+    		    #if PIBOY
+    			    batteryStatusPath = batteryRootPath + "/amps";
+    			    batteryCapacityPath = batteryRootPath + "/percent";
+    		    #else
+     			    batteryStatusPath = batteryRootPath + "/status";
+     			    batteryCapacityPath = batteryRootPath + "/capacity";
+    		    #endif
 				}
 			}
 
@@ -484,9 +506,33 @@ namespace Utils
 			else
 			{
 				ret.hasBattery = true;
+    		#if PIBOY
+        	 	std::string battlvl;
+        	 	std::string battsts;
+        	 	std::ifstream rfile;
+
+        	 	// Battery Charging
+        	 	rfile.open(batteryStatusPath);
+					if (std::getline(rfile, battsts)) LOG(LogDebug) << "Batt Charging: " + battsts;
+					int output = Utils::String::toInteger(battsts);
+	       	 	rfile.close();
+
+        	 	if (output < 0)
+				    ret.isCharging = false;
+				else
+				    ret.isCharging = true;
+
+        	 	// Battery Level
+        	 	rfile.open(batteryCapacityPath);
+					if (std::getline(rfile, battlvl)) LOG(LogDebug) << "Batt Lvl: " + battlvl;
+					int intBattLvl = Utils::String::toInteger(battlvl);
+        	 	rfile.close();
+        	 	ret.level = intBattLvl;
+    		#else
 				std::string chargerStatus;
 				chargerStatus = Utils::String::replace(Utils::FileSystem::readAllText(batteryStatusPath), "\n", "");
 				ret.isCharging = ((chargerStatus != "Not charging") && (chargerStatus != "Discharging"));
+    		#endif
 				// If reading from fuel gauge, we have to calculate remaining charge
 				if (batteryCapacityPath.length() <= 1)
 				{
@@ -507,7 +553,11 @@ namespace Utils
 						file.close();
 						if (!buffer.empty())
 						{
-							ret.level = Utils::String::toInteger(buffer);
+						#if PIBOY
+						//do nothing
+						#else
+						ret.level = Utils::String::toInteger(buffer);
+						#endif
 						}
 						else
 						{
