diff --git a/es-app/src/services/HttpServerThread.cpp b/es-app/src/services/HttpServerThread.cpp
index 41acbf0e6..f68ee63c6 100644
--- a/es-app/src/services/HttpServerThread.cpp
+++ b/es-app/src/services/HttpServerThread.cpp
@@ -573,7 +573,7 @@ void HttpServerThread::run()
 
 		mWindow->displayNotificationMessage(req.body);
 	});
-	
+#ifdef BATOCERA_DISKS
 	mHttpServer->Post("/storage/event", [this](const httplib::Request& req, httplib::Response& res)
 	{
 		if (!isAllowed(req, res))
@@ -595,6 +595,7 @@ void HttpServerThread::run()
 
         res.set_content("OK", "text/plain");
 	});
+#endif
 
 	mHttpServer->Post("/launch", [this](const httplib::Request& req, httplib::Response& res)
 	{
