diff --git a/libretro/main.cpp b/libretro/main.cpp
index a785aa767..ebfd5093d 100644
--- a/libretro/main.cpp
+++ b/libretro/main.cpp
@@ -1520,7 +1520,7 @@ void retro_init(void)
 		environ_cb(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, &system_base);
 
 		FileSystem::FindResultsArray results;
-		if (FileSystem::FindFiles(Path::Combine(system_base, "/pcsx2/bios").c_str(), "*", FILESYSTEM_FIND_FILES, &results))
+		if (FileSystem::FindFiles(Path::Combine(system_base, "/ps2").c_str(), "*", FILESYSTEM_FIND_FILES, &results))
 		{
 			u32 version, region;
 			static constexpr u32 MIN_BIOS_SIZE = 4 * _1mb;
