diff --git a/CMakeLists.txt b/CMakeLists.txt
index e838030b..d49cd2fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -228,7 +228,7 @@ find_package(half 1.12.0 MODULE)
 find_package(magic_enum 0.9.7 CONFIG)
 find_package(PNG 1.6 MODULE)
 find_package(RenderDoc 1.6.0 MODULE)
-find_package(SDL3_mixer 2.8.1 CONFIG)
+find_package(SDL3_mixer 9.8.1 CONFIG)
 if (SDL3_mixer_FOUND)
     find_package(SDL3 3.1.2 CONFIG)
 endif()
