diff --git a/openbor.c b/openbor.c
index 2750f50..92ca6d7 100644
--- a/openbor.c
+++ b/openbor.c
@@ -28320,7 +28320,8 @@ int match_combo(int a[], s_player *p, int l)
         step = (step + MAX_SPECIAL_INPUTS) % MAX_SPECIAL_INPUTS;
 
         // old: !(a[l - 1 - j]&p->combokey[step])
-        if( ((a[l - 1 - j]&p->combokey[step]) ^ a[l - 1 - j]) ) // if input&combokey == 0 then not good btn
+        //if( ((a[l - 1 - j]&p->combokey[step]) ^ a[l - 1 - j]) ) // if input&combokey == 0 then not good btn
+		if ((a[l - 1 - j] & p->combokey[step]) == 0)
         {
             return 0;
         }
