diff --git a/RSDKv4/Drawing.cpp b/RSDKv4/Drawing.cpp
index 6d8b747..ce38e2c 100644
--- a/RSDKv4/Drawing.cpp
+++ b/RSDKv4/Drawing.cpp
@@ -209,7 +209,7 @@ int InitRenderDevice()
 
 #if RETRO_PLATFORM != RETRO_ANDROID && RETRO_PLATFORM != RETRO_OSX
     GLenum err = glewInit();
-    if (err != GLEW_OK) {
+    if (err != GLEW_OK && err != GLEW_ERROR_NO_GLX_DISPLAY) {
         PrintLog("glew init error:");
         PrintLog((const char *)glewGetErrorString(err));
         return false;
diff --git a/RSDKv4/Ini.cpp b/RSDKv4/Ini.cpp
index 3ec266a..b60edf8 100644
--- a/RSDKv4/Ini.cpp
+++ b/RSDKv4/Ini.cpp
@@ -124,7 +124,7 @@ int IniParser::GetFloat(const char *section, const char *key, float *dest)
         }
     }
 
-    return 0.0f;
+    return 0;
 }
 int IniParser::GetBool(const char *section, const char *key, bool *dest)
 {
diff --git a/RSDKv4/Reader.cpp b/RSDKv4/Reader.cpp
index 9771350..5cfafed 100644
--- a/RSDKv4/Reader.cpp
+++ b/RSDKv4/Reader.cpp
@@ -29,7 +29,7 @@ bool CheckRSDKFile(const char *filePath)
     char filePathBuffer[0x100];
 #if RETRO_PLATFORM == RETRO_OSX
     char pathBuf[0x100];
-    sprintf(pathBuf, "%s/%s", gamePath, filePathBuffer);
+    sprintf(pathBuf, "%s/%s", gamePath, filePath);
     sprintf(filePathBuffer, "%s", pathBuf);
 #else
     sprintf(filePathBuffer, "%s", filePath);
diff --git a/RSDKv4/Userdata.cpp b/RSDKv4/Userdata.cpp
index ed63199..adba10e 100644
--- a/RSDKv4/Userdata.cpp
+++ b/RSDKv4/Userdata.cpp
@@ -27,7 +27,7 @@ bool vsPlaying   = false;
 
 int sendCounter = 0;
 
-#if RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_LINUX
+#if RETRO_PLATFORM == RETRO_OSX
 #include <sys/stat.h>
 #include <sys/types.h>
 #endif
@@ -42,24 +42,6 @@ int disableFocusPause_Config = 0;
 
 bool useSGame = false;
 
-#if RETRO_PLATFORM == RETRO_LINUX
-std::string getXDGDataPath()
-{
-    std::string path;
-    char const *dataHome = getenv("XDG_DATA_HOME");
-    if (dataHome == NULL) {
-        char const *home = getenv("HOME");
-        path += home;
-        path += "/.local/share";
-    }
-    else {
-        path += dataHome;
-    }
-    path += "/RSDKv4";
-    return path;
-}
-#endif
-
 bool ReadSaveRAMData()
 {
     useSGame = false;
@@ -74,8 +56,6 @@ bool ReadSaveRAMData()
     sprintf(buffer, "%s/%sSData.bin", redirectSave ? modsPath : gamePath, savePath);
 #elif RETRO_PLATFORM == RETRO_iOS
     sprintf(buffer, "%s/%sSData.bin", redirectSave ? modsPath : getDocumentsPath(), savePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-    sprintf(buffer, "%s/%sSData.bin", redirectSave ? modsPath : getXDGDataPath().c_str(), savePath);
 #else
     sprintf(buffer, "%s%sSData.bin", redirectSave ? modsPath : gamePath, savePath);
 #endif
@@ -106,8 +86,6 @@ bool ReadSaveRAMData()
         sprintf(buffer, "%s/%sSGame.bin", redirectSave ? modsPath : gamePath, savePath);
 #elif RETRO_PLATFORM == RETRO_iOS
         sprintf(buffer, "%s/%sSGame.bin", redirectSave ? modsPath : getDocumentsPath(), savePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-        sprintf(buffer, "%s/%sSGame.bin", redirectSave ? modsPath : getXDGDataPath().c_str(), savePath);
 #else
         sprintf(buffer, "%s%sSGame.bin", redirectSave ? modsPath : gamePath, savePath);
 #endif
@@ -151,8 +129,6 @@ bool WriteSaveRAMData()
         sprintf(buffer, "%s/%sSData.bin", redirectSave ? modsPath : gamePath, savePath);
 #elif RETRO_PLATFORM == RETRO_iOS
         sprintf(buffer, "%s/%sSData.bin", redirectSave ? modsPath : getDocumentsPath(), savePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-        sprintf(buffer, "%s/%sSData.bin", redirectSave ? modsPath : getXDGDataPath().c_str(), savePath);
 #else
         sprintf(buffer, "%s%sSData.bin", redirectSave ? modsPath : gamePath, savePath);
 #endif
@@ -182,8 +158,6 @@ bool WriteSaveRAMData()
         sprintf(buffer, "%s/%sSGame.bin", redirectSave ? modsPath : gamePath, savePath);
 #elif RETRO_PLATFORM == RETRO_iOS
         sprintf(buffer, "%s/%sSGame.bin", redirectSave ? modsPath : getDocumentsPath(), savePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-        sprintf(buffer, "%s/%sSGame.bin", redirectSave ? modsPath : getXDGDataPath().c_str(), savePath);
 #else
         sprintf(buffer, "%s%sSGame.bin", redirectSave ? modsPath : gamePath, savePath);
 #endif
@@ -224,11 +198,6 @@ void InitUserdata()
     sprintf(modsPath, "%s/RSDKv4/", getResourcesPath());
 
     mkdir(gamePath, 0777);
-#elif RETRO_PLATFORM == RETRO_LINUX
-    sprintf(gamePath, "%s", getXDGDataPath().c_str());
-    sprintf(modsPath, "%s/", getXDGDataPath().c_str());
-
-    mkdir(getXDGDataPath().c_str(), 0755);
 #elif RETRO_PLATFORM == RETRO_ANDROID
     {
         char buffer[0x200];
@@ -259,8 +228,6 @@ void InitUserdata()
         sprintf(buffer, "%ssettings.ini", gamePath);
 #elif RETRO_PLATFORM == RETRO_OSX || RETRO_PLATFORM == RETRO_ANDROID
     sprintf(buffer, "%s/settings.ini", gamePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-    sprintf(buffer, "%s/settings.ini", getXDGDataPath().c_str());
 #else
     sprintf(buffer, BASE_PATH "settings.ini");
 #endif
@@ -663,8 +630,6 @@ void InitUserdata()
         sprintf(buffer, "%sUData.bin", gamePath);
 #elif RETRO_PLATFORM == RETRO_OSX
     sprintf(buffer, "%s/UData.bin", gamePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-    sprintf(buffer, "%s/UData.bin", getXDGDataPath().c_str());
 #else
     sprintf(buffer, "%sUData.bin", gamePath);
 #endif
@@ -824,8 +789,6 @@ void WriteSettings()
         sprintf(buffer, "%ssettings.ini", gamePath);
 #elif RETRO_PLATFORM == RETRO_OSX
     sprintf(buffer, "%s/settings.ini", gamePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-    sprintf(buffer, "%s/settings.ini", getXDGDataPath().c_str());
 #else
     sprintf(buffer, "%ssettings.ini", gamePath);
 #endif
@@ -846,8 +809,6 @@ void ReadUserdata()
     sprintf(buffer, "%s/%sUData.bin", redirectSave ? modsPath : gamePath, savePath);
 #elif RETRO_PLATFORM == RETRO_iOS
     sprintf(buffer, "%s/%sUData.bin", redirectSave ? modsPath : getDocumentsPath(), savePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-    sprintf(buffer, "%s/%sUData.bin", redirectSave ? modsPath : getXDGDataPath().c_str(), savePath);
 #else
     sprintf(buffer, "%s%sUData.bin", redirectSave ? modsPath : gamePath, savePath);
 #endif
@@ -901,8 +862,6 @@ void WriteUserdata()
     sprintf(buffer, "%s/%sUData.bin", redirectSave ? modsPath : gamePath, savePath);
 #elif RETRO_PLATFORM == RETRO_iOS
     sprintf(buffer, "%s/%sUData.bin", redirectSave ? modsPath : getDocumentsPath(), savePath);
-#elif RETRO_PLATFORM == RETRO_LINUX
-    sprintf(buffer, "%s/%sUData.bin", redirectSave ? modsPath : getXDGDataPath().c_str(), savePath);
 #else
     sprintf(buffer, "%s%sUData.bin", redirectSave ? modsPath : gamePath, savePath);
 #endif
