diff --git a/src/xrCore/LocatorAPI.cpp b/src/xrCore/LocatorAPI.cpp
index 211313010..76f48b96c 100644
--- a/src/xrCore/LocatorAPI.cpp
+++ b/src/xrCore/LocatorAPI.cpp
@@ -829,11 +829,11 @@ void CLocatorAPI::setup_fs_path(pcstr fs_name)
         else
         {
             if (strstr(Core.Params, "-shoc") || strstr(Core.Params, "-soc"))
-                pref_path = SDL_GetPrefPath("GSC Game World", "S.T.A.L.K.E.R. - Shadow of Chernobyl");
+                pref_path = "/userdata/roms/stalker/shoc";
             else if (strstr(Core.Params, "-cs"))
-                pref_path = SDL_GetPrefPath("GSC Game World", "S.T.A.L.K.E.R. - Clear Sky");
+                pref_path = "/userdata/roms/stalker/cs";
             else
-                pref_path = SDL_GetPrefPath("GSC Game World", "S.T.A.L.K.E.R. - Call of Pripyat");
+                pref_path = "/userdata/roms/stalker/cop";
         }
         SDL_strlcpy(full_current_directory, pref_path, sizeof full_current_directory);
         SDL_free(pref_path);
diff --git a/src/xrCore/xrCore.cpp b/src/xrCore/xrCore.cpp
index 5de61cd9e..a552f202c 100644
--- a/src/xrCore/xrCore.cpp
+++ b/src/xrCore/xrCore.cpp
@@ -189,11 +189,11 @@ void xrCore::Initialize(pcstr _ApplicationName, pcstr commandLine, LogCallback c
         else
         {
             if (strstr(Core.Params, "-shoc") || strstr(Core.Params, "-soc"))
-                pref_path = SDL_GetPrefPath("GSC Game World", "S.T.A.L.K.E.R. - Shadow of Chernobyl");
+                pref_path = "/userdata/roms/stalker/shoc";
             else if (strstr(Core.Params, "-cs"))
-                pref_path = SDL_GetPrefPath("GSC Game World", "S.T.A.L.K.E.R. - Clear Sky");
+                pref_path = "/userdata/roms/stalker/cs";
             else
-                pref_path = SDL_GetPrefPath("GSC Game World", "S.T.A.L.K.E.R. - Call of Pripyat");
+                pref_path = "/userdata/roms/stalker/cop";
         }
         SDL_strlcpy(ApplicationPath, pref_path, sizeof(ApplicationPath));
         SDL_free(pref_path);
