diff --git a/makefile b/makefile
index 6fd7e06..997de54 100644
--- a/makefile
+++ b/makefile
@@ -38,7 +38,8 @@ CMAKEFLAGS += \
 	-DCMAKE_DISABLE_FIND_PACKAGE_PNG=1 \
 	-DCMAKE_DISABLE_FIND_PACKAGE_PortMidi=1 \
 	-DCMAKE_DISABLE_FIND_PACKAGE_CapsImage=1 \
-	-DENABLE_SMALL_MEM=0
+	-DENABLE_SMALL_MEM=0 \
+	-DCMAKE_C_COMPILER=$(CC)
 CMAKEBUILDFLAGS += $(MULTITHREAD)
 
 ifneq ($(DEBUG),0)
@@ -124,8 +125,8 @@ $(BD)core/%.o: core/%.c hatarilib
 	$(CC) -o $@ $(CFLAGS) -c $< 
 
 hatarilib: directories
-	(cd hatari/build && export CFLAGS="$(CFLAGS)" && cmake .. $(CMAKEFLAGS))
-	(cd hatari/build && export CFLAGS="$(CFLAGS)" && cmake --build . $(CMAKEBUILDFLAGS))
+	(cd hatari/build && export CFLAGS="$(CFLAGS)" && $(CMAKE) .. $(CMAKEFLAGS))
+	(cd hatari/build && export CFLAGS="$(CFLAGS)" && $(CMAKE) --build . $(CMAKEBUILDFLAGS))
 
 clean:
 	rm -f -r $(BD)
