diff --git a/src/System/SystemSDL.cpp b/src/System/SystemSDL.cpp
index 7451844..e803507 100644
--- a/src/System/SystemSDL.cpp
+++ b/src/System/SystemSDL.cpp
@@ -99,7 +99,7 @@ fs::path SystemSDL::FindConfigurationFile() const
     filenamePath = fs::path(appDataPath);
 
 #elif defined (__linux__)
-    filenamePath = fs::path(getenv("HOME")) / ".config";
+    filenamePath = fs::path("/userdata/system/configs");
 
 #endif
 
