diff --git a/src/rt_cfg.c b/src/rt_cfg.c
index d811080..0bb8783 100644
--- a/src/rt_cfg.c
+++ b/src/rt_cfg.c
@@ -75,7 +75,7 @@ int     MUvolume         = 196;
 int     FXvolume         = 196;
 
 boolean mouseenabled     = 1;
-boolean usemouselook     = 0;
+boolean usemouselook     = 1;
 int     inverse_mouse    = 1; //set  to -1 to invert mouse
 boolean usejump          = 0;
 boolean sdl_fullscreen   = 1;
@@ -96,9 +96,9 @@ int ScreenWidthToWriteToCfg = 0;
 boolean writeNewResIntoCfg = false;
 
 
-boolean joystickenabled  = 0;
-boolean joypadenabled    = 0;
-int     joystickport     = 0;
+boolean joystickenabled  = 1;
+boolean joypadenabled    = 1;
+int     joystickport     = 1;
 int     mouseadjustment  = 5;
 int     threshold        = 1;
 int     NumVoices        = 4;
