diff --git a/src/dosutil.c b/src/dosutil.c
index 9d4a01c..9926b57 100644
--- a/src/dosutil.c
+++ b/src/dosutil.c
@@ -71,14 +71,14 @@ int setup_homedir (void)
     int err;
 
     /* try to create the root directory */
-    snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/", "/userdata/system/configs/");
     err = mkdir (ApogeePath, S_IRWXU);
 
     /* keep the shareware and registered game data separated */
 #if (SHAREWARE == 1)
-    snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/", "/userdata/system/configs/");
 #else
-    snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/darkwar/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/", "/userdata/system/configs/");
 #endif
 
     err = mkdir (ApogeePath, S_IRWXU);
