diff --git a/rott/Makefile b/rott/Makefile
index ea673d8..65ce204 100644
--- a/rott/Makefile
+++ b/rott/Makefile
@@ -5,10 +5,12 @@
 SHAREWARE   ?= 0
 SUPERROTT   ?= 1
 SITELICENSE ?= 0
+USEHOMEPATH ?= 1
 
 CPPFLAGS += -DSHAREWARE=$(SHAREWARE)
 CPPFLAGS += -DSUPERROTT=$(SUPERROTT)
 CPPFLAGS += -DSITELICENSE=$(SITELICENSE)
+CPPFLAGS += -DUSEHOMEPATH=$(USEHOMEPATH)
 
 ROTT ?= rott
 
diff --git a/rott/dosutil.c b/rott/dosutil.c
index 0811641..892e617 100644
--- a/rott/dosutil.c
+++ b/rott/dosutil.c
@@ -129,14 +129,14 @@ int setup_homedir (void)
 	int err;
 
 	/* try to create the root directory */
-	snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/", "/userdata/system/configs/");
 	err = mkdir (ApogeePath, S_IRWXU);
 	
 /* keep the shareware and registered game data separated */
 #if (SHAREWARE == 1)
-	snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/", "/userdata/system/configs/");
 #else
-	snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/darkwar/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/darkwar/", "/userdata/system/configs/");
 #endif
 
 	err = mkdir (ApogeePath, S_IRWXU);
diff --git a/rott/rt_ted.c b/rott/rt_ted.c
index 9b6da98..a105770 100644
--- a/rott/rt_ted.c
+++ b/rott/rt_ted.c
@@ -1445,7 +1445,7 @@ void ReadROTTMap
 
 #if (SHAREWARE==0) && (USEHOMEPATH==1)
    char buf[MAX_PATH];
-   snprintf(buf, MAX_PATH, "%s/.rott/%s", getenv("HOME"), filename);
+   snprintf(buf, MAX_PATH, "%suserdata/roms/rott/%s", "/", filename);
 
    CheckRTLVersion( buf );
    filehandle = SafeOpenRead( buf );
diff --git a/rott/w_wad.c b/rott/w_wad.c
index 4664c5a..9adc376 100644
--- a/rott/w_wad.c
+++ b/rott/w_wad.c
@@ -92,7 +92,7 @@ void W_AddFile (char *_filename)
         char buf[MAX_PATH+100];//bna++
 
 #if (SHAREWARE==0) && (USEHOMEPATH==1)
-        snprintf(filename, MAX_PATH, "%s/RetroPie/roms/ports/rott/%s", getenv("HOME"), _filename);
+        snprintf(filename, MAX_PATH, "%suserdata/roms/rott/%s", "/", _filename);
 
         filename[sizeof (filename) - 1] = '\0';
         FixFilePath(filename);
