diff --git a/rott/dosutil.c b/rott/dosutil.c
index 0811641..b587c7e 100644
--- a/rott/dosutil.c
+++ b/rott/dosutil.c
@@ -129,14 +129,14 @@ int setup_homedir (void)
 	int err;
 
 	/* try to create the root directory */
-	snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/", "/userdata/system/configs/");
 	err = mkdir (ApogeePath, S_IRWXU);
 	
 /* keep the shareware and registered game data separated */
 #if (SHAREWARE == 1)
-	snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/", "/userdata/system/configs/");
 #else
-	snprintf (ApogeePath, sizeof (ApogeePath), "%s/.rott/darkwar/", getenv ("HOME"));
+	snprintf (ApogeePath, sizeof (ApogeePath), "%s/rott/", "/userdata/system/configs/");
 #endif
 
 	err = mkdir (ApogeePath, S_IRWXU);
