diff --git a/Makefile b/Makefile
index 1ce3e86..ab51a7c 100644
--- a/Makefile
+++ b/Makefile
@@ -88,6 +88,8 @@ else ifneq (,$(findstring rpi,$(platform)))
 	else ifneq (,$(findstring rpi3,$(platform)))
 		CFLAGS += -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard -marm
 		LDFLAGS += -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard -marm
+	else ifneq (,$(findstring rpi4,$(platform)))
+		CFLAGS = -mcpu=cortex-a72 -mtune=cortex-a72
 	endif
 # evercade armv7
 else ifneq (,$(findstring evercade,$(platform)))
