diff --git a/src/Ryujinx.UI.Common/Configuration/ConfigurationState.cs b/src/Ryujinx.UI.Common/Configuration/ConfigurationState.cs
index 50b3569a1..dba5c82eb 100644
--- a/src/Ryujinx.UI.Common/Configuration/ConfigurationState.cs
+++ b/src/Ryujinx.UI.Common/Configuration/ConfigurationState.cs
@@ -790,19 +790,19 @@ namespace Ryujinx.UI.Common.Configuration
 
         public void LoadDefault()
         {
-            Logger.EnableFileLog.Value = true;
+            Logger.EnableFileLog.Value = false;
             Graphics.BackendThreading.Value = BackendThreading.Auto;
             Graphics.ResScale.Value = 1;
             Graphics.ResScaleCustom.Value = 1.0f;
             Graphics.MaxAnisotropy.Value = -1.0f;
             Graphics.AspectRatio.Value = AspectRatio.Fixed16x9;
-            Graphics.GraphicsBackend.Value = DefaultGraphicsBackend();
+            Graphics.GraphicsBackend.Value = GraphicsBackend.Vulkan;
             Graphics.PreferredGpu.Value = string.Empty;
             Graphics.ShadersDumpPath.Value = string.Empty;
             Logger.EnableDebug.Value = false;
-            Logger.EnableStub.Value = true;
-            Logger.EnableInfo.Value = true;
-            Logger.EnableWarn.Value = true;
+            Logger.EnableStub.Value = false;
+            Logger.EnableInfo.Value = false;
+            Logger.EnableWarn.Value = false;
             Logger.EnableError.Value = true;
             Logger.EnableTrace.Value = false;
             Logger.EnableGuest.Value = true;
@@ -814,9 +814,9 @@ namespace Ryujinx.UI.Common.Configuration
             System.TimeZone.Value = "UTC";
             System.SystemTimeOffset.Value = 0;
             System.EnableDockedMode.Value = true;
-            EnableDiscordIntegration.Value = true;
-            CheckUpdatesOnStart.Value = true;
-            ShowConfirmExit.Value = true;
+            EnableDiscordIntegration.Value = false;
+            CheckUpdatesOnStart.Value = false;
+            ShowConfirmExit.Value = false;
             IgnoreApplet.Value = false;
             RememberWindowState.Value = true;
             ShowTitleBar.Value = !OperatingSystem.IsWindows();
@@ -869,8 +869,8 @@ namespace Ryujinx.UI.Common.Configuration
             UI.GridSize.Value = 2;
             UI.ApplicationSort.Value = 0;
             UI.IsAscendingOrder.Value = true;
-            UI.StartFullscreen.Value = false;
-            UI.ShowConsole.Value = true;
+            UI.StartFullscreen.Value = true;
+            UI.ShowConsole.Value = false;
             UI.WindowStartup.WindowSizeWidth.Value = 1280;
             UI.WindowStartup.WindowSizeHeight.Value = 760;
             UI.WindowStartup.WindowPositionX.Value = 0;
