diff --git a/code/rd-vanilla/qgl.h b/code/rd-vanilla/qgl.h
index c9215dbd..f49690d7 100644
--- a/code/rd-vanilla/qgl.h
+++ b/code/rd-vanilla/qgl.h
@@ -34,7 +34,7 @@ along with this program; if not, see <http://www.gnu.org/licenses/>.
 #	include <OpenGL/gl.h>
 #elif defined( __linux__ )
 #	include <GL/gl.h>
-#	include <GL/glx.h>
+//#	include <GL/glx.h>
 // bk001129 - from cvs1.17 (mkv)
 #	if defined(__FX__)
 #		include <GL/fxmesa.h>
