diff -Naur NVIDIA-Linux-x86_64-390.157.orig/kernel/nvidia-drm/nvidia-drm-drv.c NVIDIA-Linux-x86_64-390.157/kernel/nvidia-drm/nvidia-drm-drv.c
--- NVIDIA-Linux-x86_64-390.157.orig/kernel/nvidia-drm/nvidia-drm-drv.c	2022-10-12 10:30:31.000000000 +0100
+++ NVIDIA-Linux-x86_64-390.157/kernel/nvidia-drm/nvidia-drm-drv.c	2023-05-27 20:21:41.529046038 +0100
@@ -762,7 +762,11 @@
 
     nv_drm_driver.dumb_create      = nv_drm_dumb_create;
     nv_drm_driver.dumb_map_offset  = nv_drm_dumb_map_offset;
-    nv_drm_driver.dumb_destroy     = nv_drm_dumb_destroy;
+// Rel. commit "drm: remove dumb_destroy callback" (Christian König, 26 Jan 2023)
+// NB: No resources are leaked, the kernel releases the same resources by default
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
+     nv_drm_driver.dumb_destroy     = nv_drm_dumb_destroy;
+#endif
 
 #if defined(NV_DRM_DRIVER_HAS_GEM_PRIME_CALLBACKS)
     nv_drm_driver.gem_vm_ops       = &nv_drm_gem_vma_ops;
