diff --git a/Makefile b/Makefile
index d843de0..79afdee 100644
--- a/Makefile
+++ b/Makefile
@@ -8,8 +8,8 @@
 DEFINES = -DBERMUDA_POSIX -DBERMUDA_VORBIS -DBERMUDA_ZLIB
 VORBIS_LIBS = -lvorbisfile -lvorbis -logg
 
-SDL_CFLAGS = `sdl2-config --cflags`
-SDL_LIBS = `sdl2-config --libs` -lSDL2_mixer
+SDL_CFLAGS = `pkg-config sdl2 --cflags`
+SDL_LIBS = `pkg-config sdl2 --libs` -lSDL2_mixer
 
 CXXFLAGS = -g -O -Wall $(SDL_CFLAGS) $(DEFINES)
 
