diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac11a34ac..269a7d1c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@
 #   make
 #
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 ## Project name to use as command prefix.
 
Submodule external/SDL_ttf contains modified content
diff --git a/external/SDL_ttf/CMakeLists.txt b/external/SDL_ttf/CMakeLists.txt
index 2ff93c2..20c034b 100644
--- a/external/SDL_ttf/CMakeLists.txt
+++ b/external/SDL_ttf/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Taken from https://github.com/Microsoft/vcpkg/blob/2ad118a2532e99a78c41dffa1b9648e7eca20a95/ports/sdl2-ttf/CMakeLists.txt
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 project(SDL2_TTF C)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/mk/cmake)
Submodule external/discord-sdk contains modified content
diff --git a/external/discord-sdk/CMakeLists.txt b/external/discord-sdk/CMakeLists.txt
index 5dad9e9..8cfbbd4 100644
--- a/external/discord-sdk/CMakeLists.txt
+++ b/external/discord-sdk/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 3.2.0)
+cmake_minimum_required (VERSION 3.5)
 project (DiscordRPC)
 
 include(GNUInstallDirs)
Submodule external/fmt contains modified content
diff --git a/external/fmt/CMakeLists.txt b/external/fmt/CMakeLists.txt
index e5e2e317..ecea03b8 100644
--- a/external/fmt/CMakeLists.txt
+++ b/external/fmt/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1...3.18)
+cmake_minimum_required(VERSION 3.5)
 
 # Fallback for using newer policies on CMake <3.12.
 if(${CMAKE_VERSION} VERSION_LESS 3.12)
diff --git a/external/fmt/test/add-subdirectory-test/CMakeLists.txt b/external/fmt/test/add-subdirectory-test/CMakeLists.txt
index e08d0cf7..11519089 100644
--- a/external/fmt/test/add-subdirectory-test/CMakeLists.txt
+++ b/external/fmt/test/add-subdirectory-test/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1...3.18)
+cmake_minimum_required(VERSION 3.5)
 
 project(fmt-test CXX)
 
diff --git a/external/fmt/test/compile-error-test/CMakeLists.txt b/external/fmt/test/compile-error-test/CMakeLists.txt
index 8202f279..691e81d4 100644
--- a/external/fmt/test/compile-error-test/CMakeLists.txt
+++ b/external/fmt/test/compile-error-test/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Test if compile errors are produced where necessary.
 
-cmake_minimum_required(VERSION 3.1...3.18)
+cmake_minimum_required(VERSION 3.5)
 
 include(CheckCXXSourceCompiles)
 include(CheckCXXCompilerFlag)
diff --git a/external/fmt/test/find-package-test/CMakeLists.txt b/external/fmt/test/find-package-test/CMakeLists.txt
index 93d686e6..106e2652 100644
--- a/external/fmt/test/find-package-test/CMakeLists.txt
+++ b/external/fmt/test/find-package-test/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1...3.18)
+cmake_minimum_required(VERSION 3.5)
 
 project(fmt-test)
 
diff --git a/external/fmt/test/static-export-test/CMakeLists.txt b/external/fmt/test/static-export-test/CMakeLists.txt
index 7690d7b5..d43ad6ef 100644
--- a/external/fmt/test/static-export-test/CMakeLists.txt
+++ b/external/fmt/test/static-export-test/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1...3.18)
+cmake_minimum_required(VERSION 3.5)
 
 project(fmt-link CXX)
 
Submodule external/glm contains modified content
diff --git a/external/glm/CMakeLists.txt b/external/glm/CMakeLists.txt
index b7641a28..efb6904d 100644
--- a/external/glm/CMakeLists.txt
+++ b/external/glm/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 cmake_policy(VERSION 3.2)
 
 
diff --git a/external/glm/test/cmake/CMakeLists.txt b/external/glm/test/cmake/CMakeLists.txt
index 5bc11ef4..b3218132 100644
--- a/external/glm/test/cmake/CMakeLists.txt
+++ b/external/glm/test/cmake/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 project(test_find_glm)
 
 find_package(glm REQUIRED)
Submodule external/physfs contains modified content
diff --git a/external/physfs/CMakeLists.txt b/external/physfs/CMakeLists.txt
index b3291cc..61bd4d9 100644
--- a/external/physfs/CMakeLists.txt
+++ b/external/physfs/CMakeLists.txt
@@ -11,7 +11,7 @@
 
 set(PHYSFS_VERSION 3.2.0)
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 
 project(PhysicsFS VERSION ${PHYSFS_VERSION} LANGUAGES C )
 
Submodule external/sexp-cpp contains modified content
diff --git a/external/sexp-cpp/CMakeLists.txt b/external/sexp-cpp/CMakeLists.txt
index 5d6e88b..88d49e3 100644
--- a/external/sexp-cpp/CMakeLists.txt
+++ b/external/sexp-cpp/CMakeLists.txt
@@ -14,7 +14,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.5)
 project(sexp)
 
 include(mk/cmake/TinyCMMC.cmake)
Submodule external/simplesquirrel contains modified content
diff --git a/external/simplesquirrel/CMakeLists.txt b/external/simplesquirrel/CMakeLists.txt
index 92999ec..e987047 100644
--- a/external/simplesquirrel/CMakeLists.txt
+++ b/external/simplesquirrel/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
diff --git a/external/simplesquirrel/examples/CMakeLists.txt b/external/simplesquirrel/examples/CMakeLists.txt
index f8356aa..295c219 100644
--- a/external/simplesquirrel/examples/CMakeLists.txt
+++ b/external/simplesquirrel/examples/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 # Add executables
 add_executable(example_arrays example_arrays.cpp)
diff --git a/external/simplesquirrel/tests/CMakeLists.txt b/external/simplesquirrel/tests/CMakeLists.txt
index 502bdec..c7b2fa8 100644
--- a/external/simplesquirrel/tests/CMakeLists.txt
+++ b/external/simplesquirrel/tests/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 # Add executables
 add_executable(test_classes classes.cpp)
Submodule external/tinygettext contains modified content
diff --git a/external/tinygettext/CMakeLists.txt b/external/tinygettext/CMakeLists.txt
index a01eff2..36754de 100644
--- a/external/tinygettext/CMakeLists.txt
+++ b/external/tinygettext/CMakeLists.txt
@@ -39,7 +39,7 @@ set(VERSION "0.1")
 
 ### CMake configuration
 
-cmake_minimum_required(VERSION 2.4)
+cmake_minimum_required(VERSION 3.5)
 if(COMMAND cmake_policy)
 	CMAKE_POLICY(SET CMP0003 NEW)
 endif(COMMAND cmake_policy)
diff --git a/tools/scripting_docs_gen/CMakeLists.txt b/tools/scripting_docs_gen/CMakeLists.txt
index 246010eaf..7025ca6e1 100644
--- a/tools/scripting_docs_gen/CMakeLists.txt
+++ b/tools/scripting_docs_gen/CMakeLists.txt
@@ -14,7 +14,7 @@
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 project(ScriptingDocsGen)
 set(CMAKE_CXX_STANDARD 17)
