diff --git a/CMakeLists.txt b/CMakeLists.txt
index a8bce0f91..44cb9df9d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -432,7 +432,7 @@ endif()
 if(USE_STATIC_SIMPLESQUIRREL)
   target_link_libraries(supertux2 PUBLIC simplesquirrel_static)
 else()
-  target_link_libraries(supertux2 PUBLIC simplesquirrel)
+  target_link_libraries(supertux2 PUBLIC simplesquirrel_static)
 endif()
 
 target_link_libraries(supertux2 PUBLIC
