diff --git a/mk/cmake/SuperTux/ProvideSimpleSquirrel.cmake b/mk/cmake/SuperTux/ProvideSimpleSquirrel.cmake
index 8e9c41373..80b2f4cdb 100644
--- a/mk/cmake/SuperTux/ProvideSimpleSquirrel.cmake
+++ b/mk/cmake/SuperTux/ProvideSimpleSquirrel.cmake
@@ -4,11 +4,11 @@ endif()
 
 add_subdirectory("${CMAKE_SOURCE_DIR}/external/simplesquirrel/")
 
-if(MSVC)
+#if(MSVC)
   add_library(LibSimpleSquirrel ALIAS simplesquirrel_static)
-else()
-  add_library(LibSimpleSquirrel ALIAS simplesquirrel)
-endif()
+#else()
+#  add_library(LibSimpleSquirrel ALIAS simplesquirrel)
+#endif()
 
 # Set variables
 set(SQUIRREL_PREFIX ${CMAKE_BINARY_DIR}/external/simplesquirrel/libs/squirrel)
