diff -ruN PiBoyControls-cc96ca5777c2a17b9b44c1bbc15f37cfbcdd465f.orig/Makefile PiBoyControls-cc96ca5777c2a17b9b44c1bbc15f37cfbcdd465f/Makefile
--- PiBoyControls-cc96ca5777c2a17b9b44c1bbc15f37cfbcdd465f.orig/Makefile	2020-12-13 05:12:53.000000000 +0000
+++ PiBoyControls-cc96ca5777c2a17b9b44c1bbc15f37cfbcdd465f/Makefile	2021-05-15 14:17:38.536500057 +0100
@@ -1,8 +1,7 @@
 obj-m := xpi_gamecon.o
-KVERSION := 'uname -r'
 
 all:
-	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) modules
+	$(MAKE) -C $(KVERSION)/build modules
 
 clean:
-	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) clean
+	$(MAKE) -C $(KVERSION)/build clean
diff -ruN PiBoyControls-cc96ca5777c2a17b9b44c1bbc15f37cfbcdd465f.orig/xpi_gamecon.c PiBoyControls-cc96ca5777c2a17b9b44c1bbc15f37cfbcdd465f/xpi_gamecon.c
--- PiBoyControls-cc96ca5777c2a17b9b44c1bbc15f37cfbcdd465f.orig/xpi_gamecon.c	2020-12-13 05:12:53.000000000 +0000
+++ PiBoyControls-cc96ca5777c2a17b9b44c1bbc15f37cfbcdd465f/xpi_gamecon.c	2021-05-15 14:13:10.110013858 +0100
@@ -448,8 +448,8 @@
 
 	values.flags_val = 1;
 	values.fan_val = 10;
-	values.red_val = 100;
-	values.green_val = 100;
+	values.red_val = 2;
+	values.green_val = 2;
 
 	osd();
 
