diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index aa55399a9531..8f521d22cbe0 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -685,8 +685,7 @@ Params: swap_lr                 Reverse the channel allocation, which will also
         pins_12_13              Select GPIOs 12 & 13 (default)
         pins_18_19              Select GPIOs 18 & 19
         pins_40_41              Select GPIOs 40 & 41
-        pins_40_45              Select GPIOs 40 & 45 (don't use on BCM2711 - the
-                                pins are on different controllers)
+        pins_40_45              Select GPIOs 40 & 45
 
 
 Name:   balena-fin
diff --git a/arch/arm/boot/dts/overlays/audremap-overlay.dts b/arch/arm/boot/dts/overlays/audremap-overlay.dts
index a73cc95934ba..edacc498b11c 100644
--- a/arch/arm/boot/dts/overlays/audremap-overlay.dts
+++ b/arch/arm/boot/dts/overlays/audremap-overlay.dts
@@ -22,16 +22,12 @@ __overrides__ {
 		swap_lr = <&frag0>, "swap_lr?";
 		enable_jack = <&frag0>, "enable_jack?";
 		pins_12_13 = <&frag0>,"brcm,pins:0=12",
-		             <&frag0>,"brcm,pins:4=13",
-			     <&frag0>,"brcm,function:0=4";
+		             <&frag0>,"brcm,pins:4=13";
 		pins_18_19 = <&frag0>,"brcm,pins:0=18",
-		             <&frag0>,"brcm,pins:4=19",
-			     <&frag0>,"brcm,function:0=2";
+		             <&frag0>,"brcm,pins:4=19";
 		pins_40_41 = <&frag0>,"brcm,pins:0=40",
-		             <&frag0>,"brcm,pins:4=41",
-			     <&frag0>,"brcm,function:0=4";
+		             <&frag0>,"brcm,pins:4=41";
 		pins_40_45 = <&frag0>,"brcm,pins:0=40",
-		             <&frag0>,"brcm,pins:4=45",
-			     <&frag0>,"brcm,function:0=4";
+		             <&frag0>,"brcm,pins:4=45";
 	};
 };
