diff --git a/hid-lg4ff.c b/hid-lg4ff.c
index 30e9754..5be2586 100644
--- a/hid-lg4ff.c
+++ b/hid-lg4ff.c
@@ -2477,7 +2477,7 @@ int lg4ff_init(struct hid_device *hid)
 	spin_lock_init(&entry->timer_lock);
 
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 15, 0)
-	hrtimer_init(&entry->hrtimer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
+	hrtimer_setup(&entry->hrtimer, HRTIMER_NORESTART, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
 	entry->hrtimer.function = lg4ff_timer_hires;
 #else
 	hrtimer_setup(&entry->hrtimer, lg4ff_timer_hires, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
