diff --git a/public/xash3d_mathlib.c b/public/xash3d_mathlib.c
index 082d5ca2..e12b906e 100644
--- a/public/xash3d_mathlib.c
+++ b/public/xash3d_mathlib.c
@@ -143,7 +143,6 @@ void RoundUpHullSize( vec3_t size )
 =================
 rsqrt
 =================
-*/
 float rsqrt( float number )
 {
 	int	i;
@@ -159,7 +158,7 @@ float rsqrt( float number )
 	y = y * (1.5f - (x * y * y));	// first iteration
 
 	return y;
-}
+}*/
 
 void VectorVectors( const vec3_t forward, vec3_t right, vec3_t up )
 {
diff --git a/public/xash3d_mathlib.h b/public/xash3d_mathlib.h
index 98ac4164..3d8405f3 100644
--- a/public/xash3d_mathlib.h
+++ b/public/xash3d_mathlib.h
@@ -158,7 +158,7 @@ typedef struct mplane_s mplane_t;
 typedef struct mstudiobone_s mstudiobone_t;
 typedef struct mstudioanim_s mstudioanim_t;
 
-float rsqrt( float number );
+//float rsqrt( float number );
 uint16_t FloatToHalf( float v );
 float HalfToFloat( uint16_t h );
 void RoundUpHullSize( vec3_t size );
