diff --git a/Sources/Shared/Core/Logger.h b/Sources/Shared/Core/Logger.h
index 9435c7ff..26074b39 100644
--- a/Sources/Shared/Core/Logger.h
+++ b/Sources/Shared/Core/Logger.h
@@ -242,7 +242,7 @@ namespace Death { namespace Trace {
 			std::uint64_t tsc;
 			__asm__ volatile("stck %0" : "=Q" (tsc) : : "cc");
 			return tsc;
-#	elif (defined(_M_ARM) || defined(_M_ARM64) || defined(__PPC__) || defined(__PPC64__))
+#	elif (defined(_M_ARM) || defined(_M_ARM64) || defined(__PPC__) || defined(__PPC64__) || defined(__riscv))
 			return static_cast<std::uint64_t>(std::chrono::system_clock::now().time_since_epoch().count());
 #	else
 			return __rdtsc();
@@ -1353,4 +1353,4 @@ namespace Death { namespace Trace {
 
 }}
 
-#endif
\ No newline at end of file
+#endif
