diff --git a/Sources/Shared/Core/Logger.h b/Sources/Shared/Core/Logger.h
index 9adae2cb..7e83ac16 100644
--- a/Sources/Shared/Core/Logger.h
+++ b/Sources/Shared/Core/Logger.h
@@ -248,7 +248,7 @@ namespace Death { namespace Trace {
 			__asm__ volatile("stck %0" : "=Q" (tsc) : : "cc");
 			return tsc;
 		}
-#	elif (defined(_M_ARM) || defined(_M_ARM64) || defined(__PPC64__))
+#	elif (defined(_M_ARM) || defined(_M_ARM64) || defined(__PPC64__) || defined(__riscv))
 		DEATH_ALWAYS_INLINE std::uint64_t rdtsc() noexcept {
 			return static_cast<std::uint64_t>(std::chrono::system_clock::now().time_since_epoch().count());
 		}
@@ -1312,4 +1312,4 @@ namespace Death { namespace Trace {
 
 }}
 
-#endif
\ No newline at end of file
+#endif
