diff -Naur ionfury.orig/Common.mak ionfury/Common.mak
--- ionfury.orig/Common.mak	2021-09-05 05:58:03.000000000 +0100
+++ ionfury/Common.mak	2022-01-26 21:29:07.767023933 +0000
@@ -573,8 +573,8 @@
         else
             ifeq ($(PLATFORM),DARWIN)
                 OPTOPT := -march=core2 -mmmx -msse -msse2 -msse3 -mssse3
-            else
-                OPTOPT := -march=nocona
+#            else
+#                OPTOPT := -march=nocona
             endif
         endif
     endif
@@ -601,6 +601,9 @@
     ifeq ($(PLATFORM),WII)
         OPTOPT := -mtune=750
     endif
+    ifneq (0,$(RPI4))
+        OPTOPT := -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math
+    endif  
 endif
 
 ifeq ($(PACKAGE_REPOSITORY),0)
@@ -892,7 +895,7 @@
 
 ifeq ($(RENDERTYPE),SDL)
     ifeq ($(SDL_TARGET),2)
-        SDLCONFIG := sdl2-config
+        SDLCONFIG := sdl2
         SDLNAME := SDL2
     else ifeq ($(SDL_TARGET),1)
         SDLCONFIG := sdl-config
@@ -934,8 +937,8 @@
             SDLCONFIG_CFLAGS := $(strip $(subst -Dmain=SDL_main,,$(shell $(SDLCONFIG) --cflags)))
             SDLCONFIG_LIBS := $(strip $(subst -mwindows,,$(shell $(SDLCONFIG) --libs)))
 
-            COMPILERFLAGS += $(SDLCONFIG_CFLAGS)
-            LIBS += $(SDLCONFIG_LIBS)
+            COMPILERFLAGS += `pkg-config sdl2 --cflags`
+            LIBS += `pkg-config sdl2 --libs`
         else
             ifeq ($(SDL_TARGET),1)
                 COMPILERFLAGS += -D_GNU_SOURCE=1
diff -Naur ionfury.orig/GNUmakefile ionfury/GNUmakefile
--- ionfury.orig/GNUmakefile	2021-09-05 05:58:03.000000000 +0100
+++ ionfury/GNUmakefile	2022-01-26 21:30:01.747475480 +0000
@@ -519,7 +519,7 @@
 duke3d_editor_deps := audiolib
 
 duke3d_game := eduke32
-duke3d_editor := mapster32
+duke3d_editor := 
 
 ifneq (,$(APPBASENAME))
     duke3d_game := $(APPBASENAME)
