diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 7fee08c3d3b..d92ea39ef4d 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -100,6 +100,7 @@ AddReg=\
     VersionInfo,\
     LicenseInformation,\
     NVIDIANGX, \
+    BatoceraOverrides,\
     WineNonSteamController, \
     ProtonOverrides
 
@@ -130,6 +131,7 @@ AddReg=\
     VersionInfo,\
     LicenseInformation,\
     NVIDIANGX, \
+    BatoceraOverrides,\
     WineNonSteamController, \
     ProtonOverrides
 
@@ -176,6 +178,7 @@ AddReg=\
     VersionInfo,\
     LicenseInformation,\
     NVIDIANGX, \
+    BatoceraOverrides,\
     WineNonSteamController, \
     ProtonOverrides
 
@@ -3129,3 +3132,8 @@ HKCU,Software\Wine\AppDefaults\RiftApart.exe\DllOverrides,"atiadlxx",,"builtin"
 HKCU,Software\Wine\AppDefaults\ffxvi.exe\DllOverrides,"atiadlxx",,"builtin"
 HKLM,Software\Wow6432Node\lucasarts entertainment company llc\Star Wars: Episode I Racer\v1.0,"Display Height",0x10001,480
 HKLM,Software\Wow6432Node\lucasarts entertainment company llc\Star Wars: Episode I Racer\v1.0,"Display Width",0x10001,640
+
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+
