diff --git a/src/platform/sdl2/main.cpp b/src/platform/sdl2/main.cpp
index ce56353..c64d5a7 100644
--- a/src/platform/sdl2/main.cpp
+++ b/src/platform/sdl2/main.cpp
@@ -551,11 +551,12 @@ int main(int argc, char **argv) {
 
     SDL_ShowCursor(SDL_DISABLE);
 
+    strcpy(contentDir,  "/userdata/roms/ports/traider1/");
+
     const char *home;
     if (!(home = getenv("HOME")))
         home = getpwuid(getuid())->pw_dir;
-    strcat(cacheDir, home);
-    strcat(cacheDir, "/.openlara/");
+    strcat(cacheDir, "/userdata/system/configs/opernlara/");
 
     struct stat st = {0};
     if (stat(cacheDir, &st) == -1 && mkdir(cacheDir, 0777) == -1)
