diff --git a/src/platform/sdl2/main.cpp b/src/platform/sdl2/main.cpp
index 7ba7b7f..bf8897e 100644
--- a/src/platform/sdl2/main.cpp
+++ b/src/platform/sdl2/main.cpp
@@ -579,8 +579,7 @@ int main(int argc, char **argv) {
     const char *home;
     if (!(home = getenv("HOME")))
         home = getpwuid(getuid())->pw_dir;
-    strcat(cacheDir, home);
-    strcat(cacheDir, "/.openlara/");
+    strcat(cacheDir, "/userdata/system/configs/opernlara/");
 
     struct stat st = {0};
     if (stat(cacheDir, &st) == -1 && mkdir(cacheDir, 0777) == -1)
