diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index 7d565ead..a0de721b 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -7508,10 +7508,10 @@ static void init_amiberry_dirs(const bool portable_mode, const bool materialize_
 #else
 	const std::string amiberry_dir = "amiberry";
 #endif
-	config_path = get_config_directory(portable_mode);
-	current_dir = home_dir = get_home_directory(portable_mode);
-	data_dir = get_data_directory(portable_mode);
-	plugins_dir = get_plugins_directory(portable_mode);
+	config_path = "/userdata/system/configs/amiberry/conf";
+	current_dir = "/userdata/system/configs/amiberry";
+	data_dir = "/usr/share/amiberry/data/";
+	plugins_dir = "/usr/share/amiberry/plugins/";
 	const auto visual_content_root = home_dir;
 
 #ifdef __MACH__
@@ -7623,27 +7623,27 @@ static void init_amiberry_dirs(const bool portable_mode, const bool materialize_
 	nvram_dir.append("\\Nvram\\");
 	video_dir.append("\\Videos\\");
 #else
-	controllers_path.append("/controllers/");
-	whdboot_path.append("/whdboot/");
+	controllers_path = "/userdata/system/configs/amiberry/conf/retroarch/inputs";
+	whdboot_path = "/usr/share/amiberry/whdboot/";
 	whdload_arch_path.append("/lha/");
 	floppy_path.append("/floppies/");
 	harddrive_path.append("/harddrives/");
 	cdrom_path.append("/cdroms/");
-	logfile_path.append("/amiberry.log");
-	rom_path.append("/roms/");
+	logfile_path = "/usrdata/system/logs/amiberry.log";
+	rom_path = "/userdata/bios/amiga";
 	rp9_path.append("/rp9/");
-	saveimage_dir.append("/");
-	savestate_dir.append("/savestates/");
+	saveimage_dir = "/userdata/saves/amiga/";
+	savestate_dir = "/userdata/saves/amiga/";
 	ripper_path.append("/ripper/");
 	input_dir.append("/inputrecordings/");
-	screenshot_dir.append("/screenshots/");
+	screenshot_dir = "/userdata/screenshots/";
 	nvram_dir.append("/nvram/");
 	video_dir.append("/videos/");
 #endif
 
 	apply_current_visual_asset_paths(get_visual_asset_paths_from_content_root(visual_content_root));
 
-	retroarch_file = config_path;
+	retroarch_file = controllers_path;
 #ifdef _WIN32
 	retroarch_file.append("\\retroarch.cfg");
 #elif defined(__ANDROID__)
