diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index ff7c2432..55009402 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -3075,11 +3075,11 @@ static int get_env_dir( char * path, const char *path_template, const char *envn
 static void init_amiberry_paths(void)
 {
 	strncpy(current_dir, start_path_data, MAX_DPATH - 1);
-	snprintf(config_path, MAX_DPATH, "%s/conf/", start_path_data);
-	snprintf(controllers_path, MAX_DPATH, "%s/controllers/", start_path_data);
-	snprintf(retroarch_file, MAX_DPATH, "%s/conf/retroarch.cfg", start_path_data);
-	snprintf(whdboot_path, MAX_DPATH, "%s/whdboot/", start_path_data);
-	snprintf(logfile_path, MAX_DPATH, "%s/amiberry.log", start_path_data);
+	snprintf(config_path, MAX_DPATH, "/userdata/system/configs/amiberry/conf");
+	snprintf(controllers_path, MAX_DPATH, "/userdata/system/configs/amiberry/conf/retroarch/inputs");
+	snprintf(retroarch_file, MAX_DPATH, "/userdata/system/configs/amiberry/conf/retroarch/retroarchcustom.cfg");
+	snprintf(whdboot_path, MAX_DPATH, "/userdata/system/configs/amiberry/whdboot/");
+	snprintf(logfile_path, MAX_DPATH, "/userdata/system/configs/amiberry/amiberry.log");
 
 #ifdef ANDROID
 	char afepath[MAX_DPATH];
@@ -3090,9 +3090,9 @@ static void init_amiberry_paths(void)
 		closedir(afedir);
 	}
 	else
-		snprintf(rom_path, MAX_DPATH, "%s/kickstarts/", start_path_data);
+		snprintf(rom_path, MAX_DPATH, "/userdata/bios/");
 #else
-	snprintf(rom_path, MAX_DPATH, "%s/kickstarts/", start_path_data);
+	snprintf(rom_path, MAX_DPATH, "/userdata/bios/");
 #endif
 	snprintf(rp9_path, MAX_DPATH, "%s/rp9/", start_path_data);
 #ifdef __MACH__
@@ -3101,15 +3101,15 @@ static void init_amiberry_paths(void)
 	snprintf(amiberry_conf_file, MAX_DPATH, "%s", (macos_home_directory + "/Documents/Amiberry/Configurations/amiberry.conf").c_str());
 	printf("Using configuration: %s\n", (macos_home_directory + "/Documents/Amiberry/Configurations/amiberry.conf").c_str());
 #else
-	snprintf(amiberry_conf_file, MAX_DPATH, "%s/conf/amiberry.conf", start_path_data);
+    snprintf(amiberry_conf_file, MAX_DPATH, "/userdata/system/configs/amiberry/conf/amiberry.conf");
 #endif
-	snprintf(floppy_sounds_dir, MAX_DPATH, "%s/data/floppy_sounds/", start_path_data);
-	snprintf(data_dir, MAX_DPATH, "%s/data/", start_path_data);
-	snprintf(saveimage_dir, MAX_DPATH, "%s/savestates/", start_path_data);
-	snprintf(savestate_dir, MAX_DPATH, "%s/savestates/", start_path_data);
-	snprintf(ripper_path, MAX_DPATH, "%s/ripper/", start_path_data);
-	snprintf(input_dir, MAX_DPATH, "%s/inputrecordings/", start_path_data);
-	snprintf(screenshot_dir, MAX_DPATH, "%s/screenshots/", start_path_data);
+	snprintf(floppy_sounds_dir, MAX_DPATH, "/usr/share/amiberry/data/floppy_sounds/");
+	snprintf(data_dir, MAX_DPATH, "/usr/share/amiberry/data/");
+	snprintf(saveimage_dir, MAX_DPATH, "/userdata/saves/amiga/");
+	snprintf(savestate_dir, MAX_DPATH, "/userdata/saves/amiga/");
+	snprintf(ripper_path, MAX_DPATH, "/userdata/saves/amiga/ripper/");
+	snprintf(input_dir, MAX_DPATH, "/userdata/screenshots//inputrecordings/");
+	snprintf(screenshot_dir, MAX_DPATH, "/userdata/screenshots/");
 }
 
 void load_amiberry_settings(void)
