diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index 3328d830..89ad5e51 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -4202,7 +4202,7 @@ void create_missing_amiberry_folders()
 #ifdef __MACH__
         const std::string default_roms_path = app_directory + "/Resources/roms/";
 #else
-        const std::string default_roms_path = AMIBERRY_DATADIR "/roms/";
+        const std::string default_roms_path = "/userdata/bios/amiga";
 #endif
         // copy default kickstart files, if they exist in AMIBERRY_DATADIR/roms
         if (my_existsdir(default_roms_path.c_str()))
@@ -4237,9 +4237,9 @@ static void init_amiberry_dirs()
 	const std::string amiberry_dir = "amiberry";
 #endif
 	current_dir = home_dir = get_home_directory();
-    data_dir = get_data_directory();
-    config_path = get_config_directory();
-    plugins_dir = get_plugins_directory();
+    data_dir = "/usr/share/amiberry/data/";
+    config_path = "/userdata/system/configs/amiberry/conf";
+    plugins_dir = "/usr/share/amiberry/plugins/";
 
 	std::string xdg_data_home = get_xdg_data_home();
 	xdg_data_home += "/" + amiberry_dir;
@@ -4282,25 +4282,25 @@ static void init_amiberry_dirs()
     nvram_dir.append("/Nvram/");
     video_dir.append("/Videos/");
 #else
-	controllers_path.append("/controllers/");
-	whdboot_path.append("/whdboot/");
+	controllers_path = "/userdata/system/configs/amiberry/conf/retroarch/inputs";
+	whdboot_path = "/usr/share/amiberry/whdboot/";
 	whdload_arch_path.append("/lha/");
 	floppy_path.append("/floppies/");
 	harddrive_path.append("/harddrives/");
 	cdrom_path.append("/cdroms/");
-	logfile_path.append("/amiberry.log");
-	rom_path.append("/roms/");
+	logfile_path = "/usrdata/system/logs/amiberry.log";
+	rom_path = "/userdata/bios/amiga";
 	rp9_path.append("/rp9/");
-	saveimage_dir.append("/savestates/");
-	savestate_dir.append("/savestates/");
+	saveimage_dir = "/userdata/saves/amiga/";
+	savestate_dir = "/userdata/saves/amiga/";
 	ripper_path.append("/ripper/");
 	input_dir.append("/inputrecordings/");
-	screenshot_dir.append("/screenshots/");
+	screenshot_dir = "/userdata/screenshots/";
 	nvram_dir.append("/nvram/");
 	video_dir.append("/videos/");
 #endif
 
-	retroarch_file = config_path;
+	retroarch_file = controllers_path;
 	retroarch_file.append("/retroarch.cfg");
 
 	floppy_sounds_dir = data_dir;
