diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index dac0e7a0..d4a5f043 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -4019,22 +4019,22 @@ static void init_amiberry_paths(void)
 		input_dir = screenshot_dir = nvram_dir = plugins_dir = video_dir = 
 		start_path_data;
 
-	config_path.append("/conf/");
-	controllers_path.append("/controllers/");
-	data_dir.append("/data/");
-	whdboot_path.append("/whdboot/");
+	config_path = "/userdata/system/configs/amiberry/conf/";
+	controllers_path = "/userdata/system/configs/amiberry/conf/retroarch/inputs";
+	data_dir = "/usr/share/amiberry/data/";
+	whdboot_path = "/userdata/system/configs/amiberry/whdboot/";
 	whdload_arch_path.append("/lha/");
 	floppy_path.append("/floppies/");
 	harddrive_path.append("/harddrives/");
 	cdrom_path.append("/cdroms/");
-	logfile_path.append("/amiberry.log");
-	rom_path.append("/kickstarts/");
+	logfile_path = "/usrdata/system/logs/amiberry.log";
+	rom_path = "/userdata/bios/amiga";
 	rp9_path.append("/rp9/");
-	saveimage_dir.append("/savestates/");
-	savestate_dir.append("/savestates/");
+	saveimage_dir = "/userdata/saves/amiga/";
+	savestate_dir = "/userdata/saves/amiga/";
 	ripper_path.append("/ripper/");
 	input_dir.append("/inputrecordings/");
-	screenshot_dir.append("/screenshots/");
+	screenshot_dir = "/userdata/screenshots/";
 	nvram_dir.append("/nvram/");
 	plugins_dir.append("/plugins/");
 	video_dir.append("/videos/");
@@ -4043,7 +4043,7 @@ static void init_amiberry_paths(void)
 	amiberry_conf_file.append("amiberry.conf");
 #endif
 
-	retroarch_file = config_path;
+	retroarch_file = controllers_path;
 	retroarch_file.append("retroarch.cfg");
 
 	floppy_sounds_dir = data_dir;
