diff --git a/src/include/options.h b/src/include/options.h
index ccf22c3a..ed615420 100644
--- a/src/include/options.h
+++ b/src/include/options.h
@@ -1385,7 +1385,7 @@ struct amiberry_options
 	bool default_auto_crop = false;
 	int default_width = 720;
 	int default_height = 568;
-	int default_fullscreen_mode = 0;
+	int default_fullscreen_mode = 1;
 	int default_stereo_separation = 7;
 	int default_sound_buffer = 3072;
 	int default_sound_frequency = 44100;
