diff --git a/src/osdep/amiberry_gui.cpp b/src/osdep/amiberry_gui.cpp
index 3399826c..b56fbf60 100644
--- a/src/osdep/amiberry_gui.cpp
+++ b/src/osdep/amiberry_gui.cpp
@@ -296,7 +296,6 @@ void SymlinkROMs()
 
 void RescanROMs()
 {
-	vector<string> dirs;
 	vector<string> files;
 	char path[MAX_DPATH];
 
@@ -306,28 +305,13 @@ void RescanROMs()
 	get_rom_path(path, MAX_DPATH);
 
 	load_keyring(&changed_prefs, path);
-	read_directory(path, &dirs, &files);
-
-	// Root level scan
+	read_directory(path, nullptr, &files);
 	for (auto & file : files)
 	{
-		std::string tmp_path = std::string(path).append(file);
-		scan_rom(tmp_path);
-	}
-
-	// Recursive scan
-	for (auto & dir : dirs)
-	{
-		if (dir != "..")
-		{
-			std::string full_path = std::string(path).append(dir);
-			read_directory(full_path.c_str(), nullptr, &files);
-			for (auto & file : files)
-			{
-				std::string tmp_path = full_path;
-				scan_rom(tmp_path.append("/").append(file));
-			}
-		}
+		char tmppath[MAX_DPATH];
+		strncpy(tmppath, path, MAX_DPATH - 1);
+		strncat(tmppath, file.c_str(), MAX_DPATH - 1);
+		scan_rom(tmppath);
 	}
 
 	auto id = 1;
