diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index 98fef0b7..125b4ca8 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -4515,8 +4515,10 @@ static bool locate_amiberry_conf(const bool portable_mode)
 		if (!my_existsdir(xdg_config_home.c_str()))
 			my_mkdir(xdg_config_home.c_str());
 
-		amiberry_conf_file = xdg_config_home + "/amiberry.conf";
-		amiberry_ini_file = xdg_config_home + "/amiberry.ini";
+		//amiberry_conf_file = xdg_config_home + "/amiberry.conf";
+		//amiberry_ini_file = xdg_config_home + "/amiberry.ini";
+		amiberry_conf_file = std::string("/userdata/system/configs/amiberry/conf/amiberry.conf");
+		amiberry_ini_file = std::string("/userdata/system/configs/amiberry/conf/amiberry.ini");
 	}
 	return my_existsfile2(amiberry_conf_file.c_str());
 }
