diff --git a/src/floppybridge/CommonBridgeTemplate.h b/src/floppybridge/CommonBridgeTemplate.h
index e17558cc..2dc157f3 100644
--- a/src/floppybridge/CommonBridgeTemplate.h
+++ b/src/floppybridge/CommonBridgeTemplate.h
@@ -36,7 +36,9 @@
 #include <thread>
 #include <functional>
 #include <queue>
+#include <string>
 #include <vector>
+#include <cstdint>
 #include <deque>
 #include <mutex>
 #include <condition_variable>
@@ -541,4 +543,4 @@ public:
 
 };
 
-#endif
\ No newline at end of file
+#endif
diff --git a/src/floppybridge/FloppyBridge.h b/src/floppybridge/FloppyBridge.h
index 91c409d9..a8a45a04 100644
--- a/src/floppybridge/FloppyBridge.h
+++ b/src/floppybridge/FloppyBridge.h
@@ -29,7 +29,9 @@
 // that uses this DLL. This way any other project whose source files include this file see
 // FLOPPYBRIDGE_API functions as being imported from a DLL, whereas this DLL sees symbols
 // defined with this macro as being exported.
-
+#include <string>
+#include <vector>
+#include <cstdint>
 
 #ifdef _WIN32
 
diff --git a/src/floppybridge/SerialIO.h b/src/floppybridge/SerialIO.h
index 69b4ce73..6769b258 100644
--- a/src/floppybridge/SerialIO.h
+++ b/src/floppybridge/SerialIO.h
@@ -27,6 +27,8 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
+
 
 #ifdef _WIN32
 #ifdef USING_MFC
@@ -151,4 +153,4 @@ public:
 };
  
 
-#endif
\ No newline at end of file
+#endif
diff --git a/src/floppybridge/floppybridge_config.h b/src/floppybridge/floppybridge_config.h
index 4232b907..e02dbc53 100644
--- a/src/floppybridge/floppybridge_config.h
+++ b/src/floppybridge/floppybridge_config.h
@@ -35,7 +35,6 @@
 	ROMTYPE_FLOPYBRDGEF 
 */
 
-
 // DrawBridge aka Arduino floppy reader/writer
 #define ROMTYPE_ARDUINOREADER_WRITER			ROMTYPE_FLOPYBRDGE0
 #include "ArduinoFloppyBridge.h"
@@ -118,4 +117,4 @@
 #endif
 
 
-#endif
\ No newline at end of file
+#endif
