diff --git a/Makefile b/Makefile
index 610abcaa..ee437d25 100644
--- a/Makefile
+++ b/Makefile
@@ -238,6 +238,12 @@ else ifeq ($(PLATFORM),rpi5-64-sdl2)
      CPPFLAGS += $(CPPFLAGS64)
      AARCH64 = 1
 
+# Raspberry Pi 5 (SDL2 64-bit)
+else ifeq ($(PLATFORM),rpi5-64-sdl2)
+	CPUFLAGS = -mcpu=cortex-a76+crc+simd+fp
+	CPPFLAGS += $(CPPFLAGS64)
+	AARCH64 = 1
+
 # Vero 4k (SDL2)
 else ifeq ($(PLATFORM),vero4k)
 	CPUFLAGS = -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
