diff --git a/src/caps/caps_amiberry.cpp b/src/caps/caps_amiberry.cpp
index 011a0b5e..7fd1c7e0 100644
--- a/src/caps/caps_amiberry.cpp
+++ b/src/caps/caps_amiberry.cpp
@@ -67,11 +67,7 @@ int caps_init (void)
 
 	if (init)
 		return 1;
-#ifdef ANDROID
-	UAE_DLHANDLE h = uae_dlopen_plugin(_T("libcapsimage.so"));
-#else
-	UAE_DLHANDLE h = uae_dlopen_plugin(_T("capsimg"));
-#endif
+	UAE_DLHANDLE h = uae_dlopen(_T("libcapsimage.so.5"));
 	if (!h) {
 		if (noticed)
 			return 0;
