diff --git a/src/floppybridge/CommonBridgeTemplate.h b/src/floppybridge/CommonBridgeTemplate.h
index e70bbede..2c71d62c 100644
--- a/src/floppybridge/CommonBridgeTemplate.h
+++ b/src/floppybridge/CommonBridgeTemplate.h
@@ -32,7 +32,9 @@
 #include <thread>
 #include <functional>
 #include <queue>
+#include <string>
 #include <vector>
+#include <cstdint>
 #include <deque>
 #include <mutex>
 #include <condition_variable>
diff --git a/src/floppybridge/FloppyBridge.h b/src/floppybridge/FloppyBridge.h
index c3dd4c05..7083f4db 100644
--- a/src/floppybridge/FloppyBridge.h
+++ b/src/floppybridge/FloppyBridge.h
@@ -28,7 +28,9 @@
 // that uses this DLL. This way any other project whose source files include this file see
 // FLOPPYBRIDGE_API functions as being imported from a DLL, whereas this DLL sees symbols
 // defined with this macro as being exported.
-
+#include <string>
+#include <vector>
+#include <cstdint>
 
 #ifdef _WIN32
 
diff --git a/src/floppybridge/SerialIO.h b/src/floppybridge/SerialIO.h
index abb43858..153b30a9 100644
--- a/src/floppybridge/SerialIO.h
+++ b/src/floppybridge/SerialIO.h
@@ -26,6 +26,8 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
+
 
 #ifdef _WIN32
 #ifdef USING_MFC
