diff --git a/cmake/StandardProjectSettings.cmake b/cmake/StandardProjectSettings.cmake
index aa9299d6..73001e36 100644
--- a/cmake/StandardProjectSettings.cmake
+++ b/cmake/StandardProjectSettings.cmake
@@ -74,7 +74,6 @@ if(AMIBERRY_GNU_LIKE_COMPILER)
             # GNU ld-only flags (do not work on FreeBSD or Windows)
             if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
                 list(APPEND AMIBERRY_LINK_OPTIONS
-                    "-Wl,--sort-common=descending"
                     "-Wl,--hash-style=gnu"
                 )
             endif()
