diff --git a/external/floppybridge/CMakeLists.txt b/external/floppybridge/CMakeLists.txt
index cb439f96..87d77938 100644
--- a/external/floppybridge/CMakeLists.txt
+++ b/external/floppybridge/CMakeLists.txt
@@ -26,6 +26,6 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     message(STATUS "Linux platform detected")
     add_custom_command(TARGET floppybridge POST_BUILD
-        COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:floppybridge> ../../../plugins/
+        COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:floppybridge> ../../plugins/
     )
 endif ()
