diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index acac049b15f..8a9ced0e0d4 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -73,7 +73,8 @@ AddReg=\
     SessionMgr,\
     Tapi,\
     Timezones,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.NT]
 RegisterDlls=RegisterDllsSection
@@ -100,6 +101,7 @@ AddReg=\
     Timezones,\
     VersionInfo,\
     LicenseInformation,\
+    BatoceraOverrides,\
     LutrisClient, \
     NVIDIANGX
 
@@ -129,6 +131,7 @@ AddReg=\
     Timezones,\
     VersionInfo.ntamd64,\
     LicenseInformation,\
+    BatoceraOverrides,\
     LutrisClient.ntamd64, \
     NVIDIANGX
 
@@ -156,7 +159,12 @@ AddReg=\
     Tapi,\
     Timezones,\
     VersionInfo.ntamd64,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides,\
+    
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
 
 [Wow64Install]
 RegisterDlls=RegisterDllsSection
