diff --git a/Oxygen/sonic3air/build/_cmake/CMakeLists.txt b/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
index 2a43f7db..6e356421 100644
--- a/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
+++ b/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
@@ -62,8 +62,8 @@ set(SDL_SHARED OFF)
 add_subdirectory(${WORKSPACE_DIR}/framework/external/sdl/SDL2 SDL)
 
 # zlib sources as virtual subdirectory "zlib"
-include_directories(${WORKSPACE_DIR}/framework/external/zlib/zlib)		# Included before add_subdirectory so that the examples will build
-add_subdirectory(${WORKSPACE_DIR}/framework/external/zlib/zlib zlib)
+#include_directories(${WORKSPACE_DIR}/framework/external/zlib/zlib)		# Included before add_subdirectory so that the examples will build
+#add_subdirectory(${WORKSPACE_DIR}/framework/external/zlib/zlib zlib)
 
 
 
@@ -71,7 +71,7 @@ include_directories(SDL/include)
 include_directories(${WORKSPACE_DIR}/framework/external/ogg-vorbis/libogg/include)
 include_directories(${WORKSPACE_DIR}/framework/external/ogg-vorbis/libvorbis/include)
 include_directories(${WORKSPACE_DIR}/framework/external/ogg-vorbis/libvorbis/lib)
-include_directories(build/zlib)		# Needed for zconf.h
+#include_directories(build/zlib)		# Needed for zconf.h
 include_directories(${WORKSPACE_DIR}/framework/external/zlib/zlib/contrib/minizip)
 include_directories(${WORKSPACE_DIR}/librmx/source)
 include_directories(${WORKSPACE_DIR}/librmx/source/rmxmedia/_glew)
@@ -96,13 +96,13 @@ add_library(oggvorbis ${OGGVORBIS_SOURCES})
 
 # minizip
 
-file(GLOB MINIZIP_SOURCES ${WORKSPACE_DIR}/framework/external/zlib/zlib/contrib/minizip/ioapi.c
-						  ${WORKSPACE_DIR}/framework/external/zlib/zlib/contrib/minizip/unzip.c
-						  ${WORKSPACE_DIR}/framework/external/zlib/zlib/contrib/minizip/zip.c)
+#file(GLOB MINIZIP_SOURCES ${WORKSPACE_DIR}/framework/external/zlib/zlib/contrib/minizip/ioapi.c
+#						  ${WORKSPACE_DIR}/framework/external/zlib/zlib/contrib/minizip/unzip.c
+#						  ${WORKSPACE_DIR}/framework/external/zlib/zlib/contrib/minizip/zip.c)
 
-add_library(minizip ${MINIZIP_SOURCES})
+#add_library(minizip ${MINIZIP_SOURCES})
 
-target_link_libraries(minizip zlibstatic)
+#target_link_libraries(minizip zlibstatic)
 
 
 
@@ -117,7 +117,7 @@ if (NOT CMAKE_VERSION VERSION_LESS "3.16.0")
 endif()
 
 target_link_libraries(rmxbase stdc++fs)
-target_link_libraries(rmxbase zlibstatic)
+target_link_libraries(rmxbase z)
 
 
 
