diff --git a/Oxygen/sonic3air/build/_cmake/CMakeLists.txt b/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
index 6e0943b3..a4754b6b 100644
--- a/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
+++ b/Oxygen/sonic3air/build/_cmake/CMakeLists.txt
@@ -12,9 +12,9 @@ project(Sonic3AIR)
 # Build options
 option(BUILD_OXYGEN_ENGINEAPP "Build the Oxygen App executable" ON)
 option(BUILD_OXYGEN_SERVER "Build the Oxygen server executable" OFF)
-option(BUILD_SDL_STATIC "Build SDL as a static library instead of a shared / dynamic one" ON)
+option(BUILD_SDL_STATIC "Build SDL as a static library instead of a shared / dynamic one" OFF)
 option(USE_GLES "Use OpenGLESv2" OFF)
-option(USE_DISCORD "Use Discord API" ON)
+option(USE_DISCORD "Use Discord API" OFF)
 option(USE_IMGUI "Use ImGui, required for dev mode and system windows" ON)
 
 
@@ -96,13 +96,13 @@ if (BUILD_SDL_STATIC)
 else()
 	set(SDL_STATIC OFF)
 	set(SDL_SHARED ON)
-	if (UNIX AND CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm|aarch64)")
+#	if (UNIX AND CMAKE_SYSTEM_PROCESSOR MATCHES "^(arm|aarch64)")
 		include(FindPkgConfig)
 		pkg_check_modules(SDL2 sdl2 REQUIRED)
-	endif()
+#	endif()
 endif()
 
-add_subdirectory(${WORKSPACE_DIR}/framework/external/sdl/SDL2 SDL)
+#add_subdirectory(${WORKSPACE_DIR}/framework/external/sdl/SDL2 SDL)
 
 
 # batocera - use system zlib
diff --git a/framework/external/imgui/imgui/backends/imgui_impl_sdl2.cpp b/framework/external/imgui/imgui/backends/imgui_impl_sdl2.cpp
index c2dae405..47485809 100644
--- a/framework/external/imgui/imgui/backends/imgui_impl_sdl2.cpp
+++ b/framework/external/imgui/imgui/backends/imgui_impl_sdl2.cpp
@@ -113,8 +113,8 @@
 #endif
 
 // SDL
-#include <SDL.h>
-#include <SDL_syswm.h>
+#include <SDL2/SDL.h>
+#include <SDL2/SDL_syswm.h>
 #include <stdio.h>              // for snprintf()
 #ifdef __APPLE__
 #include <TargetConditionals.h>
@@ -133,7 +133,7 @@
 #define SDL_HAS_VULKAN                      SDL_VERSION_ATLEAST(2,0,6)
 #define SDL_HAS_OPEN_URL                    SDL_VERSION_ATLEAST(2,0,14)
 #if SDL_HAS_VULKAN
-#include <SDL_vulkan.h>
+#include <SDL2/SDL_vulkan.h>
 #endif
 
 // SDL Data
