diff --git a/Core/HW/MediaEngine.cpp b/Core/HW/MediaEngine.cpp
index b87d47a8f..7095be1d2 100644
--- a/Core/HW/MediaEngine.cpp
+++ b/Core/HW/MediaEngine.cpp
@@ -770,10 +770,10 @@ inline void writeVideoLineRGBA(void *destp, const void *srcp, int width) {
 		count -= 8;
 	}
 #elif PPSSPP_ARCH(ARM_NEON)
-	int32x4_t mask = vdupq_n_u32(0x00FFFFFF);
+	uint32x4_t mask = vdupq_n_u32(0x00FFFFFF);
 	while (count >= 8) {
-		int32x4_t pixels1 = vandq_u32(vld1q_u32(src), mask);
-		int32x4_t pixels2 = vandq_u32(vld1q_u32(src + 4), mask);
+		uint32x4_t pixels1 = vandq_u32(vld1q_u32(src), mask);
+		uint32x4_t pixels2 = vandq_u32(vld1q_u32(src + 4), mask);
 		vst1q_u32(dest, pixels1);
 		vst1q_u32(dest + 4, pixels2);
 		src += 8;
