diff --git a/sdl/scrnmng.c b/sdl/scrnmng.c
index 21e1807..e9f8b88 100755
--- a/sdl/scrnmng.c
+++ b/sdl/scrnmng.c
@@ -6,8 +6,6 @@
 #include	<pccore.h>
 #include	"np2.h"
 
-SCRNMNG		scrnmng;
-
 #if defined(SUPPORT_WAB)
 #include <wab/wab.h>
 #endif
@@ -19,6 +17,22 @@ SCRNMNG		scrnmng;
 extern retro_environment_t environ_cb;
 #endif	/* __LIBRETRO__ */
 
+typedef struct {
+	BOOL	enable;
+	int		width;
+	int		height;
+	int		bpp;
+#if defined(__LIBRETRO__)
+	void*	pc98surf;
+	void*	dispsurf;
+#else	/* __LIBRETRO__ */
+	SDL_Surface* pc98surf;
+	SDL_Surface* dispsurf;
+#endif	/* __LIBRETRO__ */
+	VRAMHDL vram;
+} SCRNMNG;
+
+static SCRNMNG scrnmng;
 static SCRNSURF scrnsurf;
 
 #if !defined(__LIBRETRO__)
diff --git a/sdl/scrnmng.h b/sdl/scrnmng.h
index 93af0c7..50cd212 100755
--- a/sdl/scrnmng.h
+++ b/sdl/scrnmng.h
@@ -20,22 +20,6 @@ typedef struct {
 	int	extend;
 } SCRNSURF;
 
-typedef struct {
-	BOOL	enable;
-	int		width;
-	int		height;
-	int		bpp;
-	int		flag;
-#if defined(__LIBRETRO__)
-	void*	pc98surf;
-	void*	dispsurf;
-#else	/* __LIBRETRO__ */
-	SDL_Surface* pc98surf;
-	SDL_Surface* dispsurf;
-#endif	/* __LIBRETRO__ */
-	VRAMHDL vram;
-} SCRNMNG;
-
 enum {
 	SCRNMODE_FULLSCREEN	= 0x01,
 	SCRNMODE_HIGHCOLOR	= 0x02,
@@ -56,8 +40,6 @@ enum {
 extern "C" {
 #endif
 
-extern	SCRNMNG		scrnmng;
-
 void scrnmng_getsize(int* pw, int* ph);
 void scrnmng_setwidth(int posx, int width);
 #define scrnmng_setextend(e)
@@ -65,7 +47,7 @@ void scrnmng_setheight(int posy, int height);
 const SCRNSURF *scrnmng_surflock(void);
 void scrnmng_surfunlock(const SCRNSURF *surf);
 
-#define	scrnmng_isfullscreen()	(scrnmng.flag & SCRNFLAG_FULLSCREEN)
+#define	scrnmng_isfullscreen()	(0)
 #define	scrnmng_haveextend()	(0)
 #define	scrnmng_getbpp()		(16)
 #define	scrnmng_allflash()		
diff --git a/sdl/sysmenu.c b/sdl/sysmenu.c
index 7c8edc4..b0ba2dd 100755
--- a/sdl/sysmenu.c
+++ b/sdl/sysmenu.c
@@ -295,15 +295,6 @@ static void sys_cmd(MENUID id) {
 		case MID_SCSI3EJECT:
 			diskdrv_setsxsi(0x23, NULL);
 			break;
-#endif
-#if SDL_MAJOR_VERSION != 1
-		case MID_FULLSCREEN:
-			changescreen(scrnmode | SCRNMODE_FULLSCREEN);
-			break;
-
-		case MID_WINDOW:
-			changescreen(scrnmode & ~SCRNMODE_FULLSCREEN);
-			break;
 #endif
 		case MID_ROLNORMAL:
 			changescreen((scrnmode & ~SCRNMODE_ROTATEMASK) | 0);
@@ -1019,8 +1010,6 @@ BRESULT sysmenu_menuopen(UINT menutype, int x, int y) {
 #if defined(SUPPORT_ASYNC_CPU)
 	menusys_setcheck(MID_ASYNCCPU, (np2cfg.asynccpu & 1));
 #endif
-	menusys_setcheck(MID_FULLSCREEN, ((scrnmode & SCRNMODE_FULLSCREEN) == 1));
-	menusys_setcheck(MID_WINDOW, ((scrnmode & SCRNMODE_FULLSCREEN) == 0));
 	menusys_setcheck(MID_ROLNORMAL, ((scrnmode & SCRNMODE_ROTATEMASK) == 0));
 	menusys_setcheck(MID_ROLLEFT,   ((scrnmode & SCRNMODE_ROTATEMASK) == SCRNMODE_ROTATELEFT));
 	menusys_setcheck(MID_ROLRIGHT,  ((scrnmode & SCRNMODE_ROTATEMASK) == SCRNMODE_ROTATERIGHT));
diff --git a/sdl/sysmenu.str b/sdl/sysmenu.str
index 1229975..c0e4886 100755
--- a/sdl/sysmenu.str
+++ b/sdl/sysmenu.str
@@ -365,15 +365,10 @@ static const MSYSITEM s_vf1[] = {
 #endif
 
 static const MSYSITEM s_screen[] = {
-#if SDL_MAJOR_VERSION != 1
-		{str_window,		NULL,		MID_WINDOW,		0},
-		{str_fullscrn,		NULL,		MID_FULLSCREEN,		0},
-		{NULL,				NULL,		MID_STATIC,			MENU_SEPARATOR},
-#endif
-#if SDL_MAJOR_VERSION == 1
-		{str_window,		NULL,		MID_WINDOW,		MENU_GRAY},
+		{str_window,		NULL,		MID_WINDOW,			MENU_GRAY},
 		{str_fullscrn,		NULL,		MID_FULLSCREEN,		MENU_GRAY},
 		{NULL,				NULL,		MID_STATIC,			MENU_SEPARATOR},
+#if SDL_MAJOR_VERSION == 1
 		{str_normal,		NULL,		MID_ROLNORMAL,		MENU_GRAY},
 		{str_leftrol,		NULL,		MID_ROLLEFT,		MENU_GRAY},
 		{str_rightrol,		NULL,		MID_ROLRIGHT,		MENU_GRAY},
diff --git a/sdl/taskmng.c b/sdl/taskmng.c
index 27aa9aa..1354d16 100755
--- a/sdl/taskmng.c
+++ b/sdl/taskmng.c
@@ -240,18 +240,6 @@ void taskmng_rol(void) {
 					mousemng_toggle(MOUSEPROC_SYSTEM);
 				}
 			}
-#endif
-#if SDL_MAJOR_VERSION != 1
-			/* use command-f to toggle fullscreen/windowed mode */
-			if ((e.key.keysym.mod == KMOD_LGUI) || (e.key.keysym.mod == KMOD_RGUI)) {
-				if (e.key.keysym.scancode == SDL_SCANCODE_F) {
-					if (!scrnmng_isfullscreen()) {
-						changescreen(scrnmode | SCRNMODE_FULLSCREEN);
-					} else {
-						changescreen(scrnmode & ~SCRNMODE_FULLSCREEN);
-					}
-				}
-			}
 #endif
 			else {
 #if SDL_MAJOR_VERSION == 1
