diff -Nuar orig/kernel/nvidia/nv.c new/kernel/nvidia/nv.c
--- orig/kernel/nvidia/nv.c	2024-03-29 16:20:57.086314073 +0000
+++ new/kernel/nvidia/nv.c	2024-03-29 16:20:00.949259477 +0000
@@ -36,7 +36,7 @@
 #include "nv-kthread-q.h"
 #include "nv-pat.h"
 
-#if !defined(CONFIG_RETPOLINE)
+#if !defined(CONFIG_RETPOLINE) && !defined(CONFIG_MITIGATION_RETPOLINE)
 #include "nv-retpoline.h"
 #endif
 
diff -Nuar orig/kernel/nvidia-modeset/nvidia-modeset-linux.c new/kernel/nvidia-modeset/nvidia-modeset-linux.c
--- orig/kernel/nvidia-modeset/nvidia-modeset-linux.c	2024-03-29 16:21:01.262392555 +0000
+++ new/kernel/nvidia-modeset/nvidia-modeset-linux.c	2024-03-29 16:20:21.433644212 +0000
@@ -45,7 +45,7 @@
 #include <linux/sched/signal.h>
 #endif
 
-#if !defined(CONFIG_RETPOLINE)
+#if !defined(CONFIG_RETPOLINE) && !defined(CONFIG_MITIGATION_RETPOLINE)
 #include "nv-retpoline.h"
 #endif
 
