diff --git a/engine/Makefile b/engine/Makefile
index 7f1792ae..4a6639b3 100644
--- a/engine/Makefile
+++ b/engine/Makefile
@@ -248,23 +248,15 @@ INCS += $(INCLUDES)
 ADPCM 	        = source/adpcmlib/adpcm.o
 
 ifdef BUILD_GFX
-GFX 	        = source/gfxlib/2xSaI.o                                                             \
-                  source/gfxlib/bilinear.o                                                          \
+GFX 	        = source/gfxlib/bilinear.o                                                          \
                   source/gfxlib/dotmatrix.o                                                         \
                   source/gfxlib/gfx.o                                                               \
-                  source/gfxlib/hq2x.o                                                              \
                   source/gfxlib/scale2x.o                                                           \
                   source/gfxlib/scanline.o                                                          \
                   source/gfxlib/simple2x.o                                                          \
                   source/gfxlib/tv2x.o
 endif
 
-ifdef BUILD_MMX
-GFX 	       += source/gfxlib/2xSaImmx.o                                                          \
-                  source/gfxlib/bilinearmmx.o                                                       \
-                  source/gfxlib/hq2x16mmx.o
-endif
-
 GAME	        = source/gamelib/draw.o                                                             \
                   source/gamelib/draw16.o                                                           \
                   source/gamelib/draw32.o                                                           \
