diff --git a/src/joystick/SDL_gamepad.c b/src/joystick/SDL_gamepad.c
index dd5b4166a..c64e337fe 100644
--- a/src/joystick/SDL_gamepad.c
+++ b/src/joystick/SDL_gamepad.c
@@ -3977,13 +3977,7 @@ const char *SDL_GetGamepadName(SDL_Gamepad *gamepad)
     SDL_LockJoysticks();
     {
         CHECK_GAMEPAD_MAGIC(gamepad, NULL);
-
-        if (SDL_strcmp(gamepad->name, "*") == 0 ||
-            gamepad->joystick->steam_handle != 0) {
-            result = SDL_GetJoystickName(gamepad->joystick);
-        } else {
-            result = SDL_GetPersistentString(gamepad->name);
-        }
+        result = SDL_GetJoystickName(gamepad->joystick);
     }
     SDL_UnlockJoysticks();
 
diff --git a/src/joystick/linux/SDL_sysjoystick.c b/src/joystick/linux/SDL_sysjoystick.c
index 2bd5d6149..8895b5bdb 100644
--- a/src/joystick/linux/SDL_sysjoystick.c
+++ b/src/joystick/linux/SDL_sysjoystick.c
@@ -329,10 +329,7 @@ static bool IsJoystick(const char *path, int *fd, char **name_return, Uint16 *ve
         }
     }
 
-    name = SDL_CreateJoystickName(inpid.vendor, inpid.product, NULL, product_string);
-    if (!name) {
-        goto error;
-    }
+    name = SDL_strdup(product_string);
 
     if (!IsVirtualJoystick(inpid.vendor, inpid.product, inpid.version, name) &&
         SDL_JoystickHandledByAnotherDriver(&SDL_LINUX_JoystickDriver, inpid.vendor, inpid.product, inpid.version, name)) {
