diff -Naur SDLPoP-819058624bcf0cada46e6ea3757e7b051d49565e.orig/src/types.h SDLPoP-819058624bcf0cada46e6ea3757e7b051d49565e/src/types.h
--- SDLPoP-819058624bcf0cada46e6ea3757e7b051d49565e.orig/src/types.h	2021-05-30 09:15:44.000000000 +0100
+++ SDLPoP-819058624bcf0cada46e6ea3757e7b051d49565e/src/types.h	2021-06-05 13:45:52.767429130 +0100
@@ -24,16 +24,16 @@
 #define STB_VORBIS_HEADER_ONLY
 #include "stb_vorbis.c"
 
-//#if !defined(_MSC_VER)
-//# include <SDL2/SDL.h>
-//# include <SDL2/SDL_image.h>
-//#else
+#if !defined(_MSC_VER)
+# include <SDL2/SDL.h>
+# include <SDL2/SDL_image.h>
+#else
 // These headers for SDL seem to be the pkgconfig/meson standard as per the
 // latest versions. If the old ones should be used, the ifdef must be used
 // to compare versions. 
 # include <SDL.h>
 # include <SDL_image.h>
-//#endif
+#endif
 
 #if SDL_BYTEORDER != SDL_LIL_ENDIAN
 #error This program is not (yet) prepared for big endian CPUs, please contact the author.
