diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index c086f911f..acd9721f1 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -143,7 +143,7 @@ if(MSVC AND NOT USE_CLANG_CL)
 		"/utf-8"
 	)
 elseif(NOT MSVC)
-	add_compile_options(-pipe -fvisibility=hidden -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse)
+	add_compile_options(-pipe -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse -fno-operator-names)
 endif()
 
 set(CONFIG_REL_NO_DEB $<OR:$<CONFIG:Release>,$<CONFIG:MinSizeRel>>)
