diff --git a/cmake/BuildParameters.cmake b/cmake/BuildParameters.cmake
index 86faa7193..fdc180e34 100644
--- a/cmake/BuildParameters.cmake
+++ b/cmake/BuildParameters.cmake
@@ -145,7 +145,7 @@ if(MSVC AND NOT USE_CLANG_CL)
 		"/utf-8"
 	)
 elseif(NOT MSVC)
-	add_compile_options(-pipe -fvisibility=hidden -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse)
+	add_compile_options(-pipe -pthread -fno-builtin-strcmp -fno-builtin-memcmp -mfpmath=sse -fno-operator-names)
 endif()
 
 set(CONFIG_REL_NO_DEB $<OR:$<CONFIG:Release>,$<CONFIG:MinSizeRel>>)
