diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 69791b78130..c2253b4f64e 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -72,7 +72,8 @@ AddReg=\
     SessionMgr,\
     Tapi,\
     ThemeManager,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.NT]
 RegisterDlls=RegisterDllsSection
@@ -129,6 +130,7 @@ AddReg=\
     ThemeManager,\
     VersionInfo,\
     LicenseInformation,\
+    BatoceraOverrides
     NVIDIANGX, \
     WineNonSteamController, \
     ProtonOverrides
@@ -157,7 +159,8 @@ AddReg=\
     Tapi,\
     ThemeManager,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [Wow64Install.ntx86]
 RegisterDlls=RegisterDllsSection
@@ -192,7 +195,12 @@ AddReg=\
     Misc,\
     Tapi,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
+
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
 
 [DefaultInstall.Services]
 AddService=BITS,0,BITSService
