diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index cea98a5af54..c24b9c27b84 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -102,6 +102,7 @@ AddReg=\
     WineDecorateWindows,\
     NVIDIANGX, \
     ProtonOverrides,\
+    BatoceraOverrides,\
     SteamClient
 
 [DefaultInstall.ntamd64]
@@ -133,6 +134,7 @@ AddReg=\
     WineDecorateWindows,\
     NVIDIANGX, \
     ProtonOverrides,\
+    BatoceraOverrides,\
     SteamClient.ntamd64
 
 [DefaultInstall.ntarm64]
@@ -180,6 +182,7 @@ AddReg=\
     WineDecorateWindows,\
     NVIDIANGX, \
     ProtonOverrides,\
+    BatoceraOverrides,\
     SteamClient.ntamd64
 
 [Wow64Install.ntarm]
@@ -3168,5 +3171,9 @@ HKCU,Software\Wine\AppDefaults\ffxvi.exe\DllOverrides,"atiadlxx",,"builtin"
 HKLM,Software\Wow6432Node\lucasarts entertainment company llc\Star Wars: Episode I Racer\v1.0,"Display Height",0x10001,480
 HKLM,Software\Wow6432Node\lucasarts entertainment company llc\Star Wars: Episode I Racer\v1.0,"Display Width",0x10001,640
 
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+
 [WineDecorateWindows]
 HKCU,Software\Wine\AppDefaults\DarkSoulsIII.exe\X11 Driver,"Decorated",,"N"
