diff --git a/google/protobuf/service.py b/google/protobuf/service.py
index 8002c040ffd9..38b8217919cb 100644
--- a/google/protobuf/service.py
+++ b/google/protobuf/service.py
@@ -70,12 +70,12 @@ def CallMethod(self, method_descriptor, rpc_controller,
     Postconditions:
 
     * "done" will be called when the method is complete.  This may be
-     before CallMethod() returns or it may be at some point in the future.
+      before CallMethod() returns or it may be at some point in the future.
     * If the RPC failed, the response value passed to "done" will be None.
-     Further details about the failure can be found by querying the
-     RpcController.
+      Further details about the failure can be found by querying the
+      RpcController.
     """
-   raise NotImplementedError
+    raise NotImplementedError
 
   def GetRequestClass(self, method_descriptor):
     """Returns the class of the request message for the specified method.
