diff --git a/src/d_main.c b/src/d_main.c
index 5d8211393..4f4c9f594 100644
--- a/src/d_main.c
+++ b/src/d_main.c
@@ -1823,7 +1823,7 @@ const char *D_Home(void)
 			usehome = false; // Let's NOT use home
 		else
 #endif
-			userhome = I_GetEnv("HOME"); //Alam: my new HOME for srb2
+			userhome = "/userdata/system/configs"; //Alam: my new HOME for srb2
 	}
 #ifdef _WIN32 //Alam: only Win32 have APPDATA and USERPROFILE
 	if (!userhome && usehome) //Alam: Still not?
