diff --git a/src/Makefile b/src/Makefile
index 539c2fa74..748bb4c6c 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -141,9 +141,9 @@ endif
 
 OBJDUMP_OPTS?=--wide --source --line-numbers
 
-OBJCOPY?=$(call Prefix,objcopy)
-OBJDUMP?=$(call Prefix,objdump)
-WINDRES?=$(call Prefix,windres)
+OBJCOPY:=$(OBJCOPY)
+OBJDUMP:=$(OBJDUMP)
+WINDRES:=$(WINDRES)
 
 GZIP?=gzip
 GZIP_OPTS?=-9 -f -n
diff --git a/src/Makefile.d/sdl.mk b/src/Makefile.d/sdl.mk
index a1bfa3303..58f5b6821 100644
--- a/src/Makefile.d/sdl.mk
+++ b/src/Makefile.d/sdl.mk
@@ -50,9 +50,8 @@ ifdef SDL_PKGCONFIG
 $(eval $(call Use_pkg_config,SDL))
 else
 SDL_CONFIG?=$(call Prefix,sdl2-config)
-SDL_CFLAGS?=$(shell $(SDL_CONFIG) --cflags)
-SDL_LDFLAGS?=$(shell $(SDL_CONFIG) \
-		$(if $(STATIC),--static-libs,--libs))
+SDL_CFLAGS?= `pkg-config sdl2 --cflags`
+SDL_LDFLAGS?= `pkg-config sdl2 --libs` 
 $(eval $(call Propogate_flags,SDL))
 endif
 
