diff -Naur daphne-1710367ecafe2028a29c0344646366b4bb964740.org/src/manymouse/x11_xinput2.c daphne-1710367ecafe2028a29c0344646366b4bb964740/src/manymouse/x11_xinput2.c
--- daphne-1710367ecafe2028a29c0344646366b4bb964740.org/src/manymouse/x11_xinput2.c	2021-08-23 19:30:00.000000000 +0100
+++ daphne-1710367ecafe2028a29c0344646366b4bb964740/src/manymouse/x11_xinput2.c	2021-08-28 13:40:51.231953085 +0100
@@ -15,7 +15,7 @@
 #elif ( (defined(__MACH__)) && (defined(__APPLE__)) )
 #define SUPPORT_XINPUT2 0
 #else
-#define SUPPORT_XINPUT2 1
+#define SUPPORT_XINPUT2 0
 #endif
 #endif
