diff --git a/Makefile b/Makefile
index 9012456..1cad153 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ TARGET_WEB ?= 0
 OSX_BUILD ?= 0
 
 # Enable -no-pie linker option
-NO_PIE ?= 1
+NO_PIE ?= 0
 
 # Specify the target you are building for, TARGET_BITS=0 means native
 TARGET_ARCH ?= native
@@ -562,12 +562,12 @@ ifneq ($(SDL1_USED)$(SDL2_USED),00)
     OSX_PREFIX := $(shell $(SDLCONFIG) --prefix)
     BACKEND_CFLAGS += -I$(OSX_PREFIX)/include $(shell $(SDLCONFIG) --cflags)
   else
-    BACKEND_CFLAGS += $(shell $(SDLCONFIG) --cflags)
+    BACKEND_CFLAGS += `pkg-config sdl2 --cflags`
   endif
   ifeq ($(WINDOWS_BUILD),1)
     BACKEND_LDFLAGS += $(shell $(SDLCONFIG) --static-libs) -lsetupapi -luser32 -limm32 -lole32 -loleaut32 -lshell32 -lwinmm -lversion
   else
-    BACKEND_LDFLAGS += $(shell $(SDLCONFIG) --libs)
+    BACKEND_LDFLAGS += `pkg-config sdl2 --libs`
   endif
 endif
 
@@ -674,7 +674,7 @@ else ifeq ($(HOST_OS),Haiku)
   LDFLAGS := $(BACKEND_LDFLAGS) -no-pie
 
 else
-  LDFLAGS := $(BITS) -march=$(TARGET_ARCH) -lm $(BACKEND_LDFLAGS) -lpthread -ldl
+  LDFLAGS := $(BITS) -lm $(BACKEND_LDFLAGS) -lpthread -ldl
   ifeq ($(NO_PIE), 1)
     LDFLAGS += -no-pie
   endif
