diff --git a/src/pc/audio/audio_sdl2.c b/src/pc/audio/audio_sdl2.c
index b0abea7..506a004 100644
--- a/src/pc/audio/audio_sdl2.c
+++ b/src/pc/audio/audio_sdl2.c
@@ -17,7 +17,7 @@ static bool audio_sdl_init(void) {
     want.freq = 32000;
     want.format = AUDIO_S16SYS;
     want.channels = 2;
-    want.samples = 512;
+    want.samples = 2048;
     want.callback = NULL;
     dev = SDL_OpenAudioDevice(NULL, 0, &want, &have, 0);
     if (dev == 0) {
