diff --git a/drivers/hid/hid-steam.c b/drivers/hid/hid-steam.c
index 9cd07607d082..4c487bdf30dc 100644
--- a/drivers/hid/hid-steam.c
+++ b/drivers/hid/hid-steam.c
@@ -1405,6 +1405,10 @@ static int steam_register(struct steam_device *steam)
 			steam->quirks & STEAM_QUIRK_DECK ? "Deck" : "Controller",
 			steam->serial_no);
 
+	/* start in gamepad mode */
+	steam->gamepad_mode = true;
+	steam_set_lizard_mode(steam, false);
+
 	/* ignore battery errors, we can live without it */
 	if (steam->quirks & (STEAM_QUIRK_WIRELESS | STEAM_QUIRK_IBEX))
 		steam_battery_register(steam);
@@ -2156,15 +2160,16 @@ static const struct steam_axis_mapping steam_deck_imu_mappings[] = {
 static void steam_do_deck_input_event(struct steam_device *steam,
 		struct input_dev *input, u8 *data)
 {
-	bool start_pressed;
+	bool start_pressed, start_pressed2;
 	bool lpad_touched, rpad_touched;
 
-	start_pressed = data[9] & BIT(6);
-
-	if (!start_pressed && steam->did_mode_switch) {
+	start_pressed = data[9] & BIT(5);
+	start_pressed2 = data[14] & BIT(2);
+	
+	if (!start_pressed && !start_pressed2 && steam->did_mode_switch) {
 		steam->did_mode_switch = false;
 		cancel_delayed_work(&steam->mode_switch);
-	} else if (!steam->client_opened && start_pressed && !steam->did_mode_switch) {
+	} else if (!steam->client_opened && start_pressed && start_pressed2 && !steam->did_mode_switch) {
 		hid_dbg(steam->hdev, "%s: doing mode switch\n", __func__);
 		steam->did_mode_switch = true;
 		schedule_delayed_work(&steam->mode_switch, 45 * HZ / 100);
