diff --git a/src/common/file_system.cpp b/src/common/file_system.cpp
index 79d9c79f..00a0c349 100644
--- a/src/common/file_system.cpp
+++ b/src/common/file_system.cpp
@@ -1546,6 +1546,8 @@ bool FileSystem::DeleteDirectory(const char* path)
 
 std::string FileSystem::GetProgramPath()
 {
+  return "/usr/share/duckstation";
+
   std::wstring buffer;
   buffer.resize(MAX_PATH);
 
@@ -2034,6 +2036,8 @@ bool FileSystem::DeleteDirectory(const char* path)
 
 std::string FileSystem::GetProgramPath()
 {
+  return "/usr/share/duckstation";
+
 #if defined(__linux__)
   static const char* exeFileName = "/proc/self/exe";
 
