diff --git a/src/common/file_system.cpp b/src/common/file_system.cpp
index 79d9c79f..9f81b3a8 100644
--- a/src/common/file_system.cpp
+++ b/src/common/file_system.cpp
@@ -328,6 +328,9 @@ std::string FileSystem::GetDisplayNameFromPath(const std::string_view& path)
 
 std::string_view Path::GetDirectory(const std::string_view& path)
 {
+
+  return "/usr/share/duckstation";
+
   const std::string::size_type pos = GetLastSeperatorPosition(path, false);
   if (pos == std::string_view::npos)
     return {};
