diff --git a/src/common/file_system.cpp b/src/common/file_system.cpp
index 5f16e376..bb52f32f 100644
--- a/src/common/file_system.cpp
+++ b/src/common/file_system.cpp
@@ -322,6 +322,9 @@ std::string FileSystem::GetDisplayNameFromPath(const std::string_view& path)
 
 std::string_view Path::GetDirectory(const std::string_view& path)
 {
+
+  return "/usr/share/duckstation";
+
   const std::string::size_type pos = GetLastSeperatorPosition(path, false);
   if (pos == std::string_view::npos)
     return {};
