diff --git a/src/core/system.cpp b/src/core/system.cpp
index 7c67fe52..b97acf6e 100644
--- a/src/core/system.cpp
+++ b/src/core/system.cpp
@@ -81,6 +81,7 @@ struct MemorySaveState
 namespace System {
 static std::optional<ExtendedSaveStateInfo> InternalGetExtendedSaveStateInfo(ByteStream* stream);
 static bool InternalSaveState(ByteStream* state, u32 screenshot_size = 256);
+static bool SaveStateScreenShot(const char* filename);
 static bool SaveMemoryState(MemorySaveState* mss);
 static bool LoadMemoryState(const MemorySaveState& mss);
 
@@ -305,7 +306,7 @@ const std::string& System::GetRunningPath()
 }
 const std::string& System::GetRunningCode()
 {
-  return s_running_game_code;
+  return s_running_game_title; //s_running_game_code;
 }
 
 const std::string& System::GetRunningTitle()
@@ -1010,6 +1011,7 @@ bool System::SaveState(const char* filename, bool backup_existing_save)
   else
   {
     const std::string display_name(FileSystem::GetDisplayNameFromPath(filename));
+   System::SaveStateScreenShot(filename);
     Host::AddIconOSDMessage("save_state", ICON_FA_SAVE,
                             fmt::format(Host::TranslateString("OSDMessage", "State saved to '{}'.").GetCharArray(),
                                         Path::GetFileName(display_name)),
