diff -Naur a/kernel-open/nvidia/nv.c b/kernel-open/nvidia/nv.c
--- a/kernel-open/nvidia/nv.c	2025-01-05 12:13:53.581949199 +0000
+++ b/kernel-open/nvidia/nv.c	2025-01-05 12:22:59.460202684 +0000
@@ -127,7 +127,7 @@
  * DMA_BUF namespace is added by commit id 16b0314aa746
  * ("dma-buf: move dma-buf symbols into the DMA_BUF module namespace") in 5.16
  */
-MODULE_IMPORT_NS(DMA_BUF);
+MODULE_IMPORT_NS("DMA_BUF");
 #endif  // defined(MODULE_IMPORT_NS)
 
 const NvBool nv_is_rm_firmware_supported_os = NV_TRUE;
diff -Naur a/kernel-open/nvidia/nvidia.Kbuild b/kernel-open/nvidia/nvidia.Kbuild
--- a/kernel-open/nvidia/nvidia.Kbuild	2025-01-05 12:19:48.725682325 +0000
+++ b/kernel-open/nvidia/nvidia.Kbuild	2025-01-05 12:22:39.774049799 +0000
@@ -40,13 +40,15 @@
 NVIDIA_BINARY_OBJECT := $(src)/nvidia/nv-kernel.o_binary
 NVIDIA_BINARY_OBJECT_O := nvidia/nv-kernel.o
 
-quiet_cmd_symlink = SYMLINK $@
- cmd_symlink = ln -sf $< $@
+# Rel. commit 80f289101690 "kbuild: change working directory to external module directory with M=" (Masahiro Yamada, 10 Nov 2024)
+# Ensure `$<` is absolute, since the link target is resolved relative to its path, not from where `ln` is run from.
+quiet_cmd_symlinkabs = SYMLINK $@
+ cmd_symlinkabs = ln -sf $(abspath $<) $@
 
 targets += $(NVIDIA_BINARY_OBJECT_O)
 
 $(obj)/$(NVIDIA_BINARY_OBJECT_O): $(NVIDIA_BINARY_OBJECT) FORCE
-	$(call if_changed,symlink)
+	$(call if_changed,symlinkabs)
 
 nvidia-y += $(NVIDIA_BINARY_OBJECT_O)
 
diff -Naur a/kernel-open/nvidia-modeset/nvidia-modeset.Kbuild b/kernel-open/nvidia-modeset/nvidia-modeset.Kbuild
--- a/kernel-open/nvidia-modeset/nvidia-modeset.Kbuild	2025-01-05 12:19:14.123873091 +0000
+++ b/kernel-open/nvidia-modeset/nvidia-modeset.Kbuild	2025-01-05 12:22:39.774049799 +0000
@@ -40,13 +40,15 @@
 NVIDIA_MODESET_BINARY_OBJECT := $(src)/nvidia-modeset/nv-modeset-kernel.o_binary
 NVIDIA_MODESET_BINARY_OBJECT_O := nvidia-modeset/nv-modeset-kernel.o
 
-quiet_cmd_symlink = SYMLINK $@
-cmd_symlink = ln -sf $< $@
+# Rel. commit 80f289101690 "kbuild: change working directory to external module directory with M=" (Masahiro Yamada, 10 Nov 2024)
+# Ensure `$<` is absolute, since the link target is resolved relative to its path, not from where `ln` is run from.
+quiet_cmd_symlinkabs = SYMLINK $@
+ cmd_symlinkabs = ln -sf $(abspath $<) $@
 
 targets += $(NVIDIA_MODESET_BINARY_OBJECT_O)
 
 $(obj)/$(NVIDIA_MODESET_BINARY_OBJECT_O): $(NVIDIA_MODESET_BINARY_OBJECT) FORCE
-	$(call if_changed,symlink)
+	$(call if_changed,symlinkabs)
 
 nvidia-modeset-y += $(NVIDIA_MODESET_BINARY_OBJECT_O)
 
