diff --git a/src/sdl2-jstest.c b/src/sdl2-jstest.c
index 888bbdc..ba87d3d 100644
--- a/src/sdl2-jstest.c
+++ b/src/sdl2-jstest.c
@@ -65,6 +65,7 @@ static void print_joystick_info(int joy_idx, SDL_Joystick* joy, SDL_GameControll
 {
   SDL_JoystickGUID guid = SDL_JoystickGetGUID(joy);
   char guid_str[1024];
+  int i;
   SDL_JoystickGetGUIDString(guid, guid_str, sizeof(guid_str));
 
   printf("Joystick Name:     '%s'\n", SDL_JoystickName(joy));
@@ -84,6 +85,18 @@ static void print_joystick_info(int joy_idx, SDL_Joystick* joy, SDL_GameControll
     printf("  Name:    '%s'\n", SDL_GameControllerName(gamepad));
     printf("  Mapping: '%s'\n", SDL_GameControllerMappingForGUID(guid));
   }
+
+  /* display the list of axes and buttons */
+  for(i=0; i<SDL_JoystickNumAxes(joy); i++) {
+    printf("Axis code %2d:   %2d\n", i, SDL_JoystickAxisEventCodeById(joy_idx, i));
+  }
+  for(i=0; i<SDL_JoystickNumButtons(joy); i++) {
+    printf("Button code %2d:   %2d\n", i, SDL_JoystickButtonEventCodeById(joy_idx, i));
+  }
+  for(i=0; i<SDL_JoystickNumHats(joy); i++) {
+    printf("Hat code %2d:   %2d\n", i, SDL_JoystickHatEventCodeById(joy_idx, i));
+  }
+
   printf("\n");
 }
 
