diff --git a/src/sdl2-jstest.c b/src/sdl2-jstest.c
index ba87d3d..721b991 100644
--- a/src/sdl2-jstest.c
+++ b/src/sdl2-jstest.c
@@ -69,6 +69,7 @@ static void print_joystick_info(int joy_idx, SDL_Joystick* joy, SDL_GameControll
   SDL_JoystickGetGUIDString(guid, guid_str, sizeof(guid_str));
 
   printf("Joystick Name:     '%s'\n", SDL_JoystickName(joy));
+  printf("Joystick Path:     '%s'\n", SDL_JoystickDevicePathById(joy_idx));
   printf("Joystick GUID:     %s\n", guid_str);
   printf("Joystick Number:   %2d\n", joy_idx);
   printf("Number of Axes:    %2d\n", SDL_JoystickNumAxes(joy));
