diff --git a/sdl2-jstest.c b/sdl2-jstest.c
index fb08b56..227f558 100644
--- a/sdl2-jstest.c
+++ b/sdl2-jstest.c
@@ -581,7 +581,7 @@ int main(int argc, char** argv)
     {
       int ret = SDL_GameControllerAddMappingsFromFile(SDL2_JSTEST_DATADIR "/gamecontrollerdb.txt");
       if (ret < 0) {
-        ret = SDL_GameControllerAddMappingsFromFile("gamecontrollerdb.txt");
+        ret = SDL_GameControllerAddMappingsFromFile("/usr/share/gamecontrollerdb.txt");
       }
 
       if (ret < 0)
