diff --git a/src/osd-cat.c b/src/osd-cat.c
index 75602ef..d19670c 100644
--- a/src/osd-cat.c
+++ b/src/osd-cat.c
@@ -77,17 +77,18 @@ static struct option long_options[] = {
 xosd_xft *osd;
 
 /* Default Values */
-char*     font        = "mono:size=12";
-char*     color       = "lightblue";
+char*     font        = "DejaVu Sans:size=14";
+char*     color       = "red";
 char*     bg_color    = "black";
-int       bg_alpha    = 100;
+int       bg_alpha    = 20;
 char*     padding     = "10";
 int       delay_millis = 100;
+int       end_millis = 3000;
 #if defined(HAVE_LIBXINERAMA) || defined(HAVE_LIBXRANDR)
 int       monitor = -1;
 #endif
-char*     geometry = "80cx24l+30+30*top/left";
-char*     text_align = "left";
+char*     geometry = "0cx120*top/left";
+char*     text_align = "center";
 int       nlines = 24;
 #ifdef DEBUG
 char*     debug_level = NULL;
@@ -120,7 +121,7 @@ int main(int argc, char *argv[])
   {
     int option_index = 0;
     int c =
-        getopt_long(argc, argv, "D:f:c:m:g:p:b:a:d:ht:",
+        getopt_long(argc, argv, "D:f:c:m:g:p:b:a:d:e:ht:",
                     long_options,
                     &option_index);
     if (c == -1)
@@ -147,6 +148,9 @@ int main(int argc, char *argv[])
     case 'd':
       delay_millis = atoi(optarg);
       break;
+    case 'e':
+      end_millis = atoi(optarg);
+      break;
     case 'g':
       geometry = optarg;
       break;
@@ -234,7 +238,7 @@ int main(int argc, char *argv[])
     }
     fclose(fp);
   }
-
+  usleep(end_millis * 1000);
   osd_destroy(osd);
   return EXIT_SUCCESS;
 }
