diff --git a/src/main.c b/src/main.c
index f1889ed..80d1fd6 100644
--- a/src/main.c
+++ b/src/main.c
@@ -48,9 +48,9 @@
 #if defined(__APPLE__)
 #include <strings.h>
 #define secure_zero(p, n)  secure_avpzero((p),(n))
-#elif defined(__linux__)
-#include <string.h>
-#define secure_zero(p, n)  secure_avpzero((p),(n))
+//#elif defined(__linux__)
+//#include <string.h>
+//#define secure_zero(p, n)  secure_avpzero((p),(n))
 #else
 static inline void secure_avpzero(void* p, size_t n) {
 	volatile unsigned char* vp = (volatile unsigned char*)p;
