diff --git a/src/files.c b/src/files.c
index 08c0b22..26e99e1 100644
--- a/src/files.c
+++ b/src/files.c
@@ -650,8 +650,8 @@ void InitGameDirectories(char *argv0)
     
 	localdir = (char *)malloc(strlen(homedir)+10);
 	strcpy(localdir, homedir);
-	strcat(localdir, "/");
-	strcat(localdir, ".avp");
+	strcat(localdir, "/configs/");
+	strcat(localdir, "avp");
 
 	tmp = NULL;
 	
@@ -740,13 +740,14 @@ void InitGameDirectories(char *argv0)
 #endif
 
 #if defined __linux__
-	gamedir = "/usr/local/games/AliensVsPredator/";
+	gamedir = "/userdata/roms/ports/avp/";
 #endif
 
 #if defined _WIN32
 	gamedir = ".";
 #endif
 
+	gamedir = "/userdata/roms/ports/avp/";
 	assert(gamedir != NULL);
 	
 	/* last chance sanity check */
