diff --git a/src/oglfunc.h b/src/oglfunc.h
index 62622fb..09475ce 100644
--- a/src/oglfunc.h
+++ b/src/oglfunc.h
@@ -12,7 +12,7 @@
 
 // OpenGL compatibility
 typedef GLclampf GLclampd;
-typedef GLfloat GLdouble;
+//typedef GLfloat GLdouble;
 
 #else
 #include "SDL_opengl.h"
diff --git a/src/stubs.c b/src/stubs.c
index 714fd7f..2f3cdf9 100644
--- a/src/stubs.c
+++ b/src/stubs.c
@@ -109,7 +109,7 @@ void ReloadImageIntoD3DImmediateSurface(IMAGEHEADER* iheader)
 int NumberOfLandscapePolygons;
 int FMVParticleColour;
 int WireFrameMode;
-int WaterFallBase;
+//int WaterFallBase;
 
 void InitDrawTest()
 {
