diff --git a/src/io/log.cpp b/src/io/log.cpp
index e00af2a..da5f14d 100644
--- a/src/io/log.cpp
+++ b/src/io/log.cpp
@@ -48,7 +48,7 @@ Log::Log () {
 
 	level = LL_DEBUG;
 	quiet = false;
-	logfile = fopen("openjazz.log", "w");
+	logfile = fopen("/userdata/system/logs/openjazz.log", "w");
 	color_stdout = false;
 	color_stderr = false;
 
diff --git a/src/main.cpp b/src/main.cpp
index 1f661e6..a3a54ac 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -183,11 +183,11 @@ void startUp (const char *argv0, int pathCount, char *paths[]) {
 
 	// Use hard-coded paths, if available
 
-#ifdef DATAPATH
-	firstPath = new Path(NULL, createString(DATAPATH));
-#else
-	firstPath = NULL;
-#endif
+//#ifdef DATAPATH
+//	firstPath = new Path(NULL, createString(DATAPATH));
+//#else
+	firstPath = new Path(NULL, createString("/userdata/roms/ports/openjazz"));
+//#endif
 
 #ifdef __HAIKU__
 	dev_t volume = dev_for_path("/boot");
@@ -270,18 +270,18 @@ void startUp (const char *argv0, int pathCount, char *paths[]) {
 
 	// Use the user's home directory, if available
 
-#ifdef HOMEDIR
-	#ifdef _WIN32
-	firstPath = new Path(firstPath, createString(getenv("HOME"), "\\"));
-	#else
-	firstPath = new Path(firstPath, createString(getenv("HOME"), "/."));
-	#endif
-#endif
+//#ifdef HOMEDIR
+//	#ifdef _WIN32
+//	firstPath = new Path(firstPath, createString(getenv("HOME"), "\\"));
+//	#else
+	firstPath = new Path(firstPath, createString("/userdata/system/configs/"));
+//	#endif
+//#endif
 
 
 	// Last Resort: Use the current working directory
 
-	firstPath = new Path(firstPath, createString(""));
+	//firstPath = new Path(firstPath, createString(""));
 
 
 
