diff --git a/src/main.cpp b/src/main.cpp
index 1f661e6..e64cd30 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -183,11 +183,12 @@ void startUp (const char *argv0, int pathCount, char *paths[]) {
 
 	// Use hard-coded paths, if available
 
-#ifdef DATAPATH
+/*#ifdef DATAPATH
 	firstPath = new Path(NULL, createString(DATAPATH));
 #else
 	firstPath = NULL;
-#endif
+#endif*/
+firstPath = new Path(NULL, createString("/userdata/roms/ports/openjazz"));
 
 #ifdef __HAIKU__
 	dev_t volume = dev_for_path("/boot");
