diff -Naur melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec.orig/src/ARMJIT_x64/ARMJIT_Compiler.cpp melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec/src/ARMJIT_x64/ARMJIT_Compiler.cpp
--- melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec.orig/src/ARMJIT_x64/ARMJIT_Compiler.cpp	2021-04-11 14:28:15.000000000 +0100
+++ melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec/src/ARMJIT_x64/ARMJIT_Compiler.cpp	2021-08-13 12:18:47.800403534 +0100
@@ -449,14 +449,14 @@
 void Compiler::LoadReg(int reg, X64Reg nativeReg)
 {
     if (reg != 15)
-        MOV(32, R(nativeReg), MDisp(RCPU, offsetof(ARM, R[reg])));
+        MOV(32, R(nativeReg), MDisp(RCPU, offsetof(ARM, R) + reg*4));
     else
         MOV(32, R(nativeReg), Imm32(R15));
 }
 
 void Compiler::SaveReg(int reg, X64Reg nativeReg)
 {
-    MOV(32, MDisp(RCPU, offsetof(ARM, R[reg])), R(nativeReg));
+    MOV(32, MDisp(RCPU, offsetof(ARM, R) + reg*4), R(nativeReg));
 }
 
 // invalidates RSCRATCH and RSCRATCH3
diff -Naur melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec.orig/src/dolphin/x64Emitter.cpp melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec/src/dolphin/x64Emitter.cpp
--- melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec.orig/src/dolphin/x64Emitter.cpp	2021-04-11 14:28:15.000000000 +0100
+++ melonDS-657c729e7473e9b2c09ddf2eef9dcf4ba5b247ec/src/dolphin/x64Emitter.cpp	2021-08-13 11:38:13.584229974 +0100
@@ -4,6 +4,7 @@
 
 #include <cinttypes>
 #include <cstring>
+#include <limits>
 
 #include "CPUDetect.h"
 #include "../types.h"
 