diff -Naur moonlight-embedded-7c8795febc9fad98084aa10fc5918985cbd8081e.orig/CMakeLists.txt moonlight-embedded-7c8795febc9fad98084aa10fc5918985cbd8081e/CMakeLists.txt
--- moonlight-embedded-7c8795febc9fad98084aa10fc5918985cbd8081e.orig/CMakeLists.txt	2021-04-21 18:41:51.000000000 +0100
+++ moonlight-embedded-7c8795febc9fad98084aa10fc5918985cbd8081e/CMakeLists.txt	2021-06-06 09:25:38.607031358 +0100
@@ -113,6 +113,9 @@
   target_include_directories(moonlight-rk PRIVATE ${ROCKCHIP_INCLUDE_DIRS} ${GAMESTREAM_INCLUDE_DIR} ${MOONLIGHT_COMMON_INCLUDE_DIR})
   target_link_libraries(moonlight-rk gamestream ${ROCKCHIP_LIBRARIES})
   set_property(TARGET moonlight-rk PROPERTY COMPILE_DEFINITIONS ${ROCKCHIP_DEFINITIONS})
+    if(ROCKCHIP_FOUND)
+        target_link_libraries(moonlight ${ROCKCHIP_LIBRARIES})
+    endif()
   install(TARGETS moonlight-rk DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
