diff --git a/src/character_editor.cc b/src/character_editor.cc
index e0d373d..a44136d 100644
--- a/src/character_editor.cc
+++ b/src/character_editor.cc
@@ -4,7 +4,7 @@
 #include <ctype.h>
 #include <stdio.h>
 #include <string.h>
-
+#include <climits>
 #include <algorithm>
 #include <vector>
 
diff --git a/src/input.cc b/src/input.cc
index 390a61c..b80d033 100644
--- a/src/input.cc
+++ b/src/input.cc
@@ -2,6 +2,7 @@
 
 #include <SDL.h>
 #include <lodepng.h>
+#include <climits>
 
 #include "audio_engine.h"
 #include "color.h"
diff --git a/src/inventory.cc b/src/inventory.cc
index 0d20aa2..ce74cc5 100644
--- a/src/inventory.cc
+++ b/src/inventory.cc
@@ -3,7 +3,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <string.h>
-
+#include <climits>
 #include <algorithm>
 
 #include "actions.h"
diff --git a/src/platform/git_version.h.in b/src/platform/git_version.h.in
index 99f8728..583d3c4 100644
--- a/src/platform/git_version.h.in
+++ b/src/platform/git_version.h.in
@@ -4,6 +4,6 @@
 #define _BUILD_BRANCH "@BRANCH@"
 #define _BUILD_HASH   "@HASH@"
 #define _BUILD_VER    "@LATEST_TAG@"
-#define _BUILD_DATE   @DATE@
+#define _BUILD_DATE   "@DATE@"
 
 #define CI_BUILD @CI_BUILD@
diff --git a/src/stat.cc b/src/stat.cc
index aacdb91..118e360 100644
--- a/src/stat.cc
+++ b/src/stat.cc
@@ -1,9 +1,9 @@
 #include "stat.h"
 
 #include <stdio.h>
-
+#include <climits>
 #include <algorithm>
-
+#include <climits>
 #include "art.h"
 #include "combat.h"
 #include "critter.h"
diff --git a/src/word_wrap.cc b/src/word_wrap.cc
index 7b6c19c..77b175a 100644
--- a/src/word_wrap.cc
+++ b/src/word_wrap.cc
@@ -3,6 +3,7 @@
 #include <ctype.h>
 #include <stddef.h>
 #include <string.h>
+#include <climits>
 
 #include "text_font.h"
 
