diff --git a/src/character_editor.cc b/src/character_editor.cc
index d3972dc4..560075d0 100644
--- a/src/character_editor.cc
+++ b/src/character_editor.cc
@@ -4,7 +4,7 @@
 #include <ctype.h>
 #include <stdio.h>
 #include <string.h>
-
+#include <climits>
 #include <algorithm>
 #include <vector>
 
diff --git a/src/input.cc b/src/input.cc
index 0eb1818d..c4c605a2 100644
--- a/src/input.cc
+++ b/src/input.cc
@@ -2,6 +2,7 @@
 
 #include <SDL.h>
 #include <lodepng.h>
+#include <climits>
 
 #include "audio_engine.h"
 #include "color.h"
diff --git a/src/inventory.cc b/src/inventory.cc
index 2084c990..53e80a26 100644
--- a/src/inventory.cc
+++ b/src/inventory.cc
@@ -3,7 +3,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <string.h>
-
+#include <climits>
 #include <algorithm>
 #include <cmath>
 #include <string>
diff --git a/src/platform/git_version.h.in b/src/platform/git_version.h.in
index 99f8728f..583d3c49 100644
--- a/src/platform/git_version.h.in
+++ b/src/platform/git_version.h.in
@@ -4,6 +4,6 @@
 #define _BUILD_BRANCH "@BRANCH@"
 #define _BUILD_HASH   "@HASH@"
 #define _BUILD_VER    "@LATEST_TAG@"
-#define _BUILD_DATE   @DATE@
+#define _BUILD_DATE   "@DATE@"
 
 #define CI_BUILD @CI_BUILD@
diff --git a/src/stat.cc b/src/stat.cc
index d9397547..5633610c 100644
--- a/src/stat.cc
+++ b/src/stat.cc
@@ -3,6 +3,7 @@
 #include <charconv>
 #include <stdio.h>
 
+#include <climits>
 #include <algorithm>
 #include <string_view>
 
diff --git a/src/word_wrap.cc b/src/word_wrap.cc
index a8d9b8ac..d204b600 100644
--- a/src/word_wrap.cc
+++ b/src/word_wrap.cc
@@ -3,6 +3,7 @@
 #include <ctype.h>
 #include <stddef.h>
 #include <string.h>
+#include <climits>
 
 #include "text_font.h"
 
