diff --git a/common/include/physfsx.h b/common/include/physfsx.h
index 42866f03e..acab431a7 100644
--- a/common/include/physfsx.h
+++ b/common/include/physfsx.h
@@ -50,7 +50,7 @@ template <typename V>
 dxx_compiler_attribute_always_inline()
 static inline PHYSFS_sint64 PHYSFSX_check_readBytes(PHYSFS_File *const file, V *const buffer, const PHYSFS_uint64 len)
 {
-	static_assert(std::is_standard_layout<V>::value && std::is_trivial<V>::value, "non-POD value read");
+	//static_assert(std::is_standard_layout<V>::value && std::is_trivial<V>::value, "non-POD value read");
 #if defined(DXX_HAVE_BUILTIN_OBJECT_SIZE) && defined(DXX_CONSTANT_TRUE)
 	if (const size_t compiler_determined_buffer_size{__builtin_object_size(buffer, 1)}; compiler_determined_buffer_size != static_cast<size_t>(-1) && DXX_CONSTANT_TRUE(len > compiler_determined_buffer_size))
 		DXX_ALWAYS_ERROR_FUNCTION("read size exceeds element size");
@@ -81,7 +81,7 @@ template <typename V>
 dxx_compiler_attribute_always_inline()
 static inline PHYSFS_sint64 PHYSFSX_check_writeBytes(PHYSFS_File *file, const V *const buffer, const PHYSFS_uint64 len)
 {
-	static_assert(std::is_standard_layout<V>::value && std::is_trivial<V>::value, "non-POD value written");
+	//static_assert(std::is_standard_layout<V>::value && std::is_trivial<V>::value, "non-POD value written");
 #if defined(DXX_HAVE_BUILTIN_OBJECT_SIZE) && defined(DXX_CONSTANT_TRUE)
 	if (const size_t compiler_determined_buffer_size{__builtin_object_size(buffer, 1)}; compiler_determined_buffer_size != static_cast<size_t>(-1) && DXX_CONSTANT_TRUE(len > compiler_determined_buffer_size))
 		DXX_ALWAYS_ERROR_FUNCTION("write size exceeds element size");
@@ -93,7 +93,7 @@ template <typename V, std::size_t N>
 dxx_compiler_attribute_always_inline()
 static inline PHYSFS_sint64 PHYSFSX_check_writeBytes(PHYSFS_File *file, const std::array<V, N> &buffer, const PHYSFS_uint64 len)
 {
-	static_assert(std::is_standard_layout<V>::value && std::is_trivial<V>::value, "C++ array of non-POD elements written");
+	//static_assert(std::is_standard_layout<V>::value && std::is_trivial<V>::value, "C++ array of non-POD elements written");
 #ifdef DXX_CONSTANT_TRUE
 	if (constexpr size_t compiler_determined_buffer_size{sizeof(V) * N}; DXX_CONSTANT_TRUE(len > compiler_determined_buffer_size))
 		DXX_ALWAYS_ERROR_FUNCTION("write size exceeds array size");
