diff --git a/src/bullet.c b/src/bullet.c
index 84e0328..4138645 100644
--- a/src/bullet.c
+++ b/src/bullet.c
@@ -23,8 +23,6 @@
 
 #include "data.h"
 
-Data G_DATA;
-
 void bullet_create(Bullet* bullet, const struct _weapon_data* weapon, Resources* resources) {
 
     int i;
@@ -181,4 +179,4 @@ void bullet_release(Bullet* bullet) {
     bullet->weapon = NULL;
 
     bullet->actor.w = bullet->actor.h = 0;
-}
\ No newline at end of file
+}
diff --git a/src/cli/main.c b/src/cli/main.c
index 60c9d7c..b46db10 100644
--- a/src/cli/main.c
+++ b/src/cli/main.c
@@ -10,6 +10,10 @@
 #include <stdio.h>
 #include <dirent.h>
 
+#include "res.h"
+
+Resources G_RESOURCES;
+
 int create_files(const char* srcdir, const char* dstdir) {
     DIR *d;
     struct dirent *dir;
@@ -145,4 +149,4 @@ int main(int argc, char* argv[]) {
     res_dispose(&res);
 
     return 0;
-}
\ No newline at end of file
+}
diff --git a/src/definitions.h b/src/definitions.h
index 5b555f8..a638aa8 100644
--- a/src/definitions.h
+++ b/src/definitions.h
@@ -25,7 +25,8 @@
 #define FILENAME_INI    "RAPTOR.INI"
 #define FILENAME_ICON   "ICON.BMP"
 //#define DATA_PATH       "DATA/"
-#define DATA_PATH       "../resources/dist/"
+#define DATA_PATH       "/usr/share/raptor/"
+#define SAVE_PATH       "/userdata/saves/raptor/"
 
 #define VIDEO_SCREEN_WIDTH      320
 #define VIDEO_SCREEN_HEIGHT     200
@@ -326,4 +327,4 @@ enum {
     ITEM_LAST
 };
 
-#endif  /* __definitions_h */
\ No newline at end of file
+#endif  /* __definitions_h */
diff --git a/src/game.c b/src/game.c
index 17ba605..b89709d 100644
--- a/src/game.c
+++ b/src/game.c
@@ -26,9 +26,6 @@
 #include "loader.h"
 #include "dialog.h"
 
-Data G_DATA;
-Resources G_RESOURCES;
-
 void game_add_energy(Game* game, uint32_t energy) {
     pilot_add_energy(&game->pilot, energy);
 }
diff --git a/src/main.c b/src/main.c
index a45f615..8f184ac 100644
--- a/src/main.c
+++ b/src/main.c
@@ -26,6 +26,9 @@
 #include "game.h"
 #include "definitions.h"
 
+Data G_DATA;
+Resources G_RESOURCES;
+
 void read_arguments(Game *game, int argc, char* argv[]) {
 	int i;
 
diff --git a/src/pilot.c b/src/pilot.c
index 138feae..5c19cb2 100644
--- a/src/pilot.c
+++ b/src/pilot.c
@@ -29,7 +29,7 @@ BOOL pilot_save(Pilot* pilot) {
     FILE *file;
     uint32_t index = 0;
 
-    sprintf(filename, "%s/SAVE%04d.SAV", DATA_PATH, index);
+    sprintf(filename, "%s/SAVE%04d.SAV", SAVE_PATH, index);
 
     file = fopen(filename, "wb");
     if (!file) {
@@ -50,7 +50,7 @@ BOOL pilot_load(Pilot* pilot, uint32_t index)   {
     char filename[32];
     FILE *file;
 
-    sprintf(filename, "%s/SAVE%04d.SAV", DATA_PATH, index);
+    sprintf(filename, "%s/SAVE%04d.SAV", SAVE_PATH, index);
 
     file = fopen(filename, "rb");
     if (!file) {
