diff --git a/SDL/SDLMain.cpp b/SDL/SDLMain.cpp
index 9b08c7571..a552ee3ae 100644
--- a/SDL/SDLMain.cpp
+++ b/SDL/SDLMain.cpp
@@ -817,12 +817,6 @@ static void ProcessSDLEvent(SDL_Window *window, const SDL_Event &event, InputSta
 				g_Config.iWindowWidth = new_width;
 				g_Config.iWindowHeight = new_height;
 			}
-			// Hide/Show cursor correctly toggling fullscreen
-			if (lastUIState == UISTATE_INGAME && fullscreen && !g_Config.bShowTouchControls) {
-				SDL_ShowCursor(SDL_DISABLE);
-			} else if (lastUIState != UISTATE_INGAME || !fullscreen) {
-				SDL_ShowCursor(SDL_ENABLE);
-			}
 			break;
 		}
 
@@ -1435,9 +1429,7 @@ int main(int argc, char *argv[]) {
 		printf("Init from thread error: '%s'\n", error_message.c_str());
 	}
 
-#ifdef MOBILE_DEVICE
 	SDL_ShowCursor(SDL_DISABLE);
-#endif
 
 	// Ensure that the swap interval is set after context creation (needed for kmsdrm)
 	SDL_GL_SetSwapInterval(1);
@@ -1476,7 +1468,7 @@ int main(int argc, char *argv[]) {
 			if (g_QuitRequested || g_RestartRequested)
 				break;
 
-			UpdateSDLCursor();
+			//UpdateSDLCursor();
 
 			inputTracker.MouseControl();
 			inputTracker.MouseCaptureControl();
@@ -1506,7 +1498,7 @@ int main(int argc, char *argv[]) {
 		if (g_QuitRequested || g_RestartRequested)
 			break;
 
-		UpdateSDLCursor();
+		//UpdateSDLCursor();
 
 		inputTracker.MouseControl();
 		inputTracker.MouseCaptureControl();
