diff --git a/Core/SaveState.cpp b/Core/SaveState.cpp
index 684ad2a17e..99fcb44f94 100644
--- a/Core/SaveState.cpp
+++ b/Core/SaveState.cpp
@@ -543,7 +543,7 @@ double g_lastSaveTime = -1.0;
 
 	Path GenerateSaveSlotFilename(const Path &gameFilename, int slot, const char *extension)
 	{
-		std::string filename = StringFromFormat("%s_%d.%s", GenerateFullDiscId(gameFilename).c_str(), slot, extension);
+		std::string filename = StringFromFormat("%s_%d.%s", gameFilename.WithReplacedExtension("").GetFilename().c_str(), slot, extension);
 		return GetSysDirectory(DIRECTORY_SAVESTATE) / filename;
 	}
 
diff --git a/Core/Util/PathUtil.cpp b/Core/Util/PathUtil.cpp
index fe0aaed13e..a50ca2f402 100644
--- a/Core/Util/PathUtil.cpp
+++ b/Core/Util/PathUtil.cpp
@@ -66,7 +66,7 @@ Path GetSysDirectory(PSPDirectories directoryType) {
 	case DIRECTORY_DUMP:
 		return pspDirectory / "SYSTEM/DUMP";
 	case DIRECTORY_SAVESTATE:
-		return memStickDirectory / "PPSSPP_STATE";
+		return memStickDirectory;
 	case DIRECTORY_CACHE:
 		return Path("/userdata/system/cache/ppsspp/");
 	case DIRECTORY_TEXTURES:
