diff --git a/Core/SaveState.cpp b/Core/SaveState.cpp
index fb42b68bdc..aeec69821c 100644
--- a/Core/SaveState.cpp
+++ b/Core/SaveState.cpp
@@ -532,7 +532,7 @@ namespace SaveState
 
 	Path GenerateSaveSlotFilename(const Path &gameFilename, int slot, const char *extension)
 	{
-		std::string filename = StringFromFormat("%s_%d.%s", GenerateFullDiscId(gameFilename).c_str(), slot, extension);
+		std::string filename = StringFromFormat("%s_%d.%s", gameFilename.WithReplacedExtension("").GetFilename().c_str(), slot, extension);
 		return GetSysDirectory(DIRECTORY_SAVESTATE) / filename;
 	}
 
