diff --git a/Core/SaveState.cpp b/Core/SaveState.cpp
index 2971b88068..48e786a0b3 100644
--- a/Core/SaveState.cpp
+++ b/Core/SaveState.cpp
@@ -542,7 +542,7 @@ double g_lastSaveTime = -1.0;
 
 	Path GenerateSaveSlotFilename(const Path &gameFilename, int slot, const char *extension)
 	{
-		std::string filename = StringFromFormat("%s_%d.%s", GenerateFullDiscId(gameFilename).c_str(), slot, extension);
+		std::string filename = StringFromFormat("%s_%d.%s", gameFilename.WithReplacedExtension("").GetFilename().c_str(), slot, extension);
 		return GetSysDirectory(DIRECTORY_SAVESTATE) / filename;
 	}
 
diff --git a/Core/System.cpp b/Core/System.cpp
index 475b65a92b..38d57e12e3 100644
--- a/Core/System.cpp
+++ b/Core/System.cpp
@@ -790,7 +790,7 @@ Path GetSysDirectory(PSPDirectories directoryType) {
 	case DIRECTORY_DUMP:
 		return pspDirectory / "SYSTEM/DUMP";
 	case DIRECTORY_SAVESTATE:
-		return memStickDirectory / "PPSSPP_STATE";
+		return memStickDirectory;
 	case DIRECTORY_CACHE:
 		return Path("/userdata/system/cache/ppsspp/");
 	case DIRECTORY_TEXTURES:
