diff --git a/Common/GPU/Vulkan/VulkanLoader.cpp b/Common/GPU/Vulkan/VulkanLoader.cpp
index 17596b50a3..870afdf8ae 100644
--- a/Common/GPU/Vulkan/VulkanLoader.cpp
+++ b/Common/GPU/Vulkan/VulkanLoader.cpp
@@ -258,8 +258,9 @@ typedef void *VulkanLibraryHandle;
 static VulkanLibraryHandle vulkanLibrary;
 #endif
 
-bool g_vulkanAvailabilityChecked = false;
-bool g_vulkanMayBeAvailable = false;
+// Batocera - set to true to avoid long load times
+bool g_vulkanAvailabilityChecked = true;
+bool g_vulkanMayBeAvailable = true;
 
 #define LOAD_INSTANCE_FUNC(instance, x) x = (PFN_ ## x)vkGetInstanceProcAddr(instance, #x); if (!x) {INFO_LOG(G3D, "Missing (instance): %s", #x);}
 #define LOAD_INSTANCE_FUNC_CORE(instance, x, ext_x, min_core) \
