diff --git a/Common/GPU/Vulkan/VulkanContext.cpp b/Common/GPU/Vulkan/VulkanContext.cpp
index 3d40c99ec..d1d2a0c83 100644
--- a/Common/GPU/Vulkan/VulkanContext.cpp
+++ b/Common/GPU/Vulkan/VulkanContext.cpp
@@ -970,8 +970,8 @@ VkResult VulkanContext::ReinitSurface() {
 		{
 			const VkDisplayModePropertiesKHR* mode = &mode_props[i];
 
-			if (mode->parameters.visibleRegion.width == pixel_xres &&
-			    mode->parameters.visibleRegion.height == pixel_yres)
+			if (mode->parameters.visibleRegion.width == g_display.pixel_xres &&
+			    mode->parameters.visibleRegion.height == g_display.pixel_yres)
 			{
 				display_mode = mode->displayMode;
 				mode_found = true;
@@ -1050,8 +1050,8 @@ VkResult VulkanContext::ReinitSurface() {
 		}
 	
 		// Finally, create the vulkan surface.
-		image_size.width = pixel_xres;
-		image_size.height = pixel_yres;
+		image_size.width = g_display.pixel_xres;
+		image_size.height = g_display.pixel_yres;
 
 		display.displayMode = display_mode;
 		display.imageExtent = image_size;
