diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8ca2131927..f4ecb6d1c9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1505,7 +1505,7 @@ else()
 		message(WARNING "Found SDL2_ttf <2.0.18 - this is too old, falling back to atlas")
 	endif()
 	if(SDL2_ttf_FOUND)
-		set(nativeExtraLibs ${nativeExtraLibs} SDL2_ttf::SDL2_ttf)
+		set(nativeExtraLibs ${nativeExtraLibs} SDL2_ttf)
 	elseif(SDL2_ttf_PKGCONFIG_FOUND)
 		set(nativeExtraLibs ${nativeExtraLibs} PkgConfig::SDL2_ttf_PKGCONFIG)
 	endif()
