diff --git a/Common/Render/Text/draw_text_sdl.h b/Common/Render/Text/draw_text_sdl.h
index 267028638f..1482ccba4a 100644
--- a/Common/Render/Text/draw_text_sdl.h
+++ b/Common/Render/Text/draw_text_sdl.h
@@ -9,8 +9,8 @@
 #include <fontconfig/fontconfig.h>
 #endif
 
-// SDL2_ttf's TTF_Font is a typedef of _TTF_Font.
-struct _TTF_Font;
+// SDL2_ttf's TTF_Font is a typedef of TTF_Font.
+struct TTF_Font;
 
 class TextDrawerSDL : public TextDrawer {
 public:
@@ -32,9 +32,9 @@ class TextDrawerSDL : public TextDrawer {
 	uint32_t CheckMissingGlyph(std::string_view text);
 	int FindFallbackFonts(uint32_t missingGlyph, int ptSize);
 
-	std::map<uint32_t, _TTF_Font *> fontMap_;
+	std::map<uint32_t, TTF_Font *> fontMap_;
 
-	std::vector<_TTF_Font *> fallbackFonts_;
+	std::vector<TTF_Font *> fallbackFonts_;
 	std::vector<std::pair<std::string, int>> fallbackFontPaths_; // path and font face index
 
 	std::map<int, int> glyphFallbackFontIndex_;
