diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8b455b21b..39476bdc5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1351,7 +1351,7 @@ else()
 		endif()
 	endif()
 	if(SDL2_ttf_FOUND)
-		set(nativeExtraLibs ${nativeExtraLibs} SDL2_ttf::SDL2_ttf)
+		set(nativeExtraLibs ${nativeExtraLibs} SDL2_ttf)
 	elseif(SDL2_ttf_PKGCONFIG_FOUND)
 		add_definitions(-DUSE_SDL2_TTF_PKGCONFIG)
 		set(nativeExtraLibs ${nativeExtraLibs} PkgConfig::SDL2_ttf_PKGCONFIG)
