diff -Naur devilutionx-1.4.1.orig/CMakeLists.txt devilutionx-1.4.1/CMakeLists.txt
--- devilutionx-1.4.1.orig/CMakeLists.txt	2021-11-27 19:13:35.380277400 +0000
+++ devilutionx-1.4.1/CMakeLists.txt	2022-08-27 19:41:22.035115218 +0100
@@ -283,7 +283,9 @@
 else()
   find_package(SDL2 REQUIRED)
   if(TARGET SDL2::SDL2)
-    set(SDL2_MAIN SDL2::SDL2main)
+      if(TARGET SDL2::SDL2main)
+        set(SDL2_MAIN SDL2::SDL2main)
+      endif()
   elseif(TARGET SDL2::SDL2-static)
     # On some distros, such as vitasdk, only the SDL2::SDL2-static target is available.
     # Alias to SDL2::SDL2 because some finder scripts may refer to SDL2::SDL2.
@@ -292,7 +294,9 @@
       set_target_properties(SDL2::SDL2-static PROPERTIES IMPORTED_GLOBAL TRUE)
     endif()
     add_library(SDL2::SDL2 ALIAS SDL2::SDL2-static)
-    set(SDL2_MAIN SDL2::SDL2main)
+      if(TARGET SDL2::SDL2main)
+        set(SDL2_MAIN SDL2::SDL2main)
+      endif()
   else()
     # Assume an older Debian derivate that comes with an sdl2-config.cmake
     # that only defines `SDL2_LIBRARIES` (as -lSDL2) and `SDL2_INCLUDE_DIRS`.
