diff -Naur switchres-d7064ecdf0d6b3716a441f0d395a2c91b34fee8f.orig/makefile switchres-d7064ecdf0d6b3716a441f0d395a2c91b34fee8f/makefile
--- switchres-d7064ecdf0d6b3716a441f0d395a2c91b34fee8f.orig/makefile	2021-05-30 21:02:59.000000000 +0100
+++ switchres-d7064ecdf0d6b3716a441f0d395a2c91b34fee8f/makefile	2021-06-27 08:38:42.468864518 +0100
@@ -10,19 +10,21 @@
 CROSS_COMPILE ?=
 CXX ?= g++
 AR ?= ar
-LDFLAGS = -shared
+LDFLAGS ?=
+LDFLAGS += -shared
 FINAL_CXX=$(CROSS_COMPILE)$(CXX)
 FINAL_AR=$(CROSS_COMPILE)$(AR)
-CPPFLAGS = -O3 -Wall -Wextra
+CPPFLAGS ?= 
+CPPFLAGS += -O3 -Wall -Wextra
 
 PKG_CONFIG=pkg-config
 INSTALL=install
 
 DESTDIR ?=
 PREFIX ?= /usr
-INCDIR = $(DESTDIR)$(PREFIX)/include
-LIBDIR = $(DESTDIR)$(PREFIX)/lib
-BINDIR = $(DESTDIR)$(PREFIX)/bin
+INCDIR = $(PREFIX)/include/libdrm
+LIBDIR = $(PREFIX)/lib
+BINDIR = $(PREFIX)/bin
 PKGDIR = $(LIBDIR)/pkgconfig
 
 # Linux
@@ -52,8 +54,8 @@
 CPPFLAGS += $(shell $(PKG_CONFIG) --cflags $(EXTRA_LIBS))
 endif
 
-CPPFLAGS += -fPIC
-LIBS = -ldl
+CPPFLAGS += -fPIC -I$(INCDIR)
+LIBS = -ldl -lm -lrt -lstdc++
 REMOVE = rm -f
 STATIC_LIB_EXT = a
 DYNAMIC_LIB_EXT = so