diff --git a/Makefile b/Makefile
index 17256bd..53a6c59 100644
--- a/Makefile
+++ b/Makefile
@@ -137,6 +137,20 @@ else ifeq ($(platform), libnx)
     CFLAGS	+= -march=armv8-a -mtune=cortex-a57 -mtp=soft -mcpu=cortex-a57+crc+fp+simd
     CXXFLAGS := $(ASFLAGS) $(CFLAGS) -std=gnu++11
     STATIC_LINKING = 1
+# Raspberry Pi 4 (64-bit)
+else ifeq ($(platform), rpi4_64)
+    TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+	CFLAGS += -mcpu=cortex-a72 -mtune=cortex-a72 -ffast-math -DARM
+    CFLAGS += -D_POSIX_C_SOURCE=199309L
+# Raspberry Pi 3 (64-bit)
+else ifeq ($(platform), rpi3_64)
+    TARGET := $(TARGET_NAME)_libretro.so
+	fpic := -fPIC
+    SHARED := -shared -Wl,--version-script=$(CORE_DIR)/link.T -Wl,--no-undefined
+	CFLAGS += -mcpu=cortex-a53 -mtune=cortex-a53 -ffast-math -DARM
+    CFLAGS += -D_POSIX_C_SOURCE=199309L
 else
    CC ?= gcc
    TARGET := $(TARGET_NAME)_libretro.dll
