diff --git a/3rdparty/FAudio/FAudio_internal.h b/3rdparty/FAudio/src/FAudio_internal.h
index ed255f1..70daca6 100644
--- a/3rdparty/FAudio/src/FAudio_internal.h
+++ b/3rdparty/FAudio/src/FAudio_internal.h
@@ -129,10 +129,10 @@ extern void FAudio_Log(char const *msg);
 #define FAudio_swap64BE(x) SDL_Swap64BE(x)
 
 #else
-#include <SDL_stdinc.h>
-#include <SDL_assert.h>
-#include <SDL_endian.h>
-#include <SDL_log.h>
+#include <SDL2/SDL_stdinc.h>
+#include <SDL2/SDL_assert.h>
+#include <SDL2/SDL_endian.h>
+#include <SDL2/SDL_log.h>
 
 #define FAudio_swap16LE(x) SDL_SwapLE16(x)
 #define FAudio_swap16BE(x) SDL_SwapBE16(x)
diff --git a/3rdparty/FAudio/src/FAudio_platform_sdl2.c b/3rdparty/FAudio/src/FAudio_platform_sdl2.c
index 9eaa157..da14dab 100644
--- a/3rdparty/FAudio/src/FAudio_platform_sdl2.c
+++ b/3rdparty/FAudio/src/FAudio_platform_sdl2.c
@@ -28,7 +28,7 @@
 
 #include "FAudio_internal.h"
 
-#include <SDL.h>
+#include <SDL2/SDL.h>
 
 #if !SDL_VERSION_ATLEAST(2, 24, 0)
 #error "SDL version older than 2.24.0"
