diff --git a/rpcs3/CMakeLists.txt b/rpcs3/CMakeLists.txt
index b1d10d38e..f3c12e310 100644
--- a/rpcs3/CMakeLists.txt
+++ b/rpcs3/CMakeLists.txt
@@ -11,7 +11,7 @@ include(CheckFunctionExists)
 set(CMAKE_CXX_STANDARD 20)
 
 set(ADDITIONAL_LIBS "")
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if(TRUE)
     #on some Linux distros shm_unlink and similar functions are in librt only
     list(APPEND ADDITIONAL_LIBS "rt")
 elseif(NOT WIN32 AND NOT CMAKE_CXX_FLAGS MATCHES "LIBICONV_PLUG")
