diff --git a/rpcs3/CMakeLists.txt b/rpcs3/CMakeLists.txt
index 6d78ae237..355a1aed1 100644
--- a/rpcs3/CMakeLists.txt
+++ b/rpcs3/CMakeLists.txt
@@ -16,7 +16,7 @@ include(CheckFunctionExists)
 set(CMAKE_CXX_STANDARD 20)
 
 set(ADDITIONAL_LIBS "")
-if(CMAKE_SYSTEM MATCHES "Linux")
+if(TRUE)
     #on some Linux distros shm_unlink and similar functions are in librt only
     set(ADDITIONAL_LIBS ${ADDITIONAL_LIBS} "rt")
 elseif(NOT MSVC AND NOT CMAKE_CXX_FLAGS MATCHES "LIBICONV_PLUG")
