diff --git a/3rdparty/libusb/libusb.cmake b/3rdparty/libusb/libusb.cmake
index 6de23c352..0056c95c7 100644
--- a/3rdparty/libusb/libusb.cmake
+++ b/3rdparty/libusb/libusb.cmake
@@ -2,6 +2,7 @@ include(os.cmake)
 
 include(config.cmake)
 include(FindThreads)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 include_directories(${LIBUSB_SOURCE_DIR}/libusb)
 include_directories(${LIBUSB_SOURCE_DIR}/libusb/os)
