diff --git a/rpcs3/Input/evdev_joystick_handler.cpp b/rpcs3/Input/evdev_joystick_handler.cpp
index 03f2016c2..a4b6c504e 100644
--- a/rpcs3/Input/evdev_joystick_handler.cpp
+++ b/rpcs3/Input/evdev_joystick_handler.cpp
@@ -749,7 +749,7 @@ std::vector<pad_list_entry> evdev_joystick_handler::list_devices()
 				if (is_pad_device || is_motion_device)
 				{
 					// It's a joystick or motion device.
-					std::string name = get_device_name(dev);
+					std::string name = "/dev/input/" + et.name;
 
 					if (unique_names.find(name) == unique_names.end())
 						unique_names.emplace(name, 1);
@@ -794,7 +794,7 @@ std::shared_ptr<evdev_joystick_handler::EvdevDevice> evdev_joystick_handler::add
 				continue;
 			}
 
-			std::string name = get_device_name(dev);
+			std::string name = path;
 
 			if (unique_names.find(name) == unique_names.end())
 				unique_names.emplace(name, 1);
