diff --git a/rpcs3/Emu/RSX/GL/glutils/common.h b/rpcs3/Emu/RSX/GL/glutils/common.h
index f99c3590a..94a9a26ee 100644
--- a/rpcs3/Emu/RSX/GL/glutils/common.h
+++ b/rpcs3/Emu/RSX/GL/glutils/common.h
@@ -25,6 +25,10 @@
 #define GL_COMPUTE_BUFFER_SLOT(index)          SSBO_SLOT(2 + index)
 #define GL_COMPUTE_IMAGE_SLOT(index)           SSBO_SLOT(index)
 
+#ifndef glNamedBufferStorageEXT
+#define glNamedBufferStorageEXT __glewBufferStorageEXT
+#endif
+
 //Function call wrapped in ARB_DSA vs EXT_DSA compat check
 #define DSA_CALL(func, object_name, target, ...)\
 	if (::gl::get_driver_caps().ARB_direct_state_access_supported)\
@@ -90,3 +94,4 @@ namespace gl
 		ensure(glGetError() != GL_OUT_OF_MEMORY);
 	}
 }
+
