diff --git a/rpcs3/Input/sdl_pad_handler.cpp b/rpcs3/Input/sdl_pad_handler.cpp
index f25ab2112..14ce64207 100644
--- a/rpcs3/Input/sdl_pad_handler.cpp
+++ b/rpcs3/Input/sdl_pad_handler.cpp
@@ -221,7 +221,7 @@ bool sdl_pad_handler::Init()
 
 	if (g_cfg.io.load_sdl_mappings)
 	{
-		const std::string db_path = rpcs3::utils::get_input_config_root() + "gamecontrollerdb.txt";
+		const std::string db_path = "/usr/share/sdlcontrollerdb/gamecontrollerdb.txt";
 		sdl_log.notice("Adding mappings from file '%s'", db_path);
 
 		if (fs::is_file(db_path))
