diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8c5bfd662..5d60c1b6b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,8 +111,12 @@ if(GIT_FOUND AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git")
 		OUTPUT_STRIP_TRAILING_WHITESPACE
 	)
 else()
-	set(GIT_VERSION "v0.0.0-0-g000000000")
-	set(GIT_HASH "000000000")
+    if(NOT DEFINED GIT_VERSION)
+	    set(GIT_VERSION "v0.0.0-0-g000000000")
+	endif()
+	if(NOT DEFINED GIT_HASH)
+	    set(GIT_HASH "000000000")
+	endif()
 endif()
 
 if(WINDOWS_STORE)
@@ -1991,6 +1995,27 @@ if(UNIX AND NOT APPLE AND NOT ANDROID AND NOT LIBRETRO)
 	endforeach()
 endif()
 
+if(LIBRETRO)
+	MESSAGE("Libretro build ON")
+endif()
+
+if(USE_VULKAN)
+	MESSAGE("Vulkan build ON")
+endif()
+
+if(USE_OPENGL)
+	MESSAGE("OpenGL build ON")
+endif()
+
+if(USE_GLES)
+	MESSAGE("OpenGLES build ON")
+endif()
+
+if(USE_GLES2)
+	MESSAGE("OpenGLES2 build ON")
+endif()
+
+
 if(${CMAKE_GENERATOR} MATCHES "^Xcode.*|^Visual Studio.*")
 	file(GLOB_RECURSE SRC_FILES *.h *.cpp *.c *.cc *.mm)
 	source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR} FILES ${SRC_FILES})
