diff --git a/core/input/gamepad_device.cpp b/core/input/gamepad_device.cpp
index 3fe77654..12d3e2da 100644
--- a/core/input/gamepad_device.cpp
+++ b/core/input/gamepad_device.cpp
@@ -39,6 +39,7 @@ s8 joyrx[4];
 s8 joyry[4];
 u8 rt[4];
 u8 lt[4];
+static int batocera_hotkey = 0;
 
 std::vector<std::shared_ptr<GamepadDevice>> GamepadDevice::_gamepads;
 std::mutex GamepadDevice::_gamepads_mutex;
@@ -82,12 +83,20 @@ bool GamepadDevice::gamepad_btn_input(u32 code, bool pressed)
 			{
 			case EMU_BTN_ESCAPE:
 				if (pressed)
-					dc_exit();
+				batocera_hotkey = 1;
+			else{
+				batocera_hotkey = 0;
+				gui_open_settings();
+			}
 				break;
 			case EMU_BTN_MENU:
 				if (pressed)
 					gui_open_settings();
 				break;
+			case DC_BTN_START:
+				if ((pressed) && (batocera_hotkey == 1))
+					dc_exit();
+ 				break;
 			case EMU_BTN_FFORWARD:
 				if (pressed && !gui_is_open())
 					settings.input.fastForwardMode = !settings.input.fastForwardMode;
