diff --git a/src/gui/sdlmain.cpp b/src/gui/sdlmain.cpp
index c500868b2..4417f9842 100644
--- a/src/gui/sdlmain.cpp
+++ b/src/gui/sdlmain.cpp
@@ -931,7 +931,7 @@ const char *modifier;
 #elif defined(HAIKU)
 #define DEFAULT_CONFIG_FILE             "~/config/settings/dosbox-x/dosbox-x.conf"
 #else /*linux freebsd*/
-# define DEFAULT_CONFIG_FILE            "/.dosboxrc"
+# define DEFAULT_CONFIG_FILE            "/dosboxx.conf"
 #endif
 
 #if C_SET_PRIORITY
diff --git a/src/misc/cross.cpp b/src/misc/cross.cpp
index 2ba476994..b4375352e 100644
--- a/src/misc/cross.cpp
+++ b/src/misc/cross.cpp
@@ -194,7 +194,7 @@ std::string Cross::GetPlatformConfigDir()
         ? xdg_conf_home
         : "~/.config";
 
-    dir = conf_home + "/dosbox-x";
+    dir = conf_home + "/dosbox";
     ResolveHomedir(dir);
 #endif
 
@@ -216,7 +216,7 @@ std::string Cross::GetPlatformConfigName()
 #elif defined(OS2) && defined(C_SDL2)
 #define DEFAULT_CONFIG_FILE "dosbox-x-" PACKAGE_VERSION ".conf"
 #else /* linux, freebsd */
-#define DEFAULT_CONFIG_FILE "dosbox-x-" VERSION ".conf"
+#define DEFAULT_CONFIG_FILE "dosboxx.conf"
 #endif
 
     name = DEFAULT_CONFIG_FILE;
@@ -250,7 +250,7 @@ std::string Cross::CreatePlatformConfigDir()
     const char* xdg_conf_home = getenv("XDG_CONFIG_HOME");
     const std::string conf_home =
         (xdg_conf_home && xdg_conf_home[0] == '/') ? xdg_conf_home : "~/.config";
-    path = conf_home + "/dosbox-x";
+    path = conf_home + "/dosbox";
     ResolveHomedir(path);
     mkdir(path.c_str(), 0700);
 #endif
