diff --git a/src/gui/sdl_mapper.cpp b/src/gui/sdl_mapper.cpp
index ebe31c9e6..5c0cfd4fa 100644
--- a/src/gui/sdl_mapper.cpp
+++ b/src/gui/sdl_mapper.cpp
@@ -94,6 +94,7 @@
 #define MAX_VJOY_HATS           16
 #define MAX_VJOY_AXES           8
 
+#define MOUSE_OFFSET            8
 
 class CEvent;
 class CKeyEvent;
@@ -3063,6 +3064,21 @@ public:
 	uint8_t button;
 };
 
+class CMouseMoveEvent : public CContinuousEvent {
+public:
+   CMouseMoveEvent(char const * const _entry,float _xrel, float _yrel) : CContinuousEvent(_entry) {
+       xrel=_xrel;
+       yrel=_yrel;
+        }
+   void Active(bool /*moved*/) {
+       Mouse_CursorMoved((float)xrel*MOUSE_OFFSET*GetValue()/32767.0,
+           (float)yrel*MOUSE_OFFSET*GetValue()/32767.0,
+           0, 0, true);
+
+   }
+   float xrel, yrel;
+};
+
 //! \brief Joystick axis event handling for the mapper
 class CJAxisEvent : public CContinuousEvent {
 public:
@@ -3937,6 +3953,16 @@ static CMouseButtonEvent * AddMouseButtonEvent(Bitu x,Bitu y,Bitu dx,Bitu dy,cha
 	return event;
 }
 
+static CMouseMoveEvent * AddMouseMoveEvent(Bitu x,Bitu y,Bitu dx,Bitu dy,char const * const title,char const * const entry,float xrel,float yrel) {
+   char buf[64];
+   strcpy(buf,"maxis_");
+   strcat(buf,entry);
+   CMouseMoveEvent * event=new CMouseMoveEvent(buf,xrel,yrel);
+   new CEventButton(x,y,dx,dy,title,event);
+   return event;
+}
+
+ 
 static CJAxisEvent * AddJAxisButton(Bitu x,Bitu y,Bitu dx,Bitu dy,char const * const title,Bitu stick,Bitu axis,bool positive,CJAxisEvent * opposite_axis) {
     char buf[64];
     sprintf(buf,"jaxis_%d_%d%s",(int)stick,(int)axis,positive ? "+" : "-");
@@ -4065,6 +4091,16 @@ static void CreateLayout(void) {
 
 #undef XO
 #undef YO
+#define XO 6
+#define YO 8
+   /* Mouse Buttons and Axis */
+   new CTextButton(PX(XO+0),PY(YO-1),2*BW,20,"Mouse axis");
+   AddMouseMoveEvent(PX(XO+1),PY(YO),BW,BH,"Y-","y-",0,-1);
+   AddMouseMoveEvent(PX(XO+0),PY(YO+1),BW,BH,"X-","x-",-1,0);
+   AddMouseMoveEvent(PX(XO+1),PY(YO+1),BW,BH,"Y+","y+",0,1);
+   AddMouseMoveEvent(PX(XO+2),PY(YO+1),BW,BH,"X+","x+",1,0);
+#undef XO
+#undef YO
 #define XO 0
 #define YO 3
 
