diff --git a/src/gui/sdlmain.cpp b/src/gui/sdlmain.cpp
index 27ca9ae17..f2842157f 100644
--- a/src/gui/sdlmain.cpp
+++ b/src/gui/sdlmain.cpp
@@ -819,7 +819,7 @@ const char *modifier;
 #elif defined(HAIKU)
 #define DEFAULT_CONFIG_FILE "~/config/settings/dosbox-x/dosbox-x.conf"
 #else /*linux freebsd*/
-# define DEFAULT_CONFIG_FILE            "/.dosboxrc"
+# define DEFAULT_CONFIG_FILE            "/dosboxx.conf"
 #endif
 
 #if C_SET_PRIORITY
diff --git a/src/misc/cross.cpp b/src/misc/cross.cpp
index 36e558c8c..8cadd571d 100644
--- a/src/misc/cross.cpp
+++ b/src/misc/cross.cpp
@@ -167,7 +167,7 @@ void Cross::GetPlatformConfigDir(std::string& in) {
 #elif !defined(HX_DOS)
 	const char *xdg_conf_home = getenv("XDG_CONFIG_HOME");
 	const std::string conf_home = xdg_conf_home && xdg_conf_home[0] == '/' ? xdg_conf_home: "~/.config";
-	in = conf_home + "/dosbox-x";
+	in = conf_home + "/dosbox";
 	ResolveHomedir(in);
 #endif
 	//LOG_MSG("Config dir: %s", in.c_str());
@@ -180,7 +180,7 @@ void Cross::GetPlatformConfigName(std::string& in) {
 #elif defined(MACOSX)
 #define DEFAULT_CONFIG_FILE "DOSBox-X " VERSION " Preferences"
 #else /*linux freebsd*/
-#define DEFAULT_CONFIG_FILE "dosbox-x-" VERSION ".conf"
+#define DEFAULT_CONFIG_FILE "dosboxx.conf"
 #endif
 	in = DEFAULT_CONFIG_FILE;
 }
@@ -204,7 +204,7 @@ void Cross::CreatePlatformConfigDir(std::string& in) {
 #elif !defined(HX_DOS)
 	const char *xdg_conf_home = getenv("XDG_CONFIG_HOME");
 	const std::string conf_home = xdg_conf_home && xdg_conf_home[0] == '/' ? xdg_conf_home: "~/.config";
-	in = conf_home + "/dosbox-x";
+	in = conf_home + "/dosbox";
 	ResolveHomedir(in);
 	mkdir(in.c_str(),0700);
 #endif
