diff --git a/plugins/pinmame/PinMAMEPlugin.cpp b/plugins/pinmame/PinMAMEPlugin.cpp
index 3b2002fc6..117109ce2 100644
--- a/plugins/pinmame/PinMAMEPlugin.cpp
+++ b/plugins/pinmame/PinMAMEPlugin.cpp
@@ -414,7 +414,7 @@ MSGPI_EXPORT void MSGPIAPI PinMAMEPluginLoad(const uint32_t sessionId, const Msg
       }
       #elif defined(__APPLE__) || defined(__linux__)
       if (pinmamePath.empty())
-         pinmamePath = std::filesystem::path(getenv("HOME")) / ".pinmame";
+         pinmamePath = string("/userdata/system/configs/vpinball/pinmame/");
       #endif
 
       // FIXME implement a last resort or just ask the user to define its path setup in the settings ?
