diff --git a/lib/graphics_engine/src/ge_vulkan_shader_manager.cpp b/lib/graphics_engine/src/ge_vulkan_shader_manager.cpp
index 9ddcd68b8..7a8469a2e 100644
--- a/lib/graphics_engine/src/ge_vulkan_shader_manager.cpp
+++ b/lib/graphics_engine/src/ge_vulkan_shader_manager.cpp
@@ -231,12 +231,9 @@ VkShaderModule GEVulkanShaderManager::loadShader(shaderc_shader_kind kind,
             delete include_result;
         }, &includer);
 
-    shaderc_compilation_result_t result = shaderc_compile_into_spv(compiler,
-        shader_data.c_str(), shader_data.size(), kind, shader_fullpath.c_str(),
-        "main", options);
+	shaderc_compilation_result_t result;
+	const shaderc_compilation_status status = shaderc_compile_into_spv(compiler, shader_data.c_str(), shader_data.size(), kind, shader_fullpath.c_str(), "main", options, &result);
     shaderc_compile_options_release(options);
-    shaderc_compilation_status status =
-        shaderc_result_get_compilation_status(result);
     if (status != shaderc_compilation_status_success)
         throw std::runtime_error(shaderc_result_get_error_message(result));
 
