diff --git a/yabause/CMake/Packages/external_libchdr.cmake b/yabause/CMake/Packages/external_libchdr.cmake
index f6a8db60..b91eddf0 100644
--- a/yabause/CMake/Packages/external_libchdr.cmake
+++ b/yabause/CMake/Packages/external_libchdr.cmake
@@ -29,8 +29,8 @@ endif()
 
 ExternalProject_Add(
   libchdr
-  GIT_REPOSITORY "https://github.com/devmiyax/libchdr.git"
-  GIT_TAG "074ff1614f2a685f2b5a95b0e788bff6297d5680"
+  #GIT_REPOSITORY "https://github.com/devmiyax/libchdr.git"
+  #GIT_TAG "074ff1614f2a685f2b5a95b0e788bff6297d5680"
   #PATCH_COMMAND  git apply "${CMAKE_SOURCE_DIR}/CMake/Packages/libchdr.patch"
   CMAKE_ARGS  -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/libchdr 
               -DCMAKE_BUILD_TYPE:STRING=Release 
diff --git a/yabause/src/retro_arena/CMakeLists.txt b/yabause/src/retro_arena/CMakeLists.txt
index d767aa81..a4e05ad7 100644
--- a/yabause/src/retro_arena/CMakeLists.txt
+++ b/yabause/src/retro_arena/CMakeLists.txt
@@ -31,11 +31,11 @@ include( ${CMAKE_SOURCE_DIR}/CMake/Packages/external_libpng.cmake )
 ExternalProject_Add(
      Json
      PREFIX ${CMAKE_CURRENT_BINARY_DIR}/Json
-     GIT_REPOSITORY https://github.com/nlohmann/json.git
+     #GIT_REPOSITORY https://github.com/nlohmann/json.git
      # For shallow git clone (without downloading whole history)
      #GIT_SHALLOW 1
      # For point at certain tag
-     GIT_TAG v3.2.0
+     #GIT_TAG v3.2.0
      #disables auto update on every build
      UPDATE_DISCONNECTED 1
      #disable following
