diff --git a/Makefile b/Makefile
index 633a5ca..84e5734 100755
--- a/Makefile
+++ b/Makefile
@@ -8,14 +8,14 @@
 
 CC         = gcc
 LD         = gcc
-TARGET     = vj$(EXESUFFIX)
+TARGET     = virtualjaguar
 
-CFLAGS = -Wall -Ofast -march=native -fno-common -flto -Wall -Wextra `sdl-config --cflags` -D__GCCUNIX__ -DDSP_EMU -DSCALE_TO_SCREEN
+CFLAGS += -Wall -fno-common -flto -D__GCCUNIX__ -DDSP_EMU -DSCALE_TO_SCREEN
 LDFLAGS = -flto
 
-LIBS = `sdl-config --libs` -lz $(GLLIB)
+LIBS =-lz $(GLLIB) `pkg-config sdl --libs`
 
-INCS = -I. -Isrc -Isrc/include -Isrc/musashi
+INCS = -I. -Isrc -Isrc/include -Isrc/musashi `pkg-config sdl --cflags`
 
 THECC = $(CC) $(CFLAGS) $(INCS)
 
@@ -87,4 +87,5 @@ obj/m68kops.h: obj/m68kmake$(EXESUFFIX)
 	obj/m68kmake obj src/m68k_in.c
 
 obj/m68kmake$(EXESUFFIX): src/m68kmake.c src/m68k_in.c
-	$(CC) $(WARNINGS) src/m68kmake.c -o obj/m68kmake$(EXESUFFIX)
+	#$(CC_FOR_BUILD) $(WARNINGS) src/m68kmake.c -o obj/m68kmake$(EXESUFFIX)
+	$(CC_FOR_BUILD) -o $@ $<
