diff --git a/src/settings.c b/src/settings.c
index 4694ed1..792cf8c 100755
--- a/src/settings.c
+++ b/src/settings.c
@@ -64,9 +64,9 @@ void LoadVJSettings(void)
 	vjs.p1KeyBindings[19] = SDLK_KP_DIVIDE;
 	vjs.p1KeyBindings[20] = SDLK_KP_MULTIPLY;
 
-	strcpy(vjs.jagBootPath, "./BIOS/jagboot.rom");
-	strcpy(vjs.EEPROMPath, "./EEPROMs");
-	strcpy(vjs.ROMPath, "./ROMs");
+	strcpy(vjs.jagBootPath, "/userdata/bios/jagboot.rom");
+	strcpy(vjs.EEPROMPath, "./userdata/saves/jaguar");
+	strcpy(vjs.ROMPath, "/userdata/roms/jaguar");
 	CheckForTrailingSlash(vjs.EEPROMPath);
 	CheckForTrailingSlash(vjs.ROMPath);
 
