diff --git a/src/joystick.c b/src/joystick.c
index 385f537..4e5a324 100755
--- a/src/joystick.c
+++ b/src/joystick.c
@@ -112,12 +112,14 @@ void joystick_exec(void)
 	if (keystate[vjs.p1KeyBindings[20]])
 		joypad_0_buttons[BUTTON_d] = 0x01;
 
-#ifdef GCW0
-    if (keystate[SDLK_ESCAPE] && keystate[SDLK_RETURN])
-#else
+//#ifdef GCW0
+//    if (keystate[SDLK_ESCAPE] && keystate[SDLK_RETURN])
+//#else
     if (keystate[SDLK_ESCAPE])
-#endif
+//#endif
     {
+    	SDL_QuitSubSystem(SDL_INIT_VIDEO | SDL_INIT_JOYSTICK | SDL_INIT_AUDIO | SDL_INIT_TIMER);
+    	SDL_Quit();
 		exit(0);
     }
 
