diff --git a/arch/Kconfig b/arch/Kconfig
index a3308a220f86..2c2af9e4287f 100644
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -14,7 +14,7 @@ config ARCH_CONFIGURES_CPU_MITIGATIONS
 
 if !ARCH_CONFIGURES_CPU_MITIGATIONS
 config CPU_MITIGATIONS
-	def_bool y
+	def_bool n
 endif
 
 #
diff --git a/kernel/cpu.c b/kernel/cpu.c
index a59e009e0be4..28e511bd646d 100644
--- a/kernel/cpu.c
+++ b/kernel/cpu.c
@@ -3216,7 +3216,7 @@ EXPORT_SYMBOL_GPL(cpu_mitigations_auto_nosmt);
 #else
 static int __init mitigations_parse_cmdline(char *arg)
 {
-	pr_crit("Kernel compiled without mitigations, ignoring 'mitigations'; system may still be vulnerable\n");
+	//pr_crit("Kernel compiled without mitigations, ignoring 'mitigations'; system may still be vulnerable\n");
 	return 0;
 }
 #endif
