diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 8437ee953dc..62a347dd7e1 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -76,7 +76,8 @@ AddReg=\
     Services, \
     SessionMgr,\
     SteamClient,\
-    ThemeManager
+    ThemeManager,\
+    WineBatoOverrides
 
 [DefaultInstall]
 Needs=BaseInstall
@@ -90,13 +91,15 @@ WineFakeDlls=FakeDllsWin32
 AddReg=\
     Packages,\
     VCRuntime.ntx86,\
-    VersionInfo
+    VersionInfo,\
+    WineBatoOverrides
 
 [DefaultInstall.ntarm]
 Needs=BaseInstall
 WineFakeDlls=FakeDllsWin32
 AddReg=\
-    VersionInfo
+    VersionInfo,\
+    WineBatoOverrides
 
 [DefaultInstall.ntamd64]
 Needs=BaseInstall
@@ -106,7 +109,8 @@ AddReg=\
     Packages.ntamd64,\
     SteamClient64, \
     VCRuntime.ntamd64,\
-    VersionInfo
+    VersionInfo,\
+    WineBatoOverrides
 
 [DefaultInstall.ntarm64]
 Needs=BaseInstall
@@ -117,7 +121,8 @@ AddReg=\
     SteamClient64, \
     VCRuntime.ntamd64,\
     VCRuntime.ntarm64,\
-    VersionInfo
+    VersionInfo,\
+    WineBatoOverrides
 
 [BaseWow64Install]
 RegisterDlls=RegisterDllsSection
@@ -130,7 +135,8 @@ AddReg=\
     LicenseInformation,\
     MCI,\
     Misc,\
-    VersionInfo
+    VersionInfo,\
+    WineBatoOverrides
 
 [Wow64Install.ntx86]
 Needs=BaseWow64Install
@@ -1753,3 +1759,9 @@ HKLM,Software\Khronos\OpenXR\1,"ActiveRuntime",,"C:\openxr\wineopenxr64.json"
 ;;Other app-specific overrides
 HKCU,Software\Wine\AppDefaults\Pentiment.exe\DllOverrides,"SpeechSynthesisWrapper",4,
 HKCU,Software\Wine\AppDefaults\Maine-Win64-Shipping.exe\DllOverrides,"SpeechSynthWrapper",4,
+
+[WineBatoOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+HKCU,Software\Xenia,"XEFLAGS",0x10001,"1"
+
