diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index d0a0ffd0774..ece6bb197a4 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -75,7 +75,8 @@ AddReg=\
     Tapi,\
     ThemeManager,\
     VCRuntime.ntx86,\
-    LicenseInformation
+    LicenseInformation,\
+    WineBatoOverrides
 
 [DefaultInstall.NT]
 RegisterDlls=RegisterDllsSection
@@ -106,7 +107,8 @@ AddReg=\
     VersionInfo,\
     LicenseInformation,\
     NVIDIANGX, \
-    ProtonOverrides
+    ProtonOverrides, \
+    WineBatoOverrides
 
 [DefaultInstall.ntamd64]
 RegisterDlls=RegisterDllsSection
@@ -139,7 +141,8 @@ AddReg=\
     LicenseInformation,\
     NVIDIANGX, \
     TTS, \
-    ProtonOverrides
+    ProtonOverrides, \
+    WineBatoOverrides
 
 [DefaultInstall.ntarm64]
 RegisterDlls=RegisterDllsSection
@@ -172,7 +175,8 @@ AddReg=\
     LicenseInformation,\
     NVIDIANGX, \
     TTS, \
-    ProtonOverrides
+    ProtonOverrides, \
+    WineBatoOverrides
 
 [Wow64Install.ntx86]
 RegisterDlls=RegisterDllsSection
@@ -195,7 +199,8 @@ AddReg=\
     VersionInfo,\
     LicenseInformation,\
     NVIDIANGX, \
-    ProtonOverrides
+    ProtonOverrides, \
+    WineBatoOverrides
 
 [Wow64Install.ntarm]
 RegisterDlls=RegisterDllsSection
@@ -212,7 +217,8 @@ AddReg=\
     SteamClient64,\
     Tapi,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation, \
+    WineBatoOverrides
 
 [DefaultInstall.Services]
 AddService=BITS,0,BITSService
@@ -3287,3 +3293,9 @@ HKLM,Software\Wow6432Node\lucasarts entertainment company llc\Star Wars: Episode
 ;;Other app-specific overrides
 HKCU,Software\Wine\AppDefaults\Pentiment.exe\DllOverrides,"SpeechSynthesisWrapper",,"disabled"
 HKCU,Software\Wine\AppDefaults\Maine-Win64-Shipping.exe\DllOverrides,"SpeechSynthWrapper",0x2,"disabled"
+
+[WineBatoOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+HKCU,Software\Xenia,"XEFLAGS",0x10001,"1"
+
