diff --git a/dlls/winebus.sys/main.c b/dlls/winebus.sys/main.c
index 19fe986a674..274663cc173 100644
--- a/dlls/winebus.sys/main.c
+++ b/dlls/winebus.sys/main.c
@@ -1075,7 +1075,7 @@ static NTSTATUS udev_driver_init(BOOL enable_sdl)
         .wait_code = udev_wait,
     };
 
-    bus_options.disable_hidraw = check_bus_option(L"DisableHidraw", 0);
+    bus_options.disable_hidraw = TRUE;
     if (bus_options.disable_hidraw) TRACE("UDEV hidraw devices disabled in registry\n");
     bus_options.disable_input = check_bus_option(L"DisableInput", 0);
     if (bus_options.disable_input) TRACE("UDEV input devices disabled in registry\n");
