diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index 95c1a2dc07e..f3c1b70bd7a 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -75,7 +75,8 @@ AddReg=\
     Tapi,\
     ThemeManager,\
     VCRuntime.ntx86,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.NT]
 RegisterDlls=RegisterDllsSection
@@ -169,7 +170,8 @@ AddReg=\
     VCRuntime.ntamd64,\
     VCRuntime.ntarm64,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [Wow64Install.ntx86]
 RegisterDlls=RegisterDllsSection
@@ -209,7 +211,13 @@ AddReg=\
     SteamClient64,\
     Tapi,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
+    
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+HKCU,Software\Xenia,"XEFLAGS",0x10001,"1"
 
 [DefaultInstall.Services]
 AddService=BITS,0,BITSService
