diff --git a/loader/wine.inf.in b/loader/wine.inf.in
index d0a0ffd0774..6acd79f1829 100644
--- a/loader/wine.inf.in
+++ b/loader/wine.inf.in
@@ -75,7 +75,8 @@ AddReg=\
     Tapi,\
     ThemeManager,\
     VCRuntime.ntx86,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
 
 [DefaultInstall.NT]
 RegisterDlls=RegisterDllsSection
@@ -106,7 +107,8 @@ AddReg=\
     VersionInfo,\
     LicenseInformation,\
     NVIDIANGX, \
-    ProtonOverrides
+    ProtonOverrides,\
+    BatoceraOverrides
 
 [DefaultInstall.ntamd64]
 RegisterDlls=RegisterDllsSection
@@ -139,7 +141,8 @@ AddReg=\
     LicenseInformation,\
     NVIDIANGX, \
     TTS, \
-    ProtonOverrides
+    ProtonOverrides,\
+    BatoceraOverrides
 
 [DefaultInstall.ntarm64]
 RegisterDlls=RegisterDllsSection
@@ -172,7 +175,8 @@ AddReg=\
     LicenseInformation,\
     NVIDIANGX, \
     TTS, \
-    ProtonOverrides
+    ProtonOverrides,\
+    BatoceraOverrides
 
 [Wow64Install.ntx86]
 RegisterDlls=RegisterDllsSection
@@ -195,7 +199,8 @@ AddReg=\
     VersionInfo,\
     LicenseInformation,\
     NVIDIANGX, \
-    ProtonOverrides
+    ProtonOverrides,\
+    BatoceraOverrides
 
 [Wow64Install.ntarm]
 RegisterDlls=RegisterDllsSection
@@ -212,7 +217,13 @@ AddReg=\
     SteamClient64,\
     Tapi,\
     VersionInfo,\
-    LicenseInformation
+    LicenseInformation,\
+    BatoceraOverrides
+
+[BatoceraOverrides]
+HKLM,System\CurrentControlSet\Services\winebus,"DisableHidraw",0x10001,"1"
+HKLM,System\CurrentControlSet\Services\winebus,"Enable SDL",0x10001,"1"
+HKCU,Software\Xenia,"XEFLAGS",0x10001,"1"
 
 [DefaultInstall.Services]
 AddService=BITS,0,BITSService
