diff --git a/server/Makefile.in b/server/Makefile.in
index 24ad64849bd..df09fa16e1a 100644
--- a/server/Makefile.in
+++ b/server/Makefile.in
@@ -51,6 +51,7 @@ SOURCES = \
 	wineserver.man.in \
 	winstation.c
 
-UNIX_LIBS = $(LDEXECFLAGS) $(RT_LIBS) $(INOTIFY_LIBS) $(PROCSTAT_LIBS)
+UNIX_LIBS = $(LDEXECFLAGS) $(RT_LIBS) $(INOTIFY_LIBS) $(PROCSTAT_LIBS) -flto=auto -flto-partition=one -fdevirtualize-at-ltrans $(CFLAGS)
+EXTRADEFS = -flto=auto -flto-partition=one -fdevirtualize-at-ltrans
 
 unicode_EXTRADEFS = -DBINDIR="\"${bindir}\"" -DDATADIR="\"${datadir}\""
