diff --git a/dlls/winedmo/Makefile.in b/dlls/winedmo/Makefile.in
index bee5d80f242..1cf228ffacd 100644
--- a/dlls/winedmo/Makefile.in
+++ b/dlls/winedmo/Makefile.in
@@ -12,7 +12,6 @@ SOURCES = \
 	media-converter/fossilize.c \
 	media-converter/murmur3.c \
 	media-converter/lib.c \
-	libavcodec/pcm_byte_order_reverse_bsf.c \
 	unix_demuxer.c \
 	unix_demuxer_mediaconv.c \
 	unix_media_type.c \
diff --git a/dlls/winedmo/unix_demuxer.c b/dlls/winedmo/unix_demuxer.c
index 24681e2d5df..4f53b5cffac 100644
--- a/dlls/winedmo/unix_demuxer.c
+++ b/dlls/winedmo/unix_demuxer.c
@@ -104,22 +104,6 @@ NTSTATUS demuxer_check( void *arg )
     return STATUS_SUCCESS;
 }
 
-static BOOL codec_is_big_endian_pcm(enum AVCodecID codec_id)
-{
-    switch (codec_id)
-    {
-    case AV_CODEC_ID_PCM_S16BE:
-    case AV_CODEC_ID_PCM_S24BE:
-    case AV_CODEC_ID_PCM_S32BE:
-    case AV_CODEC_ID_PCM_S64BE:
-    case AV_CODEC_ID_PCM_F32BE:
-    case AV_CODEC_ID_PCM_F64BE:
-        return TRUE;
-    default:
-        return FALSE;
-    }
-}
-
 static NTSTATUS demuxer_create_streams( struct demuxer *demuxer )
 {
     UINT i;
@@ -142,14 +126,6 @@ static NTSTATUS demuxer_create_streams( struct demuxer *demuxer )
                 continue;
             }
         }
-        else if (codec_is_big_endian_pcm(par->codec_id))
-        {
-            /* WAVEFORMATEX does not contain endianness info, so this needs to be converted here. */
-            if (av_bsf_alloc( &ff_pcm_byte_order_reverse_bsf, &stream->filter ) < 0) return STATUS_UNSUCCESSFUL;
-            avcodec_parameters_copy( stream->filter->par_in, par );
-            av_bsf_init( stream->filter );
-            continue;
-        }
 
         av_bsf_get_null_filter( &stream->filter );
         avcodec_parameters_copy( stream->filter->par_in, demuxer->ctx->streams[i]->codecpar );
diff --git a/dlls/winedmo/unix_media_type.c b/dlls/winedmo/unix_media_type.c
index edde85172b6..15ffdaaed38 100644
--- a/dlls/winedmo/unix_media_type.c
+++ b/dlls/winedmo/unix_media_type.c
@@ -185,8 +185,6 @@ static NTSTATUS audio_format_from_codec_params( const AVCodecParameters *params,
 
     format_tag = wave_format_tag_from_codec_id( params->codec_id );
     if (params->codec_id == AV_CODEC_ID_OPUS) format_tag = WAVE_FORMAT_OPUS;
-    /* Big-endian PCM in native Windows is given the usual WAVE_FORMAT_PCM tag. */
-    else if (params->codec_id == AV_CODEC_ID_PCM_S16BE) format_tag = WAVE_FORMAT_PCM;
 
     if (format_tag == WAVE_FORMAT_EXTENSIBLE || format_tag >> 16 || (channels > 2 && channel_mask != 0) ||
         params->codec_id == AV_CODEC_ID_VORBIS)
diff --git a/dlls/winedmo/unix_private.h b/dlls/winedmo/unix_private.h
index 8974afd5f72..53391118f53 100644
--- a/dlls/winedmo/unix_private.h
+++ b/dlls/winedmo/unix_private.h
@@ -44,11 +44,6 @@ extern void mediaconv_demuxer_init(void);
 extern void mediaconv_demuxer_exit(void);
 extern int mediaconv_demuxer_open( AVFormatContext **ctx, struct stream_context *context );
 
-/* libavcodec/pcm_byte_order_reverse.c */
-#ifdef HAVE_FFMPEG
-extern const AVBitStreamFilter ff_pcm_byte_order_reverse_bsf;
-#endif
-
 /* unixlib.c */
 extern int64_t unix_seek_callback( void *opaque, int64_t offset, int whence );
 extern int unix_read_callback( void *opaque, uint8_t *buffer, int size );
